{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4a9627be-7752-4af2-a7f6-361064013d5f",
   "metadata": {},
   "source": [
    "# Figure 5 panel C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "882f68f7-948d-4502-a50e-e9af9bd01ee5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.ticker import MaxNLocator\n",
    "import matplotlib.colors as mcolors\n",
    "import matplotlib.cm as cm\n",
    "from matplotlib.lines import Line2D\n",
    "from mpl_toolkits.axes_grid1.inset_locator import inset_axes\n",
    "import matplotlib.patches as mpatches \n",
    "import json\n",
    "import warnings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d8fe0067-92b8-4ff8-8e9f-33ffabe3a4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.path.append('../../')\n",
    "sys.path.append(os.path.abspath(\"./modules\"))\n",
    "from consensus_variables import *\n",
    "from analysis_utils import (\n",
    "    get_normal_maf,\n",
    "    get_dna2prot_depth,\n",
    "    find_exon,\n",
    "    get_cancer_maf_all,\n",
    "    get_o3d_gene_data,\n",
    "    get_o3d_data_cancer\n",
    ")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc0fd24c-ba06-4309-bafc-596f312e2afa",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "da9fddaa-295f-46b1-9d2f-901073e67611",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Init\n",
    "gene_order = [\"KMT2D\",\"EP300\",\"ARID1A\",\"CREBBP\",\"NOTCH2\",\"RB1\",\"RBM10\",\"TP53\",\n",
    "              \"CDKN1A\",\"FOXQ1\", \"KDM6A\", \"STAG2\", \"FGFR3\"\n",
    "             # \"PIK3CA\",\"TERT\", \"KMT2C\",\n",
    "             ]\n",
    "run_name = \"all_samples\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c4291a10-a699-47f1-aef7-3d51a6ba1d8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "pfam_domains_file = \"/data/bbg/nobackup/scratch/oncodrive3d/annotations_mane_240506/uniprot_feat.tsv\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c25414eb-480d-47a1-a972-7500f60c8426",
   "metadata": {},
   "source": [
    "### Mutations\n",
    "* __TODO: check for correct filtering__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "06a49a27-50ce-4e76-91c0-3b06341914b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/fcalvet/projects/bladder_paper/normal_bladder_paper/extended/extended10/modules/analysis_utils.py:21: DtypeWarning: Columns (49,92,112) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  maf_df = pd.read_table(path_maf)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>indel</td>\n",
       "      <td>2768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>missense</td>\n",
       "      <td>1234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>truncating</td>\n",
       "      <td>1617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>indel</td>\n",
       "      <td>1189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>missense</td>\n",
       "      <td>239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>truncating</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>indel</td>\n",
       "      <td>561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>missense</td>\n",
       "      <td>1630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>truncating</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EP300</td>\n",
       "      <td>indel</td>\n",
       "      <td>772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>EP300</td>\n",
       "      <td>missense</td>\n",
       "      <td>1911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>EP300</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EP300</td>\n",
       "      <td>truncating</td>\n",
       "      <td>415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>indel</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>missense</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>truncating</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>indel</td>\n",
       "      <td>1037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>missense</td>\n",
       "      <td>475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>truncating</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>indel</td>\n",
       "      <td>2631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>missense</td>\n",
       "      <td>969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>truncating</td>\n",
       "      <td>1131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>indel</td>\n",
       "      <td>5568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>missense</td>\n",
       "      <td>4093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>truncating</td>\n",
       "      <td>4738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>indel</td>\n",
       "      <td>518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>missense</td>\n",
       "      <td>1481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>truncating</td>\n",
       "      <td>222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>RB1</td>\n",
       "      <td>indel</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>RB1</td>\n",
       "      <td>missense</td>\n",
       "      <td>235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>RB1</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>RB1</td>\n",
       "      <td>truncating</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>indel</td>\n",
       "      <td>1591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>missense</td>\n",
       "      <td>934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>truncating</td>\n",
       "      <td>1383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>indel</td>\n",
       "      <td>802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>missense</td>\n",
       "      <td>619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>truncating</td>\n",
       "      <td>672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>TP53</td>\n",
       "      <td>indel</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>TP53</td>\n",
       "      <td>missense</td>\n",
       "      <td>559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>TP53</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>TP53</td>\n",
       "      <td>truncating</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Consequence  Count\n",
       "0   ARID1A       indel   2768\n",
       "1   ARID1A    missense   1234\n",
       "2   ARID1A  synonymous    327\n",
       "3   ARID1A  truncating   1617\n",
       "4   CDKN1A       indel   1189\n",
       "5   CDKN1A    missense    239\n",
       "6   CDKN1A  synonymous     40\n",
       "7   CDKN1A  truncating    183\n",
       "8   CREBBP       indel    561\n",
       "9   CREBBP    missense   1630\n",
       "10  CREBBP  synonymous    283\n",
       "11  CREBBP  truncating    233\n",
       "12   EP300       indel    772\n",
       "13   EP300    missense   1911\n",
       "14   EP300  synonymous    237\n",
       "15   EP300  truncating    415\n",
       "16   FGFR3       indel     28\n",
       "17   FGFR3    missense    121\n",
       "18   FGFR3  synonymous     72\n",
       "19   FGFR3  truncating      6\n",
       "20   FOXQ1       indel   1037\n",
       "21   FOXQ1    missense    475\n",
       "22   FOXQ1  synonymous    136\n",
       "23   FOXQ1  truncating     61\n",
       "24   KDM6A       indel   2631\n",
       "25   KDM6A    missense    969\n",
       "26   KDM6A  synonymous     88\n",
       "27   KDM6A  truncating   1131\n",
       "28   KMT2D       indel   5568\n",
       "29   KMT2D    missense   4093\n",
       "30   KMT2D  synonymous    944\n",
       "31   KMT2D  truncating   4738\n",
       "32  NOTCH2       indel    518\n",
       "33  NOTCH2    missense   1481\n",
       "34  NOTCH2  synonymous    223\n",
       "35  NOTCH2  truncating    222\n",
       "36     RB1       indel    137\n",
       "37     RB1    missense    235\n",
       "38     RB1  synonymous     46\n",
       "39     RB1  truncating     38\n",
       "40   RBM10       indel   1591\n",
       "41   RBM10    missense    934\n",
       "42   RBM10  synonymous    142\n",
       "43   RBM10  truncating   1383\n",
       "44   STAG2       indel    802\n",
       "45   STAG2    missense    619\n",
       "46   STAG2  synonymous    112\n",
       "47   STAG2  truncating    672\n",
       "48    TP53       indel     81\n",
       "49    TP53    missense    559\n",
       "50    TP53  synonymous     43\n",
       "51    TP53  truncating     51"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count each mutation only ones if it appears in multiple reads \n",
    "maf = get_normal_maf(somatic_maf_file, gene_order, truncating=True)\n",
    "maf_count = maf.groupby(['Gene', 'Consequence', 'Pos']).size().reset_index(name='Count')\n",
    "maf_cnsq_count = maf.groupby(['Gene', 'Consequence']).size().reset_index(name='Count')\n",
    "maf_cnsq_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13479804-aa9b-490b-b952-964dfde1bee9",
   "metadata": {},
   "source": [
    "### Get exon coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2f25805b-ee0b-428e-97b2-57ebcbcf2e71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>Ens_Transcr_ID</th>\n",
       "      <th>Ens_Gene_ID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Begin</th>\n",
       "      <th>End</th>\n",
       "      <th>Description</th>\n",
       "      <th>Full_description</th>\n",
       "      <th>Evidence</th>\n",
       "      <th>Pfam_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>P04217</td>\n",
       "      <td>ENST00000263100</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>218.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>Ig_2</td>\n",
       "      <td>Immunoglobulin domain</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF13895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A0A024QZJ5</td>\n",
       "      <td>ENST00000373997</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>58.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>RRM_1</td>\n",
       "      <td>RNA recognition motif</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A0A024QZJ5</td>\n",
       "      <td>ENST00000373997</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>138.0</td>\n",
       "      <td>199.0</td>\n",
       "      <td>RRM_1</td>\n",
       "      <td>RNA recognition motif</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A0A024QZJ5</td>\n",
       "      <td>ENST00000373997</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>233.0</td>\n",
       "      <td>296.0</td>\n",
       "      <td>RRM_1</td>\n",
       "      <td>RNA recognition motif</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A0A024QZJ5</td>\n",
       "      <td>ENST00000373997</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>439.0</td>\n",
       "      <td>514.0</td>\n",
       "      <td>DND1_DSRM</td>\n",
       "      <td>double strand RNA binding domain from DEAD END...</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF14709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189112</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>O43149</td>\n",
       "      <td>ENST00000381638</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>258.0</td>\n",
       "      <td>380.0</td>\n",
       "      <td>ANAPC10</td>\n",
       "      <td>Anaphase-promoting complex, subunit 10 (APC10)</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF03256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189123</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>O43149</td>\n",
       "      <td>ENST00000381638</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>1781.0</td>\n",
       "      <td>1812.0</td>\n",
       "      <td>ZZ</td>\n",
       "      <td>Zinc finger, ZZ type</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189133</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>O43149</td>\n",
       "      <td>ENST00000381638</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>1828.0</td>\n",
       "      <td>1869.0</td>\n",
       "      <td>ZZ</td>\n",
       "      <td>Zinc finger, ZZ type</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189155</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Q8IYH5</td>\n",
       "      <td>ENST00000370801</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>655.0</td>\n",
       "      <td>703.0</td>\n",
       "      <td>Myb_DNA-binding</td>\n",
       "      <td>Myb-like DNA-binding domain</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189160</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Q8IYH5</td>\n",
       "      <td>ENST00000370801</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>DOMAIN</td>\n",
       "      <td>818.0</td>\n",
       "      <td>864.0</td>\n",
       "      <td>ZZ</td>\n",
       "      <td>Zinc finger, ZZ type</td>\n",
       "      <td>Pfam</td>\n",
       "      <td>PF00569</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43704 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Uniprot_ID   Ens_Transcr_ID      Ens_Gene_ID    Type   Begin  \\\n",
       "6        A1BG      P04217  ENST00000263100  ENSG00000121410  DOMAIN   218.0   \n",
       "11       A1CF  A0A024QZJ5  ENST00000373997  ENSG00000148584  DOMAIN    58.0   \n",
       "12       A1CF  A0A024QZJ5  ENST00000373997  ENSG00000148584  DOMAIN   138.0   \n",
       "13       A1CF  A0A024QZJ5  ENST00000373997  ENSG00000148584  DOMAIN   233.0   \n",
       "14       A1CF  A0A024QZJ5  ENST00000373997  ENSG00000148584  DOMAIN   439.0   \n",
       "...       ...         ...              ...              ...     ...     ...   \n",
       "189112  ZZEF1      O43149  ENST00000381638  ENSG00000074755  DOMAIN   258.0   \n",
       "189123  ZZEF1      O43149  ENST00000381638  ENSG00000074755  DOMAIN  1781.0   \n",
       "189133  ZZEF1      O43149  ENST00000381638  ENSG00000074755  DOMAIN  1828.0   \n",
       "189155   ZZZ3      Q8IYH5  ENST00000370801  ENSG00000036549  DOMAIN   655.0   \n",
       "189160   ZZZ3      Q8IYH5  ENST00000370801  ENSG00000036549  DOMAIN   818.0   \n",
       "\n",
       "           End      Description  \\\n",
       "6        300.0             Ig_2   \n",
       "11       125.0            RRM_1   \n",
       "12       199.0            RRM_1   \n",
       "13       296.0            RRM_1   \n",
       "14       514.0        DND1_DSRM   \n",
       "...        ...              ...   \n",
       "189112   380.0          ANAPC10   \n",
       "189123  1812.0               ZZ   \n",
       "189133  1869.0               ZZ   \n",
       "189155   703.0  Myb_DNA-binding   \n",
       "189160   864.0               ZZ   \n",
       "\n",
       "                                         Full_description Evidence  Pfam_ID  \n",
       "6                                   Immunoglobulin domain     Pfam  PF13895  \n",
       "11                                  RNA recognition motif     Pfam  PF00076  \n",
       "12                                  RNA recognition motif     Pfam  PF00076  \n",
       "13                                  RNA recognition motif     Pfam  PF00076  \n",
       "14      double strand RNA binding domain from DEAD END...     Pfam  PF14709  \n",
       "...                                                   ...      ...      ...  \n",
       "189112     Anaphase-promoting complex, subunit 10 (APC10)     Pfam  PF03256  \n",
       "189123                               Zinc finger, ZZ type     Pfam  PF00569  \n",
       "189133                               Zinc finger, ZZ type     Pfam  PF00569  \n",
       "189155                        Myb-like DNA-binding domain     Pfam  PF00249  \n",
       "189160                               Zinc finger, ZZ type     Pfam  PF00569  \n",
       "\n",
       "[43704 rows x 11 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam = pd.read_table(pfam_domains_file)\n",
    "pfam[(pfam[\"Type\"] == \"DOMAIN\") & (pfam[\"Evidence\"] == \"Pfam\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "466b9fdc-8501-4ffa-ab7e-71a3e7c19af1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing gene: ARID1A\n",
      "Processing gene: NOTCH2\n",
      "Processing gene: KMT2D\n",
      "Processing gene: RB1\n",
      "Processing gene: CREBBP\n",
      "Processing gene: TP53\n",
      "Processing gene: EP300\n",
      "Processing gene: FGFR3\n",
      "Processing gene: FOXQ1\n",
      "Processing gene: CDKN1A\n",
      "Processing gene: KDM6A\n",
      "Processing gene: RBM10\n",
      "Processing gene: STAG2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>DNA_POS</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>PROT_POS</th>\n",
       "      <th>REVERSE_STRAND</th>\n",
       "      <th>EXON_RANK</th>\n",
       "      <th>TRANSCRIPT_ID</th>\n",
       "      <th>COVERED</th>\n",
       "      <th>DEPTH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696404</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696405</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696406</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696407</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696408</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79220</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739789</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>CTT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>458.126582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79221</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739790</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>454.101266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79222</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739791</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>GTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>449.683544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79223</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739792</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>TCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>447.215190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79224</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739793</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>CCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>444.860759</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79225 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR   DNA_POS    GENE CONTEXT  PROT_POS  REVERSE_STRAND  EXON_RANK  \\\n",
       "0       chr1  26696404  ARID1A     NaN       1.0             1.0        0.0   \n",
       "1       chr1  26696405  ARID1A     NaN       1.0             1.0        0.0   \n",
       "2       chr1  26696406  ARID1A     NaN       1.0             1.0        0.0   \n",
       "3       chr1  26696407  ARID1A     NaN       2.0             1.0        0.0   \n",
       "4       chr1  26696408  ARID1A     NaN       2.0             1.0        0.0   \n",
       "...      ...       ...     ...     ...       ...             ...        ...   \n",
       "79220  chr19   6739789  TRIP10     CTT       NaN             NaN        NaN   \n",
       "79221  chr19   6739790  TRIP10     ACT       NaN             NaN        NaN   \n",
       "79222  chr19   6739791  TRIP10     GTC       NaN             NaN        NaN   \n",
       "79223  chr19   6739792  TRIP10     TCC       NaN             NaN        NaN   \n",
       "79224  chr19   6739793  TRIP10     CCA       NaN             NaN        NaN   \n",
       "\n",
       "         TRANSCRIPT_ID  COVERED       DEPTH  \n",
       "0      ENST00000324856        0    0.000000  \n",
       "1      ENST00000324856        0    0.000000  \n",
       "2      ENST00000324856        0    0.000000  \n",
       "3      ENST00000324856        0    0.000000  \n",
       "4      ENST00000324856        0    0.000000  \n",
       "...                ...      ...         ...  \n",
       "79220              NaN        1  458.126582  \n",
       "79221              NaN        1  454.101266  \n",
       "79222              NaN        1  449.683544  \n",
       "79223              NaN        1  447.215190  \n",
       "79224              NaN        1  444.860759  \n",
       "\n",
       "[79225 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26696015</td>\n",
       "      <td>26697540</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARID1A--2_ENST00000324856_ENSE00003471930</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26729651</td>\n",
       "      <td>26729863</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARID1A--3_ENST00000324856_ENSE00000902180</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26731152</td>\n",
       "      <td>26731604</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARID1A--4_ENST00000324856_ENSE00001227857</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26732676</td>\n",
       "      <td>26732792</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARID1A--5_ENST00000324856_ENSE00001349762</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26760856</td>\n",
       "      <td>26761096</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>STAG2--31_ENST00000371145_ENSE00004014452</td>\n",
       "      <td>chrX</td>\n",
       "      <td>124090575</td>\n",
       "      <td>124090764</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>STAG2--32_ENST00000371145_ENSE00001728775</td>\n",
       "      <td>chrX</td>\n",
       "      <td>124090854</td>\n",
       "      <td>124090964</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>STAG2--33_ENST00000371145_ENSE00004014445</td>\n",
       "      <td>chrX</td>\n",
       "      <td>124094018</td>\n",
       "      <td>124094144</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>STAG2--34_ENST00000371145_ENSE00004014453</td>\n",
       "      <td>chrX</td>\n",
       "      <td>124095372</td>\n",
       "      <td>124095449</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>STAG2--35_ENST00000371145_ENSE00003903284</td>\n",
       "      <td>chrX</td>\n",
       "      <td>124100574</td>\n",
       "      <td>124102656</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            ID   Chr      Start        End  \\\n",
       "0    ARID1A--1_ENST00000324856_ENSE00001907429  chr1   26696015   26697540   \n",
       "1    ARID1A--2_ENST00000324856_ENSE00003471930  chr1   26729651   26729863   \n",
       "2    ARID1A--3_ENST00000324856_ENSE00000902180  chr1   26731152   26731604   \n",
       "3    ARID1A--4_ENST00000324856_ENSE00001227857  chr1   26732676   26732792   \n",
       "4    ARID1A--5_ENST00000324856_ENSE00001349762  chr1   26760856   26761096   \n",
       "..                                         ...   ...        ...        ...   \n",
       "315  STAG2--31_ENST00000371145_ENSE00004014452  chrX  124090575  124090764   \n",
       "316  STAG2--32_ENST00000371145_ENSE00001728775  chrX  124090854  124090964   \n",
       "317  STAG2--33_ENST00000371145_ENSE00004014445  chrX  124094018  124094144   \n",
       "318  STAG2--34_ENST00000371145_ENSE00004014453  chrX  124095372  124095449   \n",
       "319  STAG2--35_ENST00000371145_ENSE00003903284  chrX  124100574  124102656   \n",
       "\n",
       "     Strand  \n",
       "0         1  \n",
       "1         1  \n",
       "2         1  \n",
       "3         1  \n",
       "4         1  \n",
       "..      ...  \n",
       "315       1  \n",
       "316       1  \n",
       "317       1  \n",
       "318       1  \n",
       "319       1  \n",
       "\n",
       "[320 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "consensus_df = pd.read_table(\"../../data/consensus.exons_splice_sites.unique.tsv\")\n",
    "depth_df = pd.read_table(f\"{deepcsa_run_dir}/annotatedepths/all_samples_indv.depths.tsv.gz\")\n",
    "\n",
    "exons_depth, exons_coord_id = get_dna2prot_depth(maf, depth_df, consensus_df)\n",
    "#exons_coord = pd.read_table(\"/data/bbg/projects/bladder_ts/data/exons/bladder_panel.cds.bed4.bed\", header=None)\n",
    "display(exons_depth)\n",
    "display(exons_coord_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dc8ff72a-096f-4922-9145-cb154b040431",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>DNA_POS</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>PROT_POS</th>\n",
       "      <th>REVERSE_STRAND</th>\n",
       "      <th>EXON_RANK</th>\n",
       "      <th>TRANSCRIPT_ID</th>\n",
       "      <th>COVERED</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>EXON_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696404</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696405</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696406</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696407</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696408</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ARID1A--1_ENST00000324856_ENSE00001907429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79220</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739789</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>CTT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>458.126582</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79221</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739790</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>454.101266</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79222</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739791</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>GTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>449.683544</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79223</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739792</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>TCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>447.215190</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79224</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6739793</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>CCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>444.860759</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79225 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR   DNA_POS    GENE CONTEXT  PROT_POS  REVERSE_STRAND  EXON_RANK  \\\n",
       "0       chr1  26696404  ARID1A     NaN       1.0             1.0        0.0   \n",
       "1       chr1  26696405  ARID1A     NaN       1.0             1.0        0.0   \n",
       "2       chr1  26696406  ARID1A     NaN       1.0             1.0        0.0   \n",
       "3       chr1  26696407  ARID1A     NaN       2.0             1.0        0.0   \n",
       "4       chr1  26696408  ARID1A     NaN       2.0             1.0        0.0   \n",
       "...      ...       ...     ...     ...       ...             ...        ...   \n",
       "79220  chr19   6739789  TRIP10     CTT       NaN             NaN        NaN   \n",
       "79221  chr19   6739790  TRIP10     ACT       NaN             NaN        NaN   \n",
       "79222  chr19   6739791  TRIP10     GTC       NaN             NaN        NaN   \n",
       "79223  chr19   6739792  TRIP10     TCC       NaN             NaN        NaN   \n",
       "79224  chr19   6739793  TRIP10     CCA       NaN             NaN        NaN   \n",
       "\n",
       "         TRANSCRIPT_ID  COVERED       DEPTH  \\\n",
       "0      ENST00000324856        0    0.000000   \n",
       "1      ENST00000324856        0    0.000000   \n",
       "2      ENST00000324856        0    0.000000   \n",
       "3      ENST00000324856        0    0.000000   \n",
       "4      ENST00000324856        0    0.000000   \n",
       "...                ...      ...         ...   \n",
       "79220              NaN        1  458.126582   \n",
       "79221              NaN        1  454.101266   \n",
       "79222              NaN        1  449.683544   \n",
       "79223              NaN        1  447.215190   \n",
       "79224              NaN        1  444.860759   \n",
       "\n",
       "                                         EXON_ID  \n",
       "0      ARID1A--1_ENST00000324856_ENSE00001907429  \n",
       "1      ARID1A--1_ENST00000324856_ENSE00001907429  \n",
       "2      ARID1A--1_ENST00000324856_ENSE00001907429  \n",
       "3      ARID1A--1_ENST00000324856_ENSE00001907429  \n",
       "4      ARID1A--1_ENST00000324856_ENSE00001907429  \n",
       "...                                          ...  \n",
       "79220                                        NaN  \n",
       "79221                                        NaN  \n",
       "79222                                        NaN  \n",
       "79223                                        NaN  \n",
       "79224                                        NaN  \n",
       "\n",
       "[79225 rows x 11 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exons_depth[\"EXON_ID\"] = exons_depth.apply(lambda x: find_exon(x, exons_coord_id), axis=1)\n",
    "exons_depth"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b03f9567-bafa-4ecb-97e8-f7e333248b49",
   "metadata": {},
   "source": [
    "### Others"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fc9801e1-3dfd-414d-be6f-866d6edaea4c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> site_selection:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Selection</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>1000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>1001</td>\n",
       "      <td>2.139958</td>\n",
       "      <td>0.373307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>1002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>1003</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>1004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22549</th>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ENST00000313244</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22550</th>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ENST00000313244</td>\n",
       "      <td>5</td>\n",
       "      <td>2.433402</td>\n",
       "      <td>0.336978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22551</th>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ENST00000313244</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22552</th>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ENST00000313244</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22553</th>\n",
       "      <td>TRIP10</td>\n",
       "      <td>ENST00000313244</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22554 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         GENE          Feature  Protein_position  Selection   p_value\n",
       "0      ARID1A  ENST00000324856              1000   0.000000  1.000000\n",
       "1      ARID1A  ENST00000324856              1001   2.139958  0.373307\n",
       "2      ARID1A  ENST00000324856              1002   0.000000  1.000000\n",
       "3      ARID1A  ENST00000324856              1003   0.000000  1.000000\n",
       "4      ARID1A  ENST00000324856              1004   0.000000  1.000000\n",
       "...       ...              ...               ...        ...       ...\n",
       "22549  TRIP10  ENST00000313244                 4   0.000000  1.000000\n",
       "22550  TRIP10  ENST00000313244                 5   2.433402  0.336978\n",
       "22551  TRIP10  ENST00000313244                 6   0.000000  1.000000\n",
       "22552  TRIP10  ENST00000313244                 7   0.000000  1.000000\n",
       "22553  TRIP10  ENST00000313244                 8   0.000000  1.000000\n",
       "\n",
       "[22554 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> exon_selection:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>sample</th>\n",
       "      <th>impact</th>\n",
       "      <th>mutations</th>\n",
       "      <th>dnds</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>lower</th>\n",
       "      <th>upper</th>\n",
       "      <th>exon_rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A--10_ENST00000324856_ENSE00001349753</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>missense</td>\n",
       "      <td>16</td>\n",
       "      <td>0.715234</td>\n",
       "      <td>1.663548e-01</td>\n",
       "      <td>0.416336</td>\n",
       "      <td>1.138208</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARID1A--10_ENST00000324856_ENSE00001349753</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>truncating</td>\n",
       "      <td>49</td>\n",
       "      <td>28.999998</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>20.546647</td>\n",
       "      <td>40.289061</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARID1A--11_ENST00000324856_ENSE00001349752</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>missense</td>\n",
       "      <td>47</td>\n",
       "      <td>2.300000</td>\n",
       "      <td>1.251698e-06</td>\n",
       "      <td>1.679419</td>\n",
       "      <td>3.075757</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARID1A--11_ENST00000324856_ENSE00001349752</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>truncating</td>\n",
       "      <td>46</td>\n",
       "      <td>22.810938</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>16.016163</td>\n",
       "      <td>31.845361</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARID1A--12_ENST00000324856_ENSE00000872621</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>missense</td>\n",
       "      <td>62</td>\n",
       "      <td>1.689844</td>\n",
       "      <td>2.888441e-04</td>\n",
       "      <td>1.283827</td>\n",
       "      <td>2.182397</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>TP53--P53-ENST00000269305</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>truncating</td>\n",
       "      <td>84</td>\n",
       "      <td>7.395313</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>5.734699</td>\n",
       "      <td>9.430225</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>TP53--P53_tetramer-ENST00000269305</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>missense</td>\n",
       "      <td>30</td>\n",
       "      <td>1.459155</td>\n",
       "      <td>5.814385e-02</td>\n",
       "      <td>0.986482</td>\n",
       "      <td>2.073169</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>TP53--P53_tetramer-ENST00000269305</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>truncating</td>\n",
       "      <td>18</td>\n",
       "      <td>4.387500</td>\n",
       "      <td>8.940697e-07</td>\n",
       "      <td>2.596850</td>\n",
       "      <td>6.973560</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>TP53--TAD2-ENST00000269305</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>missense</td>\n",
       "      <td>28</td>\n",
       "      <td>1.758984</td>\n",
       "      <td>8.130968e-03</td>\n",
       "      <td>1.168019</td>\n",
       "      <td>2.539624</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>TP53--TAD2-ENST00000269305</td>\n",
       "      <td>all_samples</td>\n",
       "      <td>truncating</td>\n",
       "      <td>8</td>\n",
       "      <td>4.927332</td>\n",
       "      <td>5.791783e-04</td>\n",
       "      <td>2.140323</td>\n",
       "      <td>9.941370</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>653 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           gene       sample      impact  \\\n",
       "0    ARID1A--10_ENST00000324856_ENSE00001349753  all_samples    missense   \n",
       "1    ARID1A--10_ENST00000324856_ENSE00001349753  all_samples  truncating   \n",
       "2    ARID1A--11_ENST00000324856_ENSE00001349752  all_samples    missense   \n",
       "3    ARID1A--11_ENST00000324856_ENSE00001349752  all_samples  truncating   \n",
       "4    ARID1A--12_ENST00000324856_ENSE00000872621  all_samples    missense   \n",
       "..                                          ...          ...         ...   \n",
       "648                   TP53--P53-ENST00000269305  all_samples  truncating   \n",
       "649          TP53--P53_tetramer-ENST00000269305  all_samples    missense   \n",
       "650          TP53--P53_tetramer-ENST00000269305  all_samples  truncating   \n",
       "651                  TP53--TAD2-ENST00000269305  all_samples    missense   \n",
       "652                  TP53--TAD2-ENST00000269305  all_samples  truncating   \n",
       "\n",
       "     mutations       dnds        pvalue      lower      upper  exon_rank  \n",
       "0           16   0.715234  1.663548e-01   0.416336   1.138208        9.0  \n",
       "1           49  28.999998  0.000000e+00  20.546647  40.289061        9.0  \n",
       "2           47   2.300000  1.251698e-06   1.679419   3.075757       10.0  \n",
       "3           46  22.810938  0.000000e+00  16.016163  31.845361       10.0  \n",
       "4           62   1.689844  2.888441e-04   1.283827   2.182397       11.0  \n",
       "..         ...        ...           ...        ...        ...        ...  \n",
       "648         84   7.395313  0.000000e+00   5.734699   9.430225        NaN  \n",
       "649         30   1.459155  5.814385e-02   0.986482   2.073169        NaN  \n",
       "650         18   4.387500  8.940697e-07   2.596850   6.973560        NaN  \n",
       "651         28   1.758984  8.130968e-03   1.168019   2.539624        NaN  \n",
       "652          8   4.927332  5.791783e-04   2.140323   9.941370        NaN  \n",
       "\n",
       "[653 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1517215/2029447566.py:28: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  o3d_pdb_tool_df = pd.read_table(f\"{o3d_annotations}/pdb_tool_df.tsv\")\n",
      "/tmp/ipykernel_1517215/2029447566.py:29: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  disorder_df = pd.read_table(f\"{o3d_datasets}/confidence.tsv\")\n"
     ]
    }
   ],
   "source": [
    "# Per-site selection\n",
    "site_selection = os.path.join(deepcsa_run_dir, f\"sitecomparison/{run_name}.aminoacid.comparison.tsv.gz\")\n",
    "site_selection = pd.read_table(site_selection).rename(\n",
    "    columns={\"OBS/EXP\" : \"Selection\"}).drop(\n",
    "    columns=[\"OBSERVED_MUTS\", \"EXPECTED_MUTS\"])\n",
    "site_selection.loc[site_selection.Selection < 0, \"Selection\"] = 0\n",
    "site_selection = site_selection[site_selection.Protein_position != \"-\"].reset_index(drop=True)\n",
    "site_selection.Protein_position = site_selection.Protein_position.astype(int)\n",
    "print(\"> site_selection:\")\n",
    "display(site_selection)\n",
    "\n",
    "# Exon selection\n",
    "exon_selection = os.path.join(deepcsa_run_dir, f\"omega/output_mle.{run_name}.tsv\")                    # exon selection + confidence (missense and truncating)\n",
    "exon_selection = pd.read_table(exon_selection)\n",
    "exon_selection = exon_selection[exon_selection.gene.str.contains(\"--\")]\n",
    "exon_selection = exon_selection[exon_selection.impact.isin([\"missense\", \"truncating\"])].reset_index(drop=True)\n",
    "exon_selection = exon_selection.merge(\n",
    "    exons_depth.rename(columns={\"EXON_ID\" : \"gene\", \"EXON_RANK\" : \"exon_rank\"}).dropna()[[\"gene\", \"exon_rank\"]].drop_duplicates().reset_index(drop=True), \n",
    "    how=\"left\").sort_values([\"gene\", \"exon_rank\", \"impact\"]).reset_index(drop=True)\n",
    "print(\"> exon_selection:\")\n",
    "display(exon_selection)\n",
    "\n",
    "o3d_seq_df = pd.read_table(f\"{o3d_datasets}/seq_for_mut_prob.tsv\")\n",
    "\n",
    "o3d_alt_seq_df = pd.read_table(f\"{o3d_alt_datasets}/seq_for_mut_prob.tsv\")\n",
    "\n",
    "o3d_annot_df = pd.read_table(f\"{o3d_annotations}/uniprot_feat.tsv\")\n",
    "o3d_pdb_tool_df = pd.read_table(f\"{o3d_annotations}/pdb_tool_df.tsv\")\n",
    "disorder_df = pd.read_table(f\"{o3d_datasets}/confidence.tsv\")\n",
    "o3d_df = f\"{deepcsa_run_dir}/oncodrive3d/run/{run_name}/{run_name}.3d_clustering_pos.csv\"\n",
    "o3d_df = pd.read_csv(o3d_df)[[\"Gene\", \"Pos\", \"Score\", \"Score_obs_sim\", \"pval\", \"C\", \"C_ext\"]]\n",
    "\n",
    "# Domain annotations\n",
    "domain = pd.read_table(f\"{o3d_annotations}/uniprot_feat.tsv\")\n",
    "# domain = domain[domain.Ens_Transcr_ID.isin(maf.Ens_transcript_ID.unique())]\n",
    "domain = domain[(domain.Type == \"DOMAIN\") & (domain.Evidence == \"Pfam\")].reset_index(drop=True)\n",
    "\n",
    "# Domain selection\n",
    "domain_selection = pd.read_table(os.path.join(deepcsa_run_dir, f\"omegagloballoc/output_mle.{run_name}.global_loc.tsv\"))\n",
    "domain_selection = domain_selection[domain_selection.gene.str.contains(\"--\")]\n",
    "domain_selection = domain_selection[domain_selection[\"impact\"].isin([\"missense\", \"truncating\"])]\n",
    "domain_selection[\"selection_id\"] = domain_selection.gene.str.split(\"--\").apply(lambda x: x[1].split(\"-\")[0])\n",
    "domain_selection = domain_selection[domain_selection.selection_id.isin(domain.Description.values)].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fd7c0d3",
   "metadata": {},
   "source": [
    "### Cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c980c817",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HARTWIG_WGS_BLCA_2023\n",
      "CBIOP_WXS_BLCA_VALLEN_2018\n",
      "CBIOP_WXS_BLCA_BGI\n",
      "CBIOP_WXS_BLCA_DFARBER_MSKCC_2014\n",
      "CBIOP_WXS_BLCA_UTUC_IGBMC_2021\n",
      "ICGC_WXS_BLCA_BLCA_CN_2019\n",
      "PCAWG_WGS_BLADDER_TCC\n",
      "TCGA_WXS_BLCA\n"
     ]
    }
   ],
   "source": [
    "# Cancer\n",
    "path_intogen = \"/data/bbg/projects/clustering_3d/o3d_analysys/datasets/input/cancer_202404\"\n",
    "cancer_cohort_df = pd.read_table(f\"{path_intogen}/cohorts.tsv\")\n",
    "path_cancer_vep_all = f\"{path_intogen}/vep/\"\n",
    "\n",
    "# Cancer mut\n",
    "selected_cancer_type = \"BLCA\"\n",
    "blca_cancer_df = get_cancer_maf_all(\n",
    "    cancer_cohort_df[cancer_cohort_df[\"CANCER_TYPE\"] == selected_cancer_type],\n",
    "    path_cancer_vep_all,\n",
    "    lst_genes=gene_order,\n",
    "    only_protein_pos=True,\n",
    "    truncating=True\n",
    "    )\n",
    "blca_cancer_cnsq_count = blca_cancer_df.groupby(['Gene', 'Consequence']).size().reset_index(name='Count')\n",
    "blca_cancer_df = blca_cancer_df.groupby(['Gene', 'Consequence', 'Pos']).size().reset_index(name='Count')\n",
    "\n",
    "# Cancer 3D clustering\n",
    "o3d_cancer_path = \"/data/bbg/projects/clustering_3d/o3d_analysys/datasets/output/cancer_202404/o3d_output/human_mane_raw/run_2024-07-01_16-04-14\"\n",
    "o3d_cancer_df = get_o3d_data_cancer(cancer_cohort_df[cancer_cohort_df[\"CANCER_TYPE\"] == selected_cancer_type], o3d_cancer_path, o3d_seq_df, gene_order)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dd518d56",
   "metadata": {},
   "outputs": [],
   "source": [
    "warnings.filterwarnings(\"ignore\", message=\"DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns.\", category=UserWarning)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "65c1a738-c81e-4367-931b-524515b9c0bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "warnings.filterwarnings(\"ignore\", message=\"Warning: converting a masked element to nan.\", category=UserWarning)\n",
    "warnings.filterwarnings(\"ignore\", message=\"FixedFormatter should only be used together with FixedLocator\", category=UserWarning)\n",
    "\n",
    "plot_pars = {\n",
    "    \"fsize\"                     : (12.3, 8.3),\n",
    "    \"hspace\"                    : 0.1,                    # General space between all tracks\n",
    "    \"track_title_x_coord\"       : 0.83,                   # x-coordinate (respect to protein len) for track txt title\n",
    "    \"track_title_fontsize\"      : plot_fontsizes_general_config[\"axis_labels_fontsize\"],\n",
    "    \"ylabel_fontsize\"           : plot_fontsizes_general_config[\"axis_labels_fontsize\"],\n",
    "    \"xlabel_fontsize\"           : plot_fontsizes_general_config[\"axis_labels_fontsize\"],\n",
    "    \"ylabel_pad\"                : 20,\n",
    "    \"ticksize\"                  : plot_fontsizes_general_config[\"tick_labels_fontsize\"],\n",
    "    \"legend_fontsize\"           : plot_fontsizes_general_config[\"legend_labels_fontsize\"],\n",
    "    \"legend_frameon\"            : False,\n",
    "    \"j_margin\"                  : 15,\n",
    "    \"i_margin\"                  : 45,\n",
    "    \"c_margin\"                  : 0,\n",
    "    \"y_labels_coord\"            : [-0.05, 0.5],\n",
    "    \"cnsq_bbox_to_anchor\"       : [1.1123, 0.84],\n",
    "    \"o3d_bbox_to_anchor\"        : [1.27, 1.35],\n",
    "    \"saturation_bbox_to_anchor\" : [1.086, 8.5, 1.4, 5.05],\n",
    "    \"depth_bbox_to_anchor\"      : [1.086, 6.1, 1.4, 5.05],\n",
    "    \"depth_res_bbox_to_anchor\"  : [1.086, -2, 1.4, 5.05],\n",
    "    \"depth_0_bbox_to_anchor\"    : [1.246, -2], # 3.9 5.9\n",
    "    \"sse_bbox_to_anchor\"        : [1.2645, 2.4],\n",
    "    \"domain_sel_bbox_to_anchor\" : [1.086, -3, 1.4, 4.8],\n",
    "    \"domain_y_bbox_to_anchor\"   : 5,\n",
    "    \"domain_x_bbox_to_anchor\"   : {\n",
    "                                \"ARID1A\" : 1.25,\n",
    "                                \"KDM6A\"  : 1.245,\n",
    "                                \"KMT2D\"  : 1.255,\n",
    "                                \"EP300\"  : 1.267,\n",
    "                                \"CREBBP\" : 1.267,\n",
    "                                \"NOTCH2\" : 1.241,\n",
    "                                \"KMT2C\"  : 1.255\n",
    "                                },\n",
    "    \"legend_depth_fontsize\"     : plot_fontsizes_general_config[\"legend_labels_fontsize\"],\n",
    "    \"txt_fontsize\"              : plot_fontsizes_general_config[\"custom_text_fontsize\"],\n",
    "    \"len_txt_thr\"               : 2400,                   \n",
    "    \"dpi\"                       : 300,\n",
    "    \"colors\"                    : {\n",
    "                                \"truncating\"  : \"#FB8E6F\",  # group nonsense + splicing\n",
    "                                \"nonsense\"    : \"#FB8E6F\",  \n",
    "                                \"synonymous\"  : \"#ACECBD\", \n",
    "                                \"missense\"    : \"#FBD180\", \n",
    "                                \"o3d_score\"   : \"#6DBDCC\",   \n",
    "                                \"o3d_cluster\" : \"#DAF0F8\", # \"skyblue\",  \n",
    "                                \"indel\"       : \"#ECC4F7\", \n",
    "                                \"splicing\"    : \"#A1C5DF\", \n",
    "                                \"vlines\"      : \"lightgray\",\n",
    "                                \"plddt_pacc\"  : \"#f7f3ff\",\n",
    "                                \"site_selection\" : \"#b3b3ff\",\n",
    "                                \"site_selection_hits\" : \"#4d4dff\",               # #4d4dff #1a1aff\n",
    "                                \"exon_selection\" : {\n",
    "                                                    \"missense\" : \"#FBD180\",\n",
    "                                                    \"truncating\" : \"#FB8E6F\"\n",
    "                                                    },\n",
    "                                \"depth_below_thr\" : \"#d9d9d9\"\n",
    "                                },\n",
    "    \"sse_colors\"                : {\n",
    "                                'Coil'   : \"#D5E8D4\", \n",
    "                                'Helix'  : \"#F7CAC9\", \n",
    "                                'Ladder' : \"#A7C7E7\"\n",
    "                                },\n",
    "    \"sse_lw\"                    : {\n",
    "                                \"KMT2D\"  : 1,\n",
    "                                \"EP300\"  : 1,\n",
    "                                \"ARID1A\" : 1,\n",
    "                                \"CREBBP\" : 1,\n",
    "                                \"NOTCH2\" : 1,\n",
    "                                \"KMT2C\"  : 1,\n",
    "                                \"STAG2\"  : 1,\n",
    "                                \"RB1\"    : 1,\n",
    "                                \"RBM10\"  : 1,\n",
    "                                \"KDM6A\"  : 1,\n",
    "                                \"TP53\"   : 2.2,\n",
    "                                \"FGFR3\"  : 1,\n",
    "                                \"CDKN1A\" : 4,\n",
    "                                \"FOXQ1\"  : 2.2,\n",
    "                                \"PIK3CA\" : 1,\n",
    "                                \"TERT\"   : 1,\n",
    "                                }\n",
    "}\n",
    "\n",
    "\n",
    "def get_exon_depth_saturation(gene_depth, gene_mut, dna=False):\n",
    "    \n",
    "    # Exon average depth\n",
    "    gene_depth = gene_exons_depth.copy()\n",
    "    exon_depth = gene_depth.groupby(\"EXON_RANK\").apply(\n",
    "        lambda x: 0 if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED)).reset_index().rename(columns = {0 : \"DEPTH\"})\n",
    "    exon_depth[\"START_PROT_POS\"] = gene_depth.groupby(\"EXON_RANK\").apply(lambda x: x.PROT_POS.min())\n",
    "\n",
    "    # Exon saturation\n",
    "    if dna:\n",
    "        exon_depth[\"COVERED\"] = gene_depth.groupby(\"EXON_RANK\").apply(lambda x: sum(x.COVERED)).values\n",
    "        gene_depth[\"MUTATED\"] = gene_depth.DNA_POS.isin(gene_mut.DNA_POS.unique()).astype(int)\n",
    "        exon_depth[\"MUTATED\"] = gene_depth.groupby(\"EXON_RANK\").apply(lambda x: sum(x.MUTATED)).values\n",
    "        exon_depth[\"SATURATION\"] = exon_depth[\"MUTATED\"] / exon_depth[\"COVERED\"]\n",
    "    else:\n",
    "        \n",
    "        # If an amino acids from the same codon are in two exons, consider the position only in the second one \n",
    "        # (but look at booth to know if the residue is in the panel and if it is mutated)\n",
    "        exon_depth_prot = gene_depth.groupby(\"PROT_POS\").apply(lambda x: (x.COVERED.max(), x.EXON_RANK.max())).reset_index()\n",
    "        exon_depth_prot[[\"COVERED\", \"EXON_RANK\"]] = pd.DataFrame(exon_depth_prot[0].tolist(), index=exon_depth_prot.index)\n",
    "        exon_depth_prot = exon_depth_prot.drop(columns=[0])\n",
    "\n",
    "        exon_depth[\"COVERED\"] = exon_depth_prot.groupby(\"EXON_RANK\").apply(lambda x: sum(x.COVERED)).values\n",
    "        exon_depth_prot[\"MUTATED\"] = exon_depth_prot.PROT_POS.isin(gene_mut.Pos.unique()).astype(int)\n",
    "        exon_depth[\"MUTATED\"] = exon_depth_prot.groupby(\"EXON_RANK\").apply(lambda x: sum(x.MUTATED)).values\n",
    "        exon_depth[\"SATURATION\"] = exon_depth[\"MUTATED\"] / exon_depth[\"COVERED\"]\n",
    "    \n",
    "    check_mutated_masked = exon_depth_prot[(exon_depth_prot[\"MUTATED\"] == 1) & (exon_depth_prot[\"COVERED\"] == 0)]\n",
    "    check_mutated_masked[\"GENE\"] = gene_depth.GENE.unique()[0]\n",
    "    if len(check_mutated_masked) > 0:\n",
    "        display(check_mutated_masked)\n",
    "    \n",
    "    return exon_depth\n",
    "\n",
    "\n",
    "def get_exon_mid_prot_pos(exon_info, prot_len):\n",
    "\n",
    "    lst_end_pos = []\n",
    "    for i in range(len(exon_info)):\n",
    "        exon = exon_info.iloc[i]\n",
    "        start_pos = int(exon.START_PROT_POS)\n",
    "        end_pos = int(exon_info.iloc[i+1].START_PROT_POS) if i < len(exon_info) -1 else prot_len\n",
    "        lst_end_pos.append(end_pos)\n",
    "    exon_info[\"END_PROT_POS\"] = lst_end_pos\n",
    "    exon_info[\"MID_PROT_POS\"] = (exon_info[\"START_PROT_POS\"] + exon_info[\"END_PROT_POS\"]) / 2\n",
    "    \n",
    "    return exon_info\n",
    "\n",
    "\n",
    "def get_res_coverage(dna_prot_df):\n",
    "    \n",
    "    res_depth = dna_prot_df.dropna(subset=[\"PROT_POS\"])[[\"PROT_POS\", \"COVERED\", \"DEPTH\"]].reset_index(drop=True)\n",
    "    res_depth = res_depth.groupby(\"PROT_POS\").mean().reset_index()\n",
    "    \n",
    "    return res_depth\n",
    "\n",
    "\n",
    "def add_consecutive_numbers(nums, max_n):\n",
    "\n",
    "    result = []\n",
    "    for i in range(len(nums)):\n",
    "        result.append(nums[i])\n",
    "        # Check if the current number is the start of a consecutive sequence\n",
    "        if i < len(nums) - 1 and nums[i] + 1 != nums[i + 1]:\n",
    "            result.append(nums[i] + 1)\n",
    "            \n",
    "    # Add the last consecutive number after the final element\n",
    "    result.append(nums[-1] + 1)\n",
    "    \n",
    "    return result\n",
    "\n",
    "\n",
    "def where_plus(condition):\n",
    "    \"\"\"\n",
    "    Util function to extend the color of mpl filling to the next position.\n",
    "    \"\"\"\n",
    "    \n",
    "    ix = np.where(condition)[0]\n",
    "    \n",
    "    if len(ix) > 0:\n",
    "        ix = add_consecutive_numbers(ix, max_n=len(condition))\n",
    "        if len(condition) in ix:\n",
    "            ix.remove(len(condition))\n",
    "        boolean_vector = np.zeros(len(condition), dtype=bool)\n",
    "        boolean_vector[ix] = True\n",
    "\n",
    "        return pd.Series(boolean_vector)\n",
    "    \n",
    "    else:\n",
    "        return condition\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1038025c-755c-4e06-9c17-fc440400bc2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# mut_count_horizontal_barplot(\n",
    "#         gene_mut_cnsq_count, plot_pars[\"colors\"], \n",
    "#         # title=f\"{gene}\\nUrothelium\", \n",
    "#         title=f\"\", \n",
    "#         save=True, \n",
    "#         filename=f\"plots/fig4bc/side/cnsq_count/{gene}.normal.stacked_horizontal.png\"\n",
    "#         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3508a5e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_count_track(\n",
    "    count_df, \n",
    "    gene_len,\n",
    "    axes, \n",
    "    colors_dict,\n",
    "    ax=0, \n",
    "    negative=False, \n",
    "    label_pos_track=None,\n",
    "    label_neg_track=None,\n",
    "    ymargin=None,\n",
    "    alpha=1,\n",
    "    indel=False,\n",
    "    n_batches = 10\n",
    "    ):\n",
    "\n",
    "    pos_df = pd.DataFrame({\"Pos\" : range(1, gene_len+1)})\n",
    "\n",
    "\n",
    "    # Shuffle the data and split into batches\n",
    "    shuffled_df = count_df.sample(frac=1, random_state=42).reset_index(drop=True)\n",
    "    batches = np.array_split(shuffled_df, n_batches)\n",
    "\n",
    "    legend_list = []\n",
    "    pos_df = pd.DataFrame({\"Pos\" : range(1, gene_len+1)})\n",
    "    \n",
    "    for batch_idx, batch in enumerate(batches):\n",
    "        for cnsq in ['indel', 'truncating', 'nonsense', 'missense', 'synonymous', 'splicing']:\n",
    "            \n",
    "            if (indel == False and cnsq == \"indel\") or cnsq not in batch[\"Consequence\"].unique():\n",
    "                continue\n",
    "        \n",
    "            count_cnsq_df = batch[batch[\"Consequence\"] == cnsq].reset_index(drop=True)\n",
    "            count_cnsq_df = pos_df.merge(count_cnsq_df, on=\"Pos\", how=\"left\")\n",
    "\n",
    "            if negative:\n",
    "                axes[ax].vlines(count_cnsq_df[\"Pos\"], ymin=-count_cnsq_df[\"Count\"], ymax=0, lw=1, zorder=1, alpha=0.5, color=colors_dict[\"vlines\"])\n",
    "                axes[ax].scatter(count_cnsq_df[\"Pos\"], -count_cnsq_df[\"Count\"], s=20, color='white', zorder=3, lw=0.11, ec=\"none\") # To cover the overlapping needle top part\n",
    "                if cnsq not in legend_list:\n",
    "                    axes[ax].scatter(\n",
    "                        count_cnsq_df[\"Pos\"].values, -count_cnsq_df[\"Count\"].values, zorder=4, \n",
    "                        alpha=alpha, lw=0.1, ec=\"none\", s=20, label=cnsq.capitalize(), color=colors_dict[cnsq]\n",
    "                        )\n",
    "                    legend_list.append(cnsq)\n",
    "\n",
    "                else:\n",
    "                    axes[ax].scatter(\n",
    "                        count_cnsq_df[\"Pos\"].values, -count_cnsq_df[\"Count\"].values, zorder=4, \n",
    "                        alpha=alpha, lw=0.1, ec=\"none\", s=20, color=colors_dict[cnsq]\n",
    "                        )\n",
    "\n",
    "            else:\n",
    "                axes[ax].vlines(count_cnsq_df[\"Pos\"], ymin=0, ymax=count_cnsq_df[\"Count\"], lw=1, zorder=1, alpha=0.5, color=colors_dict[\"vlines\"])\n",
    "                axes[ax].scatter(count_cnsq_df[\"Pos\"], count_cnsq_df[\"Count\"], s=20, color='white', zorder=3, lw=0.1, ec=\"none\") # To cover the overlapping needle top part\n",
    "                if cnsq not in legend_list:\n",
    "                    axes[ax].scatter(\n",
    "                        count_cnsq_df[\"Pos\"].values, count_cnsq_df[\"Count\"].values, zorder=4, \n",
    "                        alpha=alpha, lw=0.1, ec=\"none\", s=20, label=cnsq.capitalize(), color=colors_dict[cnsq]\n",
    "                        )\n",
    "                    legend_list.append(cnsq)\n",
    "                else:\n",
    "                    axes[ax].scatter(\n",
    "                        count_cnsq_df[\"Pos\"].values, count_cnsq_df[\"Count\"].values, zorder=4, \n",
    "                        alpha=alpha, lw=0.1, ec=\"none\", s=20, color=colors_dict[cnsq]\n",
    "                        )\n",
    "\n",
    "\n",
    "def get_domain_selection_gene(df, domain_gene, gene, sort_by_impact=True):\n",
    "    \n",
    "    domain_selection_gene = df.copy()\n",
    "    domain_selection_gene = df[df.gene.str.startswith(gene)]\n",
    "    domain_selection_gene = domain_selection_gene.merge(domain_gene.rename(columns={\"Description\": \"selection_id\"})[[\"selection_id\", \"Begin\", \"End\"]], how=\"inner\")\n",
    "    domain_selection_gene = domain_selection_gene.sort_values([\"Begin\", \"impact\"]).reset_index(drop=True)\n",
    "    if sort_by_impact:\n",
    "        average_dnds = domain_selection_gene.groupby(\"selection_id\").apply(lambda x: x[\"dnds\"].mean()).to_dict()\n",
    "        domain_selection_gene[\"average_dnds\"] = domain_selection_gene[\"selection_id\"].map(average_dnds)\n",
    "        domain_selection_gene = domain_selection_gene.sort_values([\"average_dnds\", \"impact\"], ascending=[False, True]).reset_index(drop=True)\n",
    "    domain_selection_gene[\"x_pos\"] = domain_selection_gene[\"selection_id\"].astype(\"category\").cat.set_categories(domain_selection_gene[\"selection_id\"].unique()).cat.codes\n",
    "    domain_selection_gene[\"x_pos\"] += np.where(domain_selection_gene[\"impact\"] == \"missense\", -0.15, 0.15)\n",
    "    \n",
    "    return domain_selection_gene\n",
    "    \n",
    "    \n",
    "def color_bar(\n",
    "            axes, \n",
    "            ax, \n",
    "            cmap,\n",
    "            norm,\n",
    "            general_ticks=False, \n",
    "            bbox_to_anchor=[1.086, -2, 1.4, 5.05],\n",
    "            label='Cbar label',\n",
    "            fontsize=plot_fontsizes_general_config[\"axis_labels_fontsize\"],\n",
    "            tick_size=10.25,\n",
    "            tick_span=0,\n",
    "            borderpad=10\n",
    "            ):\n",
    "            \n",
    "            # Color bar\n",
    "            sm = cm.ScalarMappable(cmap=cmap, norm=norm)\n",
    "            sm.set_array([])\n",
    "            inset_ax = inset_axes(\n",
    "                axes[ax], width=\"10%\", height=\"10%\", loc='center left', \n",
    "                bbox_to_anchor=bbox_to_anchor,\n",
    "                bbox_transform=axes[ax].transAxes, borderpad=borderpad\n",
    "                )\n",
    "            cbar = plt.colorbar(sm, cax=inset_ax, orientation='horizontal')  \n",
    "            cbar.set_label(label, fontsize=fontsize\n",
    "            #, labelpad=7\n",
    "            )  \n",
    "            cbar.ax.xaxis.set_label_position('top') \n",
    "\n",
    "            cbar.ax.tick_params(labelsize=tick_size)    \n",
    "            span = norm.vmax - norm.vmin\n",
    "            left_tick = norm.vmin + tick_span * span\n",
    "            # mid_tick = norm.vmin + span / 2\n",
    "            right_tick = norm.vmax - tick_span * span\n",
    "            tick_values = [left_tick,\n",
    "                            # mid_tick,\n",
    "                            right_tick]\n",
    "            cbar.set_ticks(tick_values)\n",
    "            if general_ticks:\n",
    "                cbar.set_ticklabels([tick for tick in [\"Low\", \"High\"]]) \n",
    "\n",
    "\n",
    "def get_equivalent_alt_prot_id(gene, seq_df, alt_seq_df):\n",
    "    \n",
    "    seq = seq_df[seq_df[\"Gene\"] == gene].Seq.unique()[0]\n",
    "    alt_seq = alt_seq_df[alt_seq_df[\"Gene\"] == gene].Seq.unique()[0]\n",
    "    \n",
    "    if seq == alt_seq:\n",
    "        return alt_seq_df[alt_seq_df[\"Gene\"] == gene].Uniprot_ID.unique()[0]    \n",
    "\n",
    "\n",
    "def avg_per_pos_ddg(mut_df, ddg_prot):\n",
    "    \"\"\"\n",
    "    Compute per-position average stability change upon mutations (DDG).\n",
    "    \"\"\"\n",
    "    \n",
    "    mut_df = mut_df.copy()\n",
    "    mut_df = mut_df[mut_df[\"Consequence\"] == \"missense\"].reset_index(drop=True)\n",
    "    \n",
    "    ddg_vec = np.repeat(0., len(ddg_prot.keys()))\n",
    "    for pos, df in mut_df.groupby('Pos'):\n",
    "        pos = str(pos)\n",
    "        obs_mut = df.Alt_amino_acid\n",
    "        if pos in ddg_prot:\n",
    "            ddg_pos = ddg_prot[pos]\n",
    "            ddg_pos = np.mean([ddg_pos[mut] for mut in obs_mut])\n",
    "            ddg_vec[int(pos)-1] = ddg_pos\n",
    "            \n",
    "    return pd.DataFrame({\"Pos\": ddg_prot.keys(), \"Stability_change\": ddg_vec}).astype({\"Pos\": int, \"Stability_change\": float})\n",
    "\n",
    "\n",
    "def get_ddg_gene(gene, gene_mut, o3d_annotations, o3d_seq_df, o3d_alt_seq_df):\n",
    "\n",
    "    uni_id = o3d_seq_df[o3d_seq_df[\"Gene\"] == gene].Uniprot_ID.unique()[0]    \n",
    "    ddg_path = os.path.join(o3d_annotations, \"stability_change\", f\"{uni_id}_ddg.json\")\n",
    "    alt_uni_id = get_equivalent_alt_prot_id(gene, o3d_seq_df, o3d_alt_seq_df)\n",
    "    alt_ddg_path = os.path.join(o3d_annotations, \"stability_change\", f\"{alt_uni_id}_ddg.json\")\n",
    "    if os.path.isfile(ddg_path):\n",
    "        ddg_gene = json.load(open(ddg_path))\n",
    "        ddg_gene = avg_per_pos_ddg(gene_mut, ddg_gene)\n",
    "    elif isinstance(alt_uni_id, str) and os.path.isfile(alt_ddg_path):\n",
    "        ddg_gene = json.load(open(alt_ddg_path))\n",
    "        ddg_gene = avg_per_pos_ddg(gene_mut, ddg_gene)\n",
    "    else:\n",
    "        ddg_gene = None\n",
    "        \n",
    "    return ddg_gene\n",
    "\n",
    "\n",
    "def plot_gene_selection(mut_count_df,\n",
    "                        mut_df,\n",
    "                        o3d_df,\n",
    "                        pdb_tool_df,\n",
    "                        domain_df,\n",
    "                        coverage_df,\n",
    "                        site_selection_df,\n",
    "                        exon_selection_df,\n",
    "                        domain_selection_df,\n",
    "                        gene,\n",
    "                        max_depth,\n",
    "                        gene_len,\n",
    "                        plot_pars,\n",
    "                        title,\n",
    "                        ddg_df=None,\n",
    "                        mut_cancer_df=None,\n",
    "                        o3d_cancer_df=None,\n",
    "                        thr_selection=0.0001,\n",
    "                        lst_tracks=[\"Mut_count\", \"Site_selection\", \"Res_depth\", \"Domain\"],\n",
    "                        default_track_order=False,\n",
    "                        save=False,\n",
    "                        filename=\"gene_selection.png\"):\n",
    "    \n",
    "    h_ratios_dict = {\n",
    "        \"Mut_count\": 0.20,\n",
    "        \"3d_clustering\": 0.15,\n",
    "        \"Site_selection\": 0.15,\n",
    "        \"Exon_selection\": 0.15,\n",
    "        \"Exon_saturation\": 0.03,\n",
    "        \"Exon_depth\": 0.03,\n",
    "        \"Res_depth\": 0.03,\n",
    "        \"Solvent_accessibility\": 0.05,\n",
    "        \"Stability_change\": 0.05,\n",
    "        \"Secondary_structure\": 0.03,\n",
    "        \"Domain\": 0.03,\n",
    "        \"Domain_selection\": 0.035,\n",
    "        \"Cancer_3d_clustering\": 0.15,\n",
    "        \"Cancer_mut_count\": 0.25\n",
    "    }\n",
    "\n",
    "    lst_tracks = [track.capitalize() for track in lst_tracks]\n",
    "    if default_track_order:\n",
    "        lst_tracks = [track for track in h_ratios_dict.keys() if track in lst_tracks] \n",
    "    h_ratios = [h_ratios_dict[track] for track in lst_tracks]\n",
    "    \n",
    "    fig, axes = plt.subplots(\n",
    "        len(h_ratios), 1, \n",
    "        figsize=plot_pars[\"fsize\"], \n",
    "        sharex=True, \n",
    "        gridspec_kw={'hspace': plot_pars[\"hspace\"], \n",
    "                    'height_ratios': h_ratios}\n",
    "        )\n",
    "    \n",
    "    n1_max = np.max(mut_count_df[\"Count\"])\n",
    "    print(n1_max)\n",
    "    n2_max = np.nan\n",
    "    n_max = np.nanmax((n1_max, n2_max))\n",
    "    \n",
    "    n1_3d_max = np.max(o3d_df[\"O3D_score\"])\n",
    "    n2_3d_max = np.max(o3d_cancer_df[\"O3D_score\"]) if isinstance(o3d_cancer_df, pd.DataFrame) else np.nan\n",
    "    n2_3d_max = 0 if pd.isnull(n2_3d_max) else n2_3d_max\n",
    "    n_3d_max = np.max((n1_3d_max, n2_3d_max))\n",
    "    \n",
    "    \n",
    "    # Track mut\n",
    "    # ---------\n",
    "    if \"Mut_count\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Mut_count\")\n",
    "        \n",
    "        plot_count_track(mut_count_df, gene_len=gene_len, axes=axes, ax=ax, colors_dict=plot_pars[\"colors\"])\n",
    "        axes[ax].set_ylabel('Mutations', rotation=0, va='center', ha='right', fontsize=plot_pars[\"ylabel_fontsize\"])\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        \n",
    "        axes[ax].set_ylim(0 - n_max/plot_pars[\"i_margin\"], n_max + n_max/plot_pars[\"j_margin\"])\n",
    "        axes[ax].yaxis.set_major_locator(MaxNLocator(integer=True, nbins=5))\n",
    "        axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        \n",
    "    \n",
    "    # Track site selection\n",
    "    # --------------------\n",
    "    if \"Site_selection\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Site_selection\")\n",
    "        \n",
    "        site_selection_df = pd.DataFrame({\"Protein_position\" : np.arange(gene_len)+1}).merge(site_selection_df, how=\"left\")\n",
    "        axes[ax].plot(\n",
    "            site_selection_df.Protein_position, site_selection_df.Selection, zorder=1, \n",
    "            color=plot_pars[\"colors\"][\"site_selection\"], lw=0.5\n",
    "            ) \n",
    "        axes[ax].fill_between(\n",
    "            site_selection_df.Protein_position, 0, site_selection_df.Selection, \n",
    "            color=plot_pars[\"colors\"][\"site_selection\"], alpha=0.4, zorder=0, lw=0.5\n",
    "            )\n",
    "        \n",
    "        site_selection_hits_df = pd.DataFrame({\"Protein_position\" : np.arange(gene_len)+1}).merge(\n",
    "            site_selection_df[site_selection_df[\"p_value\"] < thr_selection].reset_index(drop=True), how=\"left\")\n",
    "        axes[ax].fill_between(\n",
    "            site_selection_hits_df.Protein_position, 0, site_selection_hits_df.Selection, \n",
    "            color=plot_pars[\"colors\"][\"site_selection_hits\"], alpha=1, zorder=2, lw=1, label=\"Significant\"\n",
    "            )\n",
    "\n",
    "        n_max = np.max(site_selection_df.Selection)\n",
    "        ax_ylim_min = 0 if plot_pars[\"c_margin\"] == 0 else 0 - n_max/c_margin\n",
    "        ax_ylim_max = n_max + n_max/plot_pars[\"j_margin\"]\n",
    "        axes[ax].set_ylim(ax_ylim_min, ax_ylim_max)\n",
    "        \n",
    "        axes[ax].set_ylabel('Site selection\\nper residue', rotation=0, va='center', ha='right', fontsize=plot_pars[\"ylabel_fontsize\"])  # 'Site selection\\n(dN/dS)    '\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        ax_legend = axes[ax].legend(frameon = False)\n",
    "        # ax_legend.get_frame().set_linewidth(0.5)\n",
    "    \n",
    "    \n",
    "    # Track 3D clustering\n",
    "    # -------------------\n",
    "    if \"3d_clustering\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"3d_clustering\")\n",
    "\n",
    "        axes[ax].plot(np.array(range(prot_len-1))+1, o3d_df[\"O3D_score\"], zorder=2, color=plot_pars[\"colors\"][\"o3d_score\"], lw=1, label=\"Clustering score\")                       \n",
    "        axes[ax].fill_between(o3d_df['Pos'], 0, n_3d_max, where=(o3d_df['Cluster'] == 1), \n",
    "                            color=plot_pars[\"colors\"][\"o3d_cluster\"], alpha=1, label='Cluster', zorder=0, lw=1.5)\n",
    "\n",
    "        axes[ax].set_ylabel('3D-clustering\\n(missense mutations)', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "\n",
    "        axes[ax].set_ylim(0 - n_3d_max/(plot_pars[\"j_margin\"] * 5), n_3d_max + n_3d_max/plot_pars[\"j_margin\"])\n",
    "        axes[ax].yaxis.set_major_locator(MaxNLocator(integer=True, nbins=3))\n",
    "        axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        \n",
    "        ax_legend = axes[ax].legend()\n",
    "        ax_legend.get_frame().set_linewidth(0.5)\n",
    "        # axes[ax].legend(fontsize=plot_pars[\"legend_fontsize\"], frameon=plot_pars[\"legend_frameon\"], \n",
    "        #                 bbox_to_anchor=plot_pars[\"o3d_bbox_to_anchor\"], title = \"3D-clustering\", \n",
    "        #                 title_fontsize=plot_pars[\"legend_fontsize\"])    \n",
    "    \n",
    "\n",
    "    # Track exon selection\n",
    "    # ---------------------\n",
    "    if \"Exon_selection\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Exon_selection\")\n",
    "        \n",
    "        exon_coverage = get_exon_depth_saturation(coverage_df, mut_df)\n",
    "        exon_coverage = get_exon_mid_prot_pos(exon_coverage, gene_len)\n",
    "        exon_info = exon_coverage[[\"EXON_RANK\", \"START_PROT_POS\", \"END_PROT_POS\", \"MID_PROT_POS\"]]\n",
    "        exon_selection_df = exon_selection_df.merge(exon_info)\n",
    "        custom_legend = []\n",
    "        \n",
    "        for impact in [\"missense\", \"truncating\"]:\n",
    "            \n",
    "            exon_selection_impact = exon_selection_df[exon_selection_df[\"impact\"] == impact]\n",
    "            exon_selection_impact_hits = exon_selection_impact[exon_selection_impact[\"pvalue\"] < thr_selection].reset_index(drop=True)\n",
    "            axes[ax].scatter(\n",
    "                exon_selection_impact[\"MID_PROT_POS\"], exon_selection_impact[\"dnds\"], \n",
    "                zorder=3, color=plot_pars[\"colors\"][\"exon_selection\"][impact], s=20, lw=0.1, ec=\"black\"\n",
    "                )\n",
    "            axes[ax].scatter(\n",
    "                exon_selection_impact_hits[\"MID_PROT_POS\"], exon_selection_impact_hits[\"dnds\"], \n",
    "                zorder=3, color=plot_pars[\"colors\"][\"exon_selection\"][impact], s=20, lw=1, ec=\"black\"\n",
    "                )\n",
    "            \n",
    "            exon_selection_impact = exon_selection_impact.merge(exon_info, how=\"outer\").fillna(0).sort_values(\"EXON_RANK\").reset_index(drop=True)\n",
    "            exon_selection_impact[\"impact\"] = impact\n",
    "            custom_legend.append(Line2D([0], [0], linestyle='-', marker='o', color=plot_pars[\"colors\"][\"exon_selection\"][impact], \n",
    "                                        label=impact.capitalize(), markersize=7.5, markeredgewidth=0.1, markeredgecolor=\"black\")) \n",
    "            \n",
    "            for i, exon in exon_selection_impact.iterrows():\n",
    "                start_pos = exon.START_PROT_POS\n",
    "                end_pos = exon.END_PROT_POS\n",
    "                start_dnds = 0 if i == 0 else exon_selection_impact.iloc[i-1].dnds\n",
    "                dnds = exon.dnds\n",
    "                axes[ax].vlines(start_pos, ymin=start_dnds, ymax=dnds, lw=1, zorder=1, alpha=1, color=plot_pars[\"colors\"][\"exon_selection\"][impact])\n",
    "                axes[ax].hlines(dnds, xmin=start_pos, xmax=end_pos, lw=1, zorder=1, alpha=1, color=plot_pars[\"colors\"][\"exon_selection\"][impact])    \n",
    "\n",
    "        axes[ax].set_ylabel('Exon selection\\n(dN/dS)', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])   \n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        \n",
    "        custom_legend.append(Line2D([0], [0], linestyle='-', marker='o', color=\"white\", \n",
    "                                    label=\"Significant\", markersize=7.5, markeredgewidth=1.1, markeredgecolor=\"black\")) \n",
    "        ax_legend = axes[ax].legend(handles=custom_legend)\n",
    "        ax_legend.get_frame().set_linewidth(0.5)\n",
    "    \n",
    "    \n",
    "    # Track exon saturation\n",
    "    # ---------------------\n",
    "    if \"Exon_saturation\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Exon_saturation\")\n",
    "        \n",
    "        norm = mcolors.Normalize(vmin=0, vmax=1)\n",
    "        cmap = plt.get_cmap(\"hot_r\")\n",
    "        colors = cmap(norm(exon_coverage[\"SATURATION\"].values))\n",
    "        \n",
    "        for i in range(len(exon_coverage)):\n",
    "            exon = exon_coverage.iloc[i]\n",
    "            start_pos = int(exon.START_PROT_POS)\n",
    "            end_pos = int(exon.END_PROT_POS)\n",
    "            \n",
    "            if exon.DEPTH == 0:\n",
    "                axes[ax].fill_between(\n",
    "                    [start_pos, end_pos],\n",
    "                    -0.2, 1.2,\n",
    "                    color = plot_pars[\"colors\"][\"depth_below_thr\"],\n",
    "                    label=\" \")\n",
    "            else:\n",
    "                axes[ax].fill_between(\n",
    "                    [start_pos, end_pos], \n",
    "                    -0.2, 1.2, color=colors[i])\n",
    "                \n",
    "            axes[ax].vlines(start_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=1, alpha=1, color=\"black\")\n",
    "            if i == len(exon_coverage)-1:\n",
    "                axes[ax].vlines(end_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=1, alpha=1, color=\"black\")\n",
    "        \n",
    "        axes[ax].set_yticks([])  \n",
    "        axes[ax].set_ylabel('Exon saturation', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])   \n",
    "        axes[ax].set_ylim(-0.2, 1.2)\n",
    "        \n",
    "        color_bar(\n",
    "            axes, \n",
    "            ax, \n",
    "            cmap,\n",
    "            norm,\n",
    "            general_ticks=False, \n",
    "            bbox_to_anchor=plot_pars[\"depth_res_bbox_to_anchor\"],\n",
    "            label='Exon saturation',\n",
    "            fontsize=plot_pars[\"legend_fontsize\"],\n",
    "            tick_size=plot_pars[\"ticksize\"],\n",
    "            tick_span=0,\n",
    "            borderpad=0)      \n",
    "        \n",
    "    \n",
    "    # Track exon depth\n",
    "    # ----------------\n",
    "    if \"Exon_depth\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Exon_depth\")\n",
    "    \n",
    "        norm = mcolors.Normalize(vmin=200, vmax=max_depth)\n",
    "        cmap = plt.get_cmap(\"viridis_r\")\n",
    "        colors = cmap(norm(exon_coverage[\"DEPTH\"].values))\n",
    "        for i in range(len(exon_coverage)):\n",
    "            exon = exon_coverage.iloc[i]\n",
    "            start_pos = int(exon.START_PROT_POS)\n",
    "            end_pos = int(exon.END_PROT_POS)\n",
    "            \n",
    "            if exon.DEPTH == 0:\n",
    "                axes[ax].fill_between(\n",
    "                    [start_pos, end_pos],\n",
    "                    -0.2, 1.2,\n",
    "                    color = plot_pars[\"colors\"][\"depth_below_thr\"],\n",
    "                    label=\" \")\n",
    "            else:\n",
    "                axes[ax].fill_between(\n",
    "                    [start_pos, end_pos], \n",
    "                    -0.2, 1.2, color=colors[i])\n",
    "                \n",
    "            axes[ax].vlines(start_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=1, alpha=1, color=\"black\")\n",
    "            if i == len(exon_coverage)-1:\n",
    "                axes[ax].vlines(end_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=1, alpha=1, color=\"black\")\n",
    "            # if \"Res_depth\" in lst_tracks:\n",
    "            #     axes[ax+1].vlines(start_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=3, alpha=1, color=\"black\")\n",
    "            #     if i == len(exon_coverage)-1:\n",
    "            #         axes[ax+1].vlines(end_pos, ymin=-0.2, ymax=1.2, lw=0.4, zorder=3, alpha=1, color=\"black\")\n",
    "\n",
    "        axes[ax].set_yticks([])  \n",
    "        axes[ax].set_ylabel('Exon depth', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])    \n",
    "        axes[ax].set_ylim(-0.2, 1.2)\n",
    "\n",
    "        # Legend\n",
    "        if (exon_coverage['DEPTH'] == 0).any():\n",
    "            gray_patch = mpatches.Patch(color=plot_pars[\"colors\"][\"depth_below_thr\"])\n",
    "\n",
    "            legend_depth = axes[ax].legend(handles=[gray_patch],\n",
    "                                        fontsize=plot_pars[\"legend_fontsize\"], \n",
    "                                        frameon=plot_pars[\"legend_frameon\"], \n",
    "                                        bbox_to_anchor=plot_pars[\"depth_0_bbox_to_anchor\"],\n",
    "                                        title=\"Depth < 200  \", title_fontsize=plot_pars[\"legend_depth_fontsize\"])\n",
    "\n",
    "            plt.setp(legend_depth.get_title(), fontsize=plot_pars[\"legend_fontsize\"])\n",
    "            handle = legend_depth.get_patches()[0]  # For patches (like bars in bar plots)\n",
    "            handle.set_edgecolor('black')  \n",
    "            handle.set_linewidth(0.75)\n",
    "        \n",
    "        if \"Res_depth\" not in lst_tracks:\n",
    "            color_bar(\n",
    "                axes, \n",
    "                ax, \n",
    "                cmap,\n",
    "                norm,\n",
    "                general_ticks=False, \n",
    "                bbox_to_anchor=plot_pars[\"depth_res_bbox_to_anchor\"],\n",
    "                label='Depth',\n",
    "                fontsize=plot_pars[\"legend_fontsize\"],\n",
    "                tick_size=plot_pars[\"ticksize\"],\n",
    "                tick_span=0,\n",
    "                borderpad=0)\n",
    "    \n",
    "    \n",
    "    # Track per-residue depth\n",
    "    # -----------------------\n",
    "    if \"Res_depth\" in lst_tracks:\n",
    "        ax = lst_tracks.index(\"Res_depth\")\n",
    "        \n",
    "        res_coverage = get_res_coverage(coverage_df)\n",
    "        res_coverage[\"DEPTH\"] = res_coverage[\"DEPTH\"].apply(lambda x: np.nan if x == 0 else x)\n",
    "        norm = mcolors.Normalize(vmin=200, vmax=max_depth)\n",
    "        cmap = plt.get_cmap(\"viridis_r\")\n",
    "        colors = cmap(norm(res_coverage[\"DEPTH\"].values))\n",
    "\n",
    "        # Fill for depth = 0\n",
    "        axes[ax].fill_between(\n",
    "            res_coverage['PROT_POS'], -0.2, 1.2, where=where_plus(res_coverage['DEPTH'].fillna(0) == 0), \n",
    "            color=plot_pars[\"colors\"][\"depth_below_thr\"], alpha=1, label=' '\n",
    "            )\n",
    "\n",
    "        # Fill for depth > 0\n",
    "        for i in range(len(res_coverage) - 1):\n",
    "            axes[ax].fill_between(\n",
    "                [res_coverage[\"PROT_POS\"].iloc[i], res_coverage[\"PROT_POS\"].iloc[i+1]],\n",
    "                -0.2, 1.2,\n",
    "                color=colors[i]\n",
    "            )\n",
    "\n",
    "        axes[ax].set_yticks([])  \n",
    "        axes[ax].set_ylabel('Residue depth', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])    \n",
    "        axes[ax].set_ylim(-0.2, 1.2)\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])   \n",
    "        \n",
    "        color_bar(\n",
    "            axes, \n",
    "            ax, \n",
    "            cmap,\n",
    "            norm,\n",
    "            general_ticks=False, \n",
    "            bbox_to_anchor=plot_pars[\"depth_res_bbox_to_anchor\"],\n",
    "            label='Depth',\n",
    "            fontsize=plot_pars[\"legend_fontsize\"],\n",
    "            tick_size=plot_pars[\"ticksize\"],\n",
    "            tick_span=0,\n",
    "            borderpad=0)\n",
    "        \n",
    "        if \"Exon_depth\" not in lst_tracks and (res_coverage['DEPTH'].fillna(0) == 0).any():\n",
    "            gray_patch = mpatches.Patch(color=plot_pars[\"colors\"][\"depth_below_thr\"])\n",
    "\n",
    "            legend_depth = axes[ax].legend(handles=[gray_patch],\n",
    "                                        fontsize=plot_pars[\"legend_fontsize\"], \n",
    "                                        frameon=plot_pars[\"legend_frameon\"], \n",
    "                                        bbox_to_anchor=plot_pars[\"depth_0_bbox_to_anchor\"],\n",
    "                                        title=\"Depth < 200  \", title_fontsize=plot_pars[\"legend_depth_fontsize\"])\n",
    "\n",
    "            plt.setp(legend_depth.get_title(), fontsize=plot_pars[\"legend_fontsize\"])\n",
    "            handle = legend_depth.get_patches()[0]  # For patches (like bars in bar plots)\n",
    "            handle.set_edgecolor('black')  \n",
    "            handle.set_linewidth(0.75)\n",
    "    \n",
    "    \n",
    "    # Track pACC\n",
    "    # ----------\n",
    "    if \"Solvent_accessibility\" in lst_tracks and isinstance(pdb_tool_df, pd.DataFrame):\n",
    "        ax = lst_tracks.index(\"Solvent_accessibility\")\n",
    "\n",
    "        max_pacc = np.max(pdb_tool_df[\"pACC\"].fillna(0))\n",
    "        axes[ax].fill_between(\n",
    "            pdb_tool_df[\"Pos\"], 0, pdb_tool_df[\"pACC\"].fillna(0),                  \n",
    "            zorder=2, color=sns.color_palette(\"pastel\")[4], alpha=0.35\n",
    "            )\n",
    "        axes[ax].plot(\n",
    "            pdb_tool_df['Pos'], pdb_tool_df[\"pACC\"].fillna(0), \n",
    "            label=\"pACC\", zorder=3, color=sns.color_palette(\"tab10\")[4], lw=0.5\n",
    "            )      \n",
    "        axes[ax].set_ylabel('Solvent\\naccessibility', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        axes[ax].set_ylim(0 - max_pacc/30, max_pacc + max_pacc/10)\n",
    "        \n",
    "        \n",
    "    # Plot stability change\n",
    "    # ---------------------\n",
    "    if \"Stability_change\" in lst_tracks and isinstance(ddg_df, pd.DataFrame):\n",
    "        ax = lst_tracks.index(\"Stability_change\")\n",
    "        \n",
    "        max_value, min_value = ddg_df[\"Stability_change\"].max(), ddg_df[\"Stability_change\"].min()\n",
    "        axes[ax].fill_between(ddg_df['Pos'], 0, ddg_df[\"Stability_change\"], zorder=1,             \n",
    "                                color=\"white\")     \n",
    "        axes[ax].fill_between(ddg_df['Pos'], 0, ddg_df[\"Stability_change\"], zorder=1,             \n",
    "                                color=sns.color_palette(\"pastel\")[4], alpha=0.35)      \n",
    "        axes[ax].plot(ddg_df['Pos'], ddg_df[\"Stability_change\"], \n",
    "                        label=\"Stability change\", zorder=2, color=sns.color_palette(\"tab10\")[4], lw=0.5)    \n",
    "        axes[ax].set_ylabel('Stability change (kcal/mol)', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "\n",
    "        \n",
    "    # Track SSE\n",
    "    # ---------\n",
    "    if \"Secondary_structure\" in lst_tracks and isinstance(pdb_tool_df, pd.DataFrame):\n",
    "        ax = lst_tracks.index(\"Secondary_structure\")\n",
    "        \n",
    "        for n, sse in enumerate(('Coil', 'Helix', 'Ladder')):   \n",
    "            axes[ax].fill_between(\n",
    "                pdb_tool_df[\"Pos\"], 0, 1, where=(pdb_tool_df['SSE'] == sse),\n",
    "                zorder=2, color=plot_pars[\"sse_colors\"][sse], \n",
    "                alpha=1, label=sse, lw=plot_pars[\"sse_lw\"][gene]\n",
    "                )\n",
    "        axes[ax].set_yticks([])  \n",
    "        axes[ax].legend(fontsize=plot_pars[\"legend_fontsize\"], frameon=plot_pars[\"legend_frameon\"], \n",
    "                        bbox_to_anchor=plot_pars[\"sse_bbox_to_anchor\"], title = \"Secondary structure\", \n",
    "                        title_fontsize=plot_pars[\"legend_fontsize\"], \n",
    "                        handleheight=0.65, handlelength=2) \n",
    "        axes[ax].set_ylabel('Secondary structure', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].set_ylim(0, 1)\n",
    "                    \n",
    "        \n",
    "    # Track Domain\n",
    "    # ------------\n",
    "    if \"Domain\" in lst_tracks and isinstance(domain_df, pd.DataFrame):\n",
    "        ax = lst_tracks.index(\"Domain\")\n",
    "        \n",
    "        domain_color_dict = {}\n",
    "        for n, name in enumerate(domain_df[\"Description\"].unique()):\n",
    "            domain_color_dict[name] = f\"C{n}\"\n",
    "\n",
    "        n = 0\n",
    "        added_domain = []\n",
    "        for i, row in domain_df.iterrows():\n",
    "            if pd.Series([row[\"Description\"], row[\"Begin\"], row[\"End\"]]).isnull().any():\n",
    "                continue\n",
    "\n",
    "            name = row[\"Description\"]\n",
    "            start = int(row[\"Begin\"])\n",
    "            end = int(row[\"End\"])\n",
    "            \n",
    "            if name not in added_domain and (gene_len >= plot_pars[\"len_txt_thr\"] or gene == \"KDM6A\"):\n",
    "                axes[ax].fill_between(range(start, end + 1), -0.5, 0.45, alpha=0.5, color=domain_color_dict[name], label=name, lw=0.5)\n",
    "            else:\n",
    "                axes[ax].fill_between(range(start, end + 1), -0.5, 0.45, alpha=0.5, color=domain_color_dict[name], lw=0.5)\n",
    "            \n",
    "            if name not in added_domain:\n",
    "                if gene_len < plot_pars[\"len_txt_thr\"] and gene != \"KDM6A\":\n",
    "                    y = -0.04\n",
    "                    axes[ax].text(((start + end) / 2)+0.5, y, name, ha='center', va='center', fontsize=plot_fontsizes_general_config[\"custom_text_fontsize\"], color=\"black\")\n",
    "                added_domain.append(name)\n",
    "        axes[ax].set_yticks([])  \n",
    "        if (gene_len >= plot_pars[\"len_txt_thr\"] or gene == \"KDM6A\") and len(added_domain) > 0:\n",
    "            pass \n",
    "            # axes[ax].legend(fontsize=plot_pars[\"legend_fontsize\"], frameon=plot_pars[\"legend_frameon\"], \n",
    "            #                 bbox_to_anchor=(plot_pars[\"domain_x_bbox_to_anchor\"][gene], plot_pars[\"domain_y_bbox_to_anchor\"]), title = \"Domain\", \n",
    "            #                 title_fontsize=plot_pars[\"legend_fontsize\"],\n",
    "            #                 handleheight=0.67, handlelength=2) \n",
    "        axes[ax].set_ylabel('Domain', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].set_ylim(-0.5, 0.45)\n",
    "        \n",
    "        \n",
    "    # Track Domain Selection\n",
    "    # ----------------------\n",
    "    if \"Domain_selection\" in lst_tracks and isinstance(domain_selection_df, pd.DataFrame):\n",
    "        ax = lst_tracks.index(\"Domain_selection\")\n",
    "        \n",
    "        norm = mcolors.Normalize(vmin=1, vmax=domain_selection_df.dnds.max())\n",
    "        cmap = plt.get_cmap(\"Reds\")\n",
    "        colors = cmap(norm(domain_selection_df[\"dnds\"].values))\n",
    "\n",
    "        for i, row in domain_selection_df.iterrows():\n",
    "            if pd.Series([row[\"dnds\"], row[\"Begin\"], row[\"End\"]]).isnull().any():\n",
    "                continue\n",
    "\n",
    "            dnds = row[\"dnds\"]\n",
    "            start = int(row[\"Begin\"])\n",
    "            end = int(row[\"End\"])\n",
    "            y = (0.02, 0.5) if row[\"impact\"] == \"missense\" else (-0.5, -0.04)\n",
    "            \n",
    "            axes[ax].fill_between(range(start, end + 1), y[0], y[1], color=colors[i])\n",
    "            \n",
    "        # xlim = axes[ax].get_xlim()\n",
    "        # axes[ax].hlines(0, 0-(gene_len*0.2), gene_len+(gene_len*0.2),  lw=0.7, zorder=1, alpha=1, color=\"black\")\n",
    "        # axes[ax].set_xlim(xlim)\n",
    "\n",
    "        axes[ax].set_yticks([0.25, -0.25])\n",
    "        axes[ax].set_yticklabels([\"M\", \"T\"])\n",
    "        axes[ax].set_ylabel('Domain selection', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].set_ylim(-0.5, 0.5)\n",
    "        \n",
    "        color_bar(\n",
    "            axes, \n",
    "            ax, \n",
    "            cmap,\n",
    "            norm,\n",
    "            general_ticks=False, \n",
    "            bbox_to_anchor=plot_pars[\"domain_sel_bbox_to_anchor\"],\n",
    "            label='Domain selection',\n",
    "            fontsize=plot_pars[\"legend_fontsize\"],\n",
    "            tick_size=plot_pars[\"ticksize\"],\n",
    "            tick_span=0,\n",
    "            borderpad=0)\n",
    "        \n",
    "        \n",
    "    # Track Cancer 3D score\n",
    "    # ----------------------\n",
    "    if \"Cancer_3d_clustering\" in lst_tracks and len(o3d_cancer_df) > 0:\n",
    "        ax = lst_tracks.index(\"Cancer_3d_clustering\")\n",
    "        \n",
    "        axes[ax].plot(np.array(range(prot_len-1))+1, -o3d_cancer_df[\"O3D_score\"], zorder=2, color=plot_pars[\"colors\"][\"o3d_score\"], lw=1)                       \n",
    "        axes[ax].fill_between(\n",
    "            o3d_cancer_df['Pos'], 0, -n_3d_max, where=(o3d_cancer_df['Cluster'] == 1), \n",
    "            color=plot_pars[\"colors\"][\"o3d_cluster\"], alpha=1, label='Cluster', zorder=0, lw=1.5\n",
    "            )\n",
    "        axes[ax].set_ylabel('Tumors 3D-clustering\\n(missense mutations)', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        axes[ax].set_ylim(-n_3d_max - n_3d_max/plot_pars[\"j_margin\"], 0 + n_3d_max/(plot_pars[\"j_margin\"] * 5))\n",
    "        axes[ax].yaxis.set_major_locator(MaxNLocator(integer=True, nbins=3))\n",
    "        axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "        axes[ax].set_yticklabels(abs(axes[ax].get_yticks()).astype(int))\n",
    "        \n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "        \n",
    "\n",
    "    # Track Cancer mut\n",
    "    # ----------------\n",
    "    if \"Cancer_mut_count\" in lst_tracks and len(mut_cancer_df) > 0:\n",
    "        ax = lst_tracks.index(\"Cancer_mut_count\")\n",
    "        \n",
    "        plot_count_track(mut_cancer_df, gene_len=gene_len, axes=axes, ax=ax, negative=True, colors_dict=plot_pars[\"colors\"])\n",
    "        axes[ax].set_ylabel('Tumors mutations', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=0, va='center', ha='right')\n",
    "        axes[ax].yaxis.set_label_coords(plot_pars[\"y_labels_coord\"][0], plot_pars[\"y_labels_coord\"][1])\n",
    "        #axes[ax].set_ylim(-n_max - n_max/plot_pars[\"j_margin\"], 0 + n_max/plot_pars[\"j_margin\"])\n",
    "        axes[ax].yaxis.set_major_locator(MaxNLocator(integer=True, nbins=5))\n",
    "        axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "        axes[ax].set_yticklabels(abs(axes[ax].get_yticks()).astype(int))\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "            \n",
    "    \n",
    "    # Legend\n",
    "    # ======\n",
    "    \n",
    "    handles_0, labels_0 = axes[0].get_legend_handles_labels()\n",
    "    handles, labels = [], []\n",
    "    for h, l in zip(handles_0, labels_0): \n",
    "        if l not in labels:  \n",
    "            handles.append(h)\n",
    "            labels.append(l)\n",
    "    fig.legend(\n",
    "        handles, labels, loc='upper right', fontsize=plot_pars[\"legend_fontsize\"], \n",
    "        frameon=plot_pars[\"legend_frameon\"], bbox_to_anchor=plot_pars[\"cnsq_bbox_to_anchor\"],\n",
    "        title=\"Mutation consequence\", title_fontsize=plot_pars[\"legend_fontsize\"]\n",
    "        )\n",
    "    \n",
    "    if gene is not None:\n",
    "        fig.suptitle(title, y=0.93)\n",
    "\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "    plt.show()\n",
    "    \n",
    "    if \"Domain\" in lst_tracks and isinstance(domain_df, pd.DataFrame):\n",
    "        return domain_color_dict\n",
    "    \n",
    "    \n",
    "def plot_domain_selection(\n",
    "    df, \n",
    "    gene, \n",
    "    color_map, \n",
    "    title=\"Domain selection\", \n",
    "    show_domain_legend=False, \n",
    "    legend2_coord=(1.05, 1), \n",
    "    figsize=(5, 2.5),\n",
    "    save=False, \n",
    "    filename=\"domain_selection.png\"\n",
    "    ):\n",
    "    \n",
    "    df = df.drop(columns=[\"Begin\", \"End\"]).drop_duplicates()\n",
    "    custom_markers = {\"missense\": \"o\", \"truncating\": \"D\"}  # 'o' = Circle, 'D' = Rotated Square\n",
    "    custom_size = {\"missense\": 30, \"truncating\": 25}  # Larger for missense\n",
    "    df[\"color\"] = df[\"selection_id\"].map(color_map) \n",
    "    df[\"edge_width\"] = df[\"pvalue\"].apply(lambda p: 1 if p < 0.0001 else 0.2)\n",
    "\n",
    "    fig = plt.figure(figsize = figsize)\n",
    "\n",
    "    # H-line and V-line\n",
    "    plt.xlim(df[\"x_pos\"].min() - 0.5, df[\"x_pos\"].max() + 0.5) \n",
    "    plt.hlines(1, *plt.xlim(), lw=1, zorder=1, alpha=1, color=\"gray\", linestyles=\"dashed\")\n",
    "    plt.vlines(df[\"x_pos\"], ymin=df[\"lower\"], ymax=df[\"upper\"], lw=1, zorder=1, alpha=1, color=\"black\")\n",
    "\n",
    "    # Scatter plot\n",
    "    for selection_id, color in color_map.items():\n",
    "        for impact, marker in custom_markers.items():\n",
    "            subset = df[(df[\"selection_id\"] == selection_id) & (df[\"impact\"] == impact)]\n",
    "            plt.scatter(\n",
    "                subset[\"x_pos\"], subset[\"dnds\"], lw=0, edgecolor=\"black\",\n",
    "                color=\"white\", marker=marker, s=custom_size[impact]\n",
    "            )\n",
    "            plt.scatter(\n",
    "                subset[\"x_pos\"], subset[\"dnds\"],\n",
    "                color=color, marker=marker, s=custom_size[impact], \n",
    "                edgecolor=\"black\", lw=subset[\"edge_width\"], alpha=0.7\n",
    "            )\n",
    "\n",
    "    ax = plt.gca()\n",
    "    ax.set_xticks(range(len(df[\"selection_id\"].unique()))) \n",
    "    ax.set_xticklabels(df[\"selection_id\"].unique(), rotation=45, ha=\"right\") \n",
    "    ax.yaxis.set_major_locator(MaxNLocator(integer=True))\n",
    "    plt.gca().spines['top'].set_visible(False)\n",
    "    plt.gca().spines['right'].set_visible(False)\n",
    "    # plt.xlabel(\"Domain ID\")\n",
    "    plt.ylabel(\"dN/dS per\\ndomain\")\n",
    "    plt.title(title)\n",
    "\n",
    "    # Legends\n",
    "    selection_legend = [mpatches.Patch(color=color, label=selection_id) for selection_id, color in color_map.items()]\n",
    "    custom_size_legend = {\"missense\": 30, \"truncating\": 25}\n",
    "    impact_legend = [\n",
    "        plt.scatter(\n",
    "            [], [], marker=custom_markers[impact], s=custom_size_legend[impact], \n",
    "            color=\"white\", edgecolor=\"black\", label=impact\n",
    "            ) for impact in custom_markers\n",
    "        ]\n",
    "\n",
    "    legend1 = plt.legend(handles=selection_legend, title=\"Domain ID\", bbox_to_anchor=(1.05, 0.65), loc=\"upper left\", frameon=False)\n",
    "    legend2 = plt.legend(handles=impact_legend, title=\"Impact\" if show_domain_legend else None, bbox_to_anchor=legend2_coord, loc=\"upper left\", frameon=False)\n",
    "    \n",
    "    if show_domain_legend:\n",
    "        plt.gca().add_artist(legend1)\n",
    "        \n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "\n",
    "    plt.show()\n",
    "    \n",
    "    \n",
    "def mut_count_side_barplot(\n",
    "    df, \n",
    "    color_dict, \n",
    "    figsize=(3.4, 2.3), \n",
    "    max_count=None,\n",
    "    title=None,\n",
    "    save=False, \n",
    "    filename=\"mut_count_side_barplot.png\"\n",
    "    ):\n",
    "    \n",
    "    title = title\n",
    "\n",
    "    order = [\"truncating\", \"missense\", \"synonymous\"]\n",
    "    df = df.set_index(\"Consequence\").loc[order].reset_index()\n",
    "\n",
    "    fig, ax = plt.subplots(figsize=figsize)\n",
    "    bars = ax.bar(df[\"Consequence\"], df[\"Count\"], color=[color_dict[c] for c in df[\"Consequence\"]])\n",
    "    \n",
    "    for bar in bars:\n",
    "        height = bar.get_height()\n",
    "        ax.text(bar.get_x() + bar.get_width() / 2, height, str(height),\n",
    "                ha='center', va='bottom', fontsize=plot_fontsizes_general_config[\"custom_text_fontsize\"])\n",
    "\n",
    "\n",
    "    plt.ylabel(\"Number of mutations\")\n",
    "    plt.ylim(0, max_count if isinstance(max_count, int) else np.max(df[\"Count\"]))\n",
    "    ax.set_title(title, pad=15)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.set_xticklabels([label.capitalize() for label in df[\"Consequence\"]])\n",
    "\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "    plt.show()\n",
    "\n",
    "def mut_count_horizontal_barplot(\n",
    "    df,\n",
    "    color_dict,\n",
    "    figsizee=(1.625, 0.32),\n",
    "    title=None,\n",
    "    save=False,\n",
    "    filename=\"mut_count_horizontal_barplot.png\"\n",
    "    ):\n",
    "    order = [\"truncating\", \"missense\", \"synonymous\"]\n",
    "    df = df.set_index(\"Consequence\").loc[order]#.reset_index()\n",
    "    \n",
    "    fig, ax = plt.subplots(figsize=figsizee)\n",
    "    bottom = 0 #np.zeros(len(df))  # Track bottom for stacking\n",
    "    \n",
    "    for consequence in order:\n",
    "        bars = ax.barh(0, df.loc[consequence], left=bottom, color=color_dict[consequence], label=consequence.capitalize())\n",
    "\n",
    "        for bar in bars:\n",
    "            width = 0.75\n",
    "            if df.loc[consequence].item() > 0:  # Only label nonzero values\n",
    "                ax.text(bottom + df.loc[consequence].item() / 2 , width, str(int(df.loc[consequence])),\n",
    "                        ha='center', va='center', fontsize=plot_fontsizes_general_config[\"custom_text_fontsize\"])\n",
    "        \n",
    "        bottom += df.loc[consequence].item()  # Update bottom for stacking\n",
    "    \n",
    "    ax.set_xlabel(\"Number of mutations\", fontsize=plot_fontsizes_general_config[\"custom_text_fontsize\"])\n",
    "    # ax.set_title(title if title else \"Mutation Count per Sample\", pad=10)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['left'].set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.set_yticks([])  # Remove y-axis ticks\n",
    "    ax.set_yticklabels([])  # Remove y-axis ticks\n",
    "    # ax.legend()\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "75637b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def get_selection_groups(df, thr_selection):\n",
    "    df = df.copy().rename(columns={\"Protein_position\": \"Pos\"})\n",
    "    g0 = df[df[\"Selection\"] == 0]\n",
    "    g1 = df[(df[\"Selection\"] != 0) & (df[\"p_value\"] >= thr_selection)]\n",
    "\n",
    "    g2 = df[df[\"p_value\"] < thr_selection]\n",
    "    g2 = g2.sort_values(\"Selection\").reset_index(drop=True)\n",
    "\n",
    "    g0[\"Group\"] = \"G0\"\n",
    "    g1[\"Group\"] = \"G1\"\n",
    "    g2[\"Group\"] = \"G2\"\n",
    "\n",
    "    return pd.concat((g0, g1, g2)).reset_index(drop=True)\n",
    "\n",
    "\n",
    "def plot_feat_by_selection_group(df, figsize=(1.74, 1.4), save=False, gene = '', filename=\"selection_groups_feat.png\",):\n",
    "\n",
    "    color_dict = {\n",
    "    \"G0\": \"gray\", \n",
    "    \"G1\": \"#6baed6\",  \n",
    "    \"G2\": \"blue\"\n",
    "}\n",
    "    rename_labels = {\n",
    "        \"G0\": f\"Not observed\\n(N={df[df['Group'] == 'G0'].shape[0]})\",\n",
    "        \"G1\": f\"Not significant\\n(N={df[df['Group'] == 'G1'].shape[0]})\",\n",
    "        \"G2\": f\"Significant\\n(N={df[df['Group'] == 'G2'].shape[0]})\"\n",
    "    }\n",
    "\n",
    "    fig, axis = plt.subplots(1, 1, figsize=figsize, sharey=False)\n",
    "\n",
    "    # Violin + Jitter plot for pACC\n",
    "    sns.violinplot(ax=axis, x=\"Group\", y=\"pACC\", data=df, color=\"#d9d9d9\", inner=None)\n",
    "    sns.stripplot(ax=axis, x=\"Group\", y=\"pACC\", data=df, \n",
    "                palette=color_dict, jitter=True, size=3, alpha=0.5)\n",
    "    axis.set_title(f\"{gene}\")\n",
    "    axis.set_xlabel(None)\n",
    "    axis.set_ylabel(\"Solvent accessibility\")\n",
    "    axis.set_ylim(-2,102)\n",
    "\n",
    "    # Overlay median values\n",
    "    medians = df.groupby(\"Group\")[\"pACC\"].median()\n",
    "    for i, median in enumerate(medians):\n",
    "        axis.scatter(i, median, zorder=3, color='black', marker='x', s=40, label=\"Median\" if i == 0 else \"\")\n",
    "        \n",
    "    axis.spines['top'].set_visible(False)\n",
    "    axis.spines['right'].set_visible(False)\n",
    "    \n",
    "    axis.set_xticklabels([rename_labels[label] for label in color_dict.keys()])\n",
    "\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "    \n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e95187d",
   "metadata": {},
   "source": [
    "### Extended and supplementary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "7526d7ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:32: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  all_exons_depth = exons_depth[[\"GENE\", \"EXON_RANK\", \"DEPTH\", \"COVERED\"]].groupby(\"GENE\").apply(\n",
      "/tmp/ipykernel_1517215/2993950153.py:74: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  average_dnds = domain_selection_gene.groupby(\"selection_id\").apply(lambda x: x[\"dnds\"].mean()).to_dict()\n",
      "/tmp/ipykernel_1517215/802346302.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  g0[\"Group\"] = \"G0\"\n",
      "/tmp/ipykernel_1517215/802346302.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  g1[\"Group\"] = \"G1\"\n",
      "/tmp/ipykernel_1517215/802346302.py:33: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  sns.stripplot(ax=axis, x=\"Group\", y=\"pACC\", data=df,\n",
      "/tmp/ipykernel_1517215/802346302.py:48: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.\n",
      "  axis.set_xticklabels([rename_labels[label] for label in color_dict.keys()])\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAL4AAACmCAYAAABk8TVrAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAQLdJREFUeJztnXl8VPW5/9/nzL5nspGEJOyLQEVwA0FBWmmttVV73TeqLLVad+212KK91qWKrVZxAQGLrVcLaq/bz4qKVkVaEBAMAcISEhKyz0xmX875/XFyDgkkZEIyMyj5vF7zgsycOeeZmc/3Oc/3WQVZlmX60Y/jDGKmBehHPzKBfuL347hEP/H7cVyin/j9OC7RT/x+HJfoJ34/jkv0E78fxyX6id+P4xL9xO/HcQl9pgXoR+eYN28e27dvZ+/evZjNZgoKCqioqECv1zN48GAEQSAcDvPQQw8xffp05s6dy+bNm7FYLNjtdt566y02bdrEL37xCyRJ4vzzz2f+/PmZ/ljHDuR+HNNYsGCBvGzZMlmWZXnZsmXyggULtNfKysrk008/XZZlWZ40aZIcCoU6vPf73/++vHHjRlmWZXn69Onyrl270iHyNwL9ps43GC0tLTgcDmKxGJWVlVx88cVMnTqVd955B4C33nqLk046iUAggM/nw2q1ZljiYwf9ps43DMuXL2fNmjWIoojb7ebpp5/G7/fzy1/+kttuuw2fz8dZZ53FpEmTyM7OZuPGjVx00UWMGTOGrKysTIt/zKBf43/DMGvWLNasWcOHH37IqlWrGDlyJE6nk1/+8peYzWby8/M58cQT2bVrFwATJkxgz549TJ48mYcffjjD0h876Cf+twBbtmzhe9/7HpIk0draytdff83IkSM599xzqa6uBsDlcmVYymML/abOtwAnnXQS5557LpMmTUKn0/HQQw/hcrm46667uPjiizGZTOTn5/PCCy9kWtRjBoIs9xei9OP4Q7+p04/jEv3E78dxiX7i9+O4RD/x+3Fcop/4/Tgu0U/8fhyX+EYSX5ZlfD4f/Z7Yfhwt0hbA+uijj3jllVe44IILWLlyJYIg8Oyzz7JixQrWrVuHzWbjscceS+pcra2tuFwuvF4vTqfziMdu3ryZ//u//8PlcvHd734Xg8HQ4fUtW7awZ88eotEoiUQCSZIAsFqtWK1WEokETqeTsWPHdhn9rKmp4V//+hdOp5PZs2djs9mS+hw9wZIlS6irq2PSzB+z2aunNQbZRtALsMcPMQmMIuSYwaoHUYBiGwx3iqz/9zoqKyu57rrrGDhwYJ/L1lPs3LmT//3f/wXgjDPO4Lvf/W7aZUiLxt+9ezcbNmwgHA7z0ksvsWTJEs4880w+/PBDVq9ezTPPPIPT6WTnzp2dvj8SieDz+To8kkFtbS1vv/02RqORqVOnHkb6uro6KioqCAQCxGIxjfQAwWCQxsZGPB4PNTU1rFu3jmAw2Ol1ioqKOPHEE/H5fKxatarDefoCjY2N1NbWklNYTFmrnvoQhOJQ2QrbvBBMQEyGQAKqA7CnFfYHYKdX+Xvw4MGAssiPBYRCIe3/XX2nqUZaiD906FDuvPNOAI18hYWF1NXVHfZ3Z1BD8OqjpKSk22tGo1Fee+01EokEkyZNwm63H3ZMJBIhkUgc8TyyLBOLxfB4POzYsaPL40aPHk1xcTGVlZV89tln3crXE5SXlwNQOLCEWLs1Jclw6BKTgIik3AF8MQjEZfLz8zGZTJSXlx8T5mF7srdfBOlE2m18nU4HKNq4oKDgsL87wz333IPX69UeVVVV3V7ngw8+oLm5mRNOOIHCwsJOjykoKMBsNicltyRJ1NTUdEkcQRA49dRTsVqtfPzxx9TW1iZ13mSwfft2RFFk6MABFFvBIIIgKI+uEJUUMyjLCKIoUlRURGtrKwcOHOgzuY4WgUCg0/+nE2kn/oUXXsi8efP45JNPmDFjBmeccQY33XQTPp+P4cOHd/oek8mE0+ns8DgS9u/fz/r163G5XIwdO7bL44xGI2eccQYWi6XD86J4+NciiiKCICAcgW1Go5HTTjsNWZZ55513+kS7hkIhampqyM3NxWQ0MMQpkG9WfjijDixdiCPJkGeGQqtygLr41XTlTKI92f1+f0ZkSGt25vLlywE477zztOeuu+46rrvuuj69zgcffADAySefrN1ROoMsy3g8HnQ6HaIoIssygiCg0+nQ6/VEo1HtWFEUGTJkSLfXHjBgAIMGDaKyspKysrIjLrxkoN7d8vPziUsy/66X2eOHhAQCIAtgbtP+sgRRQEZZGA3hg+fJz88HYN++fb2Spy+gkt3lcuH3+7XvPZ34Rrozj4Tq6moqKyspKioiLy+v22P37NlDLBZDlmVEUdQ0e3utL4oiFoulyzvSoRg3bhyCIPDpp5/2WuvX1NQAkJubSzghUxtU7PcEij1vEGCAFUZlgc3YJi+KVyeSgHjbJsBkMuFwONi/f3/G7Xyfz4fRaMRutxOPxwmHw92/qY/xrSP+xo0bARg1alS3x/r9fqLRqOaFUTe6kiQRi8W042RZJhKJ4PV6k5LBbrdTXFxMfX19r239+vp6QNGOsiwTTigaHRTiGwXlR6wOgC968PmEDNkmZT+gwuVyEQ6HM2ZXq1Drf9Ua4GS9dH2JbxXxZVlmx44dWK3WbrU9QF5eHpFIRNOAgiAgyzKSJB3m7REEgXg8nrQsqgvxSJ6gZNDS0oLBYMBsNiMjYNFDe6dsDKgMQFNE+b/q5THqYHIeHUwIh8MBQHNzc69k6g3C4TCRSKQD8ZNVKH2JbxXxm5ubCQaD5OXlJWUzZmdnk5ubi9lsxmq1anb+oX54URSxWq09KtZWZUjGA3Uk+P1+bfNt0wsUWECvU+x7EQglFLIfarzYdFDuE5DamTXqeTK1oQTweDwA2Gw2LdCnPpdOfKuIr36B3Xl9VIiiSGFhIRaLhXg83iXpHQ4Hp556KiaTKWlZDAYDFoul1z9qJBLRYh06UWBaocAAC5h1Xf94qscH5A5+f6PRqJ0zU2hpaQEUc1AlvvpcOvGtIr76g6o/cDLIzs4mFosdFrlVodPptIXRE1MHFPL3hmSyLJNIJDp4pow6kQk5IkPs4DBCZz4rQQCHHuwGoYONr56nu6BdKtGe+GpQsZ/4vYRer3hne/LDtrS0dEl69Vxer5eysjJ27drVo3SEQ0nbU6hxg0O9MMU2GOwQGOeG0/Ig16RscvUoJpAA1EegwCIjtjP5VNk7i1OkC+r+wm63YzQaMZlMNDU1pV2ObxXxVRMnWRtWkiTq6uqOGDaXZVnz6gSDwaQ1uCRJBIPBXrf1MJlMHTxMANVBaIyANwp7/ZBjBJ2guDhlIC5DOK7k7LSHep6e3BH7Gs3NzQiCoJk5DocDj8fT5/lN3eFbRfycnBx0Ol3SXotwOKxtgnfu3HmYd0H15zc1NbF582b0en3SpPF6vUiSpAWOjhYWi+WwxdYcVhZjbRBaIlAVVPJzVMgoKQv1IYhLB+8W6nky2UqwsbERm82m3QkdDgeSJKXd3PlWEd9gMFBcXExLS0tSQZFYLEYgEKCiooKXX36Z5cuXa+RXTYyWlhYWL17MsmXL+OKLL5KWRfXfq27No4XL5SIUCnUw3yw6qA9Da0wJZkXaPDvt/Vh6UXmitd3NQvXfJ7v572uoMYT211f/39jYmFZZvlXEByVLUpblpNyIfr+frKwshgwZgsvloqWlRSO/LMt4vV6WLVtGS0sLbreb7OzspLIJZVmmsrISnU7HiBEjevV53G430DG/JSbJeCMgy4p2V5eEru1h14OjzdlvarfF8Pv9CIKQsR6aKrm/ccR/6623OuSvHIsYO3YsOp2OioqKbkPzJpMJo9FIUVERc+bMwe12a+Tft28fy5cv10g/d+5ccnNzkzJ1Ghoa8Pl8jB49ukcu0M6gBuJUt6g/KrHDpxBe3bfqUH5IGaUIxWUEpxFK7AJWvXKQupDdbrfmBEg3VHKrgTQ4SPx0b3B7RPy9e/dy6aWXcs0117Bq1aqM5VIfCTabjbFjx+Lz+bpNF8jJySEvLw+n08nQoUO58cYbyc7OpqWlhaVLl2qkv/7668nOzqawsDAp4m/fvh2A0047rdefR03VVm3g5ohMQlI0uUFUTBq7QVkEoqj49w0ijHYJfMd90PgJhUJEIpEuU7/Tgc40vtVqRafT0dDQkFZZekT8m266iddff51f//rXLFu2jNLSUq6++mo2bNiQKvmOCpMnTwagrKzsiFpfEAQKCgo46aSTGDFiBKeddho///nPOxzz05/+lIEDB2K1WpOyjVtaWqipqaG0tJTi4uLefRAOEr+5uRlZlqkLK3k4UUnR6mPcyiIw6xQTJ5RQXotKSqKaClWjdlWbkA50RnxRFLHb7TQ2NqY1ea5H97zHH3+c1atXk5WVxTXXXMOrr75KLBZj2rRpbNq0KUUi9hz5+fmMGTOGsrIyampquq0zlSQJt9vN7t27Wbx4cYfXVq1aRU5ODqNGjSInJ6fba6vlfdOmTTv6D9AOasPX5uZmWqMScQlcJmVDa9fD1AEiO7wSu3xQE1Ty8EXAF5XxRQWy2iwtlfh9sRiPFg0NDZjN5g53Tb9fxGx24/V6aW1tTdvGu0fEDwQCvP7665rdumHDBk4++WTeeuutHl10w4YNLF68GK/Xq/VyV4vPexPwaY9p06axbds2vvrqKwoLC7sM2ng8Hqqqqti1axd/+MMfaGpqIicnh8suu4yXX36Z5uZmFi9ezIIFCw6r2T0UajbmkCFDeu3NaY+SkhLq6+sJtPoQBRdZbbyx65WMzJaIgCcqoxPAqlM8PDGpzbPThoaGBnQ6HUVFRX0mV08Qj8fxeDwd3LsffWRj61YzjY0nk5MToKmp6dgi/gcffMCnn37Kiy++qN2OJEni9ddfZ/PmzT3WIv/5z38oLy9HEARaW1t56623+Mtf/sKHH37IOeecc9jxkUikgy87mTTW3NxcJkyYwJdffsnevXsZOnRop8dVVVVRVlbGn/70J5qbm8nOzmbevHm4XC7mzp3L888/T3NzMw888AAjR47s8jyyLGt3vb7uGlBaWsqGDRvwNdUzuDSLmoCMXoShToFdPllJUTAopYYGUe2wIGA3KLZOLBajpaWF4uLijG1s1djKwQxRHVu3KmWfer2JXbuG0tTUlFSxT18gKRt/4sSJTJ8+ncmTJzN9+nSmTZvGjBkzePvtt4/qoqeccgqrV6/mV7/6FV9//TXQ98XmAGeddRYGg4EtW7Z0mWfj9/uprKzE4/Hgdru57rrryMnJISsrC6fTqW1sm5ub2b17d5fXqqqqoqWlhXHjxvW5HV1aWgooWnuARWBCrsh3skWsegFRgIQkE5cgnFDIf3IuDHEe/GmbmpqQZVk7TyagmlqqRtfpZARBUaImkwmdLpFWz05SxH/zzTeZNm0aI0eOZM2aNXz88cd89NFHLF269KguunDhQkBx1VVWVgJ9X2wOinaZPHky4XC4y7z4nJwcxowZw7XXXssNN9zAgAEDsNvtuN1uXC4XQ4YM4fbbb+eWW27hrLPO6vQciUSCr776Cp1Ox9lnn52UbD2B0+nE7XbT0NBw2AZwiEMgIimaPd+ilCJu9wrs8ErasarHZNCgQX0uW7Jon6MD4HJJTJkSxGCQycnRM2rUjrTWCSR13zvppJMAmD59ep/URl588cVcffXVmEwmVqxYwbx580gkEjz//POdHm8ymY7aHz558mTWr1/Ptm3bGDZs2GHnGTRoEE1NTYwdOxaDwcCwYcOIx+MkEglcLpc2V+pI2L17N4FAgEmTJqUsOFRaWsrmzZvx+Xwd8n+seoFhTtjbKlMdULw5AjLNEYGmCOSaFeILgpDRjW37rEwVEyaEmTBBibD/4x/RY4/4f/zjHzvNEhQEoUsteCRcdNFFXHTRRdrfV155ZY/PkSyMRiNnnnkm7733HuXl5YwfP77D67Isd8gNTyQSGI1GwuEwbrebcDiMLCu9aTrz4cfjccrKyjAajUyZMiVln6OkpITNmzfT2NioEV+SZXb5oCkic6DNo6NDrbWVkWWBRCJBc3MzAwYM6HUwrTdoX4DSGWw2Gx6PJ22F50kR/7777gNIeyV8X2HixIl8/vnnVFRUHBZNVbMC1c/m9Xq1JK5QKERpaSkOh6NLr9Du3bsJh8OceeaZKU3+UrV1U1MTw4YNA5QuCk2RNjtZpwSyopLi53cYlHaCnhYviUQio9oeFOJbLJYuvXY2m42mpib8fn+HyG6qkBTxn3jiCR5//HHOPvtsjSDqyvzwww9TKmBfQK/Xc8YZZ/Dee++xc+dOxo0bp71mtVq1VAVRFHG5XB3SgCVJ6pL0kiRRXl6O0Wjk9NNPT+lnyM3NPSx3PdEuI9NlhKgkkGOSyTELjHApMqvHZ7Jnptrk90hmYPv622OG+I8//jigNH5NJJTdd2/TbdONCRMm8Mknn1BRUcEJJ5zQQfO43W4CgQCCIGA0GrXOBgMGDDhiPn1VVRWhUIhJkyYd1pSqr6FGmfft20c8Hkev15NnUbR+KAEuo8BIp1KhZdQdvDOrtnUmI7bBYJBEInHE70h9rbW1tctj+hI9SllYunQp06ZN47bbbmPq1Klax9tvAgwGAxMnTiQSiXTwCqmZlNFolEAgQFlZGa2trQQCAc2F2dWmq6KiAoBTTz01LZ+hoKBASzYDMIgCI11g08noRZmGMOzwyZR7JMJxxQTyer3o9fqkos6pgloYZLFY2LTJzCuvuPjnP+1EIgcXaLoL4XsUzVi6dCn/+te/tFYbZ511FpdddlmqZOtzTJgwgc8++4w9e/ZokdVYLKZ1Z4jH4/j9foxGI4lEgmg0it1uZ//+/Vgslg4ay+fz0djYyPDhw9OW5qtmavp8Po3Ie1pl9rcVooRiMiV2pSi9wicz1n3w2EyWGwYCAaqqiti5cxwtLVmUlMSor9djMMicfbaSbq32ME1Xz5+kiK+2nSsqKmL16tVMnDiRr776KunOYscK3G43gwcPZu/evQSDQcxmM9u3b8fn8xEOh5EkSeuOfGg3tVgs1oH4avzhxBNPTJv8Ktnba8X6kFJsEoorJo8tDHqdTEsE8vQR4vE4ubm5aZOxM2zfHmfnzuGYzQ6amvSIIpSWxmhtPfj9qg6HdLUNT4r4CxYsAJSd99/+9jdefvllZFnOWPi7Nxg7dix79+6lurqarKwszYxRia56eWRZ1jZcZrP5MDdcVVUVBoOBkSNHpk12tSilPfFtBoF9AZm4rKQqNESUtAWLDrY0JZBEfcYKT1TU1ytRc7dbwu+XiUQEBEFm7NiDVXKqqzhd7QSTYu6tt97K+PHj+fjjjzs8/010b44cOZK3336bmpoazb0WjUY1La+Ez5WmsQ6Hg8LCQtxud4fNcGtrK62trYwePbrbxLW+hM1mQxTFDlpxqAN2eEESwKaHYFxJW0CAhoiAJBp6XfDeWxQW+jAao5hMIt/5TpiJE0OMGxcmO/ugW0r9HtPV8ycp4m/evJnx48ezZs2aw8h+NAGsTMJut2t5QaNHjyYQCGh5PDqdDlmWMZlMGAwGrULrUN+zmlOUblNPFEVsNhvhcJjGsIw3qhSem0QlG1NA0fRqVqacSCAmYp0OxUgnDIYQp566gdGjcxk2LE5OzuHtX9RmvccU8a+55hpAibD6fD4EQeCpp55i9uzZKRUuVRg8eDC1tbUEg0FsNptWSabX6zGZTFp7O5PJ1GlQSnV39mXqcbKwWq3U+aNU+BSvzW6fUn1l1imlhyNcSh/NUBz0UoCQHM9oVwVQ9kcmU4wRI6K43Z33PBIEAb1e3+OmXUeLHm31f/azn2G327n33nu5/fbbueeee1IlV0qhZncGg0EcDgcmk0n74vPy8hgzZgzFxcUMGzaMWCxGbW2tVgEFSlDIZrNlxHY2m81EBQOyLOOPKS0C42259yYdFFgUN6fDoATY4tacpKe+pArt76hHgiiKaSN+j3ankiRhNBpxuVyMGDEiI11u+wJqMMfn83HKKaeQk5NDc3Mz4XCYRCJBa2srxcXFRKNRdu/erW14Y7EYWVlZBINBRowYkZE9jtFoRBdpoDkiE0oI6NqGQjgNSltwQVA6KoNAiyQhGSxp3Yd0BrU1SncuVVEU09besEfEv+GGG7j77rv5n//5Hx599FF+9atfpUqulMLhcGgNXY1GI/n5+TQ0NHDgwAFkWaa+vp5wOExeXh6xWIxwOIxOp9Oiu6BEdTMBnU6HLhbEqZcxiAJ5JqVz2mC7QIldIC5DdVBGkhXPlC4azLj3rX0b9iOhs0TIVKFH38gPf/hDJk+ejF6vR5IkTj755FTJlVIIgkBubi7V1dXE43HNlJEkCUFQMhpbWlooLCykpaVF00IOh0Or/sqUb1zVmjlGCU9CJBgHkwildjDoBIzAmCylw1og2kRjqDmjwSs4Nonfo2/k0ksvpaWlhVtvvZWJEydyww03HNVFy8vLmTNnDvPmzePpp59m9uzZzJkzJ61dfLOzs5FlmYqKCpqbm4nFYh1uyU6nE0mScLlcWK1WHA4Her1e86FnZ2enTdb2UMlTapVJSAIJWZlxu7UFwgmFNHaDov3NiWCH93wTcEwSPxKJMHToUARB4LzzzjvqfocvvPACAwcORBRFVq9e3WHgc7qgjtbxeDzaptbtdpOfn8/IkSMZOXKk5s50OBzaBA/Vh54p37hKjEgCErJi0tSHYKdP5qtmmVD8IHHaZ9JmEsnKoc4hSwd6ZOqcd955XH755fzud7/j3nvvPeoCki1btrBkyRJ27NihFWYfqeb2aIrNu4PT6exwazUajQwcOJBRo0YpHpM2zT5o0CC8Xi8Gg4G8vDzKyso0f3omoN6V6qMizVHFpIlLijuzPiTjsSlR0WAcJFHZ1Ka7E/Gh6Anx03V36hHxf/WrX3HeeedRWVnJ7Nmzky76PhR5eXnY7XaysrI6DHjuqvXFQw89xP33339U1+oKalDH4XBobkk11Xr//v3a3czlcnUo0g6Hw9jt9oyZDyqJm6MiDj14IkrgyiCCPwbhuMSWgGICNZjykXTVaXMRdoVw2EBTUzZ+v8CRYmnHLPHnz5+Px+Ph3//+NzfddBPvvvvuUaUm33LLLcybNw+n08mLL77Ybc3tPffcw+2336797fP5jnrRqVA1tuqiVKO1oVCIuro6DAYDgiDg8XjIysrSRvtEIpGMJn3FYjEknZHmqEBEUvri63RK5NYgwu5WsOiVgRCyoCNudmWU+PX18M47Q9i710I8nssVV0TJze18LyfLcp/1VeoOPSL+2rVr+fDDD5kxYwbXXnttl0TtDqeccgqvvPKK9nd3JlNvis27gsVi0VyXqrvPZrPh9/u1oc9ZWVkEAgH27t2LIAjKkOV4ZiOh8XgcyZqN26QMhjCKEIkrJYc6QWkfnmtSxgSJooiQiB42WCKd2LwZYjGFzJGISHm5ialTO8/APFK1W1+jR8TPysrilVdeIRQK8d57733jqrDaQ41mxuNxgsEgra2tWupxIpEgFAppkzvU26+6B8lkJDQajaID9KJATpsY/piMp62JtSiDTgSLTiZbCBEOezNKfJfroAtWKezver+RTuL36Cp/+ctfqK2tZcKECWzdupW//OUvqZIr5VDvIIlEQvPU6HS6DtVCh9Z+qgsgk8SPxWJYE34KrQI2PRRYBE5wKZq/bRYEsQREEgKWtrGgmWztfsopMGaMD4fDzwkn+DjxRCXt2O8X8fs72vPpNHV6RPzXX3+dQCDAokWL2LRpE2+++Waq5Eo5RFHEaDQiiqJmv+fk5KDX69Hr9TidTgwGA0VFRdocXNW2z2Sbjmg0ikGvZ5Bd4DvZIqV28MaUjmq+mLLBjSQgKsl4BGUnmUmNL4owaZKHU0/9kkmTWhBFWLfOwrJlbpYvd/PllweVyDHrznzyySdZt24dACtWrGDKlClcccUVKREsHTCbzciyzMSJE6msrKS+vl7z8KhJaO09Taqpk2niOxwOPv74Y0aOHInRXYAvpnRckCQlmLWjqpbg/p2ccNLJSKI+48M82rszo1H497+tbX8LfP65lfHjw4iinFbi9+gqFouFjRs3Eo/H2bBhQ0YJ0BcwmUxEIhF27dpFTU0NoVBIG5cTDAZpampiz549HZLU1PdlAmpZZHl5Ob/4xS+4+uqraaxThl+EEkrdbUt9La/On8XLv7uJfV9+Rsyac8wQH5Q7gE530J+v1yvPpRs9tvFXrFjBT37yE5YuXXrUvTOPFRiNRmKxmLbBDQQC+P1+qqqqqK+v13q2q1momSa+ev3S0lIKCwupqqrixuuvIdZSSygO3oZaXvvNLLx1VbjzCykdOgzaFsuxAr0ezjnHj9ksYbVKzJzZSiZCIj0ivtfrZeLEibz99ttYLJY+iaBmEmazGUmS8Pl8Wr/MeDyuzbYNh8NEIhFNY6maM1ObW5XAeXl5rFixgpKSEqqqqrj3hmvx79rI679VSO8aUMLPHllOfkEhxmBTxjX+oUlqI0ZEmTOnheuvb2Ho0Finx6YaPSL+3LlztVLDO+64gxtvvDElQqULateESCSCwWDAYDBotbaqrWmz2bS8HJV4qW4e1RVisRhRay61lhI2xfO5+fHl5BWVcGB/FU/ddiWeA1VkFZRw7cPLyc4rJCDpQE5kPHJ7aJvGzqBOcU9XekWPiC/LstbK2+12pzWbMhWwWq3a+MuSkhKys7PR6/UEg0EkSaKoqIjRo0drP5iaL5Qp4nuCUaL2fEI6C/UhaDYXcv5tD3c45ge3Pow9txAJCEo6EiZnxk2dnhSiHJMVWAsWLOD8888HFO3z29/+NiVCpQtqBFav1+P1ejViq7k4fr+/w4+V6cngkTYChwUzUdWmf+y/Oxzz7h//m0seWE5hUSF5BqgX0kemrqAuvO4KYvR6fdoWaY80/nnnncf777/P+++/z5o1a5g4cWKq5EoL1KisJElYrVbNro/H41pBSnuEw2GltC9DGt8qSujDXmKCTtvI+uqqyBpQwqUP/ZWsghK8dVX8/TezwFNLrllGH/Fm/M6sKoxkiJ+u/UiPiH/rrbcyadIkhg0bRmlpqab9v6lQMzQjkQjNzc1Eo1FkWSYQCBCJRA7rMBwKhbTeNpmAJEmYfPtJNO7j9TbSuwaUcMNjyxk2bgJXPricrIISPAeqeOFXs8gO1SHIcsY1vtokqrvaXzVJMB3o0S+4adMmvvjiC84++2zKy8szXr3fW7SfsKcWmguCgE6nQxRF9u7dS1VVlebpCYVCaWlh3RUSiQSyzkhr1XZaG2vJKijhit8vJ3tAIUPtMH5oIfc8uZyi4hLqDtSya1eF9r5MIhgMak26jgSz2Uw0Gk3LQu2RjW82m7W+k42NjWkfw97XUIkfjUbJzc3F6/VqXoVIROk7WVdXhyzLDBgwQBsPlCkkEgnCrhK+M6aU2f/zFKaCEZQWF2IUwRdXuqqdNraI019awY4dO5gyZQqrVq3KeAVWa2srZrO521x7VZH6/f6Ut27pEfGfeOIJvv76a+666y5uvvlmbrvttlTJlRZYrVYMBgPRaJSBAwdiNps5cOAAkiSRSCQwGAxa12S1i28m+1BKkoSkMxKXdZROOAtPFJrCSs9MvQgVPiUv/8TCQgoLC7VFnEmNL0kSfr8/qRpl1WnQ3RCJvkCPiD9q1ChGjRoFwBtvvNGrCzc0NHD++edz3333sXLlyj4f8JwMBEHA5XLh8/kYNmwYwWCQwsJCtm/fTjCotOUwm81kZ2cfM8Q3hFqQhBIiCYXwkQQkUAY7+4Fyj4xBlBidJWgaNpOlhz6fD0mSkirVVInf0tKS8tGkGes7sXDhQgYNGsRLL73UbbF5JBLB5/N1ePQV3G430WiUUCiEz+ejpqYGh8NBbm6u1kpQlmVtUofasTgTkCQJk/8Ag/U+7AYwtGl6naD01gkllKnmVQGZ2mD6g0Kd4dD5tkeCun9KhwndI+L/9a9/7fD3Cy+8cFQXffHFF7nwwguxWA52+UrFgOdkoLYZKS8vp7Kykrq6Onw+nzYworGxkZqaGhobG4HMEx8gS5/g5FwlFdmoA7te6a1j0St5+WY9BNr2h5kmvjpjNxniq8eo70klkjJ1Xn31VZ555hk2b97MCy+80CH34vrrr+/xRVevXs3atWtZu3at5rc9UrF5KmpuVai2Z0NDAwaDQRtUZjAYNPtULVBRo7yZgkpgURQx6QRcRhl/DPQ6pdwQWfm/URTIbsuja9/zPxM4cOAAkJyJaDabsVgs2ntSiaSIf8kll3DJJZewePFi5syZ0+uLrlixAoBZs2Zx8cUXp3TAc3dQNb66mTWZTIiiiMPh0PzPsViMSCTSoStEJtA+9N8agzwz2AwgyzDMKZBnFvBGldSSYBwIy2ntR9kZampqMBgMSbdjcbvd1NTU4Pf7U9revEeb2+zsbH7wgx90iK71pgnU8uXLASUinCmo86HUJDU42BfTarVqHp+qqqqUb7i6g0pgQRBwGaEhLGDRyQTjAjFJMXOcRtjaDBLKXTlmyVxOlVrTUFBQkHTbkJycHGpqaqiurmb06NEpk61HxH/wwQd55513MtYwNRVwOp3o9XoikYjmozeZTBQWFhIMBrFarVpAJVNtA1W0b7eda1Y6JX/dok48FAgnZHJMAu0Nm4TRTjyWmXiLOidMHVqXDNRjKysrU0r8Hm1uTznllG98Dv6hUF2akUhE8+JEIhEsFgsDBgzA4XBkvF+mikP7zDuNSrcFowjBuMyBkIxJlGivW41SJGMpC3v27AGUO2gwKLB1q4nduw+mLUgSlJcb2bbNhCqiWve8e/fulMrWI43f0NDA3LlztdZ735TJ5t0hKyuLhoYGQqEQFovlsPE/KvEzPURNzVzU6XT4ojIRCUw6mZqA4so0CFAbEhidJdASAasevFKAUAbSkmVZZufOnRiNRmy2bF55xYXPp3ynp50W5PTTQ/y//2dn1y5l71ZebuKCC3yIokh+fj41NTVaM69UoEfEf+2119i6dSt79+5l7NixfepWzBTi8bi2Z1EnpAwZMqSDTXosBK/gYAVYi2SkyaPY8DqUQJZdr/TPrAnKOA0yAywiFr2AQa/H25Z8l862hwcOHMDn8zFo0CAaGgwa6QEqKoycemqI3buN2nPV1QZCIQGrVaaoqIiamhq2b9/O6aefnhL5emTqzJ8/n2eeeYb777+ff/3rX1x11VUpESqdUCd/A9oGVx09qeJYI75POihfAgG3UcCkU7qoVfnh8zrY2CThiyojWeUMZGhu27YNgOLiYrKypA4F5tnZCUQRsrIO7kbs9gRms3LMwIEDEQRBO0cq0CPir127lqeffhqHw8E111xDVVVVquRKG9TeOqAOKTvcbRoMBjsE2zIF1b1qMyraU5ZlogkYZJfRiRCOQzShtBY8EITGcPrnx6pylZWVodfrKSgowOGQOP98H8OGRTjxxDAzZiiK5Mc/9jF6dIQRIyL85CetWrcFs9lMXl4eVVVVKRs3ddy2EATFzNm2bVsHe759EC0Wi2npDJmM2KpQ2xoW2kQSAZnaEIjIVPiUvjqBuJK6gKwkrwXjkrZY05lSrXabHjx4sHY3LSmJU1Li73Cc0ylxzjn+zk7BoEGDqK+vZ+vWrUyZMqXPZexVC8EXX3yxzwVKJ5qamggGg1qoPJFIaIugtbWV7du3U1FRQTQazVg//PYIBoNIriK2eQUa2/rihxJQE4K9SioRgvoQQJIPtkJpPxQ61di0aRPQu3GoJSUl6HQ6Nm/enJK06h5p/AsuuIBx48Zx+eWXc+aZZ/a5MOmGagaoWlEN9CQSCWpra7X05HT2dDwSAsEg8exipLZJKJ6IQvw2JY9eVJrGAuSYwagT0LfluKv7lFQjGo3y9ddfY7PZemURGAwGiouLqaysTEnwsEcaf/Xq1fz85z/n888/54ILLjjqGVjHCtxuN0OHDsXpdCKKIqIoEolE2LFjB/X19dpsLMhcgbmKRCJBKBhEZzBwIKQMcY7JB3Px7QZwGCDfAgVWGOmEIQ5BK+5IF/G3bt1KNBrVRkb1BsOGDQPgyy+/7AvROqBHxPf7/Xz11Vds3boVvV7PiSee2OcCpRulpaWcdtppWK1WYrEYjY2NxONx7HY7oigSi8W0IFcmocYSrKKMTlB89m4j5JjAqVcyM0/PE7hwsI6fDNIxNluHSXewMD4dgUdZllm/fj2CIDBkyJBOj0kkIBxObkHk5ubidDopKyvr84XbI1Pn8ssv58c//jF/+tOfyMnJ6VNBMg21j6YoilrzUrfbjdFoZP/+/RnvE6rWAziMIlaLQpzmsBLEEkUYZFdGfh4Klfjq+1OJ6upq6urqKC0t7bQTxYEDet5800E4LDJ8eIQf/MB/xPaBgiAwfPhwvvzySzZu3MjUqVP7TNakiK/2wf+v//ovAN5++23ttWuuuabPhMkkDAYDgUAAnU6H16sMUygsLNR+wEy7MlW3XqE5QUKvDIIQBCi0KO3BK/3KVJRhDgm36WD1lVrrmg6N/+9//xuA4cOHd/r62rVWwmHFyKioMLFvX4RBg44cVR48eDBbtmxh/fr1TJ48uc/2WkmZOmovSejYBi7TRcx9CYPBoBWXZ2VlkZeXp21uofueMKmGSvyoyUUwriSoWfWQkBXiA3ijMv86ILO+UaY5ovw2oihisVhS5g9X4fP52LZtG1lZWV3OCBOEjnxJpkuLwWBgyJAhmpetr5DUr3nttdcCysbl3nvvxePxYLPZePzxx/tMkExDp9ORSCQ6VAxZLBatiCPTxPd4PEg6Ax7BjgnFZdnStsH1RECQQRLAaVAWw26fTHaeoqRsNhuNjY0pHbWzfv16ZFlm5MiRXW5qp04N8tZbOvx+kTFjIhQXJ5dDNGLECHbs2MG6desYM2ZMn8jbo1/zxhtv5KWXXqKkpITKykouu+wy1q5d2+OLfvLJJzz77LNEIhFmzJjBxo0bM1Js3h7qj2W1WgkEArS2tjJ8+HBtIWTanenxeAABg1ExuYJtKfZZBvBGFJ+9JCvmjiTLyO3IZ7PZaGhowOv1piQQF4vF+PLLLzGZTEd0O+bmJpg1y4Mk9awnvt1up6ioiOrqamprayksLOy1zD2ebF5cXAyg/Xs08Pl8rFixgt/85jfcfffdGS02V6FuatU2geocXtWcy1T3NBUejweLXqDIpixAAQGzDpojioY36hTPTjQBDWEoaLe3VINvyuLpe5SVlREKhRg2bFhSCuJovsoRI0YAyp2lL9AjEW6++WbOOOMMrrjiCqZOncovfvGLo7roj370I6LRKE899RSXXHIJkLlicxXqD6amKDQ0NLBu3ToaGxvTntl4KGRZxuv1YrPZGOwQOCVXYGKukrKgTjtMtJnPhRalJNHbzopQS/hSRXzVhan63VMBtTZi69atfdJmsEfEv+KKK/j888955JFH+OCDD7j66quP6qItLS3ccsstLFiwoMNkc7UF+aG455578Hq92iMVyXFqKw6z2ax17fV4PJr/OJMa3+/3a7EFUIpPvm5RCjn0gvLINil+fV9cydBsbJeTpmp8dVp7X+LAgQPU1NRQVFTUZZBv3z4Dr7ziYuVKJw0NR2cyqgsrHo+zZcuW3ogMJEn8L774gjPOOIN4PM7SpUs577zzOP/881m5cuVRXfSOO+5g37593HPPPZx55pnMmzePTz75hBkzZnR6vMlkwul0dnj0NVRTJxQKEQ6HNV9+MkMNUg3VI9M+XygugUmvuDQTKKkL3hj4okpUt8ovs8MrdXhfKjw7al7O0KFDO309GhV46aUs1qyxsXLlZyxZ0so779j5+9+d7NhxML26traWjz/++IjXGjx4MKIoatfsDZLa3C5YsIB//vOf6HQ6FixYwPr168nLy+Occ87RfPs9waGzs1SvUSahanyXy4UgCIe1tj4WiK9qVE9Eid7GEhBvy9UBJS1ZkhWPTyQBe3wyAywyjjZffl8TP5FIsHXrVsxmc5d366YmHeXlJpqb36Oi4jK2bi1Blt/A4RhIXZ2evDwP4fB+rr76ampra1m0aBHTpk3r9Fwmk0nb5NbV1fWq9jvp+7fdbmft2rUMHTqUgoKCDsOQvw1QiW0ymcjPz6ewsJBRo0aRl5enLYpMQd3MWywWGkMSn9dJHAgpFVeCCAad0mJELesQBKW7mkFUFoDqy+9rp8DevXsJhUKUlpZ2aQoqBScJLJaxGI0lxGJ7eP31n9Dauh9ZFti9+wBXX301VVVVFBYWMnLkyCNec9CgQYCyoe4NkiL+8OHDefDBB7n77ru55ZZbaG1t5ZFHHmH8+PG9uvixBPWHEwRBy8NvT/hM2/gABpOFcq+smDNhpeCkfRqyVQ9ZRrCIyr8uo/IvKIsmEAj0adBRDSgdycOXmxvnzDMDnHRSHj/+8T9wuwfh8+3l9dd/QjC4lrvvvoKqqipKSkpYsWJFt65KVen2NpiVlKnz5JNP8tFHHzFz5kxOOeUUtm3bhtlsZtGiRb26+LEEURQRBIGCggKqq6uxWq14PB6NdJn042sbbKMZYgcLTiRZSUN2GpXAlSQrWZpRScnSPCFLxCAeTF1oamoiEon02VyDXbt2YTQaj5i3pdPBhRf6qKoyYLHYuPHGF7niiqupqdnL0qXKYJFkSQ+K6TlgwABqamrw+XxHvd9LSo3pdDq+973vccoppwBwwgkncMstt2Q8f6UvoRJbkiRMJpP2t5qWnEmNr7rvXFYTJr2SrqATFFPHaYDx2TAqS8BtUnLw7QYBgyhopIeDtQd9NXHE6/Xi8XjIz8/v9rvR62HIkBgFBXEKCgr5xS/+1OH1P/zhDz0KSqm2/d69e3sqtobMRmWOIahEb096OJiclkmNr2aNGvQ6xrsVk0YnKOZNVAJ/DOIJmUBc2fjGEjLZpo57EvVzqJv23qKmpgagx1m677/fzEMP3dHhubvvvpva2tqkz6FesyfvORT9xEfR8i0tLUiSRDAYZPjw4RQVFVFcXEwsFtNcm5lCNBrVFp5RL1JiFyiwKB2SExJs88CXTYrNr2vje1bHRhGad6qvpgqqqRw96TxRW1vL/PmXEwjsxWQawgknfEBubilVVVWaVycZqLURvemqfNwTPxwO8/bbb2tf+rZt24jFYmRlZVFXV6dtCPft25cxGdsnl0USSqFhTFKitVFJefhibV0WJMW9GTykm4i6Se+rza0aDEu2sWttbS1XX301Pt9eDIYhDB/+Hk7nJP77v1/VJrQnS369Xo/Vau1VQO64Jn48Hmfjxo1UV1drfWfi8Tgej4dIJEIsFtMCWOkq3esMaue6uCRT1iLTGlOS1KJtgyCiCSWKG0ooGZuVfvBFU9saXC1eT3ajvGPHDmpqarHZBjNkyHvIcjEjR4aZMiWXFStWUFJSQm1tLTt27EjqfCaTqVcF9JnNtc0w1LGeJpOpQ72B0+nEZDKh1+u1fPxM9s3U6XRIkkQoAREJQnG0/pgybf9v2+wG44rt/2kduE0STqOi29TP0Vd7lWSHNoOyKAsKzuGHP1xBVtYYYrE8QqEoY8ZEqK7WM3RoEStWKAPrugpeHYreDoM+rjW+2glZFEXN65GXl4fD4UCn0zF06FDttb7KAz8amEwm4vE4RkFpCBuIQzihuDQFlGJzo9iWkoxiAoXisKX5oFmjkqSvSih7Yjp9+KGNf/7TgSj+kKqqoWRnJzAYlCqsjz+2s3KlC7e7KGnSq9ftTVDxuCa+wWCgqKgIh8Ohtadu30tevRNkcqgzHKybleMxCq1KMpq7bfMal5VFEIq3eXpQktZ04sFUBjjozemrbhHqAkpG61ZUKMcWF8fIyUkwdGiU4uIoFosiYWurjgMHemZ8dNX1Llkc18QHpZI/JydHK9A4dGzOsdBMSg3SBINBsk0CLpOgkFo+GLnVCeA2KH59lxGKLJBnOagRQ6EQOp2u0yLw3siUzN4nJ0fZPwkCDBkS5fvfb2X48IMLRqeTycpKfniFLMsdGoEdDY5rGx8gPz+feDzO/v37O9w6JUnSis4zTXx1Ufr9fkqzs8k2StQFFa2ekJTsTEmG+hA4zFBsg0KrSHE7sdWhyX2Vc6T60n0+X7dVXT/8YSvr1lmJRgVOPjmE0QjTpvkxmyWamnQUFMTpSWVnKBQiFov1at913Gt8UPplTpw4UcvKTCQS7N69m127dgGZzcyEjiTzRWUaIxBuc2eq9ycZCMrgjyq5+HaDkrcPipkTiUT6tCWM2mM0mdGcNpvMjBkBfvADP3l5imY3GqGoKMbevUb+8x8rf/2ri7q65Dbe6jW7GhaYDI57ja8iGo1iNpsJhUIEg0FCoRDxeFy7rapDIzIBNUTv8Xgwh9sILcvEOtlXBhIQjsHeVokso4heFDR/d1epw0eDgoICjEZjUhMKo1GBlhaRrCwJk0kR+uOPbbzzjp2KCiNOZ5zvfCdKWZmZAQO6N53Ua/amN+cxQfxNmzbx+OOPIwgCTz75ZNq7llVXV1NRUYEoivj9fs1FpzZhampq4t133+XUU0/NyDAMm82Gy+WiqamJEkGmLijTfITMg9oIBFtARmasu2805KFQvV7l5eV4vd4ufzOfT2TVKid+vw6rVeKii7zYbBKbNpnZscPYtvE1sW+fiZNO6j6PSJIk9u/fj81m61XR+TFh6vz5z39m8eLFXH/99bz66quHvZ7qYnM12Ul1GyYSCYqKijRPiJrLnuq5TEdCSUkJ4XAYpxTQ+ugcCVEJWqMy9aGDof3eNAjoDGPHjgWOnCz29dcm/H7FhAkGRbZuNaPXQyQi0NSkRxSVnP1IRKC5uXs9XFdXRyQSYcyYMd98d6aaEdlVwXmqi83VBC6161gkEiE7O1t73mAwIAjCYZNS0gn1tt5YfwCHQfHdd0UTdTyQKIAgSzQ2NlJQUNDnptrIkSOxWCzs2bOny5Giqsuy/d+iCDNntmI2y5hMMk5nArtdwuns3rNTUVEBwEknndQr2Y8JU8disRCLxbosOD90srnX66W0tLTPNH9JSYmWWx6NRmlqasJkMpFIJIjH41gsFvLz8/v0mj1Fbm4u4XCYPXv2cNLJRayN6vELIlbktjbhAiIyMUnAIMrkGmWydRKRuv0EAgHGjh2bEtlHjRrFF198QVlZWaeNYocM8bNvX5z9+40UFkYZNsyD3w/Dh/u5444oS5YUEgqJTJ3qZdKkOo5U1KfedQcOHIjVaj3i53E4HEe8IwjyMdAHcMOGDTz99NNEIhGef/75bt2H1dXV34rBc/1IHbxe7xH9/McE8XsKSZKoqanpdlUfLXw+n5YxmIqODn2Nb5q8kHqZu+PGMWHq9BSiKPb5Rq0zpKqVSarwTZMXMifzMbG57Uc/0o1+4vfjuEQ/8TuByWRiwYIFGZ+Ckiy+afJC5mX+Rm5u+9GP3qJf4/fjuEQ/8ftxXOK4Ib7H40nrdO/e4FiRtTd9a1IJtadPb/CtIP59993H/fffD8CaNWtYvnz5Ycc88cQT1NfXd3mOWbNmpUi6jufurawbN27k3Xff7faaDz74IADf+973+Nvf/nZYh+ruUFlZyXPPPXfEY8rLy7nsssuYPXs2v/nNb7RrHgnvvvsuGzdu5He/+x233XZbUu85FPPnz+/xew7FNzKA1Rk2bNjQYWDAZ599xpIlS0gkEtx77718+umnuFwubr31VgAWLVpEWVkZwWCQZ555hp07d3LHHXcQCAT485//zOzZszGbzZx77rkUFBTwt7/9jdbWVm6//XYWLVpEPB6nuLiYK6+8ElmWee211ygoKGDTpk34fD6efPJJHn30UQKBAOvWreuVrL/97W9pbW3F5XIxffp06urqeOONN3j33XdpbGzk/PPPZ926dRQWFvLVV1/x3HPPsWPHDv75z3/S0NDA+PHj+fvf/87nn3/OSy+9RFNTE4sWLeKee+7BaDRiNpu56aabmDNnDqeddhput5vCwkI+/PBDbr/99i4DTB999BGzZs3iBz/4AStXruSNN96gpaWFm2++maysLGpqavjlL3/JokWLKCwsZNKkSUQiEQKBAKtWreKuu+5i9erVhEIh7rrrLiKRCDNnziQQCLB+/Xr27dvHkiVLuPnmmxk7dixbtmzhscce48svv+Tjjz/uUXH6ofhWaHyABx54gN/+9rdaluDSpUtZvHgxCxcu5Nlnn2XKlClccMEF2vHr1q3jqaee4txzz+W9996jpKSEhQsXMnz4cMrKyqivr2f69OmcdtppLFq0CKvVisvl4osvviAUCvHYY49x+eWX8/rrr7Ny5UouueQSFi9ejMlkQpIkNm7cSEtLC48++uhhE+B7KmtVVRXDhg3TxiaBMmv4ueee47LLLgOUcrw777yT73//+3z99dcAzJw5kwkTJmi5Ty+++CJPPfUUTzzxBJIkcfnllzNlyhQ2bNgAwIQJE/j973/Ppk2bmDJlCjNmzDhiVHX27NmUlZVx0003sXv3bnQ6Ha+88gq33norf/zjH7XPN336dB577DHef/99QEm4mzBhAldddRWgLKAZM2awePFixo0bx4QJEzjvvPOw2Wxs27YNWZb5zW9+o83BmjBhQq9ID98i4judTubOnctjjz0GHGx70dVEk0NfV/3Jat+Z3//+9+h0Ou677z5isRj33nsvt99+u9ZmxOVyMXLkSPbs2UNVVRVDhgzB7Xbz8MMPM2vWrA6Ziof2sumprDfccANjx47tkKEajUYPO9ZqtWIwGA4rmFehtkP0+/20tLSwZMkSTjjhBK12Ve3AoNfrk8qBWrJkCbNnz+app57SKtXUa/RELrXSDWD37t08+uij6HQ6TjzxRGRZ7jBku6tz9BTfGlMH4Nxzz+WVV14B4Prrr2fu3LmIosivf/1rPvvsM5599lkefvhhACZPnszNN9+MJEk89thjLFy4kN///vd4PB5GjRrFtddeS35+PpMmTWLEiBHMmTOHSCTCwoULO1xzyJAhGrHPOecc5syZQygUYtmyZeTl5XHnnXeydevWXsm6cuVKgsFgh3kEP/rRj7jhhhuoqqrSNGd3+NnPfsa8efPw+/088cQT+Hw+li1bRk1NzWH9cdxuN2vWrDni5JEzzzyTa665BofDgdlsxmQycdlll3H77bf3qL535syZ3Hjjjbz99tv86Ec/Ijs7m3feeYfdu3d32s/Ibrfz2muvcdFFFyV9jUPRH8D6hmLZsmXs3LmT5uZmHnjggS6niacb27dvZ/HixSQSCaZNm9bBZDuW0E/8fhyX+NbY+N1BlmUefPBBZsyYod3Wu3JhVldX893vfheA+vp6Zs+ezU033cRHH30EKDWskyZNAuCFF15IqtNAquRtaGjg0ksv5aqrruLll19m//79XHnllVxzzTVUVlaycOFCrrrqKk4//XSee+65lMnbW7nLy8u59NJLufTSS/noo4/4xz/+0amJ2Fc4boj/5ptvMmnSJIxGYwf/9BdffMHs2bOZPXs2t912G6AUv6t931944QUGDhxIKBTihBNOAGDhwoXaELJLLrmEp556KmPy1tXV8eCDD7JixQree+89nn76aR566CEeeOABnn/+ee644w6WLl3K+PHjmTt3bsrk7a3cjY2NLFq0iOeee4433niDH//4xyxZsiRlch43xF+9ejWtra1cdtllfPbZZ9qQaFmWicfj2gPgkUceweFwAEogZ+bMmcyfP58nn3ySF198kQsvvFDzNDgcjpT0zk9W3nHjxjFs2DAefPBBZs2aRUNDA0VFRR0K919++WUuuugiBEFImby9lXvq1KlkZWVx//338/Of/xxBEGhubk6ZnN8qr053UL0vf/jDH7jjjjswGo1MnjyZyZMnd/me3Nxc7HY7WVlZSJLE6tWrWbt2LWvXrmXVqlX89Kc/TdkssGTklWWZ+fPnc8455zB9+nTNE5NIJLTC/U8++aSD9kz17LKjkTsSiXDnnXcyd+5c7c6aSjmPG+Kfc845WnuQwsJCZs6cyZo1a7p93/XXX8/dd9+N0+lk/vz52gTvWbNm8dOf/hSfz8fAgQMzJu+KFSt499132bdvH59++ik/+9nPmD9/fgfXa3u/eqrk7a3csiyzadMmHnnkEaZMmcK8efN6NGaox5CPIzz66KN9fs4lS5bIzc3NfX5eWf7myauiL+R+7bXX5IqKij6QpnP0uzP7cVziuNnc9qMf7dFP/H4cl+gnfj+OS/QTvx/HJfqJ34/jEv3E78dxiX7i9+O4RD/x+3Fcop/4/Tgu8f8BFC+JxitqesQAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 174x140 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:33: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
      "/tmp/ipykernel_1517215/2945031149.py:32: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  all_exons_depth = exons_depth[[\"GENE\", \"EXON_RANK\", \"DEPTH\", \"COVERED\"]].groupby(\"GENE\").apply(\n",
      "/tmp/ipykernel_1517215/2993950153.py:74: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  average_dnds = domain_selection_gene.groupby(\"selection_id\").apply(lambda x: x[\"dnds\"].mean()).to_dict()\n",
      "/tmp/ipykernel_1517215/802346302.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  g0[\"Group\"] = \"G0\"\n",
      "/tmp/ipykernel_1517215/802346302.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  g1[\"Group\"] = \"G1\"\n",
      "/tmp/ipykernel_1517215/802346302.py:33: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  sns.stripplot(ax=axis, x=\"Group\", y=\"pACC\", data=df,\n",
      "/tmp/ipykernel_1517215/802346302.py:48: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.\n",
      "  axis.set_xticklabels([rename_labels[label] for label in color_dict.keys()])\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAL4AAACmCAYAAABk8TVrAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAPLVJREFUeJztnXl4VOX1+D/3zr4kk5UQQgIkLLIIElRURBCXqrhBWwtaISpIrVYRtf1ZF+ym9qu0xSouIIJQLZYKVtEqyKYsFlmULSQQshJIQpKZySSZ7d7fH8MMGZJJZsIsSPJ5njyQmTtzz2TOPfe85z2LIMuyTDfddDHEWAvQTTexoFvxu+mSdCt+N12SbsXvpkvSrfjddEm6Fb+bLkm34nfTJelW/G66JN2K302XRBlrAbrxZ+PGjUyZMoULLrjA99ioUaP497//Td++fREEAbvdzuDBg3nzzTc5duwY06ZNw+l0MnLkSF555RVEUeSpp55i3bp16HQ63n77bXJyctiyZQuPPvooCoWCe++9l5kzZ8bwk8YYuZtzig0bNsjTp09v9XifPn38fr/jjjvkzz77TJ41a5a8atUqWZZlOS8vT/7yyy/lb7/9Vr7ppptkWZbl7du3y5MnT5ZlWZZzc3PlsrIy2W63y5dccolcXV0dyY9yTtNt8X+AOJ1OGhoaiIuLY8GCBb7Hjh8/Tnx8PFu2bOG6664DYPTo0ezevRuLxYIkSfTu3RuAMWPGsGXLFm677baYfY5Y0q345yD//e9/GT9+vO/3++67D4Dx48cjCAKCIDBx4kTGjBkDwIkTJxg7diwmk4mcnBz++9//kpGR4Xu9LMtYLBbi4+N9j8XFxWG1WqPzgc5BuhX/HOSGG25gyZIlfo8988wzbNy4sc3j09LSKCgoYPny5Tz66KPk5ub6KbUoisTHx/s9ZrVaMZlMkRD/B0F3VOcHzr333suOHTsAfIp8+eWXs3btWgC2b9/OkCFDfNa+vLwch8PBV199xSWXXBIboc8Bui3+OciZrk5SUlLAY2fPns2DDz6IUqlEr9ezcOFCevXqxfDhw7nssssQBMF39/j73//Oj3/8Y1wuF/fddx89e/aM8Cc5dxFkubsQpZuuR7er002XpFvxu+mSdCt+N12SbsXvpkvSrfjddEm6Fb+bLskPUvG9W/DdkdhuOkvUNrA2bNjAihUruP3221m5ciWCIPDGG2+wbNkyvvnmGwwGAy+//HJQ7+XdbjebzX75J8G8btmyZZw8eZLBgwej0+n49ttvg3qtWq3m1ltvxel0sn79emw2G+PGjeOqq64K+vydwe12s3TpUioqKhgycjT/kzKxudt/jQhcmAij4ptZu/YLJEni3nvvJS0tLaKydsSXX37J1q1bufLKK9m0aSs22wQyMy+nf3+44QYQo2iGo3KqoqIidu7cSXNzM8uXL2fRokWMHTuW9evXs27dOl5//XXi4+MpLCxs8/V2ux2LxeL3EypWq5V3333Xp/RDhgwJWukBHA4H27ZtQ6/XM2HCBAwGA5s2bWLTpk0hyxIKX375JRUVFfTt25dDyo6VHkAC9tVBlaxj9OjRuFwuVq5cid1uj6isHdHY2AhAfHw8paWZHD6sp7YW/vc/2LMnurJERfGzs7N5/PHHAVCpVACkp6dz4sSJVr+3xQsvvIDJZPL9ZGZmhnT+hoYGli1bRm1tLUOGDOHCCy/slJvk/eK8ym80Gtm8eTObN28O+b2CIT8/n2+++Yb4+HhGjRqF1Rn8a92A1SHTq1cvBg8eTG1tLWvWrImpe2iz2QDQarVIkh6n8/QHamiIrixR9/EVCgUAlZWV9OzZs9XvbfHkk09iNpt9P2VlZUGfr6mpieXLl/ss/bBhwxAEwXfeUGh5wen1eq6++mqMRiObNm1i+/btIb9fe9TX1/Pxxx+jVCoZM2YMsqjALQX/eoUAPfWe/w8bNozU1FT279/P7t27wypnKDQ2NqJQKFAqleTk1CHLTQDExcGIEdGVJeqKP2nSJGbNmsXmzZuZMGECV1xxBQ899BAWi4X+/fu3+RqNRkN8fLzfTzC4XC5WrFhBdXU1AwcO5MILL0QQBMCzQPbebYJFq9X6/a7X6xk/fjx6vZ61a9eyb9++kN4vEJIksXr1apqbm8nNzSU+Pp4Gp0wQXs5p2UTQKzyfVRRFLrvsMtRqNV988QU1NTVhkTNUbDYbGo0GQRBITXUzatRWpk+XePBBiHaGdFSzM71ZghMnTvQ9du+993LvvfeG/VyyLPPJJ59QVlZGnz59uOiii3xKD9Dc3BzyeyqVrf9cBoOBcePGsW7dOv7zn/+QmJjoVwTSGXbs2EFZWRmZmZn07dsX8FgoFRCsl64WwCmf/rx6vZ5LLrmELVu28PHHHzN9+nTEaK4m8Vh8g8EAeIyISlVNWlozWq0+qnLADzScGQy7du1i7969pKSkcMkll/gpPXjWGmc+1hF1dXVtPh4fH8+YMWOQJIl//etfNDU1dVpus9nMhg0b0Gg0jBo1yiejUSWgD+EGZZVAI/r787179yYrK4vy8nJ27drVaRk7g9PpxOFw+O6aGo0GOO33R5vzUvHr6upYu3YtGo2GK664ok1/XqVShezqtPclpaWlceGFF2K1Wvn8889DltnLunXrcDqdXHTRRT7lAHDLoA5hWeKQwN7GmmDkyJGoVCo2bNhwVhdoqHgDA97P5P3X+3i0OS8V36s8ubm56HS6No8RBMEvqhAMktT+6nLQoEEkJyezd+/ekBbgXo4fP86BAwdITk6mT58+fs81uyG0+xM0OVtHcLRaLUOHDqW5uZlt27aFLGNn8Sq4Wq0GwOk0cPhwNps3y4T4NYSF807xjx8/Tn5+PikpKe2GPWVZxu0OZbnYsXUSRZGRI0cCBKyPbY+tW7cC+C3CveiV4AwhqgPQ4Gr78f79+6PT6dixY0fUYvveu4tGo8HhgPXr+1Ja2puvv9bw739HRQQ/zjvF37lzJwBDhgxp14d3OBwdWvAzOTOq0xbJycmkpaVRXFzMyZMng35vm83GwYMHMZlM9OjRo9XzSlEgpePT++EKELJXKBT0798fh8PB/v37Q3vTTtLS4lssChwOz4dxOp2UlkZFBD/OK8WXZZlDhw6h1WqDqicNdXHb0uduD28k5tChQ0G/96FDh5AkiX79+gWUa1CCEJK70+gI/Fy/fv0Aoqb43iiaJzTtxmTyXJUul4tTokSV80rx6+vrsdlspKWldajUSqUyZIsfrGvkvehC8fOPHDkC4Gv41Ob7tr1cCYhRHfg5nU5HUlISpaWlIa91OoPX1VGr1ajVMHlyPX37ljJs2DEmTYr46VtxXil+bW0tQFAbXJ35soONAmm1WtRqdcDwZ1tUVFSg0+l8ce622HRcJtiEAxWQZWz/601NTUWSJI4fPx60nJ3Fu5bw/g0TExVkZxfTr18lbWyPRJzzSvEdDs+9PRgF7czmTSjZjSqVyidPRzidzg4bPEmyTHEIuXkS4OrgjuY1EF6DEUm8ro73u/H+W1am4sABcAVYiEeK86qvjleZg3FhlEolgiCElLTlDcUFgyRJQV9cLZO3AiEKQkhRHTdQ2QimdpYl3lBvNDaRzrT4oihSXNyf6uosLBbIzIR77oleavJ5ZfGNRiNAUBszoigGFaVpSbA5Lm63m+bmZp88HeE6Ze7aS5xzSjJiiIH8jo73ns8VBXPb1t34+PFevnVTWRlEM4UoJMX/5JNPgr59x4LU1FQgcGrBmYSq+MHm93irw9oKS7aFNweovcWziBzyBlZcB/dz7/naykEKNw6HA0EQ/O6CRqPDJ4NaDUHaibAQ0icuLi7mZz/7GXFxcdx2223cdNNNAXdGY4FaraZnz55UV1fjcrk6/EJD7RYc7EVfVVUFEHTdgHdB295mkkIUEYXQNtwsTkhv53nvhazXRy5JzG6HkyehsVFCoVD4Rdtyc4+wb18G2dkwdixEUIxWhGTxH3roIVatWsVvf/tb3nnnHbKysrj77rt9m0bnAjk5Objd7qAiFaHe4oPNbSkvL0cQBLKzs4M6XqVSYTAYOrwQdSEa5o4uk4ZT1R8JCQmhvXGQ1NfDa6/BW2/B2rWDcDj8TbrR6GbIkH1Mm0bUY/kh/Sn/8pe/sG7dOhISEpg2bRoffPABTqeTcePGsSfatWMBGDp0KFu2bKG4uLjdmHioMXwIzjJarVZqamrIzs5uNzR5Jj169ODo0aM4nc6AUSl1iCuypA722+rr633njgS7doG3SrSpSUFFhf/9R6FQ4HK5kGU55M3EsyUkxbfZbKxatcq3g7lz505GjRrFJ598EtJJd+7cycKFCzGbzQwfPpwjR474is87UxnVkrS0NDIyMjh27Bg2my2g8nUmjt9e12Ivhw8fBvDl7ARLRkYGR48e5eTJkwF3ndN0UBFCQmV7Pr4sy9TU1JCYmBgxV6elFyzLMhqNv7HxfteSJJ319x4qQSn+l19+yddff83SpUt94T9Jkli1ahXfffddu5a1LXbs2EF+fj6CIGC1Wvnkk0949913Wb9+vW+ETUvsdruf/9tRsfkll1zC6tWrOXToELm5uW0eI4qiL5xZWFhIjx492oyjm81mqqqqGDBgQIfhSYfDQVFREfHx8QwaNKjdY8+kb9++fP311xw/fjyg4itFT4ZmMAFYFXDQDBentv18XV0dDoeDoUOHhiRnKFx6KVRWwtGjkJZWTb9+/jXVLcPP0Vb8oG6eubm5jB8/nssvv5zx48czbtw4JkyYwJo1azp10osvvph169bxm9/8xpcrEs5i86FDh2IymSgqKgoYiVGpVGg0GgoLC3n//fdZsmQJZrPZ7xiz2cySJUt4//33OXz4cIcpCwUFBbhcLkaPHh3yF5mZmYlarebYsWNt7i1Iskx9CAE1J+1/uRUVFQAByz3DgUIBkyfDY4/BhRce4kwPzuvedMbtPFuCUvyPP/6YcePGMXDgQDZu3MimTZvYsGEDixcv7tRJ582bB3jCjyUlJUB4i81FUeTKK6/E7XZz4MCBgMdptVqfpa+rq/NTfq/S19XVYTKZ6NmzZ7uujt1up6CgAIPBwKhRo9qVry2USiX9+/fHarW2ugDBY+mVQnDW3ktSgP02WZYpLy8/VfSdE7KsnUGSpFZ+fCwVPyhX56KLLgJODx87W376059y9913o9FoWLZsGbNmzcLtdvPWW2+1ebxGowk6M9LLiBEj2LJlC0eOHGHQoEGtfH2Xy0VzczMmk4m8vDyfki9ZsoRJkyaxatUq6urqSExMJC8vj+TkZFJSUgKeLz8/H6fTyYQJE0Ku7PIydOhQDhw4QElJSatIiyAIVIdQJiwCmgA3nfr6eiwWC0OGDOm0rKFiterYsWM4hYVJXHhhM1dddbq2IRYtT4JS/L/+9a9tbu8LgtCpTmKTJ09m8uTJvt/vuuuukN+jIxQKBePHj2f16tXs27eP0aNHtzrG6wadqfzeO5lX6U0mE5Ik4XA42ty3aGxspLCwEJPJFHBNEQz9+/dHq9VSUlLChRde6LemsDklzKdcnZJdm0nKGkBcSusovbWmktrSQgaOuooUXdtG6ujRo4Cn4CVaFBT0w2bTIkkC332nIzvbEfVITkuCUvznnnsOCD1/PdYMGzaMbdu2UVJSwgUXXOC3eD0zV8dkMjFp0iQ/923SpEm+1zidTqqqqlqVBIInp93tdjNu3Liz2gVVKpUMGzaMb7/9luPHj9OrVy/fc5IMyB6l//SFhzCmpHP7H5b4Kb+1ppLVz+TRUFNJyrOvIlwwvtU53G43JSUlGAyGiPr3rc8rnvF7bHUpKB9//vz59OnTh+nTp5OXl0deXp7v/+cygiAwYcIEZFlm7969rZ5vmb5sNptZtWqV3/OrVq3y87fbSoWwWCwcPXqU1NTUsFhQbxi0qKjI73GdErQKSMoagDElHcuJMlY/k4e1phI4rfSWE2UYU9JR9hqAso1knYqKChwOByNGjIhqe5H+/UtRKj3BgX79HGRmOn1GJxYGNahP/pe//AXwNH5du3Yt//znP9mwYQPr16+PqHDhICcnh6ysLCoqKlqVAnqTyFouZBMTE7n33ntJTExsteBti/379yPLMldffXVYFKlnz56kp6dz7Ngxv51iSQZBhLhTlj4+LdOn/JX5u31KH5+Wye1/WII2OR13G4tGb8FLqPsMZ0tyspWrr97FPffUMXGiNaoNYtsipNMvXryYcePG8eijj3LllVfyz3/+M1JyhQ1BEHyjM88ss1OpVK2UPi8vj6ysLPLy8lopvzcJzovFYqG0tJRevXoxcODAsMmcm5uLLMs+Xxw8NbfqU3H8M5X/w9/e5af0cSnpyEDlGZtdVquVqqoq+vXrF9RmXDgRBAGlUsJolPAaeK/Fj3ZjK+iE4n/11Vf84x//YOPGjbzyyiuRkius9OnThz59+lBZWenbppckiebmZqqqqjCbzX4LWTi94E1MTMRsNlNbW9sq3Hrw4EEAxo4dG9bb9bBhw1Cr1Rw5cuS0cggCJvXpcGZcSjrXPvKi3+uufeRFP5+/8Yz2Il73KVrW3umENWtg8WIoLU1vFRw55xW/tLTUZ9nWrVvHyZMn+eqrr6K6ODpbxowZA5wuAPe2FxkwYABTp071U3ovXuWfOnUqQ4cO9UtzaGpqorS0lNTUVAYMGBBWWdVqNcOGDaOxsdG3qSfJMjrl6d461ppK1s3/f36vWzf///l8fgBFi29XkiSOHj2KXq/nggsuCKu8gdi40dMCfNMm2LDhUnbv9s9EO+d9/Llz5zJ37lwMBgPvvfceTzzxBO+++25U8rjDRXZ2NsnJyZSWlmK321EoFD4ff8CAAQHL/kwmEwMGDPDdIbwUFRUhSRKjR4+OyBfntcped0cUBOxuj8VvuZCNT8tk8vP/8PP5rTWVntSGFgb22LFj2O12hg8fHrX0gNpaT4FJSQk0NenIz+/LkSOn9w28G1exsPhBae7s2bMZMWJEqyEIP6TwpiAIjBo1ii+++IKSkhJycnJwOp2o1eqg8uydTqdP8b3+t1qtjliuS3p6OqmpqZSXl+NwOFCpVEhya6X3+vS3/2GJ7/HVz+Rx+x+WoOt72u0pLi4GPBt70eKii+Cjjzz/Vyrd6PVN1NWZ8CRU/ABcne+++w7Al67g/elMt7BY4u2NX1paisvlQpKkoLfL3W63rynSyZMnsdlsDBkyJKQ63FAQBIHhw4cjSRLl5eVIMqhEqC0tpKGm0k/pwX/B21BTSX1pIae6hGO3230pIZFKQW6LQYNg9mwYNgyysqrRaOxkZ592F2ORjuwlKIs/bdo0wLPDarFYEASBV199lRkzZkRUuHBjMBjo27cvR48eRZIkZFkOuhhFFEWfi+BN8Bo8eHDEZAVPCsOXX35JaWkp2dnZaBXQJ/cqbnry1TZ3br3KX1taSN/cq7Cd0rHy8nIkSWLYsGERlbctLr0UsrJg/vwj6HRVJCWd7lQRSkF+uAnprPfccw9Go5Gnn36aOXPm8OSTT0ZKrojhXYieOHEipL6RGo3GF848fvw4SqXS1zEtUphMJnr37k1VVRV2ux2T2mOp+uRe1Wa6AniUv2/uVWgUnjRm8Cg+eNoqxoKePSErqxadrnWyUawsfkiKL0kSarXat+Brb2PnXMWrrFVVVSElR0mS5OuVU19fT2ZmZlQW94MGDUKWZSorK3FI/pGaQCgAkxp66ERfqkV6enq7fXu6GiEp/gMPPMCvf/1r5s6dy0svvcRvfvObSMkVMVJTU1Gr1dTW1qLVaoO+1UqSRENDg6/5UqjFN53Fe4eqrKzEoBTQKoL40gTopQedUqCqqgpJksIecv2hE5Li33TTTbzwwgunptZJnco7jzWiKJKWlobFYmkz4SzQa7zt/bwbYME0pQ0HKSkpxMXFUVVVRaLGs8DtaDkuy1B+qkeUdx8g2ML3SBFoIRurKYwhKf7PfvYz6urqmD17Nrm5uTzwwAOdOml+fj4zZ85k1qxZvPbaa8yYMYOZM2eG3K++s3h7RnoXfR1hNBq58sorT7W4tvjeIxoIgkDfvn1pbm7G3WSjqYMBESKeghVJBodboqqqCpVK5ZfpGQvaUnBRFGNShAIhKr7dbic7OxtBEJg4cWJITVFb8vbbb5ORkYEoiqxbt85v4HM08OapeK13RzidTo4fP44syzQ0NCAIQsRacrRFVlYWAMUnG3BL7VdhCXjWAaIIuN2YzWYyMjKiXtN6Ji6Xq5UMoigiy/K5W4jiZeLEiUydOpXf//73PP30050uINm7dy+LFi2ioKCAa665Bmi/5jbUYvOO8CptsH/wpqYmDh8+TGZmJjabjfj4+KgqktdamxtsiB00RBDwzMrqqYXaU4Yp1tYePPsgbSk+eC6KaFWCeQlJ8X/zm98wceJESkpKmDFjRsgTxr2kpqZiNBpJSEjwG/Ac6At64YUX+N3vftepc7WFyWQKeciz1Wr1pQq3V4IYCVJTUxFFEbu1Fpeu485LahEMKgFLXT0QvfVIe3h3yVvijYq110soUoSk+E899RT19fX873//46GHHuKzzz7rVGryI488wqxZs4iPj2fp0qUd1tw++eSTzJkzx/e7xWLp9EUHpwtQXC5X0B2TZVmmoqICWZaJi4vr9Lk7g0KhICUlhUpbY4f9M12A1Ql9jDLFxZ5wczR3a9tClmUcDgcJCQk0NwvU1ChITHT7FN/hcES0jWFbhKT427ZtY/369UyYMIHp06cHVNSOuPjii1mxYoXv945cps4Um7eHwWDo1I6htzAk2Mnq4SQ5OZmKwmJkWSKYpdn+enA0NCEIQtRz78/E66Y6nQbeey8Bm01ErZYYMMBjQJqamqK6ZoIQF7cJCQmsWLGCpqYmPv/885hbks4iCAJGo7HNlhft4b0dR9vig2dBLohKhA4ajIh4XB1Zhlqnws+djBXeHKcTJ1Kx2Twq53CIlJV59Cea83a9hKT47777LpWVlYwcOZJ9+/bx7rvvRkquiBMXF+dLTw4Wr6WPleI7DMkdumUCkKjxTDV3NjeSmJgYHQHbwTt4Ij7e/28dHy/6PR9NQlL8VatWYbPZWLBgAXv27OHjjz+OlFwRJz4+PqQwWksfNBaujslkQhLUuNvx8gVAr/AUpZsEO6qm2pinKdhsUFvr6cp8wQXNjBjRRGKimwsusDNqlCd3x9u1OZqE5OO/8sorfPPNNwAsW7aMMWPGcOedd0ZEsEjTcoHbEQqFguTkZJ+vGgvFj4+PRxYk3LIQcAdLBhyyJ08nWbIiQEwV/5NP4NtvobLSiEaTjMGgP9VIyuP61NV5BnOcbXi6M4Rk8XU6Hbt378blcrFz586wLjijjVd5g43o1NXV0dTU5FsfRBuDMQ67seOwpEuG0gbIt4jIghiTixTgxAmP0gPYbHaOHGndNt37e7AbieEkZB9/2bJl3HbbbSxevLjTvTPPBUKxhJIk4XK5sFgsxMXFxSSHvNEFsqbjC06SQZLA6pRxahNjsh4BT8NYLx6DIbcyGJ6Zt+qoTF08k5C+QbPZTG5uLmvWrEGn08XkFhUuQrWETqcTh8MRM9dBoRDQyk7/Qto2kAG7BHanG4j+noOXlBSYMMFzAUiSmeHDS9rcpIqLi6O2tjbqOTshKf7999/v65X52GOP8eCDD0ZEqGjgVYhQCyFipUgmrYoM6jzmvANcMtjdMqrmupi5OgBXXQW//rWD3NwN9O3btqp5+5Ke2ewr0oSk+LIs+7a/ExMTo5ZNGQm8t91gIzsqlQpBEGKm+AC9dTKiu5kOm4XLoG46iVqpjFhNcLDU1FQDgV1L7+OB8rQiRUhRnblz53LLLbcAnlv/s88+GxGhooF39GSwt1jvBRKLha0Xo9EANgmP4ge+U9kliLfUYDQaY94Jo7LS0+enrf2EpiaBHTsG8r//qVGrHUSzJDjk7MyJEyf6fveOtfwh4mlppwx6hKcgCAiCENJAt3CjNxgRpCBn7aoTyDAEX1McKbyF+W2lTWzbpqemRo3dfpzt2xXk50OUel2FpvizZ89m+/btVFdX43Q6SU9P98X1f4gkJiYGdYtt2VI82slULUkxGZEVbd+hvLOxPPZdxm5MQ6+Prt/cFuXl5ajV6jZdxOZmz11Xo9FgsViwWt14KoYjT0g+/p49e9i+fTtXX301+fn5IU8GP9cI1v91uVy+sZSxHGhdK2s9LZNlf/fFW3Ul4FEbAU9xh6yPbbqCzWajtraWlJSUNl2uiy5qRqXyGBO93kpSUsezicNFSBZfq9VSXFxMY2MjNTU1UV+JxxJv38xYbtq5RDWi246MDlD4LLyMJ5Ij4qm+0uOi2dGAUhud8shAeLu3BSrT7NXLxbRpdRw6ZObgwd0cP24iJycjKrKFZPHnz5/P/v37eeKJJ3j44Yd59NFHIyVXxJFluVPJUbFU/GEZCQhuJ4Is4Z35ILf4AXDLoJDdiK5m0uNiu7Pu7fvZXhavXi8zZIgJhULyXSjRICSLP2jQIN/81tWrV5/Viaurq7nlllt47rnnWLlyZdgGPAeLy+UKemHbklgqfpJRj7H2MHaVCvTJNJ4R1XQDGgE0kgOHq5nEuNitR2RZpqioCLVa3WGuvUajITExkdLS0qhVY8Ws3fG8efPo06cPy5cvZ/ny5WEd8BwMKpWqU0XOsYyLNzol5PieOEVNwFu1WwKLrMSlMcZ0IX7y5EnMZjNZWVmIoojbDTt36rBYRAQBJEkgM9PBBRd4jE/Pnj2pq6ujpKQkKu3nQ3J1/vGPf/j9/vbbb3fqpEuXLmXSpEnodDrf1R3OAc/BEmqM2xvdiRX/2V+F3ZCCU1DiCnDNuoFm1Li0iTQT3TrWlhQWFgKe7xVg61Y933yj56uvDCxdmsju3VrWro3jyBG133He10WaoCz+Bx98wOuvv853333H22+/7dfQ/7777gv5pOvWrWPbtm1s27bN5260V2we7ppbwBehCSUXPJbzAGRZpq7RiaRQAwpkGVQCnDH0BBkQJRdILhxC7OQtKChAEASfQldVeWRpaBCRJGhsFNHp3FRVKcjJ8ZRWqtVqCgoKuOGGGyJuYIL6y9xxxx3ccccdLFy4kJkzZ571SZctWwZAXl4eP/3pTyMy4LkjBEHAZDJRXV0d9Gti6eYIgkAvk4bDxxV4btQyTrm1cgiAS1CgdNnpaYqNq2Oz2SgrKyMlJcX3vfXt6+DYMRUJCW5qaxXExUmYzSI1NQrKypRkZrro1asXxcXFHD9+3HfBRIqQTEJSUhI33HCD36LwbJpALVmyBMBvNziaDBw4kMOHDwd9fCx3bQFuuiCFbYfKcEgCgcYGioDSZcdQe4TkuNj8XQsLC5FlmYyM06HJUaOaMZk8yq7XSxQWati3T0txsYaSEjW33WYhIyOD4uJi8vPzzy3Ff/755/n0009JS0vr+OAfAC1nWgXC2zfT20gqlpjtbmRRAZIQsH+mDAiSy5dUFwu8Q/FaNtZ1OAQaGwVUKhmHQ6C2VoFe7/kUsixQUaHi4ot7olQqOXjwIOPHj4+o/CEp/sUXX4zFYjlvFD+YFoharZbevXtTUFAQ0yiJLMtsL66jGTW+USdnIODpsKB02kAbm4u0sbGRoqIikpKSfHdISYLVq+MpKVHx/fdaNBqJjAwX1dVKBg/2ROvS010olUrS09MpKyujqqoqonoWkuJXV1dz//33+/JWBEH4QQx5DkQwHQiampo4evQosizHNIbvcEl8X2lFaicrU8ajZA6ljiQhuEkv4ebQoUNIkuQXfGhsFCkpUbFvn4bychWCIJOR4SI11cXgwc306+egTx/P3TcrK4uysjL2799/7ij+hx9+yL59+yguLmbo0KFhCyvGigsvvJC1a9e2e4wsyz6XqDMbXuHCJcs4XLInDwcIlJbsBlBokFWxySnat28fcLrRLYBWK+FwgNstoFTKCIJMXZ2CUaOauOYaGy09mp49Pe7O/v37ufrqqyPm7nS6heCvfvUrPv300x/EdPNAGI1GdDpdhw2NvOHbWBbe6FQKesSpsTbZcUiyf1FrCyRAQMCtjn7dgMViobi4mNTUVD+3UKmESZMs1NQoSUx0Iwgy/fvbyc1tor5eJDFRanGskt69e1NcXExFRUXEBnCEtIG1bds2XnvtNeLi4pg2bRplZWURESpayLIcUopEcnJyBKVpH1EQuKxPgsc6yhLKAE6PDMiCwpfLE0281r7lwA2rVaSsTEVOjpNZs2q54YYGJk+2YDTKbNxo5L33Eigu9t9o877++++/j5isXbKFoJeCgoKgIjteYpmSDLD3mBWnLIKowIUYsABRliU0RNctk2WZ77//HlEUfS7wsWNKli9PYPXqeN59NwGVSuaaaxpISXFjs3kMjiQJ7N/vn97eo0cPdDod+/fvD3oqZaicVQvBpUuXRkSoaFFSUhJSvk4sXR2HW+KYxX7qC2uZj9kaQQBNhwODwktlZSXV1dVkZGT4Nvr27tXicgk0Nwts3apn8eIkli9PaFUvbzT6/11FUfRNgSkoKIiIvCEp/u23305xcTFTp07lsccei3lc+2yx2+0hpSHEcjGvFAUSdCqPnZfcCMhoxdZfoACIzmYMUnT7UXqHgPfrd7p/v9Ho0fDqagVOpyeG39QkYrEoGDPGRo8enqjO5Ze3XmN5p1N63zfchLS4XbduHfn5+Xz00UfMmzeP9PR0Xn/99YgIFg00Gk2Hro43dKtWq2PagFUUBH50QQqlNRYcsoQRF0qlAlzQfMqCCnhC/OqGKtBGb7yO0+lk79696PV6vxDkJZc0YrOJNDZ6FhzJyR7LrtNJ5OY2k5sbuH44Pj6elJQUjhw5gtlsDns/o5AsfkNDA99//z379u1DqVQyfPjwsAoTTZqbm1EqlR0ubr2ukEaj8RVOx4q9lVaPgyNLuBGRZNAoPDn4SgF0CkjVgsLdjDuKDZoOHDiA3W6nX79+fl3m1Gq4/voGHn+8huuvbyAuztMsdsSI4Arms7OzkWWZPXv2hF3mkCz+1KlTufXWW/nb3/4W0whHOBBFEaVSSXx8PM3NHX8RLpcrZhP6ABodbgpqbNglAZRaGhFJEcGgAoUW7G6wOqDJBW5dAlVKNw6XhFoZkm3rFLt27UIQBD83pyUKBVxzjQ2XC1rWmDidnlBnoFB9ZmYmu3fvZvfu3YwdOzasrRuDUnxvH/yf/OQnAKxZs8b33LRp08ImTDRRq9UMHDiQoqKioI5XKBQxzcVXiAJul4wLwacpTjckGcDhhppmcEqgFMGt0mOXGymua2JgamQT606cOEF5eTnp6ekBk/hqaxX85z9xWCwK+ve386MfNbB+vYGDB7UYDBI332whLa114ECpVNK3b18KCwspLCz0Vf+Fg6AUv2Xko+WXH6vhvOGiR48eQYczY91RQqMUMWqVCFb7qb1bAbML9tV6dmt99yIJBI0JR2MjYhQu1G9PtURur2rqP/+JY+NGA263wPHjSurqFHz+eRyCANnZDrZuNTBpUttVdTk5ORQWFrJz587oK/706dMBzwbF008/TX19PQaDgb/85S9hEyRWtLdr6+221taoylhwQQ8DxdVm7LIIgoCIZ9jbmeZHFhQ4ZAV9EiN7sTY3N7N3714MBkO7kxU3bDBQVqZGofBkZjY1nW4BWlSkZtiwwK6myWSiR48eHDlyhJMnT4bNxQ7Jx3/wwQdZvnw5mZmZlJSUMGXKFLZt2xbySTdv3swbb7yB3W5nwoQJ7N69O+rF5uBR+vYKm2VZ9sXuRVGMufKP7ZfIF3tLAI9Ct7XiEE49o7BVIwQ5KK6zfP/99zidToYMGRLQ/66sVOJygSjKuFwCDoenrYgse4pQlEqZMWPaD73279+fqqoqdu7cyfXXXx8W2UNSfLvd7sudOJscCovFwrJly9i7dy9jxozBZrNFvdgcPJmEffr0oba2tsPNqfr6+pgPSq602HHKChAk2vvqNLITTcMJbDZbxNqay7LMjh07UCgUARe1AC6XwIABTkRRwO2GxEQXNptIRYUSnU5i9OhGUlLa/9tnZGSg0+nYs2cP48ePD0slXEjm4OGHH+aKK67gzjvv5Morr+SXv/xlp056880343A4ePXVV7njjjuA2BSbNzc3U1dXF9Si9Vyw+Ccb7bgEhaebWgC0IgzgBEqHDavVGjFZjh49Sm1tLVlZWe2ma2dkODEa1zBo0GEGDrSTnu7G4RBRqWSSk93U1FSydGn7XoMoiuTk5GC32335QGdLSIp/5513snXrVv785z/z5Zdfcvfdd3fqpHV1dTzyyCPMnTvXb7J5ID/xySefxGw2+37ClRwny3LQ0zhMJlNEFSkYFIJwajBE+xeqSuPJKYrk4I5gFrUAX321iXfeyWP9+psZM+YQJpOEyyUgilBeXsmKFbfz8ssz+PLLTe2+T3Z2NqIosmPHjrAEVYJS/O3bt3PFFVfgcrlYvHgxEydO5JZbbmHlypWdOuljjz1GaWkpTz75JGPHjmXWrFls3ryZCRMmtHm8RqMhPj7e7yccaLVav2iNKIo+669UKjEajT7fNTk5OaSEtkhgb25GcDto27s/dYwEVUICDkNqxGZLWSwWCgoKSEpK6nB4dFbWIEymDCoqyvjDH36GQlFCaqoLu72c/PwbaW4+isHQG5ttRLvvo9Pp6N27N1VVVZSXl5/1ZwjKx587dy5ffPEFCoWCuXPn8u2335Kamsp1113ni+2Hwpmzs7xRo2jTo0cP33wrwG+DSpIknE6n70JoaGiI+aadwtGAwtEAKg2BchZloFlQ4TCkUnLSxhURkGP37t3IstyhtS8sVPPmmyNITV1LY+P1nDhRzD/+MYkbb1xAScmDOBwlxMf3ZdKkj9BoUoD2W73k5ORQWlrKrl27ztrdDdrVMRqNbNu2jezsbHr27IlCoYjJfNJwkpCQEHDCiSzLuFwun+LHx8fHtK8OAI1mlE0WdIIUsJm2UgC1UokMnLSE3zWTZZnvvvsOlUrVrvLV1ir4+GMj+/ZpsVj60qPHWjSaftTUlLJs2c2YzSUYDB6lT0zsxdChHe+ep6amYjQaOXjwoF+wozMEpfj9+/fn+eef59e//jWPPPIIVquVP//5z4wY0f7t6VzHaDQG9INlWfa7GzQ0NMR8w66u9iQKZwN2FAFDmZLsSVvQKBU01xwLuwylpaWYzWYyMzPbNQRms4jdLqLXe/9mWeTkLPQ75u675zNkSOop16fjAIM3LcLpdJKfn382HyM4xX/llVe49NJL+dvf/saPf/xjysvL0Wq1LFiw4KxOHms6GgXUUtGPHTvGgQMHoiFWQCpq6mlMzMYRoAjF0xvf0zE5TglNlrqg8pBCYf/+/YB/TW1bZGS46NXLRY8enp/k5CLKymb4HbNixcMcOFBNZaWKzz6Lo6qq46iZ97xeOTpLUIqvUCi49tprufjiiwEYPHgwjzzySFS62kaSEydOBL1glSQppqOPHC6Jw04TslpPW1EdEc+SV8LTK19p8NTchnOGgSzLFBQUoNFoAva896JWy0yZYmb69Fqysw9z8OCNWK3FxMX1Zfbs1fTqlUldXQmrVt2G1VqBLHt67XSE0WgkMTGR4uLisyr+j/6k4nOI7777LuiqKkEQotLFNxDmRjtNkhgwhu9VehlPslqT0hP5qqmpCZsM1dXVWK1W0tPTg8qUlGXYurWW//73VpqaitHp+jJ58kcMGTKK995bRlpaFhZLMatW3YbDUUbv3sGVGfbq1Qu3231W/fS7tOJXVFT4uQJarTbgZlZKSkpMK7CcNjOi244gtL/OEE792EQdbpU+rFPDvfsnHVl7L2azSGXlIZqaytFo+jF06H+Jj+9F376e+WmLFy8nKakPDQ3lDBq001ex1RHeWu/S0tLOfRBi2B8/1siyTENDg5/F9yajud3uVgvZWI89spjr0ZrLcSdn0Z5zJuHpoKyWRBz65KC6xQXL8eOeGVXBhnUTE90MG3Y1sBytdghZWUncfLOF1FQXO3Zo+fzzYdx0038wmw8iCFcBwUWhvJVwZzMbt8sqvtvtRpIkPwX3xu3bit40NTVRVlYWM6tvsViQFEqUCG1mZLZEBmRRRBDEsO7e1tbWIghC0LN+1Wr4yU/MDBt2BRqNxNChdYgifPRRHIcOacjP12AwZDN0aAbl5cEX+ahUKvR6/VkZoy7r6njj9G09HoizubWeLTabDYchDcepoW/t4Y3saER3p+Z8BaKxsRG1Wh1SzpLRKHPxxU307u1k3z4tZWVKyspUqNUyGo2MzSbidHoyNkNBo9F02AisPbqsxe/MdJNYZmfamuxIShUaUaa5nW7J4LH4BiWoVGqareHz8V0uV8jlf7IMe/ZoWLMmDqNRRqHwGBezWUGPHq5TacmN5OaGpsQKheKseu50WcUXBIGMjIyQNkJa9nuPNpLLidLpQimcDl22h1s+NSQijA2ZlEplSL2FnE54/30TmzYZqK9XoNHIZGY6T+Xmeyaj9OnjIDHRjSR5B5cGh9vtPqud9C6r+ABpaWkdKn7LztB1dXUxa5GuUIjoao/iTh+ASgFud/tq4pLBLYiow1igHRcXR1VVFS6Xq0Olc7vhX/8y8d57iTQ0iDQ0eDMyVej1EoMG2VEqPW0Ey8uVZGc7uesuMxpNcMrf2NgYMN0kGLqsjy9JEidOnOiwqMHr84faZzPcaDQanLpEpFN60ZGTJsvgksWwbjKmpKQABJX1efKkgiNH1J4J6/JpmbRaz1SUffu07Nqlo6xMSX6+lq++MnDkSHCyNjU1YbfbzyppsMsqviiKfklowRDu7f9QMBgMOHVJNLuhyd2xq9PoBlejOegITDB4I1rBhBENBgmDQSI52Y1GI6FWy+h0EmazAptNgcUiYrWK2O0CJ08qOXHCU4QeDN7zd5Q20R5dVvHBMwMrFD8xnBGSUElKSgJRwBGkHywj49AkdJgvHwrZ2dkoFArKyso6TNgzGGQmTbKQmelAkkChkGhs9NTcCoKMJAk4nZ5++Z6LQiYpKbj1g3cjbcCAAZ3+LOeE4u/Zs4dp06Yxffp0zGZz1M5rsVhobGwM+vhojpw/k9QePWjSJIEU7Fcmg9PebveDUNFoNAwaNAiz2dxhKoTDAf/7n56yMjWNjSJms5KmJgWNjQokSUCSwOUCp9MTzhwypJmcnI7zpmw2m280rNf16gznhOL//e9/Z+HChdx333188MEHrZ632+1YLBa/n7PF6XSGPEw4lvUHstoAuvjAbcdavQBEWQr7YIXRo0cDpwe8BaKkRE1VlRKNRqaxUcTtbtmPydN1wWSSSEhwo1TC+PENqNUd383y8/ORZdknR2c5JxRfkiQ0Gk3AgvNIFJt7WwgG6+MLgnBWt9azRa0UEXxD34Jwd2RQ4wp7CLZ3797069ePysrKdn197ammtUOG2NFoJIRTE0pVKpmkJBdpaS5SU92kp7tIS3MycGDHmZZWq5UjR46QnJzMkCFDzupznBPhTJ1Oh9PpDFhwfuZkc7PZTFZW1llb/ssuu4yNGzdy8uRJJElqd3Nm4MCB9O7dO6IF3B0xOt7J9rpTLQTFDpS/voz+SeqIrEtGjx7NwYMH2bp1KxMmTGgz2pWYCMOGOTl0SM/Uqc1s3Gji+HEN8fEu7r77BCNH2liyJI3GRgXXXltHYmI97d1QZVnm66+/pqmpidGjR3d4942Li2vXqAlyrMuKgJ07d/Laa69ht9t56623OhykXF5e/oMfPNdNZDGbze02JTgnFD9UJEni2LFjHV7VncFisZCZmUlZWdkPYvBFt7xt05FunBOuTqiIohixaXhewtnGJBp0yxsa58Titptuok234nfTJelW/DPQaDTMnTu33X6Q5xLd8naOH+TitptuzpZui99Nl6Rb8bvpknQZxa+vrw8pIS1WnCtyVlZWxlqEVhw7Fr6WiOeF4j/33HP87ne/A2Djxo0sWbKk1THz589vtxNaXl5ehKQ7/d5nK+fu3bv57LPPOjzf888/D8C1117Le++916o7dUeUlJTw5ptvBnw+Pz+fKVOmMGPGDJ555hnf+drjs88+Y/fu3fz+97/n0UcfDeo1Z/LUU0+F/JpA/CA3sNpi586d7N271/f7li1bWLRoEW63m6effpqvv/4ak8nE7NmzAViwYAEHDhygsbGR119/ncLCQh577DFsNht///vfmTFjBlqtlhtvvJGePXvy3nvvYbVamTNnDgsWLMDlctG7d2/uuusuZFnmww8/pGfPnuzZsweLxcIrr7zCSy+9hM1m45tvvum0nM8++yxWqxWTycT48eM5ceIEq1ev5rPPPqOmpoZbbrmFb775hvT0dL7//nvefPNNCgoK+OKLL6iurmbEiBH861//YuvWrSxfvpyTJ0+yYMECnnzySdRqNVqtloceeoiZM2dy6aWXkpiYSHp6OuvXr2fOnDltbjJt2LCBvLw8brjhBlauXMnq1aupq6vj4YcfJiEhgWPHjvGrX/2KBQsWkJ6ezmWXXYbdbsdms/Hvf/+bJ554gnXr1tHU1MQTTzyB3W7n+uuvx2az8e2331JaWsqiRYt4+OGHGTp0KHv37uXll19m165dbNq0iXHjxp21vpwXFh/gj3/8I88++6yvGHrx4sUsXLiQefPm8cYbbzBmzBhuv/123/HffPMNr776KjfeeCOff/45mZmZzJs3j/79+3PgwAGqqqoYP348l156KQsWLECv12Mymdi+fTtNTU28/PLLTJ06lVWrVrFy5UruuOMOFi5ciEajQZIkdu/eTV1dHS+99JLfBPhQ5SwrKyMnJ8c3Mgk8c4bffPNNpkyZAnhK8R5//HF+9KMf+ZqpXn/99YwcOdKX97R06VJeffVV5s+fjyRJTJ06lTFjxrBz504ARo4cyZ/+9Cf27NnDmDFjmDBhQsCd1RkzZnDgwAEeeughioqKUCgUrFixgtmzZ/PXv/7V99nGjx/Pyy+/zNq1awFP6eLIkSP5+c9/DnguoAkTJrBw4UKGDRvGyJEjmThxIgaDgYMHDyLLMs8884wvK3bkyJFhUXo4jxQ/Pj6e+++/n5dffhk4XSvr7YZ8Zt7Gmc9748reTMM//elPKBQKnnvuOZxOJ08//TRz5szxpcOaTCYGDhzI0aNHKSsro1+/fiQmJvLiiy+Sl5fnNxCtZfZiqHI+8MADDB061C871dssteWxer0elUoVsPuz0+n0dY+rq6tj0aJFDB482FehpdfrgeDarixatIgZM2bw6quvIssyjY2NvvcPRSaXy+X7/EVFRbz00ksoFAqGDx+OLMvodJ6RRu29R2c5b1wdgBtvvJEVK1YAcN9993H//fcjiiK//e1v2bJlC2+88QYvvvgiAJdffjkPP/wwkiTx8ssvM2/ePP70pz9RX1/PoEGDmD59Oj169OCyyy5jwIABzJw5E7vdzrx58/zO2a9fP59iX3fddcycOZOmpibeeecdUlNTefzxx1sNLAtFzpUrV9LY2Og3i+Dmm2/mgQceoKyszGc9O+Kee+5h1qxZNDQ0MH/+fCwWC++88w7Hjh1rVUaYmJjIxo0bOXHiRJtdJcaOHcu0adOIi4tDq9Wi0WiYMmUKc+bMCakA/Prrr+fBBx9kzZo13HzzzSQlJfHpp59SVFTUZr690Wjkww8/ZPLkyUGfIxDdG1g/QN555x0KCwupra3lj3/841mV4IWLQ4cOsXDhQtxuN+PGjfNz185FuhW/my7JeePjt4Usyzz//PNMmDDBdzsPFLYsLy/nmmuuATx986dMmUJeXh41NTXMmzePn//854wePZo333yTpUuXMmPGDB555BEA/vrXv4Y19h6s3AcOHOD222/nF7/4BXv37qW0tJQZM2aQl5fnC3tu2LCBX/ziFwC8/fbbvo7HkSBYuZubm5k1axZz5szhn//8JxUVFdx1111MmzaNkpISPvroo7DNsw3Eea34H3/8MZdddhlqtdovLr19+3ZmzJjBjBkzePTRRwFPwXtCQgLgqeR//fXXmThxItu2beOxxx5j8eLFjBgxgvvvv5/k5GRfCNJisXDLLbfw9ttvR13uzz77jGeeeYbXXnuN1157jQMHDrBr1y5KSkrIycmhqKiInTt3+voB3XHHHbz66qthk7Ozcq9atQq9Xk99fT0XXXQRr732Gi+88AJ//OMfeeutt7j11ltZtGhRxOSE81zx161bh9VqZcqUKWzZssXXj8XbKdn7A/DnP//Z15LuiiuuoKCggHnz5vkWle+//z6TJ09GEARuvvlm3nnnHYqKitBqtfTv358dO3ZEXe4777yT+fPn88wzz+B0OsnMzGTTpk28/vrrfPDBB2RnZ/P444/73jcuLi6iHZ+DlbukpISLL76Y+fPnM2/ePKqrq+nVq5ev2YAgCGEdaNEW51VUpy28EZf/+7//47HHHkOtVnP55Zdz+eWXB3zNzp07GTFiBGvWrOHFF1/kpZdeYvPmzT4r9PXXX3PPPfcQHx/P2rVrmThxYtjngQUj9549e5g7dy5paWk88cQTLFiwgGeffZbU1NSAihPpuWXByL1o0SKMRiN6vR6lUklaWhonTpzA7Xb7mg1EWs7zWvGvu+46X2/M9PR0rr/+ejZu3Njh66xWKzNmzECWZX71q18B+MWod+zYwfLly7FarbzyyiscOnTINxgvmnKnp6fz8MMPI4oif/jDH6iqquKXv/wlRqORp59+utXxFosloh2fg5X7Jz/5CbNmzWLNmjXcd9999OzZk6eeesoXLpYkyed2Rgz5POell16K+Dn+9re/yQ6HI6zvGQm5Fy1aJNfW1ob9fVsSDrk//PBD+fDhw2GQJjDd4cxuuiTn9eK2m24C0a343XRJuhW/my5Jt+J30yXpVvxuuiTdit9Nl6Rb8bvpknQrfjddkm7F76ZL8v8BEozPkoFkCwwAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 174x140 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "indels = False\n",
    "\n",
    "for gene in [\"TP53\", \"EP300\"]:\n",
    "    \n",
    "    # Data\n",
    "    # ====\n",
    "    \n",
    "    # Mut\n",
    "    gene_mut = maf[maf[\"Gene\"] == gene].drop(columns=\"Gene\").reset_index(drop=True)\n",
    "    gene_mut_count = maf_count[maf_count[\"Gene\"] == gene].drop(columns=\"Gene\").reset_index(drop=True)\n",
    "    gene_mut_cnsq_count = maf_cnsq_count[maf_cnsq_count[\"Gene\"] == gene].drop(columns=\"Gene\").reset_index(drop=True)\n",
    "    gene_cancer_mut = blca_cancer_df[blca_cancer_df[\"Gene\"] == gene].drop(columns=\"Gene\").reset_index(drop=True)\n",
    "    gene_cancer_cnsq_count = blca_cancer_cnsq_count[blca_cancer_cnsq_count[\"Gene\"] == gene]\n",
    "    \n",
    "    # Oncodrive3D\n",
    "    o3d_gene_df = get_o3d_gene_data(gene, o3d_seq_df, o3d_df)\n",
    "    uni_id = o3d_seq_df[o3d_seq_df[\"Gene\"] == gene].Uniprot_ID.values[0]\n",
    "    pdb_tool_gene = o3d_pdb_tool_df[o3d_pdb_tool_df[\"Uniprot_ID\"] == uni_id].reset_index(drop=True)\n",
    "    o3d_cancer_gene_df = o3d_cancer_df[o3d_cancer_df[\"Gene\"] == gene].reset_index(drop=True)\n",
    "    \n",
    "    if indels == False:\n",
    "        gene_mut_count = gene_mut_count[gene_mut_count[\"Consequence\"] != \"indel\"].reset_index(drop=True)\n",
    "        gene_mut = gene_mut[gene_mut[\"Consequence\"] != \"indel\"].reset_index(drop=True)\n",
    "        gene_cancer_mut = gene_cancer_mut[gene_cancer_mut[\"Consequence\"] != \"indel\"].reset_index(drop=True)\n",
    "        gene_mut_cnsq_count = gene_mut_cnsq_count[gene_mut_cnsq_count[\"Consequence\"] != \"indel\"].reset_index(drop=True) \n",
    "        gene_cancer_cnsq_count = gene_cancer_cnsq_count[gene_cancer_cnsq_count[\"Consequence\"] != \"indel\"].reset_index(drop=True) \n",
    "        \n",
    "    prot_len = int(exons_depth[exons_depth[\"GENE\"] == gene].PROT_POS.max())\n",
    "    domain_gene = domain[domain[\"Gene\"] == gene].reset_index(drop=True)\n",
    "    \n",
    "    # Max exons depth across all exons\n",
    "    all_exons_depth = exons_depth[[\"GENE\", \"EXON_RANK\", \"DEPTH\", \"COVERED\"]].groupby(\"GENE\").apply(\n",
    "        lambda x: x.groupby(\"EXON_RANK\", group_keys=False).apply(lambda x: np.nan if sum(x.COVERED) == 0 else sum(x.DEPTH) / sum(x.COVERED))).drop(\n",
    "        columns=[\"GENE\", \"EXON_RANK\", \"DEPTH\", \"COVERED\"]).reset_index().rename(columns = {0 : \"DEPTH\"})\n",
    "    max_depth = all_exons_depth.DEPTH.max()\n",
    "\n",
    "    gene_exons_depth = exons_depth[exons_depth[\"GENE\"] == gene].sort_values(\"DNA_POS\")\n",
    "    gene_exons_depth = gene_exons_depth.dropna(subset=[\"EXON_RANK\"]).reset_index(drop=True)\n",
    "    \n",
    "    exon_selection_gene = exon_selection[exon_selection.gene.str.startswith(gene)]\n",
    "    exon_selection_gene = exon_selection_gene.sort_values(\"exon_rank\").reset_index(drop=True).rename(columns={\"exon_rank\" : \"EXON_RANK\"})\n",
    "    exon_selection_gene = exon_selection_gene[[\"EXON_RANK\", \"impact\", \"dnds\", \"lower\", \"upper\", \"pvalue\"]]\n",
    "\n",
    "    site_selection_gene = site_selection[site_selection[\"GENE\"] == gene].sort_values(\"Protein_position\").reset_index(drop=True)\n",
    "\n",
    "    domain_selection_gene = get_domain_selection_gene(domain_selection, domain_gene, gene)\n",
    "    \n",
    "    ddg_gene = get_ddg_gene(gene, gene_mut, o3d_annotations, o3d_seq_df, o3d_alt_seq_df)\n",
    "    \n",
    "    \n",
    "    # Plot\n",
    "    # ====\n",
    "    \n",
    "    plot_pars[\"fsize\"] = (12.3, 11)\n",
    "    lst_tracks=[\n",
    "        \"Mut_count\", \n",
    "        \"3d_clustering\", \n",
    "        \"Site_selection\", \n",
    "        \"Exon_selection\", \n",
    "        \"Solvent_accessibility\", \n",
    "        \"Res_depth\", \n",
    "        \"Domain\", \n",
    "        \"Cancer_mut_count\"\n",
    "        ]\n",
    "    if gene in [\"KDM6A\", \"STAG2\"]:\n",
    "        plot_pars[\"fsize\"] = (12.3, 9)\n",
    "        [lst_tracks.remove(track) for track in [\n",
    "            \"3d_clustering\", \n",
    "            \"Cancer_3d_clustering\",\n",
    "            \"Solvent_accessibility\", \n",
    "            \"Stability_change\",\n",
    "            \"Secondary_structure\",\n",
    "            ] if track in lst_tracks\n",
    "        ]\n",
    "\n",
    "    # Selection groups feat\n",
    "    # =====================\n",
    "    site_selection_gene_grouped = get_selection_groups(site_selection_gene, thr_selection=0.00001)\n",
    "    site_selection_gene_grouped = site_selection_gene_grouped.merge(pdb_tool_gene[[\"Pos\", \"pACC\"]])\n",
    "    site_selection_gene_grouped = site_selection_gene_grouped.merge(ddg_gene) \n",
    "    plot_feat_by_selection_group(site_selection_gene_grouped, save=True, gene = gene, filename=f\"plots/ExtendedFig10bc.{gene}.selection_groups.pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "860eb754",
   "metadata": {},
   "source": [
    "### Main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65d787ea-f4f9-4e74-8b01-be395c340bdd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e26ef404",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "notebooks_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
