{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b8eb3226-f371-4e3e-9937-1e8469488164",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "from scipy.stats import norm\n",
    "import tabix\n",
    "from matplotlib.patches import Patch\n",
    "from matplotlib.lines import Line2D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b2a7ebbd-7e91-454e-81af-73daf8f3fe6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', None)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cda004ed-c127-4195-b13c-b6e2d94305e3",
   "metadata": {},
   "source": [
    "# Define gene order and plot and data directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "89b16b29-7a11-471d-8211-5a0d7b5e568b",
   "metadata": {},
   "outputs": [],
   "source": [
    "plots_dir = \"./plots\"\n",
    "data_dir = \"./data\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "237a5af0-8259-4ce6-b8e1-0aafd18ba50e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append('../../')\n",
    "from consensus_variables import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c563909-3728-4eca-a3ce-0db4a78a78cc",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6c9c2934-0937-4ca0-ad8c-c2eab4887c7c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3300149/2123141286.py:1: DtypeWarning: Columns (49,92,112) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  somatic_maf_df = pd.read_csv(somatic_maf_file, sep = \"\\t\", header = 0)\n"
     ]
    }
   ],
   "source": [
    "somatic_maf_df = pd.read_csv(somatic_maf_file, sep = \"\\t\", header = 0)\n",
    "\n",
    "maf_df_f = somatic_maf_df[~(somatic_maf_df['FILTER'].str.contains(\"not_in_exons\"))].reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "38bebaf4-e850-4da3-b9ff-eb228340b577",
   "metadata": {},
   "outputs": [],
   "source": [
    "snvs_maf = maf_df_f[(maf_df_f['TYPE'] == 'SNV')\n",
    "                    & (maf_df_f[\"canonical_CDS_position\"] != '-')\n",
    "                   ].reset_index(drop = True)\n",
    "snvs_maf[\"canonical_CDS_position\"] = snvs_maf[\"canonical_CDS_position\"].astype(int)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19f10a30-326d-4c1f-81af-bc00a0d2a7c2",
   "metadata": {},
   "source": [
    "### Omega"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1dd985e0-9d0f-450b-af80-7a6f38fd926b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_missing_positions_filtered(df):\n",
    "    \"\"\"\n",
    "    Fills missing cDNA, CDS, and Protein positions with the closest non-missing value, considering only rows with non-missing values for the specific column.\n",
    "\n",
    "    Args:\n",
    "        df: A Pandas DataFrame with columns 'CHROM', 'POS', 'Consequence', 'cDNA_position', 'CDS_position', and 'Protein_position'.\n",
    "\n",
    "    Returns:\n",
    "        A Pandas DataFrame with additional columns 'closest_cDNA_position', 'closest_CDS_position', and 'closest_Protein_position'.\n",
    "    \"\"\"\n",
    "\n",
    "    df_copy = df.copy()\n",
    "\n",
    "    for col in ['canonical_cDNA_position', 'canonical_CDS_position', 'canonical_Protein_position']:\n",
    "        df_copy[f'closest_{col}'] = df_copy[col].replace('-', pd.NA)\n",
    "        missing_indices = df_copy.index[df_copy[f'closest_{col}'].isna()]\n",
    "        \n",
    "        for i in missing_indices:\n",
    "            # Find the nearest non-missing value on either side\n",
    "            prev_non_missing = df_copy.loc[:i, f'closest_{col}'].last_valid_index()\n",
    "            next_non_missing = df_copy.loc[i:, f'closest_{col}'].first_valid_index()\n",
    "\n",
    "            if prev_non_missing is not None and next_non_missing is not None:\n",
    "                prev_pos_diff = abs(df_copy.loc[prev_non_missing, 'POS'] - df_copy.loc[i, 'POS'])\n",
    "                next_pos_diff = abs(df_copy.loc[next_non_missing, 'POS'] - df_copy.loc[i, 'POS'])\n",
    "\n",
    "                if prev_pos_diff < next_pos_diff:\n",
    "                    df_copy.loc[i, f'closest_{col}'] = df_copy.loc[prev_non_missing, f'closest_{col}']\n",
    "                else:\n",
    "                    df_copy.loc[i, f'closest_{col}'] = df_copy.loc[next_non_missing, f'closest_{col}']\n",
    "            elif prev_non_missing is not None:\n",
    "                df_copy.loc[i, f'closest_{col}'] = df_copy.loc[prev_non_missing, f'closest_{col}']\n",
    "            elif next_non_missing is not None:\n",
    "                df_copy.loc[i, f'closest_{col}'] = df_copy.loc[next_non_missing, f'closest_{col}']\n",
    "\n",
    "    return df_copy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2c31c6ab-7c0d-408b-a287-f47c5377a6b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# upd_snvs_maf = fill_missing_positions_filtered(snvs_maf)\n",
    "# upd_snvs_maf['closest_canonical_cDNA_position'].astype(int)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de9535e9-e7d1-4e4b-9343-c3a7056109f0",
   "metadata": {},
   "source": [
    "Truncating"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7d4af4d6-2a90-4f39-b20f-4176c1dd73b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD;repetitive_variant</td>\n",
       "      <td>SAMPLE=P19_0020_BDO_01;TYPE=SNV;DP=6467;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6467:1:6466,1:0.0002:3943,2523:1,0:6469:64...</td>\n",
       "      <td>6469</td>\n",
       "      <td>1</td>\n",
       "      <td>6468</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>6467</td>\n",
       "      <td>1</td>\n",
       "      <td>6466</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>22</td>\n",
       "      <td>0.003389</td>\n",
       "      <td>11254</td>\n",
       "      <td>1</td>\n",
       "      <td>11253</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>30</td>\n",
       "      <td>0.002659</td>\n",
       "      <td>4785</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.574818</td>\n",
       "      <td>0.940644</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>P19_0020_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>rs2124784810,COSV61370777</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD;repetitive_variant</td>\n",
       "      <td>SAMPLE=P19_0024_BDO_01;TYPE=SNV;DP=7764;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7764:1:7763,1:0.0001:4406,3357:1,0:7768:77...</td>\n",
       "      <td>7768</td>\n",
       "      <td>1</td>\n",
       "      <td>7767</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>7764</td>\n",
       "      <td>1</td>\n",
       "      <td>7763</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>19</td>\n",
       "      <td>0.002440</td>\n",
       "      <td>11469</td>\n",
       "      <td>1</td>\n",
       "      <td>11468</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>25</td>\n",
       "      <td>0.002175</td>\n",
       "      <td>3701</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.677304</td>\n",
       "      <td>0.958316</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>P19_0024_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>rs2124784810,COSV61370777</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD;repetitive_variant</td>\n",
       "      <td>SAMPLE=P19_0051_BDO_01;TYPE=SNV;DP=8449;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:8449:1:8448,1:0.0001:5757,2691:1,0:8451:84...</td>\n",
       "      <td>8451</td>\n",
       "      <td>1</td>\n",
       "      <td>8450</td>\n",
       "      <td>0.000118</td>\n",
       "      <td>8449</td>\n",
       "      <td>1</td>\n",
       "      <td>8448</td>\n",
       "      <td>0.000118</td>\n",
       "      <td>18</td>\n",
       "      <td>0.002125</td>\n",
       "      <td>22718</td>\n",
       "      <td>2</td>\n",
       "      <td>22716</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>31</td>\n",
       "      <td>0.001363</td>\n",
       "      <td>14267</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.743992</td>\n",
       "      <td>0.968330</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>P19_0051_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>rs2124784810,COSV61370777</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD;repetitive_variant</td>\n",
       "      <td>SAMPLE=P19_0050_BTR_01;TYPE=SNV;DP=10553;VD=1;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:10553:1:10552,1:0.0001:7137,3415:1,0:10556...</td>\n",
       "      <td>10556</td>\n",
       "      <td>1</td>\n",
       "      <td>10555</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>10553</td>\n",
       "      <td>1</td>\n",
       "      <td>10552</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>35</td>\n",
       "      <td>0.003305</td>\n",
       "      <td>23434</td>\n",
       "      <td>1</td>\n",
       "      <td>23433</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>52</td>\n",
       "      <td>0.002214</td>\n",
       "      <td>12878</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.450457</td>\n",
       "      <td>0.920742</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>P19_0050_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>rs2124784810,COSV61370777</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD;repetitive_variant</td>\n",
       "      <td>SAMPLE=P19_0051_BTR_01;TYPE=SNV;DP=9736;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:9736:1:9735,1:0.0001:6230,3505:1,0:9737:97...</td>\n",
       "      <td>9737</td>\n",
       "      <td>1</td>\n",
       "      <td>9736</td>\n",
       "      <td>0.000103</td>\n",
       "      <td>9736</td>\n",
       "      <td>1</td>\n",
       "      <td>9735</td>\n",
       "      <td>0.000103</td>\n",
       "      <td>23</td>\n",
       "      <td>0.002357</td>\n",
       "      <td>20234</td>\n",
       "      <td>2</td>\n",
       "      <td>20232</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>41</td>\n",
       "      <td>0.002022</td>\n",
       "      <td>10497</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.962439</td>\n",
       "      <td>0.995849</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>P19_0051_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>rs2124784810,COSV61370777</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729669</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1545</td>\n",
       "      <td>1156</td>\n",
       "      <td>386</td>\n",
       "      <td>Q/*</td>\n",
       "      <td>Cag/Tag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729669_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729669</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29067</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100595</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0024_BTR_01;TYPE=SNV;DP=2434;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2434:1:2433,1:0.0004:1234,1199:1,0:2435:24...</td>\n",
       "      <td>2435</td>\n",
       "      <td>1</td>\n",
       "      <td>2434</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>2434</td>\n",
       "      <td>1</td>\n",
       "      <td>2433</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>6</td>\n",
       "      <td>0.002458</td>\n",
       "      <td>3914</td>\n",
       "      <td>2</td>\n",
       "      <td>3912</td>\n",
       "      <td>0.000511</td>\n",
       "      <td>7</td>\n",
       "      <td>0.001785</td>\n",
       "      <td>1479</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1.244251</td>\n",
       "      <td>1.028834</td>\n",
       "      <td>chrX:124100595_T&gt;C</td>\n",
       "      <td>P19_0024_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100595</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4118</td>\n",
       "      <td>3805</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/Q</td>\n",
       "      <td>Taa/Caa</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100595</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4118</td>\n",
       "      <td>3805</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/Q</td>\n",
       "      <td>Taa/Caa</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TTA&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T[T&gt;C]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100595_T&gt;C</td>\n",
       "      <td>X</td>\n",
       "      <td>124100595</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29068</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100595</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=P19_0007_BTR_01;TYPE=SNV;DP=1763;VD=2;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:1763:2:1761,2:0.0011:876,885:1,1:1766:1764...</td>\n",
       "      <td>1766</td>\n",
       "      <td>2</td>\n",
       "      <td>1764</td>\n",
       "      <td>0.001133</td>\n",
       "      <td>1763</td>\n",
       "      <td>2</td>\n",
       "      <td>1761</td>\n",
       "      <td>0.001134</td>\n",
       "      <td>7</td>\n",
       "      <td>0.003948</td>\n",
       "      <td>5205</td>\n",
       "      <td>2</td>\n",
       "      <td>5203</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>16</td>\n",
       "      <td>0.003065</td>\n",
       "      <td>3439</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.339289</td>\n",
       "      <td>0.862555</td>\n",
       "      <td>chrX:124100595_T&gt;A</td>\n",
       "      <td>P19_0007_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100595</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4118</td>\n",
       "      <td>3805</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/K</td>\n",
       "      <td>Taa/Aaa</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100595</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4118</td>\n",
       "      <td>3805</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/K</td>\n",
       "      <td>Taa/Aaa</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TTA&gt;A</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T[T&gt;A]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100595_T&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124100595</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29069</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100596</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0015_BTR_01;TYPE=SNV;DP=2645;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2645:1:2644,1:0.0004:1354,1290:0,1:2645:26...</td>\n",
       "      <td>2645</td>\n",
       "      <td>1</td>\n",
       "      <td>2644</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>2645</td>\n",
       "      <td>1</td>\n",
       "      <td>2644</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>5</td>\n",
       "      <td>0.001887</td>\n",
       "      <td>4661</td>\n",
       "      <td>1</td>\n",
       "      <td>4660</td>\n",
       "      <td>0.000215</td>\n",
       "      <td>7</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>2016</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.567475</td>\n",
       "      <td>0.932928</td>\n",
       "      <td>chrX:124100596_A&gt;C</td>\n",
       "      <td>P19_0015_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100596</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4119</td>\n",
       "      <td>3806</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/S</td>\n",
       "      <td>tAa/tCa</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100596</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4119</td>\n",
       "      <td>3806</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/S</td>\n",
       "      <td>tAa/tCa</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TTA&gt;G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T[T&gt;G]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100596_T&gt;G</td>\n",
       "      <td>X</td>\n",
       "      <td>124100596</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29070</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100596</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0042_BDO_01;TYPE=SNV;DP=2984;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2984:1:2983,1:0.0003:1749,1234:1,0:2985:29...</td>\n",
       "      <td>2985</td>\n",
       "      <td>1</td>\n",
       "      <td>2984</td>\n",
       "      <td>0.000335</td>\n",
       "      <td>2984</td>\n",
       "      <td>1</td>\n",
       "      <td>2983</td>\n",
       "      <td>0.000335</td>\n",
       "      <td>5</td>\n",
       "      <td>0.001672</td>\n",
       "      <td>4414</td>\n",
       "      <td>1</td>\n",
       "      <td>4413</td>\n",
       "      <td>0.000227</td>\n",
       "      <td>8</td>\n",
       "      <td>0.001809</td>\n",
       "      <td>1429</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.676257</td>\n",
       "      <td>0.953389</td>\n",
       "      <td>chrX:124100596_A&gt;C</td>\n",
       "      <td>P19_0042_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100596</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4119</td>\n",
       "      <td>3806</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/S</td>\n",
       "      <td>tAa/tCa</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100596</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4119</td>\n",
       "      <td>3806</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/S</td>\n",
       "      <td>tAa/tCa</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TTA&gt;G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T[T&gt;G]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100596_T&gt;G</td>\n",
       "      <td>X</td>\n",
       "      <td>124100596</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29071</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100597</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0015_BDO_01;TYPE=SNV;DP=3029;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3029:1:3028,1:0.0003:1482,1546:0,1:3030:30...</td>\n",
       "      <td>3030</td>\n",
       "      <td>1</td>\n",
       "      <td>3029</td>\n",
       "      <td>0.000330</td>\n",
       "      <td>3029</td>\n",
       "      <td>1</td>\n",
       "      <td>3028</td>\n",
       "      <td>0.000330</td>\n",
       "      <td>4</td>\n",
       "      <td>0.001318</td>\n",
       "      <td>4884</td>\n",
       "      <td>1</td>\n",
       "      <td>4883</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002043</td>\n",
       "      <td>1854</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.620393</td>\n",
       "      <td>0.943794</td>\n",
       "      <td>chrX:124100597_A&gt;T</td>\n",
       "      <td>P19_0015_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100597</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4120</td>\n",
       "      <td>3807</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/Y</td>\n",
       "      <td>taA/taT</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100597</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>4120</td>\n",
       "      <td>3807</td>\n",
       "      <td>1269</td>\n",
       "      <td>*/Y</td>\n",
       "      <td>taA/taT</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ATT&gt;A</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>A[T&gt;A]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100597_T&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124100597</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10727 rows  149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT                   FILTER  \\\n",
       "7      chr1   26729669   C   T  pSTD;repetitive_variant   \n",
       "8      chr1   26729669   C   T  pSTD;repetitive_variant   \n",
       "9      chr1   26729669   C   T  pSTD;repetitive_variant   \n",
       "10     chr1   26729669   C   T  pSTD;repetitive_variant   \n",
       "11     chr1   26729669   C   T  pSTD;repetitive_variant   \n",
       "...     ...        ...  ..  ..                      ...   \n",
       "29067  chrX  124100595   T   C                     pSTD   \n",
       "29068  chrX  124100595   T   A                     PASS   \n",
       "29069  chrX  124100596   A   C                     pSTD   \n",
       "29070  chrX  124100596   A   C                     pSTD   \n",
       "29071  chrX  124100597   A   T                     pSTD   \n",
       "\n",
       "                                                    INFO  \\\n",
       "7      SAMPLE=P19_0020_BDO_01;TYPE=SNV;DP=6467;VD=1;A...   \n",
       "8      SAMPLE=P19_0024_BDO_01;TYPE=SNV;DP=7764;VD=1;A...   \n",
       "9      SAMPLE=P19_0051_BDO_01;TYPE=SNV;DP=8449;VD=1;A...   \n",
       "10     SAMPLE=P19_0050_BTR_01;TYPE=SNV;DP=10553;VD=1;...   \n",
       "11     SAMPLE=P19_0051_BTR_01;TYPE=SNV;DP=9736;VD=1;A...   \n",
       "...                                                  ...   \n",
       "29067  SAMPLE=P19_0024_BTR_01;TYPE=SNV;DP=2434;VD=1;A...   \n",
       "29068  SAMPLE=P19_0007_BTR_01;TYPE=SNV;DP=1763;VD=2;A...   \n",
       "29069  SAMPLE=P19_0015_BTR_01;TYPE=SNV;DP=2645;VD=1;A...   \n",
       "29070  SAMPLE=P19_0042_BDO_01;TYPE=SNV;DP=2984;VD=1;A...   \n",
       "29071  SAMPLE=P19_0015_BDO_01;TYPE=SNV;DP=3029;VD=1;A...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "7      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "9      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "10     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "11     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "...                                                  ...   \n",
       "29067  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29068  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29069  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29070  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29071  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                  SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "7      0/1:6467:1:6466,1:0.0002:3943,2523:1,0:6469:64...   6469          1   \n",
       "8      0/1:7764:1:7763,1:0.0001:4406,3357:1,0:7768:77...   7768          1   \n",
       "9      0/1:8449:1:8448,1:0.0001:5757,2691:1,0:8451:84...   8451          1   \n",
       "10     0/1:10553:1:10552,1:0.0001:7137,3415:1,0:10556...  10556          1   \n",
       "11     0/1:9736:1:9735,1:0.0001:6230,3505:1,0:9737:97...   9737          1   \n",
       "...                                                  ...    ...        ...   \n",
       "29067  0/1:2434:1:2433,1:0.0004:1234,1199:1,0:2435:24...   2435          1   \n",
       "29068  0/1:1763:2:1761,2:0.0011:876,885:1,1:1766:1764...   1766          2   \n",
       "29069  0/1:2645:1:2644,1:0.0004:1354,1290:0,1:2645:26...   2645          1   \n",
       "29070  0/1:2984:1:2983,1:0.0003:1749,1234:1,0:2985:29...   2985          1   \n",
       "29071  0/1:3029:1:3028,1:0.0003:1482,1546:0,1:3030:30...   3030          1   \n",
       "\n",
       "       REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "7           6468  0.000155      6467             1          6466  0.000155   \n",
       "8           7767  0.000129      7764             1          7763  0.000129   \n",
       "9           8450  0.000118      8449             1          8448  0.000118   \n",
       "10         10555  0.000095     10553             1         10552  0.000095   \n",
       "11          9736  0.000103      9736             1          9735  0.000103   \n",
       "...          ...       ...       ...           ...           ...       ...   \n",
       "29067       2434  0.000411      2434             1          2433  0.000411   \n",
       "29068       1764  0.001133      1763             2          1761  0.001134   \n",
       "29069       2644  0.000378      2645             1          2644  0.000378   \n",
       "29070       2984  0.000335      2984             1          2983  0.000335   \n",
       "29071       3029  0.000330      3029             1          3028  0.000330   \n",
       "\n",
       "       numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "7         22  0.003389     11254             1         11253  0.000089   \n",
       "8         19  0.002440     11469             1         11468  0.000087   \n",
       "9         18  0.002125     22718             2         22716  0.000088   \n",
       "10        35  0.003305     23434             1         23433  0.000043   \n",
       "11        23  0.002357     20234             2         20232  0.000099   \n",
       "...      ...       ...       ...           ...           ...       ...   \n",
       "29067      6  0.002458      3914             2          3912  0.000511   \n",
       "29068      7  0.003948      5205             2          5203  0.000384   \n",
       "29069      5  0.001887      4661             1          4660  0.000215   \n",
       "29070      5  0.001672      4414             1          4413  0.000227   \n",
       "29071      4  0.001318      4884             1          4883  0.000205   \n",
       "\n",
       "       numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND    VAF_ND  \\\n",
       "7            30   0.002659      4785             0  0.000000   \n",
       "8            25   0.002175      3701             0  0.000000   \n",
       "9            31   0.001363     14267             1  0.000070   \n",
       "10           52   0.002214     12878             0  0.000000   \n",
       "11           41   0.002022     10497             1  0.000095   \n",
       "...         ...        ...       ...           ...       ...   \n",
       "29067         7   0.001785      1479             1  0.000676   \n",
       "29068        16   0.003065      3439             0  0.000000   \n",
       "29069         7   0.001500      2016             0  0.000000   \n",
       "29070         8   0.001809      1429             0  0.000000   \n",
       "29071        10   0.002043      1854             0  0.000000   \n",
       "\n",
       "       VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "7                       False                      False        0.574818   \n",
       "8                       False                      False        0.677304   \n",
       "9                       False                      False        0.743992   \n",
       "10                      False                      False        0.450457   \n",
       "11                      False                      False        0.962439   \n",
       "...                       ...                        ...             ...   \n",
       "29067                   False                      False        1.244251   \n",
       "29068                   False                      False        0.339289   \n",
       "29069                   False                      False        0.567475   \n",
       "29070                   False                      False        0.676257   \n",
       "29071                   False                      False        0.620393   \n",
       "\n",
       "       VAF_distortion_sq              MUT_ID        SAMPLE_ID METHOD  \\\n",
       "7               0.940644   chr1:26729669_C>T  P19_0020_BDO_01    med   \n",
       "8               0.958316   chr1:26729669_C>T  P19_0024_BDO_01    med   \n",
       "9               0.968330   chr1:26729669_C>T  P19_0051_BDO_01    med   \n",
       "10              0.920742   chr1:26729669_C>T  P19_0050_BTR_01    med   \n",
       "11              0.995849   chr1:26729669_C>T  P19_0051_BTR_01    med   \n",
       "...                  ...                 ...              ...    ...   \n",
       "29067           1.028834  chrX:124100595_T>C  P19_0024_BTR_01    med   \n",
       "29068           0.862555  chrX:124100595_T>A  P19_0007_BTR_01    med   \n",
       "29069           0.932928  chrX:124100596_A>C  P19_0015_BTR_01    med   \n",
       "29070           0.953389  chrX:124100596_A>C  P19_0042_BDO_01    med   \n",
       "29071           0.943794  chrX:124100597_A>T  P19_0015_BDO_01    med   \n",
       "\n",
       "       PROJECT_NAME        Location Allele             Gene          Feature  \\\n",
       "7               NaN   chr1:26729669      T  ENSG00000117713  ENST00000324856   \n",
       "8               NaN   chr1:26729669      T  ENSG00000117713  ENST00000324856   \n",
       "9               NaN   chr1:26729669      T  ENSG00000117713  ENST00000324856   \n",
       "10              NaN   chr1:26729669      T  ENSG00000117713  ENST00000324856   \n",
       "11              NaN   chr1:26729669      T  ENSG00000117713  ENST00000324856   \n",
       "...             ...             ...    ...              ...              ...   \n",
       "29067           NaN  chrX:124100595      C  ENSG00000101972  ENST00000371145   \n",
       "29068           NaN  chrX:124100595      A  ENSG00000101972  ENST00000371145   \n",
       "29069           NaN  chrX:124100596      C  ENSG00000101972  ENST00000371145   \n",
       "29070           NaN  chrX:124100596      C  ENSG00000101972  ENST00000371145   \n",
       "29071           NaN  chrX:124100597      T  ENSG00000101972  ENST00000371145   \n",
       "\n",
       "      Feature_type  Consequence cDNA_position CDS_position Protein_position  \\\n",
       "7       Transcript  stop_gained          1545         1156              386   \n",
       "8       Transcript  stop_gained          1545         1156              386   \n",
       "9       Transcript  stop_gained          1545         1156              386   \n",
       "10      Transcript  stop_gained          1545         1156              386   \n",
       "11      Transcript  stop_gained          1545         1156              386   \n",
       "...            ...          ...           ...          ...              ...   \n",
       "29067   Transcript    stop_lost          4118         3805             1269   \n",
       "29068   Transcript    stop_lost          4118         3805             1269   \n",
       "29069   Transcript    stop_lost          4119         3806             1269   \n",
       "29070   Transcript    stop_lost          4119         3806             1269   \n",
       "29071   Transcript    stop_lost          4120         3807             1269   \n",
       "\n",
       "      Amino_acids   Codons         Existing_variation IMPACT DISTANCE STRAND  \\\n",
       "7             Q/*  Cag/Tag  rs2124784810,COSV61370777   HIGH        -      1   \n",
       "8             Q/*  Cag/Tag  rs2124784810,COSV61370777   HIGH        -      1   \n",
       "9             Q/*  Cag/Tag  rs2124784810,COSV61370777   HIGH        -      1   \n",
       "10            Q/*  Cag/Tag  rs2124784810,COSV61370777   HIGH        -      1   \n",
       "11            Q/*  Cag/Tag  rs2124784810,COSV61370777   HIGH        -      1   \n",
       "...           ...      ...                        ...    ...      ...    ...   \n",
       "29067         */Q  Taa/Caa                          -   HIGH        -      1   \n",
       "29068         */K  Taa/Aaa                          -   HIGH        -      1   \n",
       "29069         */S  tAa/tCa                          -   HIGH        -      1   \n",
       "29070         */S  tAa/tCa                          -   HIGH        -      1   \n",
       "29071         */Y  taA/taT                          -   HIGH        -      1   \n",
       "\n",
       "      FLAGS  SYMBOL SYMBOL_SOURCE     HGNC_ID CANONICAL             ENSP  \\\n",
       "7         -  ARID1A          HGNC  HGNC:11110       YES  ENSP00000320485   \n",
       "8         -  ARID1A          HGNC  HGNC:11110       YES  ENSP00000320485   \n",
       "9         -  ARID1A          HGNC  HGNC:11110       YES  ENSP00000320485   \n",
       "10        -  ARID1A          HGNC  HGNC:11110       YES  ENSP00000320485   \n",
       "11        -  ARID1A          HGNC  HGNC:11110       YES  ENSP00000320485   \n",
       "...     ...     ...           ...         ...       ...              ...   \n",
       "29067     -   STAG2          HGNC  HGNC:11355       YES  ENSP00000360187   \n",
       "29068     -   STAG2          HGNC  HGNC:11355       YES  ENSP00000360187   \n",
       "29069     -   STAG2          HGNC  HGNC:11355       YES  ENSP00000360187   \n",
       "29070     -   STAG2          HGNC  HGNC:11355       YES  ENSP00000360187   \n",
       "29071     -   STAG2          HGNC  HGNC:11355       YES  ENSP00000360187   \n",
       "\n",
       "       gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  \\\n",
       "7             0.0             0.0             0.0             0.0   \n",
       "8             0.0             0.0             0.0             0.0   \n",
       "9             0.0             0.0             0.0             0.0   \n",
       "10            0.0             0.0             0.0             0.0   \n",
       "11            0.0             0.0             0.0             0.0   \n",
       "...           ...             ...             ...             ...   \n",
       "29067         0.0             0.0             0.0             0.0   \n",
       "29068         0.0             0.0             0.0             0.0   \n",
       "29069         0.0             0.0             0.0             0.0   \n",
       "29070         0.0             0.0             0.0             0.0   \n",
       "29071         0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  \\\n",
       "7                 0.0             0.0             0.0             0.0   \n",
       "8                 0.0             0.0             0.0             0.0   \n",
       "9                 0.0             0.0             0.0             0.0   \n",
       "10                0.0             0.0             0.0             0.0   \n",
       "11                0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29067             0.0             0.0             0.0             0.0   \n",
       "29068             0.0             0.0             0.0             0.0   \n",
       "29069             0.0             0.0             0.0             0.0   \n",
       "29070             0.0             0.0             0.0             0.0   \n",
       "29071             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  \\\n",
       "7                 0.0         0.0             0.0             0.0   \n",
       "8                 0.0         0.0             0.0             0.0   \n",
       "9                 0.0         0.0             0.0             0.0   \n",
       "10                0.0         0.0             0.0             0.0   \n",
       "11                0.0         0.0             0.0             0.0   \n",
       "...               ...         ...             ...             ...   \n",
       "29067             0.0         0.0             0.0             0.0   \n",
       "29068             0.0         0.0             0.0             0.0   \n",
       "29069             0.0         0.0             0.0             0.0   \n",
       "29070             0.0         0.0             0.0             0.0   \n",
       "29071             0.0         0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  \\\n",
       "7                 0.0             0.0             0.0             0.0   \n",
       "8                 0.0             0.0             0.0             0.0   \n",
       "9                 0.0             0.0             0.0             0.0   \n",
       "10                0.0             0.0             0.0             0.0   \n",
       "11                0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29067             0.0             0.0             0.0             0.0   \n",
       "29068             0.0             0.0             0.0             0.0   \n",
       "29069             0.0             0.0             0.0             0.0   \n",
       "29070             0.0             0.0             0.0             0.0   \n",
       "29071             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "7                 0.0             0.0             0.0             0.0   \n",
       "8                 0.0             0.0             0.0             0.0   \n",
       "9                 0.0             0.0             0.0             0.0   \n",
       "10                0.0             0.0             0.0             0.0   \n",
       "11                0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29067             0.0             0.0             0.0             0.0   \n",
       "29068             0.0             0.0             0.0             0.0   \n",
       "29069             0.0             0.0             0.0             0.0   \n",
       "29070             0.0             0.0             0.0             0.0   \n",
       "29071             0.0             0.0             0.0             0.0   \n",
       "\n",
       "      CLIN_SIG SOMATIC PHENO canonical_Location canonical_Allele  \\\n",
       "7            -     0,1   0,1      chr1:26729669                T   \n",
       "8            -     0,1   0,1      chr1:26729669                T   \n",
       "9            -     0,1   0,1      chr1:26729669                T   \n",
       "10           -     0,1   0,1      chr1:26729669                T   \n",
       "11           -     0,1   0,1      chr1:26729669                T   \n",
       "...        ...     ...   ...                ...              ...   \n",
       "29067        -       -     -     chrX:124100595                C   \n",
       "29068        -       -     -     chrX:124100595                A   \n",
       "29069        -       -     -     chrX:124100596                C   \n",
       "29070        -       -     -     chrX:124100596                C   \n",
       "29071        -       -     -     chrX:124100597                T   \n",
       "\n",
       "        canonical_Gene canonical_Feature canonical_Feature_type  \\\n",
       "7      ENSG00000117713   ENST00000324856             Transcript   \n",
       "8      ENSG00000117713   ENST00000324856             Transcript   \n",
       "9      ENSG00000117713   ENST00000324856             Transcript   \n",
       "10     ENSG00000117713   ENST00000324856             Transcript   \n",
       "11     ENSG00000117713   ENST00000324856             Transcript   \n",
       "...                ...               ...                    ...   \n",
       "29067  ENSG00000101972   ENST00000371145             Transcript   \n",
       "29068  ENSG00000101972   ENST00000371145             Transcript   \n",
       "29069  ENSG00000101972   ENST00000371145             Transcript   \n",
       "29070  ENSG00000101972   ENST00000371145             Transcript   \n",
       "29071  ENSG00000101972   ENST00000371145             Transcript   \n",
       "\n",
       "      canonical_Consequence canonical_cDNA_position  canonical_CDS_position  \\\n",
       "7               stop_gained                    1545                    1156   \n",
       "8               stop_gained                    1545                    1156   \n",
       "9               stop_gained                    1545                    1156   \n",
       "10              stop_gained                    1545                    1156   \n",
       "11              stop_gained                    1545                    1156   \n",
       "...                     ...                     ...                     ...   \n",
       "29067             stop_lost                    4118                    3805   \n",
       "29068             stop_lost                    4118                    3805   \n",
       "29069             stop_lost                    4119                    3806   \n",
       "29070             stop_lost                    4119                    3806   \n",
       "29071             stop_lost                    4120                    3807   \n",
       "\n",
       "      canonical_Protein_position canonical_Amino_acids canonical_Codons  \\\n",
       "7                            386                   Q/*          Cag/Tag   \n",
       "8                            386                   Q/*          Cag/Tag   \n",
       "9                            386                   Q/*          Cag/Tag   \n",
       "10                           386                   Q/*          Cag/Tag   \n",
       "11                           386                   Q/*          Cag/Tag   \n",
       "...                          ...                   ...              ...   \n",
       "29067                       1269                   */Q          Taa/Caa   \n",
       "29068                       1269                   */K          Taa/Aaa   \n",
       "29069                       1269                   */S          tAa/tCa   \n",
       "29070                       1269                   */S          tAa/tCa   \n",
       "29071                       1269                   */Y          taA/taT   \n",
       "\n",
       "      canonical_IMPACT canonical_DISTANCE canonical_STRAND canonical_FLAGS  \\\n",
       "7                 HIGH                  -              1.0               -   \n",
       "8                 HIGH                  -              1.0               -   \n",
       "9                 HIGH                  -              1.0               -   \n",
       "10                HIGH                  -              1.0               -   \n",
       "11                HIGH                  -              1.0               -   \n",
       "...                ...                ...              ...             ...   \n",
       "29067             HIGH                  -              1.0               -   \n",
       "29068             HIGH                  -              1.0               -   \n",
       "29069             HIGH                  -              1.0               -   \n",
       "29070             HIGH                  -              1.0               -   \n",
       "29071             HIGH                  -              1.0               -   \n",
       "\n",
       "      canonical_SYMBOL canonical_SYMBOL_SOURCE canonical_HGNC_ID  \\\n",
       "7               ARID1A                    HGNC        HGNC:11110   \n",
       "8               ARID1A                    HGNC        HGNC:11110   \n",
       "9               ARID1A                    HGNC        HGNC:11110   \n",
       "10              ARID1A                    HGNC        HGNC:11110   \n",
       "11              ARID1A                    HGNC        HGNC:11110   \n",
       "...                ...                     ...               ...   \n",
       "29067            STAG2                    HGNC        HGNC:11355   \n",
       "29068            STAG2                    HGNC        HGNC:11355   \n",
       "29069            STAG2                    HGNC        HGNC:11355   \n",
       "29070            STAG2                    HGNC        HGNC:11355   \n",
       "29071            STAG2                    HGNC        HGNC:11355   \n",
       "\n",
       "      canonical_CANONICAL   canonical_ENSP canonical_CLIN_SIG  \\\n",
       "7                     YES  ENSP00000320485                  -   \n",
       "8                     YES  ENSP00000320485                  -   \n",
       "9                     YES  ENSP00000320485                  -   \n",
       "10                    YES  ENSP00000320485                  -   \n",
       "11                    YES  ENSP00000320485                  -   \n",
       "...                   ...              ...                ...   \n",
       "29067                 YES  ENSP00000360187                  -   \n",
       "29068                 YES  ENSP00000360187                  -   \n",
       "29069                 YES  ENSP00000360187                  -   \n",
       "29070                 YES  ENSP00000360187                  -   \n",
       "29071                 YES  ENSP00000360187                  -   \n",
       "\n",
       "      canonical_Consequence_single canonical_Consequence_broader  \\\n",
       "7                      stop_gained                      nonsense   \n",
       "8                      stop_gained                      nonsense   \n",
       "9                      stop_gained                      nonsense   \n",
       "10                     stop_gained                      nonsense   \n",
       "11                     stop_gained                      nonsense   \n",
       "...                            ...                           ...   \n",
       "29067                    stop_lost                      nonsense   \n",
       "29068                    stop_lost                      nonsense   \n",
       "29069                    stop_lost                      nonsense   \n",
       "29070                    stop_lost                      nonsense   \n",
       "29071                    stop_lost                      nonsense   \n",
       "\n",
       "      canonical_Protein_affecting Consequence_single Consequence_broader  \\\n",
       "7               protein_affecting        stop_gained            nonsense   \n",
       "8               protein_affecting        stop_gained            nonsense   \n",
       "9               protein_affecting        stop_gained            nonsense   \n",
       "10              protein_affecting        stop_gained            nonsense   \n",
       "11              protein_affecting        stop_gained            nonsense   \n",
       "...                           ...                ...                 ...   \n",
       "29067           protein_affecting          stop_lost            nonsense   \n",
       "29068           protein_affecting          stop_lost            nonsense   \n",
       "29069           protein_affecting          stop_lost            nonsense   \n",
       "29070           protein_affecting          stop_lost            nonsense   \n",
       "29071           protein_affecting          stop_lost            nonsense   \n",
       "\n",
       "       Protein_affecting CONTEXT_MUT MUTTYPE CONTEXT_MUT_SIGPRO  gnomAD_SNP  \\\n",
       "7      protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "8      protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "9      protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "10     protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "11     protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "...                  ...         ...     ...                ...         ...   \n",
       "29067  protein_affecting       TTA>C     T>C            T[T>C]A       False   \n",
       "29068  protein_affecting       TTA>A     T>A            T[T>A]A       False   \n",
       "29069  protein_affecting       TTA>G     T>G            T[T>G]A       False   \n",
       "29070  protein_affecting       TTA>G     T>G            T[T>G]A       False   \n",
       "29071  protein_affecting       ATT>A     T>A            A[T>A]T       False   \n",
       "\n",
       "      HOTSPOT_NAME          MUT_ID_pyr CHROM_ensembl  POS_ensembl REF_ensembl  \\\n",
       "7                -   chr1:26729669_C>T             1     26729669           C   \n",
       "8                -   chr1:26729669_C>T             1     26729669           C   \n",
       "9                -   chr1:26729669_C>T             1     26729669           C   \n",
       "10               -   chr1:26729669_C>T             1     26729669           C   \n",
       "11               -   chr1:26729669_C>T             1     26729669           C   \n",
       "...            ...                 ...           ...          ...         ...   \n",
       "29067            -  chrX:124100595_T>C             X    124100595           T   \n",
       "29068            -  chrX:124100595_T>A             X    124100595           T   \n",
       "29069            -  chrX:124100596_T>G             X    124100596           A   \n",
       "29070            -  chrX:124100596_T>G             X    124100596           A   \n",
       "29071            -  chrX:124100597_T>A             X    124100597           A   \n",
       "\n",
       "      ALT_ensembl TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  \\\n",
       "7               T  SNV             0          False            False   \n",
       "8               T  SNV             0          False            False   \n",
       "9               T  SNV             0          False            False   \n",
       "10              T  SNV             0          False            False   \n",
       "11              T  SNV             0          False            False   \n",
       "...           ...  ...           ...            ...              ...   \n",
       "29067           C  SNV             0          False            False   \n",
       "29068           A  SNV             0          False            False   \n",
       "29069           C  SNV             0          False            False   \n",
       "29070           C  SNV             0          False            False   \n",
       "29071           T  SNV             0          False            False   \n",
       "\n",
       "       N_rich_frequency  VAF_Ns_threshold  FILTER.cohort_n_rich_uni  \\\n",
       "7                   0.0               1.1                     False   \n",
       "8                   0.0               1.1                     False   \n",
       "9                   0.0               1.1                     False   \n",
       "10                  0.0               1.1                     False   \n",
       "11                  0.0               1.1                     False   \n",
       "...                 ...               ...                       ...   \n",
       "29067               0.0               1.1                     False   \n",
       "29068               0.0               1.1                     False   \n",
       "29069               0.0               1.1                     False   \n",
       "29070               0.0               1.1                     False   \n",
       "29071               0.0               1.1                     False   \n",
       "\n",
       "       FILTER.gnomAD_SNP  FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  \\\n",
       "7                  False               False                False       False   \n",
       "8                  False               False                False       False   \n",
       "9                  False               False                False       False   \n",
       "10                 False               False                False       False   \n",
       "11                 False               False                False       False   \n",
       "...                  ...                 ...                  ...         ...   \n",
       "29067              False               False                False       False   \n",
       "29068              False               False                False       False   \n",
       "29069              False               False                False       False   \n",
       "29070              False               False                False       False   \n",
       "29071              False               False                False       False   \n",
       "\n",
       "       FILTER.pSTD  FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "7             True                       True                 False   \n",
       "8             True                       True                 False   \n",
       "9             True                       True                 False   \n",
       "10            True                       True                 False   \n",
       "11            True                       True                 False   \n",
       "...            ...                        ...                   ...   \n",
       "29067         True                      False                 False   \n",
       "29068        False                      False                 False   \n",
       "29069         True                      False                 False   \n",
       "29070         True                      False                 False   \n",
       "29071         True                      False                 False   \n",
       "\n",
       "       FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "7                              False           False   \n",
       "8                              False           False   \n",
       "9                              False           False   \n",
       "10                             False           False   \n",
       "11                             False           False   \n",
       "...                              ...             ...   \n",
       "29067                          False           False   \n",
       "29068                          False           False   \n",
       "29069                          False           False   \n",
       "29070                          False           False   \n",
       "29071                          False           False   \n",
       "\n",
       "       FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "7                               False                        False   \n",
       "8                               False                        False   \n",
       "9                               False                        False   \n",
       "10                              False                        False   \n",
       "11                              False                        False   \n",
       "...                               ...                          ...   \n",
       "29067                           False                        False   \n",
       "29068                           False                        False   \n",
       "29069                           False                        False   \n",
       "29070                           False                        False   \n",
       "29071                           False                        False   \n",
       "\n",
       "       FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "7                         False          False                    False   \n",
       "8                         False          False                    False   \n",
       "9                         False          False                    False   \n",
       "10                        False          False                    False   \n",
       "11                        False          False                    False   \n",
       "...                         ...            ...                      ...   \n",
       "29067                     False          False                    False   \n",
       "29068                     False          False                    False   \n",
       "29069                     False          False                    False   \n",
       "29070                     False          False                    False   \n",
       "29071                     False          False                    False   \n",
       "\n",
       "       FILTER.PASS  FILTER.AM_not_searched_COMPLEX  \\\n",
       "7            False                           False   \n",
       "8            False                           False   \n",
       "9            False                           False   \n",
       "10           False                           False   \n",
       "11           False                           False   \n",
       "...            ...                             ...   \n",
       "29067        False                           False   \n",
       "29068         True                           False   \n",
       "29069        False                           False   \n",
       "29070        False                           False   \n",
       "29071        False                           False   \n",
       "\n",
       "       FILTER.not_searched_COMPLEX  FILTER.MSI12  FILTER.AM_not_searched_SV  \\\n",
       "7                            False         False                      False   \n",
       "8                            False         False                      False   \n",
       "9                            False         False                      False   \n",
       "10                           False         False                      False   \n",
       "11                           False         False                      False   \n",
       "...                            ...           ...                        ...   \n",
       "29067                        False         False                      False   \n",
       "29068                        False         False                      False   \n",
       "29069                        False         False                      False   \n",
       "29070                        False         False                      False   \n",
       "29071                        False         False                      False   \n",
       "\n",
       "       FILTER.not_searched_SV  FILTER.SN1.5  FILTER.q22.5  FILTER.v1  \\\n",
       "7                       False         False         False      False   \n",
       "8                       False         False         False      False   \n",
       "9                       False         False         False      False   \n",
       "10                      False         False         False      False   \n",
       "11                      False         False         False      False   \n",
       "...                       ...           ...           ...        ...   \n",
       "29067                   False         False         False      False   \n",
       "29068                   False         False         False      False   \n",
       "29069                   False         False         False      False   \n",
       "29070                   False         False         False      False   \n",
       "29071                   False         False         False      False   \n",
       "\n",
       "       FILTER.low_mappability  FILTER.Bias  FILTER.d3  \n",
       "7                       False        False      False  \n",
       "8                       False        False      False  \n",
       "9                       False        False      False  \n",
       "10                      False        False      False  \n",
       "11                      False        False      False  \n",
       "...                       ...          ...        ...  \n",
       "29067                   False        False      False  \n",
       "29068                   False        False      False  \n",
       "29069                   False        False      False  \n",
       "29070                   False        False      False  \n",
       "29071                   False        False      False  \n",
       "\n",
       "[10727 rows x 149 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs_maf[snvs_maf[\"canonical_Consequence_broader\"].isin(['nonsense',\n",
    "                                                         \"essential_splice\"])]\n",
    "# [\n",
    "#     ['CHROM', 'POS', 'canonical_Consequence', 'canonical_cDNA_position', 'canonical_CDS_position', 'canonical_Protein_position',]].drop_duplicates().tail(50).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "635f12c1-27ce-4773-91e9-2216a77849f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fill_missing_positions_filtered(dddd)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fd0bfab-c565-4d33-84a7-6a099800c926",
   "metadata": {},
   "source": [
    "Synonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e14540e2-7b6c-4b5c-a146-340e4a2c0a92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729692</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0009_BTR_01;TYPE=SNV;DP=5730;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:5730:1:5729,1:0.0002:3059,2670:0,1:5733:57...</td>\n",
       "      <td>5733</td>\n",
       "      <td>1</td>\n",
       "      <td>5732</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>5730</td>\n",
       "      <td>1</td>\n",
       "      <td>5729</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>22</td>\n",
       "      <td>0.003823</td>\n",
       "      <td>9764</td>\n",
       "      <td>1</td>\n",
       "      <td>9763</td>\n",
       "      <td>0.000102</td>\n",
       "      <td>25</td>\n",
       "      <td>0.002554</td>\n",
       "      <td>4031</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.587157</td>\n",
       "      <td>0.942038</td>\n",
       "      <td>chr1:26729692_G&gt;A</td>\n",
       "      <td>P19_0009_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729692</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1568</td>\n",
       "      <td>1179</td>\n",
       "      <td>393</td>\n",
       "      <td>Q</td>\n",
       "      <td>caG/caA</td>\n",
       "      <td>rs2124784918</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729692</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1568</td>\n",
       "      <td>1179</td>\n",
       "      <td>393</td>\n",
       "      <td>Q</td>\n",
       "      <td>caG/caA</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>GCT&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>G[C&gt;T]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729692_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729692</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0008_BDO_01;TYPE=SNV;DP=6414;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6414:1:6413,1:0.0002:3714,2699:1,0:6414:64...</td>\n",
       "      <td>6414</td>\n",
       "      <td>1</td>\n",
       "      <td>6413</td>\n",
       "      <td>0.000156</td>\n",
       "      <td>6414</td>\n",
       "      <td>1</td>\n",
       "      <td>6413</td>\n",
       "      <td>0.000156</td>\n",
       "      <td>12</td>\n",
       "      <td>0.001867</td>\n",
       "      <td>11681</td>\n",
       "      <td>1</td>\n",
       "      <td>11680</td>\n",
       "      <td>0.000086</td>\n",
       "      <td>17</td>\n",
       "      <td>0.001453</td>\n",
       "      <td>5267</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.549097</td>\n",
       "      <td>0.935992</td>\n",
       "      <td>chr1:26729704_G&gt;C</td>\n",
       "      <td>P19_0008_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggC</td>\n",
       "      <td>rs1191450683</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggC</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>TCC&gt;G</td>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>T[C&gt;G]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704_C&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BTR_01;TYPE=SNV;DP=4454;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4454:1:4453,1:0.0002:2533,1920:0,1:4454:44...</td>\n",
       "      <td>4454</td>\n",
       "      <td>1</td>\n",
       "      <td>4453</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>4454</td>\n",
       "      <td>1</td>\n",
       "      <td>4453</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>6</td>\n",
       "      <td>0.001345</td>\n",
       "      <td>7381</td>\n",
       "      <td>1</td>\n",
       "      <td>7380</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>16</td>\n",
       "      <td>0.002163</td>\n",
       "      <td>2927</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.603441</td>\n",
       "      <td>0.943289</td>\n",
       "      <td>chr1:26729704_G&gt;T</td>\n",
       "      <td>P19_0002_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggT</td>\n",
       "      <td>rs1191450683</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggT</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>TCC&gt;A</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>T[C&gt;A]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704_C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BDO_01;TYPE=SNV;DP=4693;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4693:1:4692,1:0.0002:2341,2351:0,1:4693:46...</td>\n",
       "      <td>4693</td>\n",
       "      <td>1</td>\n",
       "      <td>4692</td>\n",
       "      <td>0.000213</td>\n",
       "      <td>4693</td>\n",
       "      <td>1</td>\n",
       "      <td>4692</td>\n",
       "      <td>0.000213</td>\n",
       "      <td>6</td>\n",
       "      <td>0.001277</td>\n",
       "      <td>7005</td>\n",
       "      <td>1</td>\n",
       "      <td>7004</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>10</td>\n",
       "      <td>0.001426</td>\n",
       "      <td>2312</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.669950</td>\n",
       "      <td>0.954762</td>\n",
       "      <td>chr1:26729704_G&gt;A</td>\n",
       "      <td>P19_0002_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggA</td>\n",
       "      <td>rs1191450683</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1580</td>\n",
       "      <td>1191</td>\n",
       "      <td>397</td>\n",
       "      <td>G</td>\n",
       "      <td>ggG/ggA</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>TCC&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729704_C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729704</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729713</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0050_BDO_01;TYPE=SNV;DP=9411;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:9411:1:9410,1:0.0001:5155,4255:0,1:9413:94...</td>\n",
       "      <td>9413</td>\n",
       "      <td>1</td>\n",
       "      <td>9412</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>9411</td>\n",
       "      <td>1</td>\n",
       "      <td>9410</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>21</td>\n",
       "      <td>0.002226</td>\n",
       "      <td>18825</td>\n",
       "      <td>1</td>\n",
       "      <td>18824</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>29</td>\n",
       "      <td>0.001538</td>\n",
       "      <td>9412</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.500027</td>\n",
       "      <td>0.929585</td>\n",
       "      <td>chr1:26729713_A&gt;G</td>\n",
       "      <td>P19_0050_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729713</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1589</td>\n",
       "      <td>1200</td>\n",
       "      <td>400</td>\n",
       "      <td>P</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>-</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729713</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1589</td>\n",
       "      <td>1200</td>\n",
       "      <td>400</td>\n",
       "      <td>P</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>ATG&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]G</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729713_T&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>26729713</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29014</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095377</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0045_BTR_01;TYPE=SNV;DP=1925;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:1925:1:1924,1:0.0005:1029,895:1,0:1925:192...</td>\n",
       "      <td>1925</td>\n",
       "      <td>1</td>\n",
       "      <td>1924</td>\n",
       "      <td>0.000519</td>\n",
       "      <td>1925</td>\n",
       "      <td>1</td>\n",
       "      <td>1924</td>\n",
       "      <td>0.000519</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2675</td>\n",
       "      <td>1</td>\n",
       "      <td>2674</td>\n",
       "      <td>0.000374</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000374</td>\n",
       "      <td>750</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.719626</td>\n",
       "      <td>0.958308</td>\n",
       "      <td>chrX:124095377_A&gt;G</td>\n",
       "      <td>P19_0045_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095377</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4024</td>\n",
       "      <td>3711</td>\n",
       "      <td>1237</td>\n",
       "      <td>P</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>-</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095377</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4024</td>\n",
       "      <td>3711</td>\n",
       "      <td>1237</td>\n",
       "      <td>P</td>\n",
       "      <td>ccA/ccG</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>ATG&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]G</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095377_T&gt;C</td>\n",
       "      <td>X</td>\n",
       "      <td>124095377</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29043</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095395</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0019_BTR_01;TYPE=SNV;DP=5919;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:5919:1:5918,1:0.0002:2959,2959:1,0:5919:59...</td>\n",
       "      <td>5919</td>\n",
       "      <td>1</td>\n",
       "      <td>5918</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>5919</td>\n",
       "      <td>1</td>\n",
       "      <td>5918</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>10472</td>\n",
       "      <td>1</td>\n",
       "      <td>10471</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000286</td>\n",
       "      <td>4553</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.565222</td>\n",
       "      <td>0.938363</td>\n",
       "      <td>chrX:124095395_G&gt;A</td>\n",
       "      <td>P19_0019_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095395</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4042</td>\n",
       "      <td>3729</td>\n",
       "      <td>1243</td>\n",
       "      <td>E</td>\n",
       "      <td>gaG/gaA</td>\n",
       "      <td>COSV99494589</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chrX:124095395</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4042</td>\n",
       "      <td>3729</td>\n",
       "      <td>1243</td>\n",
       "      <td>E</td>\n",
       "      <td>gaG/gaA</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>TCT&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095395_C&gt;T</td>\n",
       "      <td>X</td>\n",
       "      <td>124095395</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29048</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095401</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0003_BDO_01;TYPE=SNV;DP=7668;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7668:1:7667,1:0.0001:3866,3801:1,0:7668:76...</td>\n",
       "      <td>7668</td>\n",
       "      <td>1</td>\n",
       "      <td>7667</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>7668</td>\n",
       "      <td>1</td>\n",
       "      <td>7667</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>17</td>\n",
       "      <td>0.002212</td>\n",
       "      <td>18363</td>\n",
       "      <td>1</td>\n",
       "      <td>18362</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>33</td>\n",
       "      <td>0.001794</td>\n",
       "      <td>10695</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.417579</td>\n",
       "      <td>0.911054</td>\n",
       "      <td>chrX:124095401_A&gt;T</td>\n",
       "      <td>P19_0003_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095401</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4048</td>\n",
       "      <td>3735</td>\n",
       "      <td>1245</td>\n",
       "      <td>T</td>\n",
       "      <td>acA/acT</td>\n",
       "      <td>-</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095401</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4048</td>\n",
       "      <td>3735</td>\n",
       "      <td>1245</td>\n",
       "      <td>T</td>\n",
       "      <td>acA/acT</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>CTG&gt;A</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>C[T&gt;A]G</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095401_T&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124095401</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29050</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095410</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0019_BTR_01;TYPE=SNV;DP=5726;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:5726:1:5725,1:0.0002:2757,2968:1,0:5726:57...</td>\n",
       "      <td>5726</td>\n",
       "      <td>1</td>\n",
       "      <td>5725</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>5726</td>\n",
       "      <td>1</td>\n",
       "      <td>5725</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>18</td>\n",
       "      <td>0.003134</td>\n",
       "      <td>10086</td>\n",
       "      <td>1</td>\n",
       "      <td>10085</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>27</td>\n",
       "      <td>0.002670</td>\n",
       "      <td>4360</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.567718</td>\n",
       "      <td>0.938590</td>\n",
       "      <td>chrX:124095410_G&gt;A</td>\n",
       "      <td>P19_0019_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095410</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4057</td>\n",
       "      <td>3744</td>\n",
       "      <td>1248</td>\n",
       "      <td>K</td>\n",
       "      <td>aaG/aaA</td>\n",
       "      <td>rs750598294</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>chrX:124095410</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4057</td>\n",
       "      <td>3744</td>\n",
       "      <td>1248</td>\n",
       "      <td>K</td>\n",
       "      <td>aaG/aaA</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>GCT&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>G[C&gt;T]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095410_C&gt;T</td>\n",
       "      <td>X</td>\n",
       "      <td>124095410</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29064</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100585</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0008_BTR_01;TYPE=SNV;DP=2633;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2633:1:2632,1:0.0004:1466,1166:1,0:2633:26...</td>\n",
       "      <td>2633</td>\n",
       "      <td>1</td>\n",
       "      <td>2632</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>2633</td>\n",
       "      <td>1</td>\n",
       "      <td>2632</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>15</td>\n",
       "      <td>0.005665</td>\n",
       "      <td>5136</td>\n",
       "      <td>1</td>\n",
       "      <td>5135</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>20</td>\n",
       "      <td>0.003879</td>\n",
       "      <td>2503</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.512656</td>\n",
       "      <td>0.921799</td>\n",
       "      <td>chrX:124100585_G&gt;T</td>\n",
       "      <td>P19_0008_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100585</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4108</td>\n",
       "      <td>3795</td>\n",
       "      <td>1265</td>\n",
       "      <td>V</td>\n",
       "      <td>gtG/gtT</td>\n",
       "      <td>-</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100585</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>4108</td>\n",
       "      <td>3795</td>\n",
       "      <td>1265</td>\n",
       "      <td>V</td>\n",
       "      <td>gtG/gtT</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>non_protein_affecting</td>\n",
       "      <td>ACA&gt;A</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>A[C&gt;A]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100585_C&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124100585</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2970 rows  149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT    FILTER  \\\n",
       "31     chr1   26729692   G   A      pSTD   \n",
       "35     chr1   26729704   G   C      pSTD   \n",
       "36     chr1   26729704   G   T      pSTD   \n",
       "37     chr1   26729704   G   A      pSTD   \n",
       "38     chr1   26729713   A   G      pSTD   \n",
       "...     ...        ...  ..  ..       ...   \n",
       "29014  chrX  124095377   A   G      pSTD   \n",
       "29043  chrX  124095395   G   A      pSTD   \n",
       "29048  chrX  124095401   A   T      pSTD   \n",
       "29050  chrX  124095410   G   A      pSTD   \n",
       "29064  chrX  124100585   G   T  p10;pSTD   \n",
       "\n",
       "                                                    INFO  \\\n",
       "31     SAMPLE=P19_0009_BTR_01;TYPE=SNV;DP=5730;VD=1;A...   \n",
       "35     SAMPLE=P19_0008_BDO_01;TYPE=SNV;DP=6414;VD=1;A...   \n",
       "36     SAMPLE=P19_0002_BTR_01;TYPE=SNV;DP=4454;VD=1;A...   \n",
       "37     SAMPLE=P19_0002_BDO_01;TYPE=SNV;DP=4693;VD=1;A...   \n",
       "38     SAMPLE=P19_0050_BDO_01;TYPE=SNV;DP=9411;VD=1;A...   \n",
       "...                                                  ...   \n",
       "29014  SAMPLE=P19_0045_BTR_01;TYPE=SNV;DP=1925;VD=1;A...   \n",
       "29043  SAMPLE=P19_0019_BTR_01;TYPE=SNV;DP=5919;VD=1;A...   \n",
       "29048  SAMPLE=P19_0003_BDO_01;TYPE=SNV;DP=7668;VD=1;A...   \n",
       "29050  SAMPLE=P19_0019_BTR_01;TYPE=SNV;DP=5726;VD=1;A...   \n",
       "29064  SAMPLE=P19_0008_BTR_01;TYPE=SNV;DP=2633;VD=1;A...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "31     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "35     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "36     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "37     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "38     GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "...                                                  ...   \n",
       "29014  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29043  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29048  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29050  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29064  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                  SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "31     0/1:5730:1:5729,1:0.0002:3059,2670:0,1:5733:57...   5733          1   \n",
       "35     0/1:6414:1:6413,1:0.0002:3714,2699:1,0:6414:64...   6414          1   \n",
       "36     0/1:4454:1:4453,1:0.0002:2533,1920:0,1:4454:44...   4454          1   \n",
       "37     0/1:4693:1:4692,1:0.0002:2341,2351:0,1:4693:46...   4693          1   \n",
       "38     0/1:9411:1:9410,1:0.0001:5155,4255:0,1:9413:94...   9413          1   \n",
       "...                                                  ...    ...        ...   \n",
       "29014  0/1:1925:1:1924,1:0.0005:1029,895:1,0:1925:192...   1925          1   \n",
       "29043  0/1:5919:1:5918,1:0.0002:2959,2959:1,0:5919:59...   5919          1   \n",
       "29048  0/1:7668:1:7667,1:0.0001:3866,3801:1,0:7668:76...   7668          1   \n",
       "29050  0/1:5726:1:5725,1:0.0002:2757,2968:1,0:5726:57...   5726          1   \n",
       "29064  0/1:2633:1:2632,1:0.0004:1466,1166:1,0:2633:26...   2633          1   \n",
       "\n",
       "       REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "31          5732  0.000174      5730             1          5729  0.000175   \n",
       "35          6413  0.000156      6414             1          6413  0.000156   \n",
       "36          4453  0.000225      4454             1          4453  0.000225   \n",
       "37          4692  0.000213      4693             1          4692  0.000213   \n",
       "38          9412  0.000106      9411             1          9410  0.000106   \n",
       "...          ...       ...       ...           ...           ...       ...   \n",
       "29014       1924  0.000519      1925             1          1924  0.000519   \n",
       "29043       5918  0.000169      5919             1          5918  0.000169   \n",
       "29048       7667  0.000130      7668             1          7667  0.000130   \n",
       "29050       5725  0.000175      5726             1          5725  0.000175   \n",
       "29064       2632  0.000380      2633             1          2632  0.000380   \n",
       "\n",
       "       numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "31        22  0.003823      9764             1          9763  0.000102   \n",
       "35        12  0.001867     11681             1         11680  0.000086   \n",
       "36         6  0.001345      7381             1          7380  0.000135   \n",
       "37         6  0.001277      7005             1          7004  0.000143   \n",
       "38        21  0.002226     18825             1         18824  0.000053   \n",
       "...      ...       ...       ...           ...           ...       ...   \n",
       "29014      0  0.000000      2675             1          2674  0.000374   \n",
       "29043      1  0.000169     10472             1         10471  0.000095   \n",
       "29048     17  0.002212     18363             1         18362  0.000054   \n",
       "29050     18  0.003134     10086             1         10085  0.000099   \n",
       "29064     15  0.005665      5136             1          5135  0.000195   \n",
       "\n",
       "       numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND  VAF_ND  \\\n",
       "31           25   0.002554      4031             0     0.0   \n",
       "35           17   0.001453      5267             0     0.0   \n",
       "36           16   0.002163      2927             0     0.0   \n",
       "37           10   0.001426      2312             0     0.0   \n",
       "38           29   0.001538      9412             0     0.0   \n",
       "...         ...        ...       ...           ...     ...   \n",
       "29014         1   0.000374       750             0     0.0   \n",
       "29043         3   0.000286      4553             0     0.0   \n",
       "29048        33   0.001794     10695             0     0.0   \n",
       "29050        27   0.002670      4360             0     0.0   \n",
       "29064        20   0.003879      2503             0     0.0   \n",
       "\n",
       "       VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "31                      False                      False        0.587157   \n",
       "35                      False                      False        0.549097   \n",
       "36                      False                      False        0.603441   \n",
       "37                      False                      False        0.669950   \n",
       "38                      False                      False        0.500027   \n",
       "...                       ...                        ...             ...   \n",
       "29014                   False                      False        0.719626   \n",
       "29043                   False                      False        0.565222   \n",
       "29048                   False                      False        0.417579   \n",
       "29050                   False                      False        0.567718   \n",
       "29064                   False                      False        0.512656   \n",
       "\n",
       "       VAF_distortion_sq              MUT_ID        SAMPLE_ID METHOD  \\\n",
       "31              0.942038   chr1:26729692_G>A  P19_0009_BTR_01    med   \n",
       "35              0.935992   chr1:26729704_G>C  P19_0008_BDO_01    med   \n",
       "36              0.943289   chr1:26729704_G>T  P19_0002_BTR_01    med   \n",
       "37              0.954762   chr1:26729704_G>A  P19_0002_BDO_01    med   \n",
       "38              0.929585   chr1:26729713_A>G  P19_0050_BDO_01    med   \n",
       "...                  ...                 ...              ...    ...   \n",
       "29014           0.958308  chrX:124095377_A>G  P19_0045_BTR_01    med   \n",
       "29043           0.938363  chrX:124095395_G>A  P19_0019_BTR_01    med   \n",
       "29048           0.911054  chrX:124095401_A>T  P19_0003_BDO_01    med   \n",
       "29050           0.938590  chrX:124095410_G>A  P19_0019_BTR_01    med   \n",
       "29064           0.921799  chrX:124100585_G>T  P19_0008_BTR_01    med   \n",
       "\n",
       "       PROJECT_NAME        Location Allele             Gene          Feature  \\\n",
       "31              NaN   chr1:26729692      A  ENSG00000117713  ENST00000324856   \n",
       "35              NaN   chr1:26729704      C  ENSG00000117713  ENST00000324856   \n",
       "36              NaN   chr1:26729704      T  ENSG00000117713  ENST00000324856   \n",
       "37              NaN   chr1:26729704      A  ENSG00000117713  ENST00000324856   \n",
       "38              NaN   chr1:26729713      G  ENSG00000117713  ENST00000324856   \n",
       "...             ...             ...    ...              ...              ...   \n",
       "29014           NaN  chrX:124095377      G  ENSG00000101972  ENST00000371145   \n",
       "29043           NaN  chrX:124095395      A  ENSG00000101972  ENST00000371145   \n",
       "29048           NaN  chrX:124095401      T  ENSG00000101972  ENST00000371145   \n",
       "29050           NaN  chrX:124095410      A  ENSG00000101972  ENST00000371145   \n",
       "29064           NaN  chrX:124100585      T  ENSG00000101972  ENST00000371145   \n",
       "\n",
       "      Feature_type         Consequence cDNA_position CDS_position  \\\n",
       "31      Transcript  synonymous_variant          1568         1179   \n",
       "35      Transcript  synonymous_variant          1580         1191   \n",
       "36      Transcript  synonymous_variant          1580         1191   \n",
       "37      Transcript  synonymous_variant          1580         1191   \n",
       "38      Transcript  synonymous_variant          1589         1200   \n",
       "...            ...                 ...           ...          ...   \n",
       "29014   Transcript  synonymous_variant          4024         3711   \n",
       "29043   Transcript  synonymous_variant          4042         3729   \n",
       "29048   Transcript  synonymous_variant          4048         3735   \n",
       "29050   Transcript  synonymous_variant          4057         3744   \n",
       "29064   Transcript  synonymous_variant          4108         3795   \n",
       "\n",
       "      Protein_position Amino_acids   Codons Existing_variation IMPACT  \\\n",
       "31                 393           Q  caG/caA       rs2124784918    LOW   \n",
       "35                 397           G  ggG/ggC       rs1191450683    LOW   \n",
       "36                 397           G  ggG/ggT       rs1191450683    LOW   \n",
       "37                 397           G  ggG/ggA       rs1191450683    LOW   \n",
       "38                 400           P  ccA/ccG                  -    LOW   \n",
       "...                ...         ...      ...                ...    ...   \n",
       "29014             1237           P  ccA/ccG                  -    LOW   \n",
       "29043             1243           E  gaG/gaA       COSV99494589    LOW   \n",
       "29048             1245           T  acA/acT                  -    LOW   \n",
       "29050             1248           K  aaG/aaA        rs750598294    LOW   \n",
       "29064             1265           V  gtG/gtT                  -    LOW   \n",
       "\n",
       "      DISTANCE STRAND FLAGS  SYMBOL SYMBOL_SOURCE     HGNC_ID CANONICAL  \\\n",
       "31           -      1     -  ARID1A          HGNC  HGNC:11110       YES   \n",
       "35           -      1     -  ARID1A          HGNC  HGNC:11110       YES   \n",
       "36           -      1     -  ARID1A          HGNC  HGNC:11110       YES   \n",
       "37           -      1     -  ARID1A          HGNC  HGNC:11110       YES   \n",
       "38           -      1     -  ARID1A          HGNC  HGNC:11110       YES   \n",
       "...        ...    ...   ...     ...           ...         ...       ...   \n",
       "29014        -      1     -   STAG2          HGNC  HGNC:11355       YES   \n",
       "29043        -      1     -   STAG2          HGNC  HGNC:11355       YES   \n",
       "29048        -      1     -   STAG2          HGNC  HGNC:11355       YES   \n",
       "29050        -      1     -   STAG2          HGNC  HGNC:11355       YES   \n",
       "29064        -      1     -   STAG2          HGNC  HGNC:11355       YES   \n",
       "\n",
       "                  ENSP  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  \\\n",
       "31     ENSP00000320485    0.000000             0.0        0.000000   \n",
       "35     ENSP00000320485    0.000000             0.0        0.000000   \n",
       "36     ENSP00000320485    0.000004             0.0        0.000000   \n",
       "37     ENSP00000320485    0.000004             0.0        0.000029   \n",
       "38     ENSP00000320485    0.000000             0.0        0.000000   \n",
       "...                ...         ...             ...             ...   \n",
       "29014  ENSP00000360187    0.000000             0.0        0.000000   \n",
       "29043  ENSP00000360187    0.000000             0.0        0.000000   \n",
       "29048  ENSP00000360187    0.000000             0.0        0.000000   \n",
       "29050  ENSP00000360187    0.000005             0.0        0.000000   \n",
       "29064  ENSP00000360187    0.000000             0.0        0.000000   \n",
       "\n",
       "       gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  \\\n",
       "31                0.0             0.0             0.0        0.000000   \n",
       "35                0.0             0.0             0.0        0.000000   \n",
       "36                0.0             0.0             0.0        0.000009   \n",
       "37                0.0             0.0             0.0        0.000000   \n",
       "38                0.0             0.0             0.0        0.000000   \n",
       "...               ...             ...             ...             ...   \n",
       "29014             0.0             0.0             0.0        0.000000   \n",
       "29043             0.0             0.0             0.0        0.000000   \n",
       "29048             0.0             0.0             0.0        0.000000   \n",
       "29050             0.0             0.0             0.0        0.000012   \n",
       "29064             0.0             0.0             0.0        0.000000   \n",
       "\n",
       "       gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  \\\n",
       "31                0.0             0.0    0.000000        0.000000   \n",
       "35                0.0             0.0    0.000000        0.000000   \n",
       "36                0.0             0.0    0.000007        0.000000   \n",
       "37                0.0             0.0    0.000000        0.000000   \n",
       "38                0.0             0.0    0.000000        0.000000   \n",
       "...               ...             ...         ...             ...   \n",
       "29014             0.0             0.0    0.000000        0.000000   \n",
       "29043             0.0             0.0    0.000000        0.000000   \n",
       "29048             0.0             0.0    0.000000        0.000000   \n",
       "29050             0.0             0.0    0.000009        0.000032   \n",
       "29064             0.0             0.0    0.000000        0.000000   \n",
       "\n",
       "       gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  \\\n",
       "31                0.0             0.0             0.0             0.0   \n",
       "35                0.0             0.0             0.0             0.0   \n",
       "36                0.0             0.0             0.0             0.0   \n",
       "37                0.0             0.0             0.0             0.0   \n",
       "38                0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29014             0.0             0.0             0.0             0.0   \n",
       "29043             0.0             0.0             0.0             0.0   \n",
       "29048             0.0             0.0             0.0             0.0   \n",
       "29050             0.0             0.0             0.0             0.0   \n",
       "29064             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  \\\n",
       "31                0.0             0.0        0.000000             0.0   \n",
       "35                0.0             0.0        0.000000             0.0   \n",
       "36                0.0             0.0        0.000015             0.0   \n",
       "37                0.0             0.0        0.000000             0.0   \n",
       "38                0.0             0.0        0.000000             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29014             0.0             0.0        0.000000             0.0   \n",
       "29043             0.0             0.0        0.000000             0.0   \n",
       "29048             0.0             0.0        0.000000             0.0   \n",
       "29050             0.0             0.0        0.000000             0.0   \n",
       "29064             0.0             0.0        0.000000             0.0   \n",
       "\n",
       "       gnomADg_SAS_AF       CLIN_SIG SOMATIC PHENO canonical_Location  \\\n",
       "31                0.0              -       -     -      chr1:26729692   \n",
       "35                0.0              -       -     -      chr1:26729704   \n",
       "36                0.0              -       -     -      chr1:26729704   \n",
       "37                0.0              -       -     -      chr1:26729704   \n",
       "38                0.0              -       -     -      chr1:26729713   \n",
       "...               ...            ...     ...   ...                ...   \n",
       "29014             0.0              -       -     -     chrX:124095377   \n",
       "29043             0.0              -       1     1     chrX:124095395   \n",
       "29048             0.0              -       -     -     chrX:124095401   \n",
       "29050             0.0  likely_benign       -     1     chrX:124095410   \n",
       "29064             0.0              -       -     -     chrX:124100585   \n",
       "\n",
       "      canonical_Allele   canonical_Gene canonical_Feature  \\\n",
       "31                   A  ENSG00000117713   ENST00000324856   \n",
       "35                   C  ENSG00000117713   ENST00000324856   \n",
       "36                   T  ENSG00000117713   ENST00000324856   \n",
       "37                   A  ENSG00000117713   ENST00000324856   \n",
       "38                   G  ENSG00000117713   ENST00000324856   \n",
       "...                ...              ...               ...   \n",
       "29014                G  ENSG00000101972   ENST00000371145   \n",
       "29043                A  ENSG00000101972   ENST00000371145   \n",
       "29048                T  ENSG00000101972   ENST00000371145   \n",
       "29050                A  ENSG00000101972   ENST00000371145   \n",
       "29064                T  ENSG00000101972   ENST00000371145   \n",
       "\n",
       "      canonical_Feature_type canonical_Consequence canonical_cDNA_position  \\\n",
       "31                Transcript    synonymous_variant                    1568   \n",
       "35                Transcript    synonymous_variant                    1580   \n",
       "36                Transcript    synonymous_variant                    1580   \n",
       "37                Transcript    synonymous_variant                    1580   \n",
       "38                Transcript    synonymous_variant                    1589   \n",
       "...                      ...                   ...                     ...   \n",
       "29014             Transcript    synonymous_variant                    4024   \n",
       "29043             Transcript    synonymous_variant                    4042   \n",
       "29048             Transcript    synonymous_variant                    4048   \n",
       "29050             Transcript    synonymous_variant                    4057   \n",
       "29064             Transcript    synonymous_variant                    4108   \n",
       "\n",
       "       canonical_CDS_position canonical_Protein_position  \\\n",
       "31                       1179                        393   \n",
       "35                       1191                        397   \n",
       "36                       1191                        397   \n",
       "37                       1191                        397   \n",
       "38                       1200                        400   \n",
       "...                       ...                        ...   \n",
       "29014                    3711                       1237   \n",
       "29043                    3729                       1243   \n",
       "29048                    3735                       1245   \n",
       "29050                    3744                       1248   \n",
       "29064                    3795                       1265   \n",
       "\n",
       "      canonical_Amino_acids canonical_Codons canonical_IMPACT  \\\n",
       "31                        Q          caG/caA              LOW   \n",
       "35                        G          ggG/ggC              LOW   \n",
       "36                        G          ggG/ggT              LOW   \n",
       "37                        G          ggG/ggA              LOW   \n",
       "38                        P          ccA/ccG              LOW   \n",
       "...                     ...              ...              ...   \n",
       "29014                     P          ccA/ccG              LOW   \n",
       "29043                     E          gaG/gaA              LOW   \n",
       "29048                     T          acA/acT              LOW   \n",
       "29050                     K          aaG/aaA              LOW   \n",
       "29064                     V          gtG/gtT              LOW   \n",
       "\n",
       "      canonical_DISTANCE canonical_STRAND canonical_FLAGS canonical_SYMBOL  \\\n",
       "31                     -              1.0               -           ARID1A   \n",
       "35                     -              1.0               -           ARID1A   \n",
       "36                     -              1.0               -           ARID1A   \n",
       "37                     -              1.0               -           ARID1A   \n",
       "38                     -              1.0               -           ARID1A   \n",
       "...                  ...              ...             ...              ...   \n",
       "29014                  -              1.0               -            STAG2   \n",
       "29043                  -              1.0               -            STAG2   \n",
       "29048                  -              1.0               -            STAG2   \n",
       "29050                  -              1.0               -            STAG2   \n",
       "29064                  -              1.0               -            STAG2   \n",
       "\n",
       "      canonical_SYMBOL_SOURCE canonical_HGNC_ID canonical_CANONICAL  \\\n",
       "31                       HGNC        HGNC:11110                 YES   \n",
       "35                       HGNC        HGNC:11110                 YES   \n",
       "36                       HGNC        HGNC:11110                 YES   \n",
       "37                       HGNC        HGNC:11110                 YES   \n",
       "38                       HGNC        HGNC:11110                 YES   \n",
       "...                       ...               ...                 ...   \n",
       "29014                    HGNC        HGNC:11355                 YES   \n",
       "29043                    HGNC        HGNC:11355                 YES   \n",
       "29048                    HGNC        HGNC:11355                 YES   \n",
       "29050                    HGNC        HGNC:11355                 YES   \n",
       "29064                    HGNC        HGNC:11355                 YES   \n",
       "\n",
       "        canonical_ENSP canonical_CLIN_SIG canonical_Consequence_single  \\\n",
       "31     ENSP00000320485                  -           synonymous_variant   \n",
       "35     ENSP00000320485                  -           synonymous_variant   \n",
       "36     ENSP00000320485                  -           synonymous_variant   \n",
       "37     ENSP00000320485                  -           synonymous_variant   \n",
       "38     ENSP00000320485                  -           synonymous_variant   \n",
       "...                ...                ...                          ...   \n",
       "29014  ENSP00000360187                  -           synonymous_variant   \n",
       "29043  ENSP00000360187                  -           synonymous_variant   \n",
       "29048  ENSP00000360187                  -           synonymous_variant   \n",
       "29050  ENSP00000360187      likely_benign           synonymous_variant   \n",
       "29064  ENSP00000360187                  -           synonymous_variant   \n",
       "\n",
       "      canonical_Consequence_broader canonical_Protein_affecting  \\\n",
       "31                       synonymous       non_protein_affecting   \n",
       "35                       synonymous       non_protein_affecting   \n",
       "36                       synonymous       non_protein_affecting   \n",
       "37                       synonymous       non_protein_affecting   \n",
       "38                       synonymous       non_protein_affecting   \n",
       "...                             ...                         ...   \n",
       "29014                    synonymous       non_protein_affecting   \n",
       "29043                    synonymous       non_protein_affecting   \n",
       "29048                    synonymous       non_protein_affecting   \n",
       "29050                    synonymous       non_protein_affecting   \n",
       "29064                    synonymous       non_protein_affecting   \n",
       "\n",
       "       Consequence_single Consequence_broader      Protein_affecting  \\\n",
       "31     synonymous_variant          synonymous  non_protein_affecting   \n",
       "35     synonymous_variant          synonymous  non_protein_affecting   \n",
       "36     synonymous_variant          synonymous  non_protein_affecting   \n",
       "37     synonymous_variant          synonymous  non_protein_affecting   \n",
       "38     synonymous_variant          synonymous  non_protein_affecting   \n",
       "...                   ...                 ...                    ...   \n",
       "29014  synonymous_variant          synonymous  non_protein_affecting   \n",
       "29043  synonymous_variant          synonymous  non_protein_affecting   \n",
       "29048  synonymous_variant          synonymous  non_protein_affecting   \n",
       "29050  synonymous_variant          synonymous  non_protein_affecting   \n",
       "29064  synonymous_variant          synonymous  non_protein_affecting   \n",
       "\n",
       "      CONTEXT_MUT MUTTYPE CONTEXT_MUT_SIGPRO  gnomAD_SNP HOTSPOT_NAME  \\\n",
       "31          GCT>T     C>T            G[C>T]T       False            -   \n",
       "35          TCC>G     C>G            T[C>G]C       False            -   \n",
       "36          TCC>A     C>A            T[C>A]C       False            -   \n",
       "37          TCC>T     C>T            T[C>T]C       False            -   \n",
       "38          ATG>C     T>C            A[T>C]G       False            -   \n",
       "...           ...     ...                ...         ...          ...   \n",
       "29014       ATG>C     T>C            A[T>C]G       False            -   \n",
       "29043       TCT>T     C>T            T[C>T]T       False            -   \n",
       "29048       CTG>A     T>A            C[T>A]G       False            -   \n",
       "29050       GCT>T     C>T            G[C>T]T       False            -   \n",
       "29064       ACA>A     C>A            A[C>A]A       False            -   \n",
       "\n",
       "               MUT_ID_pyr CHROM_ensembl  POS_ensembl REF_ensembl ALT_ensembl  \\\n",
       "31      chr1:26729692_C>T             1     26729692           G           A   \n",
       "35      chr1:26729704_C>G             1     26729704           G           C   \n",
       "36      chr1:26729704_C>A             1     26729704           G           T   \n",
       "37      chr1:26729704_C>T             1     26729704           G           A   \n",
       "38      chr1:26729713_T>C             1     26729713           A           G   \n",
       "...                   ...           ...          ...         ...         ...   \n",
       "29014  chrX:124095377_T>C             X    124095377           A           G   \n",
       "29043  chrX:124095395_C>T             X    124095395           G           A   \n",
       "29048  chrX:124095401_T>A             X    124095401           A           T   \n",
       "29050  chrX:124095410_C>T             X    124095410           G           A   \n",
       "29064  chrX:124100585_C>A             X    124100585           G           T   \n",
       "\n",
       "      TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  N_rich_frequency  \\\n",
       "31     SNV             0          False            False               0.0   \n",
       "35     SNV             0          False            False               0.0   \n",
       "36     SNV             0          False            False               0.0   \n",
       "37     SNV             0          False            False               0.0   \n",
       "38     SNV             0          False            False               0.0   \n",
       "...    ...           ...            ...              ...               ...   \n",
       "29014  SNV             0          False            False               0.0   \n",
       "29043  SNV             0          False            False               0.0   \n",
       "29048  SNV             0          False            False               0.0   \n",
       "29050  SNV             0          False            False               0.0   \n",
       "29064  SNV             0          False            False               0.0   \n",
       "\n",
       "       VAF_Ns_threshold  FILTER.cohort_n_rich_uni  FILTER.gnomAD_SNP  \\\n",
       "31                  1.1                     False              False   \n",
       "35                  1.1                     False              False   \n",
       "36                  1.1                     False              False   \n",
       "37                  1.1                     False              False   \n",
       "38                  1.1                     False              False   \n",
       "...                 ...                       ...                ...   \n",
       "29014               1.1                     False              False   \n",
       "29043               1.1                     False              False   \n",
       "29048               1.1                     False              False   \n",
       "29050               1.1                     False              False   \n",
       "29064               1.1                     False              False   \n",
       "\n",
       "       FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  FILTER.pSTD  \\\n",
       "31                  False                False       False         True   \n",
       "35                  False                False       False         True   \n",
       "36                  False                False       False         True   \n",
       "37                  False                False       False         True   \n",
       "38                  False                False       False         True   \n",
       "...                   ...                  ...         ...          ...   \n",
       "29014               False                False       False         True   \n",
       "29043               False                False       False         True   \n",
       "29048               False                False       False         True   \n",
       "29050               False                False       False         True   \n",
       "29064               False                False        True         True   \n",
       "\n",
       "       FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "31                         False                 False   \n",
       "35                         False                 False   \n",
       "36                         False                 False   \n",
       "37                         False                 False   \n",
       "38                         False                 False   \n",
       "...                          ...                   ...   \n",
       "29014                      False                 False   \n",
       "29043                      False                 False   \n",
       "29048                      False                 False   \n",
       "29050                      False                 False   \n",
       "29064                      False                 False   \n",
       "\n",
       "       FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "31                             False           False   \n",
       "35                             False           False   \n",
       "36                             False           False   \n",
       "37                             False           False   \n",
       "38                             False           False   \n",
       "...                              ...             ...   \n",
       "29014                          False           False   \n",
       "29043                          False           False   \n",
       "29048                          False           False   \n",
       "29050                          False           False   \n",
       "29064                          False           False   \n",
       "\n",
       "       FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "31                              False                        False   \n",
       "35                              False                        False   \n",
       "36                              False                        False   \n",
       "37                              False                        False   \n",
       "38                              False                        False   \n",
       "...                               ...                          ...   \n",
       "29014                           False                        False   \n",
       "29043                           False                        False   \n",
       "29048                           False                        False   \n",
       "29050                           False                        False   \n",
       "29064                           False                        False   \n",
       "\n",
       "       FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "31                        False          False                    False   \n",
       "35                        False          False                    False   \n",
       "36                        False          False                    False   \n",
       "37                        False          False                    False   \n",
       "38                        False          False                    False   \n",
       "...                         ...            ...                      ...   \n",
       "29014                     False          False                    False   \n",
       "29043                     False          False                    False   \n",
       "29048                     False          False                    False   \n",
       "29050                     False          False                    False   \n",
       "29064                     False          False                    False   \n",
       "\n",
       "       FILTER.PASS  FILTER.AM_not_searched_COMPLEX  \\\n",
       "31           False                           False   \n",
       "35           False                           False   \n",
       "36           False                           False   \n",
       "37           False                           False   \n",
       "38           False                           False   \n",
       "...            ...                             ...   \n",
       "29014        False                           False   \n",
       "29043        False                           False   \n",
       "29048        False                           False   \n",
       "29050        False                           False   \n",
       "29064        False                           False   \n",
       "\n",
       "       FILTER.not_searched_COMPLEX  FILTER.MSI12  FILTER.AM_not_searched_SV  \\\n",
       "31                           False         False                      False   \n",
       "35                           False         False                      False   \n",
       "36                           False         False                      False   \n",
       "37                           False         False                      False   \n",
       "38                           False         False                      False   \n",
       "...                            ...           ...                        ...   \n",
       "29014                        False         False                      False   \n",
       "29043                        False         False                      False   \n",
       "29048                        False         False                      False   \n",
       "29050                        False         False                      False   \n",
       "29064                        False         False                      False   \n",
       "\n",
       "       FILTER.not_searched_SV  FILTER.SN1.5  FILTER.q22.5  FILTER.v1  \\\n",
       "31                      False         False         False      False   \n",
       "35                      False         False         False      False   \n",
       "36                      False         False         False      False   \n",
       "37                      False         False         False      False   \n",
       "38                      False         False         False      False   \n",
       "...                       ...           ...           ...        ...   \n",
       "29014                   False         False         False      False   \n",
       "29043                   False         False         False      False   \n",
       "29048                   False         False         False      False   \n",
       "29050                   False         False         False      False   \n",
       "29064                   False         False         False      False   \n",
       "\n",
       "       FILTER.low_mappability  FILTER.Bias  FILTER.d3  \n",
       "31                      False        False      False  \n",
       "35                      False        False      False  \n",
       "36                      False        False      False  \n",
       "37                      False        False      False  \n",
       "38                      False        False      False  \n",
       "...                       ...          ...        ...  \n",
       "29014                   False        False      False  \n",
       "29043                   False        False      False  \n",
       "29048                   False        False      False  \n",
       "29050                   False        False      False  \n",
       "29064                   False        False      False  \n",
       "\n",
       "[2970 rows x 149 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs_maf[snvs_maf[\"canonical_Consequence_broader\"] == 'synonymous']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e8302c7-dd63-466a-95dd-67a1647dc857",
   "metadata": {},
   "source": [
    "______"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fef2e840-49df-4aa9-81d7-6ee40026540b",
   "metadata": {},
   "source": [
    "Missense"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "96d7c90f-68de-47bc-8cfd-cb7f9f7f4d94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729654</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0038_BDO_01;TYPE=SNV;DP=6080;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6080:1:6079,1:0.0002:4345,1734:1,0:6082:60...</td>\n",
       "      <td>6082</td>\n",
       "      <td>1</td>\n",
       "      <td>6081</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>6080</td>\n",
       "      <td>1</td>\n",
       "      <td>6079</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>27</td>\n",
       "      <td>0.004420</td>\n",
       "      <td>13495</td>\n",
       "      <td>1</td>\n",
       "      <td>13494</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>42</td>\n",
       "      <td>0.003103</td>\n",
       "      <td>7413</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.450685</td>\n",
       "      <td>0.916196</td>\n",
       "      <td>chr1:26729654_T&gt;C</td>\n",
       "      <td>P19_0038_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729654</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1530</td>\n",
       "      <td>1141</td>\n",
       "      <td>381</td>\n",
       "      <td>S/P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>COSV61375919</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:26729654</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1530</td>\n",
       "      <td>1141</td>\n",
       "      <td>381</td>\n",
       "      <td>S/P</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ATC&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729654_T&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>26729654</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729655</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0051_BTR_01;TYPE=SNV;DP=9195;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:9195:1:9194,1:0.0001:6295,2899:1,0:9195:91...</td>\n",
       "      <td>9195</td>\n",
       "      <td>1</td>\n",
       "      <td>9194</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>9195</td>\n",
       "      <td>1</td>\n",
       "      <td>9194</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>40</td>\n",
       "      <td>0.004331</td>\n",
       "      <td>19756</td>\n",
       "      <td>1</td>\n",
       "      <td>19754</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>60</td>\n",
       "      <td>0.003028</td>\n",
       "      <td>10561</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.465428</td>\n",
       "      <td>0.922679</td>\n",
       "      <td>chr1:26729655_C&gt;G</td>\n",
       "      <td>P19_0051_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729655</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1531</td>\n",
       "      <td>1142</td>\n",
       "      <td>381</td>\n",
       "      <td>S/C</td>\n",
       "      <td>tCc/tGc</td>\n",
       "      <td>rs1557591022,COSV61385429</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>0,1</td>\n",
       "      <td>0,1</td>\n",
       "      <td>chr1:26729655</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1531</td>\n",
       "      <td>1142</td>\n",
       "      <td>381</td>\n",
       "      <td>S/C</td>\n",
       "      <td>tCc/tGc</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCC&gt;G</td>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>T[C&gt;G]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729655_C&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>26729655</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729658</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0033_BDO_01;TYPE=SNV;DP=5590;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:5590:1:5589,1:0.0002:3846,1743:1,0:5591:55...</td>\n",
       "      <td>5591</td>\n",
       "      <td>1</td>\n",
       "      <td>5590</td>\n",
       "      <td>0.000179</td>\n",
       "      <td>5590</td>\n",
       "      <td>1</td>\n",
       "      <td>5589</td>\n",
       "      <td>0.000179</td>\n",
       "      <td>24</td>\n",
       "      <td>0.004274</td>\n",
       "      <td>11306</td>\n",
       "      <td>1</td>\n",
       "      <td>11305</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>30</td>\n",
       "      <td>0.002646</td>\n",
       "      <td>5715</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.494516</td>\n",
       "      <td>0.924551</td>\n",
       "      <td>chr1:26729658_G&gt;T</td>\n",
       "      <td>P19_0033_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729658</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1534</td>\n",
       "      <td>1145</td>\n",
       "      <td>382</td>\n",
       "      <td>S/I</td>\n",
       "      <td>aGt/aTt</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729658</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1534</td>\n",
       "      <td>1145</td>\n",
       "      <td>382</td>\n",
       "      <td>S/I</td>\n",
       "      <td>aGt/aTt</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ACT&gt;A</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>A[C&gt;A]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729658_C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>26729658</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729660</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0023_BDO_01;TYPE=SNV;DP=6130;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6130:1:6129,1:0.0002:3482,2647:0,1:6130:61...</td>\n",
       "      <td>6130</td>\n",
       "      <td>1</td>\n",
       "      <td>6129</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>6130</td>\n",
       "      <td>1</td>\n",
       "      <td>6129</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>7</td>\n",
       "      <td>0.001141</td>\n",
       "      <td>9077</td>\n",
       "      <td>1</td>\n",
       "      <td>9076</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>11</td>\n",
       "      <td>0.001210</td>\n",
       "      <td>2947</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.675333</td>\n",
       "      <td>0.956927</td>\n",
       "      <td>chr1:26729660_C&gt;A</td>\n",
       "      <td>P19_0023_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729660</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1536</td>\n",
       "      <td>1147</td>\n",
       "      <td>383</td>\n",
       "      <td>P/T</td>\n",
       "      <td>Cca/Aca</td>\n",
       "      <td>COSV61386063</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:26729660</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1536</td>\n",
       "      <td>1147</td>\n",
       "      <td>383</td>\n",
       "      <td>P/T</td>\n",
       "      <td>Cca/Aca</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCC&gt;A</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>T[C&gt;A]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729660_C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>26729660</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729663</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0050_BTR_01;TYPE=SNV;DP=10233;VD=1;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:10233:1:10232,1:0.0001:7068,3164:1,0:10235...</td>\n",
       "      <td>10235</td>\n",
       "      <td>1</td>\n",
       "      <td>10234</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>10233</td>\n",
       "      <td>1</td>\n",
       "      <td>10232</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>76</td>\n",
       "      <td>0.007371</td>\n",
       "      <td>23128</td>\n",
       "      <td>1</td>\n",
       "      <td>23127</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>87</td>\n",
       "      <td>0.003748</td>\n",
       "      <td>12893</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.442537</td>\n",
       "      <td>0.918873</td>\n",
       "      <td>chr1:26729663_A&gt;G</td>\n",
       "      <td>P19_0050_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729663</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000374152</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>752</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>M/V</td>\n",
       "      <td>Atg/Gtg</td>\n",
       "      <td>rs999411101</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000363267</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729663</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1539</td>\n",
       "      <td>1150</td>\n",
       "      <td>384</td>\n",
       "      <td>M/V</td>\n",
       "      <td>Atg/Gtg</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ATT&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]T</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729663_T&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>26729663</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29058</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095441</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0028_BTR_01;TYPE=SNV;DP=2771;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2771:1:2770,1:0.0004:1324,1446:0,1:2772:27...</td>\n",
       "      <td>2772</td>\n",
       "      <td>1</td>\n",
       "      <td>2771</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>2771</td>\n",
       "      <td>1</td>\n",
       "      <td>2770</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>10</td>\n",
       "      <td>0.003595</td>\n",
       "      <td>5608</td>\n",
       "      <td>1</td>\n",
       "      <td>5607</td>\n",
       "      <td>0.000178</td>\n",
       "      <td>13</td>\n",
       "      <td>0.002313</td>\n",
       "      <td>2836</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.494294</td>\n",
       "      <td>0.918370</td>\n",
       "      <td>chrX:124095441_G&gt;A</td>\n",
       "      <td>P19_0028_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095441</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4088</td>\n",
       "      <td>3775</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gat/Aat</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095441</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4088</td>\n",
       "      <td>3775</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gat/Aat</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCC&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095441_C&gt;T</td>\n",
       "      <td>X</td>\n",
       "      <td>124095441</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29059</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095442</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=P19_0046_BDO_01;TYPE=SNV;DP=3888;VD=2;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3888:2:3886,2:0.0005:1665,2221:1,1:3888:38...</td>\n",
       "      <td>3888</td>\n",
       "      <td>2</td>\n",
       "      <td>3886</td>\n",
       "      <td>0.000514</td>\n",
       "      <td>3888</td>\n",
       "      <td>2</td>\n",
       "      <td>3886</td>\n",
       "      <td>0.000514</td>\n",
       "      <td>15</td>\n",
       "      <td>0.003843</td>\n",
       "      <td>6554</td>\n",
       "      <td>2</td>\n",
       "      <td>6552</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>21</td>\n",
       "      <td>0.003194</td>\n",
       "      <td>2666</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.593226</td>\n",
       "      <td>0.935491</td>\n",
       "      <td>chrX:124095442_A&gt;G</td>\n",
       "      <td>P19_0046_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095442</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4089</td>\n",
       "      <td>3776</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095442</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4089</td>\n",
       "      <td>3776</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ATC&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095442_T&gt;C</td>\n",
       "      <td>X</td>\n",
       "      <td>124095442</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29060</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095442</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0042_BTR_01;TYPE=SNV;DP=2599;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2599:1:2598,1:0.0004:1160,1438:1,0:2599:25...</td>\n",
       "      <td>2599</td>\n",
       "      <td>1</td>\n",
       "      <td>2598</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>2599</td>\n",
       "      <td>1</td>\n",
       "      <td>2598</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>8</td>\n",
       "      <td>0.003069</td>\n",
       "      <td>4779</td>\n",
       "      <td>1</td>\n",
       "      <td>4778</td>\n",
       "      <td>0.000209</td>\n",
       "      <td>9</td>\n",
       "      <td>0.001880</td>\n",
       "      <td>2180</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.543838</td>\n",
       "      <td>0.928104</td>\n",
       "      <td>chrX:124095442_A&gt;G</td>\n",
       "      <td>P19_0042_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095442</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4089</td>\n",
       "      <td>3776</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095442</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4089</td>\n",
       "      <td>3776</td>\n",
       "      <td>1259</td>\n",
       "      <td>D/G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ATC&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>A[T&gt;C]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095442_T&gt;C</td>\n",
       "      <td>X</td>\n",
       "      <td>124095442</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29062</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095448</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0041_BDO_01;TYPE=SNV;DP=6186;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6186:1:6185,1:0.0002:2519,3666:0,1:6186:61...</td>\n",
       "      <td>6186</td>\n",
       "      <td>1</td>\n",
       "      <td>6185</td>\n",
       "      <td>0.000162</td>\n",
       "      <td>6186</td>\n",
       "      <td>1</td>\n",
       "      <td>6185</td>\n",
       "      <td>0.000162</td>\n",
       "      <td>10</td>\n",
       "      <td>0.001614</td>\n",
       "      <td>9445</td>\n",
       "      <td>1</td>\n",
       "      <td>9444</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>12</td>\n",
       "      <td>0.001269</td>\n",
       "      <td>3259</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.654950</td>\n",
       "      <td>0.953765</td>\n",
       "      <td>chrX:124095448_C&gt;T</td>\n",
       "      <td>P19_0041_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095448</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>4095</td>\n",
       "      <td>3782</td>\n",
       "      <td>1261</td>\n",
       "      <td>S/L</td>\n",
       "      <td>tCa/tTa</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095448</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>4095</td>\n",
       "      <td>3782</td>\n",
       "      <td>1261</td>\n",
       "      <td>S/L</td>\n",
       "      <td>tCa/tTa</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>TCA&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>T[C&gt;T]A</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095448_C&gt;T</td>\n",
       "      <td>X</td>\n",
       "      <td>124095448</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29063</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100574</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0013_BDO_01;TYPE=SNV;DP=4382;VD=1;A...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4382:1:4381,1:0.0002:2835,1546:1,0:4382:43...</td>\n",
       "      <td>4382</td>\n",
       "      <td>1</td>\n",
       "      <td>4381</td>\n",
       "      <td>0.000228</td>\n",
       "      <td>4382</td>\n",
       "      <td>1</td>\n",
       "      <td>4381</td>\n",
       "      <td>0.000228</td>\n",
       "      <td>11</td>\n",
       "      <td>0.002504</td>\n",
       "      <td>6196</td>\n",
       "      <td>1</td>\n",
       "      <td>6195</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>15</td>\n",
       "      <td>0.002415</td>\n",
       "      <td>1814</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.707230</td>\n",
       "      <td>0.960328</td>\n",
       "      <td>chrX:124100574_G&gt;T</td>\n",
       "      <td>P19_0013_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100574</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>4097</td>\n",
       "      <td>3784</td>\n",
       "      <td>1262</td>\n",
       "      <td>V/F</td>\n",
       "      <td>Gtt/Ttt</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100574</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>4097</td>\n",
       "      <td>3784</td>\n",
       "      <td>1262</td>\n",
       "      <td>V/F</td>\n",
       "      <td>Gtt/Ttt</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>ACC&gt;A</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>A[C&gt;A]C</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100574_C&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124100574</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15191 rows  149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT FILTER  \\\n",
       "0      chr1   26729654   T   C   pSTD   \n",
       "1      chr1   26729655   C   G   pSTD   \n",
       "2      chr1   26729658   G   T   pSTD   \n",
       "3      chr1   26729660   C   A   pSTD   \n",
       "4      chr1   26729663   A   G   pSTD   \n",
       "...     ...        ...  ..  ..    ...   \n",
       "29058  chrX  124095441   G   A   pSTD   \n",
       "29059  chrX  124095442   A   G   PASS   \n",
       "29060  chrX  124095442   A   G   pSTD   \n",
       "29062  chrX  124095448   C   T   pSTD   \n",
       "29063  chrX  124100574   G   T   pSTD   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      SAMPLE=P19_0038_BDO_01;TYPE=SNV;DP=6080;VD=1;A...   \n",
       "1      SAMPLE=P19_0051_BTR_01;TYPE=SNV;DP=9195;VD=1;A...   \n",
       "2      SAMPLE=P19_0033_BDO_01;TYPE=SNV;DP=5590;VD=1;A...   \n",
       "3      SAMPLE=P19_0023_BDO_01;TYPE=SNV;DP=6130;VD=1;A...   \n",
       "4      SAMPLE=P19_0050_BTR_01;TYPE=SNV;DP=10233;VD=1;...   \n",
       "...                                                  ...   \n",
       "29058  SAMPLE=P19_0028_BTR_01;TYPE=SNV;DP=2771;VD=1;A...   \n",
       "29059  SAMPLE=P19_0046_BDO_01;TYPE=SNV;DP=3888;VD=2;A...   \n",
       "29060  SAMPLE=P19_0042_BTR_01;TYPE=SNV;DP=2599;VD=1;A...   \n",
       "29062  SAMPLE=P19_0041_BDO_01;TYPE=SNV;DP=6186;VD=1;A...   \n",
       "29063  SAMPLE=P19_0013_BDO_01;TYPE=SNV;DP=4382;VD=1;A...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "1      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "2      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "3      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "4      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "...                                                  ...   \n",
       "29058  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29059  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29060  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29062  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "29063  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                  SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "0      0/1:6080:1:6079,1:0.0002:4345,1734:1,0:6082:60...   6082          1   \n",
       "1      0/1:9195:1:9194,1:0.0001:6295,2899:1,0:9195:91...   9195          1   \n",
       "2      0/1:5590:1:5589,1:0.0002:3846,1743:1,0:5591:55...   5591          1   \n",
       "3      0/1:6130:1:6129,1:0.0002:3482,2647:0,1:6130:61...   6130          1   \n",
       "4      0/1:10233:1:10232,1:0.0001:7068,3164:1,0:10235...  10235          1   \n",
       "...                                                  ...    ...        ...   \n",
       "29058  0/1:2771:1:2770,1:0.0004:1324,1446:0,1:2772:27...   2772          1   \n",
       "29059  0/1:3888:2:3886,2:0.0005:1665,2221:1,1:3888:38...   3888          2   \n",
       "29060  0/1:2599:1:2598,1:0.0004:1160,1438:1,0:2599:25...   2599          1   \n",
       "29062  0/1:6186:1:6185,1:0.0002:2519,3666:0,1:6186:61...   6186          1   \n",
       "29063  0/1:4382:1:4381,1:0.0002:2835,1546:1,0:4382:43...   4382          1   \n",
       "\n",
       "       REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "0           6081  0.000164      6080             1          6079  0.000164   \n",
       "1           9194  0.000109      9195             1          9194  0.000109   \n",
       "2           5590  0.000179      5590             1          5589  0.000179   \n",
       "3           6129  0.000163      6130             1          6129  0.000163   \n",
       "4          10234  0.000098     10233             1         10232  0.000098   \n",
       "...          ...       ...       ...           ...           ...       ...   \n",
       "29058       2771  0.000361      2771             1          2770  0.000361   \n",
       "29059       3886  0.000514      3888             2          3886  0.000514   \n",
       "29060       2598  0.000385      2599             1          2598  0.000385   \n",
       "29062       6185  0.000162      6186             1          6185  0.000162   \n",
       "29063       4381  0.000228      4382             1          4381  0.000228   \n",
       "\n",
       "       numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "0         27  0.004420     13495             1         13494  0.000074   \n",
       "1         40  0.004331     19756             1         19754  0.000051   \n",
       "2         24  0.004274     11306             1         11305  0.000088   \n",
       "3          7  0.001141      9077             1          9076  0.000110   \n",
       "4         76  0.007371     23128             1         23127  0.000043   \n",
       "...      ...       ...       ...           ...           ...       ...   \n",
       "29058     10  0.003595      5608             1          5607  0.000178   \n",
       "29059     15  0.003843      6554             2          6552  0.000305   \n",
       "29060      8  0.003069      4779             1          4778  0.000209   \n",
       "29062     10  0.001614      9445             1          9444  0.000106   \n",
       "29063     11  0.002504      6196             1          6195  0.000161   \n",
       "\n",
       "       numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND  VAF_ND  \\\n",
       "0            42   0.003103      7413             0     0.0   \n",
       "1            60   0.003028     10561             0     0.0   \n",
       "2            30   0.002646      5715             0     0.0   \n",
       "3            11   0.001210      2947             0     0.0   \n",
       "4            87   0.003748     12893             0     0.0   \n",
       "...         ...        ...       ...           ...     ...   \n",
       "29058        13   0.002313      2836             0     0.0   \n",
       "29059        21   0.003194      2666             0     0.0   \n",
       "29060         9   0.001880      2180             0     0.0   \n",
       "29062        12   0.001269      3259             0     0.0   \n",
       "29063        15   0.002415      1814             0     0.0   \n",
       "\n",
       "       VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "0                       False                      False        0.450685   \n",
       "1                       False                      False        0.465428   \n",
       "2                       False                      False        0.494516   \n",
       "3                       False                      False        0.675333   \n",
       "4                       False                      False        0.442537   \n",
       "...                       ...                        ...             ...   \n",
       "29058                   False                      False        0.494294   \n",
       "29059                   False                      False        0.593226   \n",
       "29060                   False                      False        0.543838   \n",
       "29062                   False                      False        0.654950   \n",
       "29063                   False                      False        0.707230   \n",
       "\n",
       "       VAF_distortion_sq              MUT_ID        SAMPLE_ID METHOD  \\\n",
       "0               0.916196   chr1:26729654_T>C  P19_0038_BDO_01    med   \n",
       "1               0.922679   chr1:26729655_C>G  P19_0051_BTR_01    med   \n",
       "2               0.924551   chr1:26729658_G>T  P19_0033_BDO_01    med   \n",
       "3               0.956927   chr1:26729660_C>A  P19_0023_BDO_01    med   \n",
       "4               0.918873   chr1:26729663_A>G  P19_0050_BTR_01    med   \n",
       "...                  ...                 ...              ...    ...   \n",
       "29058           0.918370  chrX:124095441_G>A  P19_0028_BTR_01    med   \n",
       "29059           0.935491  chrX:124095442_A>G  P19_0046_BDO_01    med   \n",
       "29060           0.928104  chrX:124095442_A>G  P19_0042_BTR_01    med   \n",
       "29062           0.953765  chrX:124095448_C>T  P19_0041_BDO_01    med   \n",
       "29063           0.960328  chrX:124100574_G>T  P19_0013_BDO_01    med   \n",
       "\n",
       "       PROJECT_NAME        Location Allele             Gene          Feature  \\\n",
       "0               NaN   chr1:26729654      C  ENSG00000117713  ENST00000324856   \n",
       "1               NaN   chr1:26729655      G  ENSG00000117713  ENST00000324856   \n",
       "2               NaN   chr1:26729658      T  ENSG00000117713  ENST00000324856   \n",
       "3               NaN   chr1:26729660      A  ENSG00000117713  ENST00000324856   \n",
       "4               NaN   chr1:26729663      G  ENSG00000117713  ENST00000374152   \n",
       "...             ...             ...    ...              ...              ...   \n",
       "29058           NaN  chrX:124095441      A  ENSG00000101972  ENST00000371145   \n",
       "29059           NaN  chrX:124095442      G  ENSG00000101972  ENST00000371145   \n",
       "29060           NaN  chrX:124095442      G  ENSG00000101972  ENST00000371145   \n",
       "29062           NaN  chrX:124095448      T  ENSG00000101972  ENST00000371145   \n",
       "29063           NaN  chrX:124100574      T  ENSG00000101972  ENST00000371145   \n",
       "\n",
       "      Feature_type                             Consequence cDNA_position  \\\n",
       "0       Transcript                        missense_variant          1530   \n",
       "1       Transcript                        missense_variant          1531   \n",
       "2       Transcript                        missense_variant          1534   \n",
       "3       Transcript                        missense_variant          1536   \n",
       "4       Transcript                              start_lost           752   \n",
       "...            ...                                     ...           ...   \n",
       "29058   Transcript                        missense_variant          4088   \n",
       "29059   Transcript                        missense_variant          4089   \n",
       "29060   Transcript                        missense_variant          4089   \n",
       "29062   Transcript  missense_variant,splice_region_variant          4095   \n",
       "29063   Transcript  missense_variant,splice_region_variant          4097   \n",
       "\n",
       "      CDS_position Protein_position Amino_acids   Codons  \\\n",
       "0             1141              381         S/P  Tcc/Ccc   \n",
       "1             1142              381         S/C  tCc/tGc   \n",
       "2             1145              382         S/I  aGt/aTt   \n",
       "3             1147              383         P/T  Cca/Aca   \n",
       "4                1                1         M/V  Atg/Gtg   \n",
       "...            ...              ...         ...      ...   \n",
       "29058         3775             1259         D/N  Gat/Aat   \n",
       "29059         3776             1259         D/G  gAt/gGt   \n",
       "29060         3776             1259         D/G  gAt/gGt   \n",
       "29062         3782             1261         S/L  tCa/tTa   \n",
       "29063         3784             1262         V/F  Gtt/Ttt   \n",
       "\n",
       "              Existing_variation    IMPACT DISTANCE STRAND FLAGS  SYMBOL  \\\n",
       "0                   COSV61375919  MODERATE        -      1     -  ARID1A   \n",
       "1      rs1557591022,COSV61385429  MODERATE        -      1     -  ARID1A   \n",
       "2                              -  MODERATE        -      1     -  ARID1A   \n",
       "3                   COSV61386063  MODERATE        -      1     -  ARID1A   \n",
       "4                    rs999411101      HIGH        -      1     -  ARID1A   \n",
       "...                          ...       ...      ...    ...   ...     ...   \n",
       "29058                          -  MODERATE        -      1     -   STAG2   \n",
       "29059                          -  MODERATE        -      1     -   STAG2   \n",
       "29060                          -  MODERATE        -      1     -   STAG2   \n",
       "29062                          -  MODERATE        -      1     -   STAG2   \n",
       "29063                          -  MODERATE        -      1     -   STAG2   \n",
       "\n",
       "      SYMBOL_SOURCE     HGNC_ID CANONICAL             ENSP  gnomADe_AF  \\\n",
       "0              HGNC  HGNC:11110       YES  ENSP00000320485    0.000000   \n",
       "1              HGNC  HGNC:11110       YES  ENSP00000320485    0.000000   \n",
       "2              HGNC  HGNC:11110       YES  ENSP00000320485    0.000000   \n",
       "3              HGNC  HGNC:11110       YES  ENSP00000320485    0.000000   \n",
       "4              HGNC  HGNC:11110         -  ENSP00000363267    0.000012   \n",
       "...             ...         ...       ...              ...         ...   \n",
       "29058          HGNC  HGNC:11355       YES  ENSP00000360187    0.000000   \n",
       "29059          HGNC  HGNC:11355       YES  ENSP00000360187    0.000000   \n",
       "29060          HGNC  HGNC:11355       YES  ENSP00000360187    0.000000   \n",
       "29062          HGNC  HGNC:11355       YES  ENSP00000360187    0.000000   \n",
       "29063          HGNC  HGNC:11355       YES  ENSP00000360187    0.000000   \n",
       "\n",
       "       gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "0                 0.0             0.0             0.0             0.0   \n",
       "1                 0.0             0.0             0.0             0.0   \n",
       "2                 0.0             0.0             0.0             0.0   \n",
       "3                 0.0             0.0             0.0             0.0   \n",
       "4                 0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29058             0.0             0.0             0.0             0.0   \n",
       "29059             0.0             0.0             0.0             0.0   \n",
       "29060             0.0             0.0             0.0             0.0   \n",
       "29062             0.0             0.0             0.0             0.0   \n",
       "29063             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "0                 0.0        0.000000             0.0             0.0   \n",
       "1                 0.0        0.000000             0.0             0.0   \n",
       "2                 0.0        0.000000             0.0             0.0   \n",
       "3                 0.0        0.000000             0.0             0.0   \n",
       "4                 0.0        0.000026             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29058             0.0        0.000000             0.0             0.0   \n",
       "29059             0.0        0.000000             0.0             0.0   \n",
       "29060             0.0        0.000000             0.0             0.0   \n",
       "29062             0.0        0.000000             0.0             0.0   \n",
       "29063             0.0        0.000000             0.0             0.0   \n",
       "\n",
       "       gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "0             0.0             0.0             0.0             0.0   \n",
       "1             0.0             0.0             0.0             0.0   \n",
       "2             0.0             0.0             0.0             0.0   \n",
       "3             0.0             0.0             0.0             0.0   \n",
       "4             0.0             0.0             0.0             0.0   \n",
       "...           ...             ...             ...             ...   \n",
       "29058         0.0             0.0             0.0             0.0   \n",
       "29059         0.0             0.0             0.0             0.0   \n",
       "29060         0.0             0.0             0.0             0.0   \n",
       "29062         0.0             0.0             0.0             0.0   \n",
       "29063         0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "0                 0.0             0.0             0.0             0.0   \n",
       "1                 0.0             0.0             0.0             0.0   \n",
       "2                 0.0             0.0             0.0             0.0   \n",
       "3                 0.0             0.0             0.0             0.0   \n",
       "4                 0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "29058             0.0             0.0             0.0             0.0   \n",
       "29059             0.0             0.0             0.0             0.0   \n",
       "29060             0.0             0.0             0.0             0.0   \n",
       "29062             0.0             0.0             0.0             0.0   \n",
       "29063             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF CLIN_SIG SOMATIC PHENO  \\\n",
       "0                 0.0             0.0             0.0        -       1     1   \n",
       "1                 0.0             0.0             0.0        -     0,1   0,1   \n",
       "2                 0.0             0.0             0.0        -       -     -   \n",
       "3                 0.0             0.0             0.0        -       1     1   \n",
       "4                 0.0             0.0             0.0        -       -     -   \n",
       "...               ...             ...             ...      ...     ...   ...   \n",
       "29058             0.0             0.0             0.0        -       -     -   \n",
       "29059             0.0             0.0             0.0        -       -     -   \n",
       "29060             0.0             0.0             0.0        -       -     -   \n",
       "29062             0.0             0.0             0.0        -       -     -   \n",
       "29063             0.0             0.0             0.0        -       -     -   \n",
       "\n",
       "      canonical_Location canonical_Allele   canonical_Gene canonical_Feature  \\\n",
       "0          chr1:26729654                C  ENSG00000117713   ENST00000324856   \n",
       "1          chr1:26729655                G  ENSG00000117713   ENST00000324856   \n",
       "2          chr1:26729658                T  ENSG00000117713   ENST00000324856   \n",
       "3          chr1:26729660                A  ENSG00000117713   ENST00000324856   \n",
       "4          chr1:26729663                G  ENSG00000117713   ENST00000324856   \n",
       "...                  ...              ...              ...               ...   \n",
       "29058     chrX:124095441                A  ENSG00000101972   ENST00000371145   \n",
       "29059     chrX:124095442                G  ENSG00000101972   ENST00000371145   \n",
       "29060     chrX:124095442                G  ENSG00000101972   ENST00000371145   \n",
       "29062     chrX:124095448                T  ENSG00000101972   ENST00000371145   \n",
       "29063     chrX:124100574                T  ENSG00000101972   ENST00000371145   \n",
       "\n",
       "      canonical_Feature_type                   canonical_Consequence  \\\n",
       "0                 Transcript                        missense_variant   \n",
       "1                 Transcript                        missense_variant   \n",
       "2                 Transcript                        missense_variant   \n",
       "3                 Transcript                        missense_variant   \n",
       "4                 Transcript                        missense_variant   \n",
       "...                      ...                                     ...   \n",
       "29058             Transcript                        missense_variant   \n",
       "29059             Transcript                        missense_variant   \n",
       "29060             Transcript                        missense_variant   \n",
       "29062             Transcript  missense_variant,splice_region_variant   \n",
       "29063             Transcript  missense_variant,splice_region_variant   \n",
       "\n",
       "      canonical_cDNA_position  canonical_CDS_position  \\\n",
       "0                        1530                    1141   \n",
       "1                        1531                    1142   \n",
       "2                        1534                    1145   \n",
       "3                        1536                    1147   \n",
       "4                        1539                    1150   \n",
       "...                       ...                     ...   \n",
       "29058                    4088                    3775   \n",
       "29059                    4089                    3776   \n",
       "29060                    4089                    3776   \n",
       "29062                    4095                    3782   \n",
       "29063                    4097                    3784   \n",
       "\n",
       "      canonical_Protein_position canonical_Amino_acids canonical_Codons  \\\n",
       "0                            381                   S/P          Tcc/Ccc   \n",
       "1                            381                   S/C          tCc/tGc   \n",
       "2                            382                   S/I          aGt/aTt   \n",
       "3                            383                   P/T          Cca/Aca   \n",
       "4                            384                   M/V          Atg/Gtg   \n",
       "...                          ...                   ...              ...   \n",
       "29058                       1259                   D/N          Gat/Aat   \n",
       "29059                       1259                   D/G          gAt/gGt   \n",
       "29060                       1259                   D/G          gAt/gGt   \n",
       "29062                       1261                   S/L          tCa/tTa   \n",
       "29063                       1262                   V/F          Gtt/Ttt   \n",
       "\n",
       "      canonical_IMPACT canonical_DISTANCE canonical_STRAND canonical_FLAGS  \\\n",
       "0             MODERATE                  -              1.0               -   \n",
       "1             MODERATE                  -              1.0               -   \n",
       "2             MODERATE                  -              1.0               -   \n",
       "3             MODERATE                  -              1.0               -   \n",
       "4             MODERATE                  -              1.0               -   \n",
       "...                ...                ...              ...             ...   \n",
       "29058         MODERATE                  -              1.0               -   \n",
       "29059         MODERATE                  -              1.0               -   \n",
       "29060         MODERATE                  -              1.0               -   \n",
       "29062         MODERATE                  -              1.0               -   \n",
       "29063         MODERATE                  -              1.0               -   \n",
       "\n",
       "      canonical_SYMBOL canonical_SYMBOL_SOURCE canonical_HGNC_ID  \\\n",
       "0               ARID1A                    HGNC        HGNC:11110   \n",
       "1               ARID1A                    HGNC        HGNC:11110   \n",
       "2               ARID1A                    HGNC        HGNC:11110   \n",
       "3               ARID1A                    HGNC        HGNC:11110   \n",
       "4               ARID1A                    HGNC        HGNC:11110   \n",
       "...                ...                     ...               ...   \n",
       "29058            STAG2                    HGNC        HGNC:11355   \n",
       "29059            STAG2                    HGNC        HGNC:11355   \n",
       "29060            STAG2                    HGNC        HGNC:11355   \n",
       "29062            STAG2                    HGNC        HGNC:11355   \n",
       "29063            STAG2                    HGNC        HGNC:11355   \n",
       "\n",
       "      canonical_CANONICAL   canonical_ENSP canonical_CLIN_SIG  \\\n",
       "0                     YES  ENSP00000320485                  -   \n",
       "1                     YES  ENSP00000320485                  -   \n",
       "2                     YES  ENSP00000320485                  -   \n",
       "3                     YES  ENSP00000320485                  -   \n",
       "4                     YES  ENSP00000320485                  -   \n",
       "...                   ...              ...                ...   \n",
       "29058                 YES  ENSP00000360187                  -   \n",
       "29059                 YES  ENSP00000360187                  -   \n",
       "29060                 YES  ENSP00000360187                  -   \n",
       "29062                 YES  ENSP00000360187                  -   \n",
       "29063                 YES  ENSP00000360187                  -   \n",
       "\n",
       "      canonical_Consequence_single canonical_Consequence_broader  \\\n",
       "0                 missense_variant                      missense   \n",
       "1                 missense_variant                      missense   \n",
       "2                 missense_variant                      missense   \n",
       "3                 missense_variant                      missense   \n",
       "4                 missense_variant                      missense   \n",
       "...                            ...                           ...   \n",
       "29058             missense_variant                      missense   \n",
       "29059             missense_variant                      missense   \n",
       "29060             missense_variant                      missense   \n",
       "29062             missense_variant                      missense   \n",
       "29063             missense_variant                      missense   \n",
       "\n",
       "      canonical_Protein_affecting Consequence_single Consequence_broader  \\\n",
       "0               protein_affecting   missense_variant            missense   \n",
       "1               protein_affecting   missense_variant            missense   \n",
       "2               protein_affecting   missense_variant            missense   \n",
       "3               protein_affecting   missense_variant            missense   \n",
       "4               protein_affecting         start_lost            nonsense   \n",
       "...                           ...                ...                 ...   \n",
       "29058           protein_affecting   missense_variant            missense   \n",
       "29059           protein_affecting   missense_variant            missense   \n",
       "29060           protein_affecting   missense_variant            missense   \n",
       "29062           protein_affecting   missense_variant            missense   \n",
       "29063           protein_affecting   missense_variant            missense   \n",
       "\n",
       "       Protein_affecting CONTEXT_MUT MUTTYPE CONTEXT_MUT_SIGPRO  gnomAD_SNP  \\\n",
       "0      protein_affecting       ATC>C     T>C            A[T>C]C       False   \n",
       "1      protein_affecting       TCC>G     C>G            T[C>G]C       False   \n",
       "2      protein_affecting       ACT>A     C>A            A[C>A]T       False   \n",
       "3      protein_affecting       TCC>A     C>A            T[C>A]C       False   \n",
       "4      protein_affecting       ATT>C     T>C            A[T>C]T       False   \n",
       "...                  ...         ...     ...                ...         ...   \n",
       "29058  protein_affecting       TCC>T     C>T            T[C>T]C       False   \n",
       "29059  protein_affecting       ATC>C     T>C            A[T>C]C       False   \n",
       "29060  protein_affecting       ATC>C     T>C            A[T>C]C       False   \n",
       "29062  protein_affecting       TCA>T     C>T            T[C>T]A       False   \n",
       "29063  protein_affecting       ACC>A     C>A            A[C>A]C       False   \n",
       "\n",
       "      HOTSPOT_NAME          MUT_ID_pyr CHROM_ensembl  POS_ensembl REF_ensembl  \\\n",
       "0                -   chr1:26729654_T>C             1     26729654           T   \n",
       "1                -   chr1:26729655_C>G             1     26729655           C   \n",
       "2                -   chr1:26729658_C>A             1     26729658           G   \n",
       "3                -   chr1:26729660_C>A             1     26729660           C   \n",
       "4                -   chr1:26729663_T>C             1     26729663           A   \n",
       "...            ...                 ...           ...          ...         ...   \n",
       "29058            -  chrX:124095441_C>T             X    124095441           G   \n",
       "29059            -  chrX:124095442_T>C             X    124095442           A   \n",
       "29060            -  chrX:124095442_T>C             X    124095442           A   \n",
       "29062            -  chrX:124095448_C>T             X    124095448           C   \n",
       "29063            -  chrX:124100574_C>A             X    124100574           G   \n",
       "\n",
       "      ALT_ensembl TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  \\\n",
       "0               C  SNV             0          False            False   \n",
       "1               G  SNV             0          False            False   \n",
       "2               T  SNV             0          False            False   \n",
       "3               A  SNV             0          False            False   \n",
       "4               G  SNV             0          False            False   \n",
       "...           ...  ...           ...            ...              ...   \n",
       "29058           A  SNV             0          False            False   \n",
       "29059           G  SNV             0          False            False   \n",
       "29060           G  SNV             0          False            False   \n",
       "29062           T  SNV             0          False            False   \n",
       "29063           T  SNV             0          False            False   \n",
       "\n",
       "       N_rich_frequency  VAF_Ns_threshold  FILTER.cohort_n_rich_uni  \\\n",
       "0                   0.0               1.1                     False   \n",
       "1                   0.0               1.1                     False   \n",
       "2                   0.0               1.1                     False   \n",
       "3                   0.0               1.1                     False   \n",
       "4                   0.0               1.1                     False   \n",
       "...                 ...               ...                       ...   \n",
       "29058               0.0               1.1                     False   \n",
       "29059               0.0               1.1                     False   \n",
       "29060               0.0               1.1                     False   \n",
       "29062               0.0               1.1                     False   \n",
       "29063               0.0               1.1                     False   \n",
       "\n",
       "       FILTER.gnomAD_SNP  FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  \\\n",
       "0                  False               False                False       False   \n",
       "1                  False               False                False       False   \n",
       "2                  False               False                False       False   \n",
       "3                  False               False                False       False   \n",
       "4                  False               False                False       False   \n",
       "...                  ...                 ...                  ...         ...   \n",
       "29058              False               False                False       False   \n",
       "29059              False               False                False       False   \n",
       "29060              False               False                False       False   \n",
       "29062              False               False                False       False   \n",
       "29063              False               False                False       False   \n",
       "\n",
       "       FILTER.pSTD  FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "0             True                      False                 False   \n",
       "1             True                      False                 False   \n",
       "2             True                      False                 False   \n",
       "3             True                      False                 False   \n",
       "4             True                      False                 False   \n",
       "...            ...                        ...                   ...   \n",
       "29058         True                      False                 False   \n",
       "29059        False                      False                 False   \n",
       "29060         True                      False                 False   \n",
       "29062         True                      False                 False   \n",
       "29063         True                      False                 False   \n",
       "\n",
       "       FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "0                              False           False   \n",
       "1                              False           False   \n",
       "2                              False           False   \n",
       "3                              False           False   \n",
       "4                              False           False   \n",
       "...                              ...             ...   \n",
       "29058                          False           False   \n",
       "29059                          False           False   \n",
       "29060                          False           False   \n",
       "29062                          False           False   \n",
       "29063                          False           False   \n",
       "\n",
       "       FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "0                               False                        False   \n",
       "1                               False                        False   \n",
       "2                               False                        False   \n",
       "3                               False                        False   \n",
       "4                               False                        False   \n",
       "...                               ...                          ...   \n",
       "29058                           False                        False   \n",
       "29059                           False                        False   \n",
       "29060                           False                        False   \n",
       "29062                           False                        False   \n",
       "29063                           False                        False   \n",
       "\n",
       "       FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "0                         False          False                    False   \n",
       "1                         False          False                    False   \n",
       "2                         False          False                    False   \n",
       "3                         False          False                    False   \n",
       "4                         False          False                    False   \n",
       "...                         ...            ...                      ...   \n",
       "29058                     False          False                    False   \n",
       "29059                     False          False                    False   \n",
       "29060                     False          False                    False   \n",
       "29062                     False          False                    False   \n",
       "29063                     False          False                    False   \n",
       "\n",
       "       FILTER.PASS  FILTER.AM_not_searched_COMPLEX  \\\n",
       "0            False                           False   \n",
       "1            False                           False   \n",
       "2            False                           False   \n",
       "3            False                           False   \n",
       "4            False                           False   \n",
       "...            ...                             ...   \n",
       "29058        False                           False   \n",
       "29059         True                           False   \n",
       "29060        False                           False   \n",
       "29062        False                           False   \n",
       "29063        False                           False   \n",
       "\n",
       "       FILTER.not_searched_COMPLEX  FILTER.MSI12  FILTER.AM_not_searched_SV  \\\n",
       "0                            False         False                      False   \n",
       "1                            False         False                      False   \n",
       "2                            False         False                      False   \n",
       "3                            False         False                      False   \n",
       "4                            False         False                      False   \n",
       "...                            ...           ...                        ...   \n",
       "29058                        False         False                      False   \n",
       "29059                        False         False                      False   \n",
       "29060                        False         False                      False   \n",
       "29062                        False         False                      False   \n",
       "29063                        False         False                      False   \n",
       "\n",
       "       FILTER.not_searched_SV  FILTER.SN1.5  FILTER.q22.5  FILTER.v1  \\\n",
       "0                       False         False         False      False   \n",
       "1                       False         False         False      False   \n",
       "2                       False         False         False      False   \n",
       "3                       False         False         False      False   \n",
       "4                       False         False         False      False   \n",
       "...                       ...           ...           ...        ...   \n",
       "29058                   False         False         False      False   \n",
       "29059                   False         False         False      False   \n",
       "29060                   False         False         False      False   \n",
       "29062                   False         False         False      False   \n",
       "29063                   False         False         False      False   \n",
       "\n",
       "       FILTER.low_mappability  FILTER.Bias  FILTER.d3  \n",
       "0                       False        False      False  \n",
       "1                       False        False      False  \n",
       "2                       False        False      False  \n",
       "3                       False        False      False  \n",
       "4                       False        False      False  \n",
       "...                       ...          ...        ...  \n",
       "29058                   False        False      False  \n",
       "29059                   False        False      False  \n",
       "29060                   False        False      False  \n",
       "29062                   False        False      False  \n",
       "29063                   False        False      False  \n",
       "\n",
       "[15191 rows x 149 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs_maf[snvs_maf[\"canonical_Consequence_broader\"] == 'missense']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38e0e02a-f1c8-418c-b606-6f468c6ef839",
   "metadata": {},
   "source": [
    "### OncodriveFML"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4537154-df9d-4348-8308-072f1d14f3f5",
   "metadata": {},
   "source": [
    "#### Load gene/protein regions information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "096d30c0-e6d8-456c-bb15-0c84eafbb407",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_regions_df = pd.read_table(f\"{o3d_alt_datasets}/seq_for_mut_prob.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e6c4ded5-3229-4071-992e-c7094f07324f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# you should get this file on your own\n",
    "# https://cadd.gs.washington.edu/download\n",
    "tb = tabix.open(\"/data/bbg/datasets/CADD/v1.7/hg38/whole_genome_SNVs.tsv.gz\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80bebc07-2069-44fe-afbb-109c9fd15264",
   "metadata": {},
   "source": [
    "#### Query scores of each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "72f7007f-cc67-4182-983c-6d120aaf0f21",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_scores_gene(gene_info):\n",
    "    chromosome = gene_info[\"Chr\"].values[0]\n",
    "    reverse_strand = gene_info[\"Reverse_strand\"].values[0] == 1\n",
    "    exon_coords = eval(gene_info[\"Exons_coord\"].values[0])\n",
    "    if reverse_strand:\n",
    "        exon_coords = [(y, x) for x, y in exon_coords]\n",
    "\n",
    "\n",
    "    position_mutation = dict()\n",
    "    for start, stop in exon_coords:\n",
    "        for e in tb.query(str(chromosome), start, stop):\n",
    "            if int(e[1]) not in position_mutation:\n",
    "                position_mutation[int(e[1])] = dict()\n",
    "            position_mutation[int(e[1])][e[3]] = float(e[5])\n",
    "            \n",
    "    start2end_coords = { x:y for x, y in enumerate(sorted(position_mutation.keys(), reverse = not reverse_strand))}\n",
    "\n",
    "    return position_mutation, start2end_coords"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3f0b610-65da-4f3c-ab28-55de4905917d",
   "metadata": {},
   "source": [
    "#### Scores of observed mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8e62ae93-9329-4ff6-933a-a424d0ba2e20",
   "metadata": {},
   "outputs": [],
   "source": [
    "def score_mutations_from_gene(snv_data):\n",
    "    scores_muts = []\n",
    "    for ind, row in snv_data.iterrows():\n",
    "        try:\n",
    "            scores_muts.append(scores_all_mutations_in_gene[row[\"POS\"]].get(row[\"ALT\"], 0))\n",
    "            print(\"found\")\n",
    "        except:\n",
    "            scores_muts.append(0)\n",
    "            print(row[\"Consequence\"], \"not found\")\n",
    "    snv_data[\"CADDscore\"] = scores_muts\n",
    "    return snv_data\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "06698dbf-cdcf-4a3c-be7e-3f4f0ccbe1b1",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "stop_lost not found\n",
      "stop_lost not found\n",
      "stop_lost not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "missense_variant,splice_region_variant not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "missense_variant,splice_region_variant not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "missense_variant,splice_region_variant not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "splice_region_variant,synonymous_variant not found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n",
      "found\n"
     ]
    }
   ],
   "source": [
    "for gene in [\"TP53\"]:\n",
    "# for gene in gene_order:\n",
    "    gene_info = seq_regions_df[seq_regions_df[\"Gene\"] == gene]\n",
    "    scores_all_mutations_in_gene, scores_of_protein_position = get_scores_gene(gene_info)\n",
    "    \n",
    "    mutations_in_gene = snvs_maf[snvs_maf[\"canonical_SYMBOL\"] == gene].copy()\n",
    "    scored_mutations_in_gene = score_mutations_from_gene(mutations_in_gene)\n",
    "    \n",
    "    scored_mutations_in_gene.to_csv(f\"{data_dir}/mutations_scored.{gene}.tsv\", sep = '\\t', header= True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5500a622-2988-4a8c-a7f7-bef868333078",
   "metadata": {},
   "source": [
    "___"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f14da5d7-b2aa-45e0-bf56-e1020ec89466",
   "metadata": {},
   "source": [
    "### Indels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e09d8e18-260b-46c4-9f57-46c776e230e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729645</td>\n",
       "      <td>TTG</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0024_BDO_01;TYPE=Deletion;DP=7803;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7803:1:7802,1:0.0001:4759,3043:1,0:7769:77...</td>\n",
       "      <td>7769</td>\n",
       "      <td>1</td>\n",
       "      <td>7768</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>7803</td>\n",
       "      <td>1</td>\n",
       "      <td>7802</td>\n",
       "      <td>0.000128</td>\n",
       "      <td>17</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>11749</td>\n",
       "      <td>1</td>\n",
       "      <td>11748</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>24</td>\n",
       "      <td>0.002039</td>\n",
       "      <td>3980</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.661248</td>\n",
       "      <td>0.955863</td>\n",
       "      <td>chr1:26729645_TTG&gt;T</td>\n",
       "      <td>P19_0024_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729646-26729647</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_region_variant,splice_polypyrimidine_tr...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1422289765</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729646-26729647</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_region_variant,splice_polypyrimidine_tr...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>LOW</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729645_TTG&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>26729646</td>\n",
       "      <td>TG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729650</td>\n",
       "      <td>GCCAT</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0046_BTR_01;TYPE=Deletion;DP=5868;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:5868:1:5867,1:0.0002:3185,2682:1,0:5879:58...</td>\n",
       "      <td>5879</td>\n",
       "      <td>1</td>\n",
       "      <td>5878</td>\n",
       "      <td>0.000170</td>\n",
       "      <td>5868</td>\n",
       "      <td>1</td>\n",
       "      <td>5867</td>\n",
       "      <td>0.000170</td>\n",
       "      <td>23</td>\n",
       "      <td>0.003897</td>\n",
       "      <td>11107</td>\n",
       "      <td>1</td>\n",
       "      <td>11106</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>26</td>\n",
       "      <td>0.002335</td>\n",
       "      <td>5228</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.529306</td>\n",
       "      <td>0.931705</td>\n",
       "      <td>chr1:26729650_GCCAT&gt;G</td>\n",
       "      <td>P19_0046_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729651-26729654</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527-1530</td>\n",
       "      <td>1138-1141</td>\n",
       "      <td>380-381</td>\n",
       "      <td>PS/X</td>\n",
       "      <td>CCATcc/cc</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729651-26729654</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527-1530</td>\n",
       "      <td>1138-1141</td>\n",
       "      <td>380-381</td>\n",
       "      <td>PS/X</td>\n",
       "      <td>CCATcc/cc</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729650_GCCAT&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>26729651</td>\n",
       "      <td>CCAT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729650</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0023_BTR_01;TYPE=Deletion;DP=7252;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7252:1:7251,1:0.0001:4090,3161:1,0:7289:72...</td>\n",
       "      <td>7289</td>\n",
       "      <td>1</td>\n",
       "      <td>7288</td>\n",
       "      <td>0.000137</td>\n",
       "      <td>7252</td>\n",
       "      <td>1</td>\n",
       "      <td>7251</td>\n",
       "      <td>0.000138</td>\n",
       "      <td>22</td>\n",
       "      <td>0.003009</td>\n",
       "      <td>10171</td>\n",
       "      <td>1</td>\n",
       "      <td>10170</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>29</td>\n",
       "      <td>0.002843</td>\n",
       "      <td>2882</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.716645</td>\n",
       "      <td>0.963893</td>\n",
       "      <td>chr1:26729650_GC&gt;G</td>\n",
       "      <td>P19_0023_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729651</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527</td>\n",
       "      <td>1138</td>\n",
       "      <td>380</td>\n",
       "      <td>P/X</td>\n",
       "      <td>Cca/ca</td>\n",
       "      <td>COSV100253669</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:26729651</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527</td>\n",
       "      <td>1138</td>\n",
       "      <td>380</td>\n",
       "      <td>P/X</td>\n",
       "      <td>Cca/ca</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729650_GC&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>26729651</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729650</td>\n",
       "      <td>GCCAT</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=P19_0046_BDO_01;TYPE=Deletion;DP=7376;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7376:2:7374,2:0.0003:4725,2649:2,0:7402:74...</td>\n",
       "      <td>7402</td>\n",
       "      <td>2</td>\n",
       "      <td>7400</td>\n",
       "      <td>0.000270</td>\n",
       "      <td>7376</td>\n",
       "      <td>2</td>\n",
       "      <td>7374</td>\n",
       "      <td>0.000271</td>\n",
       "      <td>42</td>\n",
       "      <td>0.005642</td>\n",
       "      <td>13703</td>\n",
       "      <td>2</td>\n",
       "      <td>13701</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>61</td>\n",
       "      <td>0.004432</td>\n",
       "      <td>6301</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.540174</td>\n",
       "      <td>0.930271</td>\n",
       "      <td>chr1:26729650_GCCAT&gt;G</td>\n",
       "      <td>P19_0046_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729651-26729654</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527-1530</td>\n",
       "      <td>1138-1141</td>\n",
       "      <td>380-381</td>\n",
       "      <td>PS/X</td>\n",
       "      <td>CCATcc/cc</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729651-26729654</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant,splice_region_variant</td>\n",
       "      <td>1527-1530</td>\n",
       "      <td>1138-1141</td>\n",
       "      <td>380-381</td>\n",
       "      <td>PS/X</td>\n",
       "      <td>CCATcc/cc</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729650_GCCAT&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>26729651</td>\n",
       "      <td>CCAT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26729656</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0050_BDO_01;TYPE=Insertion;DP=7213;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7213:1:7212,1:0.0001:5024,2188:1,0:7213:72...</td>\n",
       "      <td>7213</td>\n",
       "      <td>1</td>\n",
       "      <td>7212</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>7213</td>\n",
       "      <td>1</td>\n",
       "      <td>7212</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>8</td>\n",
       "      <td>0.001108</td>\n",
       "      <td>16999</td>\n",
       "      <td>1</td>\n",
       "      <td>16998</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>19</td>\n",
       "      <td>0.001116</td>\n",
       "      <td>9786</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.424319</td>\n",
       "      <td>0.911993</td>\n",
       "      <td>chr1:26729656_C&gt;CA</td>\n",
       "      <td>P19_0050_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:26729656-26729657</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1532-1533</td>\n",
       "      <td>1143-1144</td>\n",
       "      <td>381-382</td>\n",
       "      <td>-/X</td>\n",
       "      <td>-/A</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729656-26729657</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ENST00000324856</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1532-1533</td>\n",
       "      <td>1143-1144</td>\n",
       "      <td>381-382</td>\n",
       "      <td>-/X</td>\n",
       "      <td>-/A</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000320485</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1:26729656_C&gt;CA</td>\n",
       "      <td>1</td>\n",
       "      <td>26729656</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18968</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124095450</td>\n",
       "      <td>GTAGGTTTAATT</td>\n",
       "      <td>G</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0015_BTR_01;TYPE=Deletion;DP=3069;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3069:1:3068,1:0.0003:1375,1693:1,0:3087:30...</td>\n",
       "      <td>3087</td>\n",
       "      <td>1</td>\n",
       "      <td>3086</td>\n",
       "      <td>0.000324</td>\n",
       "      <td>3069</td>\n",
       "      <td>1</td>\n",
       "      <td>3068</td>\n",
       "      <td>0.000326</td>\n",
       "      <td>13</td>\n",
       "      <td>0.004194</td>\n",
       "      <td>5242</td>\n",
       "      <td>2</td>\n",
       "      <td>5240</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>15</td>\n",
       "      <td>0.002853</td>\n",
       "      <td>2155</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000464</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1.177795</td>\n",
       "      <td>1.020790</td>\n",
       "      <td>chrX:124095450_GTAGGTTTAATT&gt;G</td>\n",
       "      <td>P19_0015_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124095451-124095461</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_donor_variant,splice_donor_5th_base_var...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095451-124095461</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_donor_variant,splice_donor_5th_base_var...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>essential_splice</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>essential_splice</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124095450_GTAGGTTTAATT&gt;G</td>\n",
       "      <td>X</td>\n",
       "      <td>124095451</td>\n",
       "      <td>TAGGTTTAATT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18969</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100569</td>\n",
       "      <td>ATTAG</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0014_BTR_01;TYPE=Deletion;DP=2971;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2971:1:2970,1:0.0003:1645,1325:1,0:2989:29...</td>\n",
       "      <td>2989</td>\n",
       "      <td>1</td>\n",
       "      <td>2988</td>\n",
       "      <td>0.000335</td>\n",
       "      <td>2971</td>\n",
       "      <td>1</td>\n",
       "      <td>2970</td>\n",
       "      <td>0.000337</td>\n",
       "      <td>32</td>\n",
       "      <td>0.010593</td>\n",
       "      <td>5630</td>\n",
       "      <td>1</td>\n",
       "      <td>5629</td>\n",
       "      <td>0.000178</td>\n",
       "      <td>32</td>\n",
       "      <td>0.005652</td>\n",
       "      <td>2641</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.530906</td>\n",
       "      <td>0.926681</td>\n",
       "      <td>chrX:124100569_ATTAG&gt;A</td>\n",
       "      <td>P19_0014_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100570-124100573</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_acceptor_variant,splice_polypyrimidine_...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100570-124100573</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_acceptor_variant,splice_polypyrimidine_...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>essential_splice</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>essential_splice</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100569_ATTAG&gt;A</td>\n",
       "      <td>X</td>\n",
       "      <td>124100570</td>\n",
       "      <td>TTAG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18970</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100577</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0045_BTR_01;TYPE=Insertion;DP=1538;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:1538:1:1537,1:0.0007:990,547:1,0:1538:1537...</td>\n",
       "      <td>1538</td>\n",
       "      <td>1</td>\n",
       "      <td>1537</td>\n",
       "      <td>0.000650</td>\n",
       "      <td>1538</td>\n",
       "      <td>1</td>\n",
       "      <td>1537</td>\n",
       "      <td>0.000650</td>\n",
       "      <td>6</td>\n",
       "      <td>0.003886</td>\n",
       "      <td>2142</td>\n",
       "      <td>1</td>\n",
       "      <td>2141</td>\n",
       "      <td>0.000467</td>\n",
       "      <td>4</td>\n",
       "      <td>0.001864</td>\n",
       "      <td>604</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.718021</td>\n",
       "      <td>0.956808</td>\n",
       "      <td>chrX:124100577_C&gt;CT</td>\n",
       "      <td>P19_0045_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100577-124100578</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4100-4101</td>\n",
       "      <td>3787-3788</td>\n",
       "      <td>1263</td>\n",
       "      <td>L/LX</td>\n",
       "      <td>ctt/cTtt</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100577-124100578</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4100-4101</td>\n",
       "      <td>3787-3788</td>\n",
       "      <td>1263</td>\n",
       "      <td>L/LX</td>\n",
       "      <td>ctt/cTtt</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100577_C&gt;CT</td>\n",
       "      <td>X</td>\n",
       "      <td>124100577</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18971</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100579</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0034_BTR_01;TYPE=Insertion;DP=1765;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:1765:1:1764,1:0.0006:1067,697:1,0:1765:176...</td>\n",
       "      <td>1765</td>\n",
       "      <td>1</td>\n",
       "      <td>1764</td>\n",
       "      <td>0.000567</td>\n",
       "      <td>1765</td>\n",
       "      <td>1</td>\n",
       "      <td>1764</td>\n",
       "      <td>0.000567</td>\n",
       "      <td>5</td>\n",
       "      <td>0.002825</td>\n",
       "      <td>2693</td>\n",
       "      <td>1</td>\n",
       "      <td>2692</td>\n",
       "      <td>0.000371</td>\n",
       "      <td>5</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>928</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.655403</td>\n",
       "      <td>0.946508</td>\n",
       "      <td>chrX:124100579_T&gt;TG</td>\n",
       "      <td>P19_0034_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100579-124100580</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4102-4103</td>\n",
       "      <td>3789-3790</td>\n",
       "      <td>1263-1264</td>\n",
       "      <td>-/X</td>\n",
       "      <td>-/G</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100579-124100580</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4102-4103</td>\n",
       "      <td>3789-3790</td>\n",
       "      <td>1263-1264</td>\n",
       "      <td>-/X</td>\n",
       "      <td>-/G</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100579_T&gt;TG</td>\n",
       "      <td>X</td>\n",
       "      <td>124100579</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18972</th>\n",
       "      <td>chrX</td>\n",
       "      <td>124100587</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0015_BTR_01;TYPE=Insertion;DP=2635;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:2635:1:2634,1:0.0004:1439,1195:0,1:2635:26...</td>\n",
       "      <td>2635</td>\n",
       "      <td>1</td>\n",
       "      <td>2634</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>2635</td>\n",
       "      <td>1</td>\n",
       "      <td>2634</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>9</td>\n",
       "      <td>0.003404</td>\n",
       "      <td>4569</td>\n",
       "      <td>1</td>\n",
       "      <td>4568</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002184</td>\n",
       "      <td>1934</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.576713</td>\n",
       "      <td>0.934685</td>\n",
       "      <td>chrX:124100587_C&gt;CA</td>\n",
       "      <td>P19_0015_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX:124100587-124100588</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4110-4111</td>\n",
       "      <td>3797-3798</td>\n",
       "      <td>1266</td>\n",
       "      <td>S/SX</td>\n",
       "      <td>tca/tcAa</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100587-124100588</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101972</td>\n",
       "      <td>ENST00000371145</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4110-4111</td>\n",
       "      <td>3797-3798</td>\n",
       "      <td>1266</td>\n",
       "      <td>S/SX</td>\n",
       "      <td>tca/tcAa</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>STAG2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11355</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000360187</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX:124100587_C&gt;CA</td>\n",
       "      <td>X</td>\n",
       "      <td>124100587</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18973 rows  149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS           REF ALT    FILTER  \\\n",
       "0      chr1   26729645           TTG   T      pSTD   \n",
       "1      chr1   26729650         GCCAT   G      pSTD   \n",
       "2      chr1   26729650            GC   G      pSTD   \n",
       "3      chr1   26729650         GCCAT   G      PASS   \n",
       "4      chr1   26729656             C  CA      pSTD   \n",
       "...     ...        ...           ...  ..       ...   \n",
       "18968  chrX  124095450  GTAGGTTTAATT   G  p10;pSTD   \n",
       "18969  chrX  124100569         ATTAG   A      pSTD   \n",
       "18970  chrX  124100577             C  CT      pSTD   \n",
       "18971  chrX  124100579             T  TG      pSTD   \n",
       "18972  chrX  124100587             C  CA      pSTD   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      SAMPLE=P19_0024_BDO_01;TYPE=Deletion;DP=7803;V...   \n",
       "1      SAMPLE=P19_0046_BTR_01;TYPE=Deletion;DP=5868;V...   \n",
       "2      SAMPLE=P19_0023_BTR_01;TYPE=Deletion;DP=7252;V...   \n",
       "3      SAMPLE=P19_0046_BDO_01;TYPE=Deletion;DP=7376;V...   \n",
       "4      SAMPLE=P19_0050_BDO_01;TYPE=Insertion;DP=7213;...   \n",
       "...                                                  ...   \n",
       "18968  SAMPLE=P19_0015_BTR_01;TYPE=Deletion;DP=3069;V...   \n",
       "18969  SAMPLE=P19_0014_BTR_01;TYPE=Deletion;DP=2971;V...   \n",
       "18970  SAMPLE=P19_0045_BTR_01;TYPE=Insertion;DP=1538;...   \n",
       "18971  SAMPLE=P19_0034_BTR_01;TYPE=Insertion;DP=1765;...   \n",
       "18972  SAMPLE=P19_0015_BTR_01;TYPE=Insertion;DP=2635;...   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "1      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "2      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "3      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "4      GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "...                                                  ...   \n",
       "18968  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "18969  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "18970  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "18971  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "18972  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                  SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "0      0/1:7803:1:7802,1:0.0001:4759,3043:1,0:7769:77...   7769          1   \n",
       "1      0/1:5868:1:5867,1:0.0002:3185,2682:1,0:5879:58...   5879          1   \n",
       "2      0/1:7252:1:7251,1:0.0001:4090,3161:1,0:7289:72...   7289          1   \n",
       "3      0/1:7376:2:7374,2:0.0003:4725,2649:2,0:7402:74...   7402          2   \n",
       "4      0/1:7213:1:7212,1:0.0001:5024,2188:1,0:7213:72...   7213          1   \n",
       "...                                                  ...    ...        ...   \n",
       "18968  0/1:3069:1:3068,1:0.0003:1375,1693:1,0:3087:30...   3087          1   \n",
       "18969  0/1:2971:1:2970,1:0.0003:1645,1325:1,0:2989:29...   2989          1   \n",
       "18970  0/1:1538:1:1537,1:0.0007:990,547:1,0:1538:1537...   1538          1   \n",
       "18971  0/1:1765:1:1764,1:0.0006:1067,697:1,0:1765:176...   1765          1   \n",
       "18972  0/1:2635:1:2634,1:0.0004:1439,1195:0,1:2635:26...   2635          1   \n",
       "\n",
       "       REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "0           7768  0.000129      7803             1          7802  0.000128   \n",
       "1           5878  0.000170      5868             1          5867  0.000170   \n",
       "2           7288  0.000137      7252             1          7251  0.000138   \n",
       "3           7400  0.000270      7376             2          7374  0.000271   \n",
       "4           7212  0.000139      7213             1          7212  0.000139   \n",
       "...          ...       ...       ...           ...           ...       ...   \n",
       "18968       3086  0.000324      3069             1          3068  0.000326   \n",
       "18969       2988  0.000335      2971             1          2970  0.000337   \n",
       "18970       1537  0.000650      1538             1          1537  0.000650   \n",
       "18971       1764  0.000567      1765             1          1764  0.000567   \n",
       "18972       2634  0.000380      2635             1          2634  0.000380   \n",
       "\n",
       "       numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "0         17  0.002183     11749             1         11748  0.000085   \n",
       "1         23  0.003897     11107             1         11106  0.000090   \n",
       "2         22  0.003009     10171             1         10170  0.000098   \n",
       "3         42  0.005642     13703             2         13701  0.000146   \n",
       "4          8  0.001108     16999             1         16998  0.000059   \n",
       "...      ...       ...       ...           ...           ...       ...   \n",
       "18968     13  0.004194      5242             2          5240  0.000382   \n",
       "18969     32  0.010593      5630             1          5629  0.000178   \n",
       "18970      6  0.003886      2142             1          2141  0.000467   \n",
       "18971      5  0.002825      2693             1          2692  0.000371   \n",
       "18972      9  0.003404      4569             1          4568  0.000219   \n",
       "\n",
       "       numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND    VAF_ND  \\\n",
       "0            24   0.002039      3980             0  0.000000   \n",
       "1            26   0.002335      5228             0  0.000000   \n",
       "2            29   0.002843      2882             0  0.000000   \n",
       "3            61   0.004432      6301             0  0.000000   \n",
       "4            19   0.001116      9786             0  0.000000   \n",
       "...         ...        ...       ...           ...       ...   \n",
       "18968        15   0.002853      2155             1  0.000464   \n",
       "18969        32   0.005652      2641             0  0.000000   \n",
       "18970         4   0.001864       604             0  0.000000   \n",
       "18971         5   0.001853       928             0  0.000000   \n",
       "18972        10   0.002184      1934             0  0.000000   \n",
       "\n",
       "       VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "0                       False                      False        0.661248   \n",
       "1                       False                      False        0.529306   \n",
       "2                       False                      False        0.716645   \n",
       "3                       False                      False        0.540174   \n",
       "4                       False                      False        0.424319   \n",
       "...                       ...                        ...             ...   \n",
       "18968                   False                      False        1.177795   \n",
       "18969                   False                      False        0.530906   \n",
       "18970                   False                      False        0.718021   \n",
       "18971                   False                      False        0.655403   \n",
       "18972                   False                      False        0.576713   \n",
       "\n",
       "       VAF_distortion_sq                         MUT_ID        SAMPLE_ID  \\\n",
       "0               0.955863            chr1:26729645_TTG>T  P19_0024_BDO_01   \n",
       "1               0.931705          chr1:26729650_GCCAT>G  P19_0046_BTR_01   \n",
       "2               0.963893             chr1:26729650_GC>G  P19_0023_BTR_01   \n",
       "3               0.930271          chr1:26729650_GCCAT>G  P19_0046_BDO_01   \n",
       "4               0.911993             chr1:26729656_C>CA  P19_0050_BDO_01   \n",
       "...                  ...                            ...              ...   \n",
       "18968           1.020790  chrX:124095450_GTAGGTTTAATT>G  P19_0015_BTR_01   \n",
       "18969           0.926681         chrX:124100569_ATTAG>A  P19_0014_BTR_01   \n",
       "18970           0.956808            chrX:124100577_C>CT  P19_0045_BTR_01   \n",
       "18971           0.946508            chrX:124100579_T>TG  P19_0034_BTR_01   \n",
       "18972           0.934685            chrX:124100587_C>CA  P19_0015_BTR_01   \n",
       "\n",
       "      METHOD  PROJECT_NAME                  Location Allele             Gene  \\\n",
       "0        med           NaN    chr1:26729646-26729647      -  ENSG00000117713   \n",
       "1        med           NaN    chr1:26729651-26729654      -  ENSG00000117713   \n",
       "2        med           NaN             chr1:26729651      -  ENSG00000117713   \n",
       "3        med           NaN    chr1:26729651-26729654      -  ENSG00000117713   \n",
       "4        med           NaN    chr1:26729656-26729657      A  ENSG00000117713   \n",
       "...      ...           ...                       ...    ...              ...   \n",
       "18968    med           NaN  chrX:124095451-124095461      -  ENSG00000101972   \n",
       "18969    med           NaN  chrX:124100570-124100573      -  ENSG00000101972   \n",
       "18970    med           NaN  chrX:124100577-124100578      T  ENSG00000101972   \n",
       "18971    med           NaN  chrX:124100579-124100580      G  ENSG00000101972   \n",
       "18972    med           NaN  chrX:124100587-124100588      A  ENSG00000101972   \n",
       "\n",
       "               Feature Feature_type  \\\n",
       "0      ENST00000324856   Transcript   \n",
       "1      ENST00000324856   Transcript   \n",
       "2      ENST00000324856   Transcript   \n",
       "3      ENST00000324856   Transcript   \n",
       "4      ENST00000324856   Transcript   \n",
       "...                ...          ...   \n",
       "18968  ENST00000371145   Transcript   \n",
       "18969  ENST00000371145   Transcript   \n",
       "18970  ENST00000371145   Transcript   \n",
       "18971  ENST00000371145   Transcript   \n",
       "18972  ENST00000371145   Transcript   \n",
       "\n",
       "                                             Consequence cDNA_position  \\\n",
       "0      splice_region_variant,splice_polypyrimidine_tr...             -   \n",
       "1               frameshift_variant,splice_region_variant     1527-1530   \n",
       "2               frameshift_variant,splice_region_variant          1527   \n",
       "3               frameshift_variant,splice_region_variant     1527-1530   \n",
       "4                                     frameshift_variant     1532-1533   \n",
       "...                                                  ...           ...   \n",
       "18968  splice_donor_variant,splice_donor_5th_base_var...             -   \n",
       "18969  splice_acceptor_variant,splice_polypyrimidine_...             -   \n",
       "18970                                 frameshift_variant     4100-4101   \n",
       "18971                                 frameshift_variant     4102-4103   \n",
       "18972                                 frameshift_variant     4110-4111   \n",
       "\n",
       "      CDS_position Protein_position Amino_acids     Codons Existing_variation  \\\n",
       "0                -                -           -          -       rs1422289765   \n",
       "1        1138-1141          380-381        PS/X  CCATcc/cc                  -   \n",
       "2             1138              380         P/X     Cca/ca      COSV100253669   \n",
       "3        1138-1141          380-381        PS/X  CCATcc/cc                  -   \n",
       "4        1143-1144          381-382         -/X        -/A                  -   \n",
       "...            ...              ...         ...        ...                ...   \n",
       "18968            -                -           -          -                  -   \n",
       "18969            -                -           -          -                  -   \n",
       "18970    3787-3788             1263        L/LX   ctt/cTtt                  -   \n",
       "18971    3789-3790        1263-1264         -/X        -/G                  -   \n",
       "18972    3797-3798             1266        S/SX   tca/tcAa                  -   \n",
       "\n",
       "      IMPACT DISTANCE STRAND FLAGS  SYMBOL SYMBOL_SOURCE     HGNC_ID  \\\n",
       "0        LOW        -      1     -  ARID1A          HGNC  HGNC:11110   \n",
       "1       HIGH        -      1     -  ARID1A          HGNC  HGNC:11110   \n",
       "2       HIGH        -      1     -  ARID1A          HGNC  HGNC:11110   \n",
       "3       HIGH        -      1     -  ARID1A          HGNC  HGNC:11110   \n",
       "4       HIGH        -      1     -  ARID1A          HGNC  HGNC:11110   \n",
       "...      ...      ...    ...   ...     ...           ...         ...   \n",
       "18968   HIGH        -      1     -   STAG2          HGNC  HGNC:11355   \n",
       "18969   HIGH        -      1     -   STAG2          HGNC  HGNC:11355   \n",
       "18970   HIGH        -      1     -   STAG2          HGNC  HGNC:11355   \n",
       "18971   HIGH        -      1     -   STAG2          HGNC  HGNC:11355   \n",
       "18972   HIGH        -      1     -   STAG2          HGNC  HGNC:11355   \n",
       "\n",
       "      CANONICAL             ENSP  gnomADe_AF  gnomADe_AFR_AF  gnomADe_AMR_AF  \\\n",
       "0           YES  ENSP00000320485         0.0             0.0             0.0   \n",
       "1           YES  ENSP00000320485         0.0             0.0             0.0   \n",
       "2           YES  ENSP00000320485         0.0             0.0             0.0   \n",
       "3           YES  ENSP00000320485         0.0             0.0             0.0   \n",
       "4           YES  ENSP00000320485         0.0             0.0             0.0   \n",
       "...         ...              ...         ...             ...             ...   \n",
       "18968       YES  ENSP00000360187         0.0             0.0             0.0   \n",
       "18969       YES  ENSP00000360187         0.0             0.0             0.0   \n",
       "18970       YES  ENSP00000360187         0.0             0.0             0.0   \n",
       "18971       YES  ENSP00000360187         0.0             0.0             0.0   \n",
       "18972       YES  ENSP00000360187         0.0             0.0             0.0   \n",
       "\n",
       "       gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  gnomADe_NFE_AF  \\\n",
       "0                 0.0             0.0             0.0             0.0   \n",
       "1                 0.0             0.0             0.0             0.0   \n",
       "2                 0.0             0.0             0.0             0.0   \n",
       "3                 0.0             0.0             0.0             0.0   \n",
       "4                 0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "18968             0.0             0.0             0.0             0.0   \n",
       "18969             0.0             0.0             0.0             0.0   \n",
       "18970             0.0             0.0             0.0             0.0   \n",
       "18971             0.0             0.0             0.0             0.0   \n",
       "18972             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  gnomADg_AFR_AF  \\\n",
       "0                 0.0             0.0         0.0             0.0   \n",
       "1                 0.0             0.0         0.0             0.0   \n",
       "2                 0.0             0.0         0.0             0.0   \n",
       "3                 0.0             0.0         0.0             0.0   \n",
       "4                 0.0             0.0         0.0             0.0   \n",
       "...               ...             ...         ...             ...   \n",
       "18968             0.0             0.0         0.0             0.0   \n",
       "18969             0.0             0.0         0.0             0.0   \n",
       "18970             0.0             0.0         0.0             0.0   \n",
       "18971             0.0             0.0         0.0             0.0   \n",
       "18972             0.0             0.0         0.0             0.0   \n",
       "\n",
       "       gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  \\\n",
       "0                 0.0             0.0             0.0             0.0   \n",
       "1                 0.0             0.0             0.0             0.0   \n",
       "2                 0.0             0.0             0.0             0.0   \n",
       "3                 0.0             0.0             0.0             0.0   \n",
       "4                 0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "18968             0.0             0.0             0.0             0.0   \n",
       "18969             0.0             0.0             0.0             0.0   \n",
       "18970             0.0             0.0             0.0             0.0   \n",
       "18971             0.0             0.0             0.0             0.0   \n",
       "18972             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  \\\n",
       "0                 0.0             0.0             0.0             0.0   \n",
       "1                 0.0             0.0             0.0             0.0   \n",
       "2                 0.0             0.0             0.0             0.0   \n",
       "3                 0.0             0.0             0.0             0.0   \n",
       "4                 0.0             0.0             0.0             0.0   \n",
       "...               ...             ...             ...             ...   \n",
       "18968             0.0             0.0             0.0             0.0   \n",
       "18969             0.0             0.0             0.0             0.0   \n",
       "18970             0.0             0.0             0.0             0.0   \n",
       "18971             0.0             0.0             0.0             0.0   \n",
       "18972             0.0             0.0             0.0             0.0   \n",
       "\n",
       "       gnomADg_SAS_AF CLIN_SIG SOMATIC PHENO        canonical_Location  \\\n",
       "0                 0.0        -       -     -    chr1:26729646-26729647   \n",
       "1                 0.0        -       -     -    chr1:26729651-26729654   \n",
       "2                 0.0        -       1     1             chr1:26729651   \n",
       "3                 0.0        -       -     -    chr1:26729651-26729654   \n",
       "4                 0.0        -       -     -    chr1:26729656-26729657   \n",
       "...               ...      ...     ...   ...                       ...   \n",
       "18968             0.0        -       -     -  chrX:124095451-124095461   \n",
       "18969             0.0        -       -     -  chrX:124100570-124100573   \n",
       "18970             0.0        -       -     -  chrX:124100577-124100578   \n",
       "18971             0.0        -       -     -  chrX:124100579-124100580   \n",
       "18972             0.0        -       -     -  chrX:124100587-124100588   \n",
       "\n",
       "      canonical_Allele   canonical_Gene canonical_Feature  \\\n",
       "0                    -  ENSG00000117713   ENST00000324856   \n",
       "1                    -  ENSG00000117713   ENST00000324856   \n",
       "2                    -  ENSG00000117713   ENST00000324856   \n",
       "3                    -  ENSG00000117713   ENST00000324856   \n",
       "4                    A  ENSG00000117713   ENST00000324856   \n",
       "...                ...              ...               ...   \n",
       "18968                -  ENSG00000101972   ENST00000371145   \n",
       "18969                -  ENSG00000101972   ENST00000371145   \n",
       "18970                T  ENSG00000101972   ENST00000371145   \n",
       "18971                G  ENSG00000101972   ENST00000371145   \n",
       "18972                A  ENSG00000101972   ENST00000371145   \n",
       "\n",
       "      canonical_Feature_type  \\\n",
       "0                 Transcript   \n",
       "1                 Transcript   \n",
       "2                 Transcript   \n",
       "3                 Transcript   \n",
       "4                 Transcript   \n",
       "...                      ...   \n",
       "18968             Transcript   \n",
       "18969             Transcript   \n",
       "18970             Transcript   \n",
       "18971             Transcript   \n",
       "18972             Transcript   \n",
       "\n",
       "                                   canonical_Consequence  \\\n",
       "0      splice_region_variant,splice_polypyrimidine_tr...   \n",
       "1               frameshift_variant,splice_region_variant   \n",
       "2               frameshift_variant,splice_region_variant   \n",
       "3               frameshift_variant,splice_region_variant   \n",
       "4                                     frameshift_variant   \n",
       "...                                                  ...   \n",
       "18968  splice_donor_variant,splice_donor_5th_base_var...   \n",
       "18969  splice_acceptor_variant,splice_polypyrimidine_...   \n",
       "18970                                 frameshift_variant   \n",
       "18971                                 frameshift_variant   \n",
       "18972                                 frameshift_variant   \n",
       "\n",
       "      canonical_cDNA_position canonical_CDS_position  \\\n",
       "0                           -                      -   \n",
       "1                   1527-1530              1138-1141   \n",
       "2                        1527                   1138   \n",
       "3                   1527-1530              1138-1141   \n",
       "4                   1532-1533              1143-1144   \n",
       "...                       ...                    ...   \n",
       "18968                       -                      -   \n",
       "18969                       -                      -   \n",
       "18970               4100-4101              3787-3788   \n",
       "18971               4102-4103              3789-3790   \n",
       "18972               4110-4111              3797-3798   \n",
       "\n",
       "      canonical_Protein_position canonical_Amino_acids canonical_Codons  \\\n",
       "0                              -                     -                -   \n",
       "1                        380-381                  PS/X        CCATcc/cc   \n",
       "2                            380                   P/X           Cca/ca   \n",
       "3                        380-381                  PS/X        CCATcc/cc   \n",
       "4                        381-382                   -/X              -/A   \n",
       "...                          ...                   ...              ...   \n",
       "18968                          -                     -                -   \n",
       "18969                          -                     -                -   \n",
       "18970                       1263                  L/LX         ctt/cTtt   \n",
       "18971                  1263-1264                   -/X              -/G   \n",
       "18972                       1266                  S/SX         tca/tcAa   \n",
       "\n",
       "      canonical_IMPACT canonical_DISTANCE canonical_STRAND canonical_FLAGS  \\\n",
       "0                  LOW                  -              1.0               -   \n",
       "1                 HIGH                  -              1.0               -   \n",
       "2                 HIGH                  -              1.0               -   \n",
       "3                 HIGH                  -              1.0               -   \n",
       "4                 HIGH                  -              1.0               -   \n",
       "...                ...                ...              ...             ...   \n",
       "18968             HIGH                  -              1.0               -   \n",
       "18969             HIGH                  -              1.0               -   \n",
       "18970             HIGH                  -              1.0               -   \n",
       "18971             HIGH                  -              1.0               -   \n",
       "18972             HIGH                  -              1.0               -   \n",
       "\n",
       "      canonical_SYMBOL canonical_SYMBOL_SOURCE canonical_HGNC_ID  \\\n",
       "0               ARID1A                    HGNC        HGNC:11110   \n",
       "1               ARID1A                    HGNC        HGNC:11110   \n",
       "2               ARID1A                    HGNC        HGNC:11110   \n",
       "3               ARID1A                    HGNC        HGNC:11110   \n",
       "4               ARID1A                    HGNC        HGNC:11110   \n",
       "...                ...                     ...               ...   \n",
       "18968            STAG2                    HGNC        HGNC:11355   \n",
       "18969            STAG2                    HGNC        HGNC:11355   \n",
       "18970            STAG2                    HGNC        HGNC:11355   \n",
       "18971            STAG2                    HGNC        HGNC:11355   \n",
       "18972            STAG2                    HGNC        HGNC:11355   \n",
       "\n",
       "      canonical_CANONICAL   canonical_ENSP canonical_CLIN_SIG  \\\n",
       "0                     YES  ENSP00000320485                  -   \n",
       "1                     YES  ENSP00000320485                  -   \n",
       "2                     YES  ENSP00000320485                  -   \n",
       "3                     YES  ENSP00000320485                  -   \n",
       "4                     YES  ENSP00000320485                  -   \n",
       "...                   ...              ...                ...   \n",
       "18968                 YES  ENSP00000360187                  -   \n",
       "18969                 YES  ENSP00000360187                  -   \n",
       "18970                 YES  ENSP00000360187                  -   \n",
       "18971                 YES  ENSP00000360187                  -   \n",
       "18972                 YES  ENSP00000360187                  -   \n",
       "\n",
       "      canonical_Consequence_single canonical_Consequence_broader  \\\n",
       "0            splice_region_variant         splice_region_variant   \n",
       "1               frameshift_variant                      nonsense   \n",
       "2               frameshift_variant                      nonsense   \n",
       "3               frameshift_variant                      nonsense   \n",
       "4               frameshift_variant                      nonsense   \n",
       "...                            ...                           ...   \n",
       "18968         splice_donor_variant              essential_splice   \n",
       "18969      splice_acceptor_variant              essential_splice   \n",
       "18970           frameshift_variant                      nonsense   \n",
       "18971           frameshift_variant                      nonsense   \n",
       "18972           frameshift_variant                      nonsense   \n",
       "\n",
       "      canonical_Protein_affecting       Consequence_single  \\\n",
       "0                       ambiguous    splice_region_variant   \n",
       "1               protein_affecting       frameshift_variant   \n",
       "2               protein_affecting       frameshift_variant   \n",
       "3               protein_affecting       frameshift_variant   \n",
       "4               protein_affecting       frameshift_variant   \n",
       "...                           ...                      ...   \n",
       "18968           protein_affecting     splice_donor_variant   \n",
       "18969           protein_affecting  splice_acceptor_variant   \n",
       "18970           protein_affecting       frameshift_variant   \n",
       "18971           protein_affecting       frameshift_variant   \n",
       "18972           protein_affecting       frameshift_variant   \n",
       "\n",
       "         Consequence_broader  Protein_affecting CONTEXT_MUT MUTTYPE  \\\n",
       "0      splice_region_variant          ambiguous           -       -   \n",
       "1                   nonsense  protein_affecting           -       -   \n",
       "2                   nonsense  protein_affecting           -       -   \n",
       "3                   nonsense  protein_affecting           -       -   \n",
       "4                   nonsense  protein_affecting           -       -   \n",
       "...                      ...                ...         ...     ...   \n",
       "18968       essential_splice  protein_affecting           -       -   \n",
       "18969       essential_splice  protein_affecting           -       -   \n",
       "18970               nonsense  protein_affecting           -       -   \n",
       "18971               nonsense  protein_affecting           -       -   \n",
       "18972               nonsense  protein_affecting           -       -   \n",
       "\n",
       "      CONTEXT_MUT_SIGPRO  gnomAD_SNP HOTSPOT_NAME  \\\n",
       "0                      -       False            -   \n",
       "1                      -       False            -   \n",
       "2                      -       False            -   \n",
       "3                      -       False            -   \n",
       "4                      -       False            -   \n",
       "...                  ...         ...          ...   \n",
       "18968                  -       False            -   \n",
       "18969                  -       False            -   \n",
       "18970                  -       False            -   \n",
       "18971                  -       False            -   \n",
       "18972                  -       False            -   \n",
       "\n",
       "                          MUT_ID_pyr CHROM_ensembl  POS_ensembl  REF_ensembl  \\\n",
       "0                chr1:26729645_TTG>T             1     26729646           TG   \n",
       "1              chr1:26729650_GCCAT>G             1     26729651         CCAT   \n",
       "2                 chr1:26729650_GC>G             1     26729651            C   \n",
       "3              chr1:26729650_GCCAT>G             1     26729651         CCAT   \n",
       "4                 chr1:26729656_C>CA             1     26729656            -   \n",
       "...                              ...           ...          ...          ...   \n",
       "18968  chrX:124095450_GTAGGTTTAATT>G             X    124095451  TAGGTTTAATT   \n",
       "18969         chrX:124100569_ATTAG>A             X    124100570         TTAG   \n",
       "18970            chrX:124100577_C>CT             X    124100577            -   \n",
       "18971            chrX:124100579_T>TG             X    124100579            -   \n",
       "18972            chrX:124100587_C>CA             X    124100587            -   \n",
       "\n",
       "      ALT_ensembl       TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  \\\n",
       "0               -   DELETION             2          False            False   \n",
       "1               -   DELETION             4          False            False   \n",
       "2               -   DELETION             1          False            False   \n",
       "3               -   DELETION             4          False            False   \n",
       "4               A  INSERTION             1          False            False   \n",
       "...           ...        ...           ...            ...              ...   \n",
       "18968           -   DELETION            11          False            False   \n",
       "18969           -   DELETION             4          False            False   \n",
       "18970           T  INSERTION             1          False            False   \n",
       "18971           G  INSERTION             1          False            False   \n",
       "18972           A  INSERTION             1          False            False   \n",
       "\n",
       "       N_rich_frequency  VAF_Ns_threshold  FILTER.cohort_n_rich_uni  \\\n",
       "0                   0.0               1.1                     False   \n",
       "1                   0.0               1.1                     False   \n",
       "2                   0.0               1.1                     False   \n",
       "3                   0.0               1.1                     False   \n",
       "4                   0.0               1.1                     False   \n",
       "...                 ...               ...                       ...   \n",
       "18968               0.0               1.1                     False   \n",
       "18969               0.0               1.1                     False   \n",
       "18970               0.0               1.1                     False   \n",
       "18971               0.0               1.1                     False   \n",
       "18972               0.0               1.1                     False   \n",
       "\n",
       "       FILTER.gnomAD_SNP  FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  \\\n",
       "0                  False               False                False       False   \n",
       "1                  False               False                False       False   \n",
       "2                  False               False                False       False   \n",
       "3                  False               False                False       False   \n",
       "4                  False               False                False       False   \n",
       "...                  ...                 ...                  ...         ...   \n",
       "18968              False               False                False        True   \n",
       "18969              False               False                False       False   \n",
       "18970              False               False                False       False   \n",
       "18971              False               False                False       False   \n",
       "18972              False               False                False       False   \n",
       "\n",
       "       FILTER.pSTD  FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "0             True                      False                 False   \n",
       "1             True                      False                 False   \n",
       "2             True                      False                 False   \n",
       "3            False                      False                 False   \n",
       "4             True                      False                 False   \n",
       "...            ...                        ...                   ...   \n",
       "18968         True                      False                 False   \n",
       "18969         True                      False                 False   \n",
       "18970         True                      False                 False   \n",
       "18971         True                      False                 False   \n",
       "18972         True                      False                 False   \n",
       "\n",
       "       FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "0                              False           False   \n",
       "1                              False           False   \n",
       "2                              False           False   \n",
       "3                              False           False   \n",
       "4                              False           False   \n",
       "...                              ...             ...   \n",
       "18968                          False           False   \n",
       "18969                          False           False   \n",
       "18970                          False           False   \n",
       "18971                          False           False   \n",
       "18972                          False           False   \n",
       "\n",
       "       FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "0                               False                        False   \n",
       "1                               False                        False   \n",
       "2                               False                        False   \n",
       "3                               False                        False   \n",
       "4                               False                        False   \n",
       "...                               ...                          ...   \n",
       "18968                           False                        False   \n",
       "18969                           False                        False   \n",
       "18970                           False                        False   \n",
       "18971                           False                        False   \n",
       "18972                           False                        False   \n",
       "\n",
       "       FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "0                         False          False                    False   \n",
       "1                         False          False                    False   \n",
       "2                         False          False                    False   \n",
       "3                         False          False                    False   \n",
       "4                         False          False                    False   \n",
       "...                         ...            ...                      ...   \n",
       "18968                     False          False                    False   \n",
       "18969                     False          False                    False   \n",
       "18970                     False          False                    False   \n",
       "18971                     False          False                    False   \n",
       "18972                     False          False                    False   \n",
       "\n",
       "       FILTER.PASS  FILTER.AM_not_searched_COMPLEX  \\\n",
       "0            False                           False   \n",
       "1            False                           False   \n",
       "2            False                           False   \n",
       "3             True                           False   \n",
       "4            False                           False   \n",
       "...            ...                             ...   \n",
       "18968        False                           False   \n",
       "18969        False                           False   \n",
       "18970        False                           False   \n",
       "18971        False                           False   \n",
       "18972        False                           False   \n",
       "\n",
       "       FILTER.not_searched_COMPLEX  FILTER.MSI12  FILTER.AM_not_searched_SV  \\\n",
       "0                            False         False                      False   \n",
       "1                            False         False                      False   \n",
       "2                            False         False                      False   \n",
       "3                            False         False                      False   \n",
       "4                            False         False                      False   \n",
       "...                            ...           ...                        ...   \n",
       "18968                        False         False                      False   \n",
       "18969                        False         False                      False   \n",
       "18970                        False         False                      False   \n",
       "18971                        False         False                      False   \n",
       "18972                        False         False                      False   \n",
       "\n",
       "       FILTER.not_searched_SV  FILTER.SN1.5  FILTER.q22.5  FILTER.v1  \\\n",
       "0                       False         False         False      False   \n",
       "1                       False         False         False      False   \n",
       "2                       False         False         False      False   \n",
       "3                       False         False         False      False   \n",
       "4                       False         False         False      False   \n",
       "...                       ...           ...           ...        ...   \n",
       "18968                   False         False         False      False   \n",
       "18969                   False         False         False      False   \n",
       "18970                   False         False         False      False   \n",
       "18971                   False         False         False      False   \n",
       "18972                   False         False         False      False   \n",
       "\n",
       "       FILTER.low_mappability  FILTER.Bias  FILTER.d3  \n",
       "0                       False        False      False  \n",
       "1                       False        False      False  \n",
       "2                       False        False      False  \n",
       "3                       False        False      False  \n",
       "4                       False        False      False  \n",
       "...                       ...          ...        ...  \n",
       "18968                   False        False      False  \n",
       "18969                   False        False      False  \n",
       "18970                   False        False      False  \n",
       "18971                   False        False      False  \n",
       "18972                   False        False      False  \n",
       "\n",
       "[18973 rows x 149 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter and somatic only\n",
    "indel_maf_df = maf_df_f.loc[\n",
    "                        (maf_df_f[\"TYPE\"].isin([\"INSERTION\", \"DELETION\"]))\n",
    "                        ].reset_index(drop = True)\n",
    "indel_maf_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5792da0-8d90-4517-b5f3-b7b1665b7da8",
   "metadata": {},
   "source": [
    "Observed frameshift indels + inframes of length >= 5 AA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fc50162e-0e67-42ea-ada6-9bdf451b80dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8909</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669611</td>\n",
       "      <td>AGTCT</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0009_BDO_01;TYPE=Deletion;DP=3873;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3873:1:3872,1:0.0003:2741,1131:1,0:3829:38...</td>\n",
       "      <td>3829</td>\n",
       "      <td>1</td>\n",
       "      <td>3828</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>3873</td>\n",
       "      <td>1</td>\n",
       "      <td>3872</td>\n",
       "      <td>0.000258</td>\n",
       "      <td>24</td>\n",
       "      <td>0.006229</td>\n",
       "      <td>7842</td>\n",
       "      <td>1</td>\n",
       "      <td>7841</td>\n",
       "      <td>0.000128</td>\n",
       "      <td>27</td>\n",
       "      <td>0.003431</td>\n",
       "      <td>4013</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.488268</td>\n",
       "      <td>0.920055</td>\n",
       "      <td>chr17:7669611_AGTCT&gt;A</td>\n",
       "      <td>P19_0009_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669612-7669615</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1318-1321</td>\n",
       "      <td>1176-1179</td>\n",
       "      <td>392-393</td>\n",
       "      <td>SD/X</td>\n",
       "      <td>tcAGAC/tc</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669612-7669615</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1318-1321</td>\n",
       "      <td>1176-1179</td>\n",
       "      <td>392-393</td>\n",
       "      <td>SD/X</td>\n",
       "      <td>tcAGAC/tc</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669611_AGTCT&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7669612</td>\n",
       "      <td>GTCT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8910</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669612</td>\n",
       "      <td>GTC</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0048_BTR_01;TYPE=Deletion;DP=4416;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4416:1:4415,1:0.0002:2387,2028:1,0:4434:44...</td>\n",
       "      <td>4434</td>\n",
       "      <td>1</td>\n",
       "      <td>4433</td>\n",
       "      <td>0.000226</td>\n",
       "      <td>4416</td>\n",
       "      <td>1</td>\n",
       "      <td>4415</td>\n",
       "      <td>0.000226</td>\n",
       "      <td>13</td>\n",
       "      <td>0.002923</td>\n",
       "      <td>6854</td>\n",
       "      <td>1</td>\n",
       "      <td>6853</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>8</td>\n",
       "      <td>0.001166</td>\n",
       "      <td>2420</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.646922</td>\n",
       "      <td>0.950691</td>\n",
       "      <td>chr17:7669612_GTC&gt;G</td>\n",
       "      <td>P19_0048_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669613-7669614</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1319-1320</td>\n",
       "      <td>1177-1178</td>\n",
       "      <td>393</td>\n",
       "      <td>D/X</td>\n",
       "      <td>GAc/c</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669613-7669614</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1319-1320</td>\n",
       "      <td>1177-1178</td>\n",
       "      <td>393</td>\n",
       "      <td>D/X</td>\n",
       "      <td>GAc/c</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669612_GTC&gt;G</td>\n",
       "      <td>17</td>\n",
       "      <td>7669613</td>\n",
       "      <td>TC</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8911</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669629</td>\n",
       "      <td>CTGTCT</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0051_BTR_01;TYPE=Deletion;DP=10669;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:10669:1:10668,1:0.0001:5456,5212:1,0:10646...</td>\n",
       "      <td>10646</td>\n",
       "      <td>1</td>\n",
       "      <td>10645</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>10669</td>\n",
       "      <td>1</td>\n",
       "      <td>10668</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>35</td>\n",
       "      <td>0.003277</td>\n",
       "      <td>22524</td>\n",
       "      <td>1</td>\n",
       "      <td>22523</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>52</td>\n",
       "      <td>0.002303</td>\n",
       "      <td>11878</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.472651</td>\n",
       "      <td>0.925227</td>\n",
       "      <td>chr17:7669629_CTGTCT&gt;C</td>\n",
       "      <td>P19_0051_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669630-7669634</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1299-1303</td>\n",
       "      <td>1157-1161</td>\n",
       "      <td>386-387</td>\n",
       "      <td>KT/X</td>\n",
       "      <td>aAGACA/a</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669630-7669634</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1299-1303</td>\n",
       "      <td>1157-1161</td>\n",
       "      <td>386-387</td>\n",
       "      <td>KT/X</td>\n",
       "      <td>aAGACA/a</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669629_CTGTCT&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7669630</td>\n",
       "      <td>TGTCT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8912</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669655</td>\n",
       "      <td>CGG</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BDO_01;TYPE=Deletion;DP=4398;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4398:1:4397,1:0.0002:2481,1916:0,1:4365:43...</td>\n",
       "      <td>4365</td>\n",
       "      <td>1</td>\n",
       "      <td>4364</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>4398</td>\n",
       "      <td>1</td>\n",
       "      <td>4397</td>\n",
       "      <td>0.000227</td>\n",
       "      <td>39</td>\n",
       "      <td>0.008856</td>\n",
       "      <td>6790</td>\n",
       "      <td>1</td>\n",
       "      <td>6789</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>52</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>2425</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.642857</td>\n",
       "      <td>0.949924</td>\n",
       "      <td>chr17:7669655_CGG&gt;C</td>\n",
       "      <td>P19_0002_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669656-7669657</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1276-1277</td>\n",
       "      <td>1134-1135</td>\n",
       "      <td>378-379</td>\n",
       "      <td>SR/SX</td>\n",
       "      <td>tcCCgc/tcgc</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669656-7669657</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1276-1277</td>\n",
       "      <td>1134-1135</td>\n",
       "      <td>378-379</td>\n",
       "      <td>SR/SX</td>\n",
       "      <td>tcCCgc/tcgc</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669655_CGG&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7669656</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8913</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669672</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0008_BDO_01;TYPE=Deletion;DP=6204;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6204:1:6203,1:0.0002:2793,3410:0,1:6287:62...</td>\n",
       "      <td>6287</td>\n",
       "      <td>1</td>\n",
       "      <td>6286</td>\n",
       "      <td>0.000159</td>\n",
       "      <td>6204</td>\n",
       "      <td>1</td>\n",
       "      <td>6203</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>34</td>\n",
       "      <td>0.005379</td>\n",
       "      <td>11791</td>\n",
       "      <td>16</td>\n",
       "      <td>11773</td>\n",
       "      <td>0.001357</td>\n",
       "      <td>46</td>\n",
       "      <td>0.003886</td>\n",
       "      <td>5504</td>\n",
       "      <td>15</td>\n",
       "      <td>0.002725</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>8.531253</td>\n",
       "      <td>1.324685</td>\n",
       "      <td>chr17:7669672_CT&gt;C</td>\n",
       "      <td>P19_0008_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669673</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1260</td>\n",
       "      <td>1118</td>\n",
       "      <td>373</td>\n",
       "      <td>K/X</td>\n",
       "      <td>aAg/ag</td>\n",
       "      <td>rs2150988776,COSV52830683</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>chr17:7669673</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>1260</td>\n",
       "      <td>1118</td>\n",
       "      <td>373</td>\n",
       "      <td>K/X</td>\n",
       "      <td>aAg/ag</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669672_CT&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7669673</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8976</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676266</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0035_BDO_01;TYPE=Deletion;DP=6842;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6842:1:6841,1:0.0001:3536,3305:0,1:7003:70...</td>\n",
       "      <td>7003</td>\n",
       "      <td>1</td>\n",
       "      <td>7002</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>6842</td>\n",
       "      <td>1</td>\n",
       "      <td>6841</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>28</td>\n",
       "      <td>0.003982</td>\n",
       "      <td>12582</td>\n",
       "      <td>1</td>\n",
       "      <td>12581</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>42</td>\n",
       "      <td>0.003327</td>\n",
       "      <td>5579</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.556589</td>\n",
       "      <td>0.937931</td>\n",
       "      <td>chr17:7676266_AG&gt;A</td>\n",
       "      <td>P19_0035_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676267</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>244</td>\n",
       "      <td>102</td>\n",
       "      <td>34</td>\n",
       "      <td>P/X</td>\n",
       "      <td>ccC/cc</td>\n",
       "      <td>rs2073488821,COSV52833841</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>chr17:7676267</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>244</td>\n",
       "      <td>102</td>\n",
       "      <td>34</td>\n",
       "      <td>P/X</td>\n",
       "      <td>ccC/cc</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>TP53_34-35_FrameshiftVariant</td>\n",
       "      <td>chr17:7676266_AG&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7676267</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8977</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676397</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0035_BDO_01;TYPE=Deletion;DP=7758;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:7758:1:7757,1:0.0001:3914,3843:0,1:7813:78...</td>\n",
       "      <td>7813</td>\n",
       "      <td>1</td>\n",
       "      <td>7812</td>\n",
       "      <td>0.000128</td>\n",
       "      <td>7758</td>\n",
       "      <td>1</td>\n",
       "      <td>7757</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>79</td>\n",
       "      <td>0.010010</td>\n",
       "      <td>14327</td>\n",
       "      <td>2</td>\n",
       "      <td>14324</td>\n",
       "      <td>0.000140</td>\n",
       "      <td>95</td>\n",
       "      <td>0.006587</td>\n",
       "      <td>6514</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1.090668</td>\n",
       "      <td>1.009777</td>\n",
       "      <td>chr17:7676397_AG&gt;A</td>\n",
       "      <td>P19_0035_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676398</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>222</td>\n",
       "      <td>80</td>\n",
       "      <td>27</td>\n",
       "      <td>P/X</td>\n",
       "      <td>cCt/ct</td>\n",
       "      <td>rs1192416464,COSV53662521</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>chr17:7676398</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>222</td>\n",
       "      <td>80</td>\n",
       "      <td>27</td>\n",
       "      <td>P/X</td>\n",
       "      <td>cCt/ct</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676397_AG&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7676398</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8978</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676550</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BDO_01;TYPE=Deletion;DP=3471;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3471:1:3470,1:0.0003:2318,1152:1,0:3452:34...</td>\n",
       "      <td>3452</td>\n",
       "      <td>1</td>\n",
       "      <td>3451</td>\n",
       "      <td>0.000290</td>\n",
       "      <td>3471</td>\n",
       "      <td>1</td>\n",
       "      <td>3470</td>\n",
       "      <td>0.000288</td>\n",
       "      <td>5</td>\n",
       "      <td>0.001446</td>\n",
       "      <td>5909</td>\n",
       "      <td>1</td>\n",
       "      <td>5908</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>11</td>\n",
       "      <td>0.001858</td>\n",
       "      <td>2457</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.584194</td>\n",
       "      <td>0.938104</td>\n",
       "      <td>chr17:7676550_AC&gt;A</td>\n",
       "      <td>P19_0002_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676551</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>186</td>\n",
       "      <td>44</td>\n",
       "      <td>15</td>\n",
       "      <td>S/X</td>\n",
       "      <td>aGt/at</td>\n",
       "      <td>rs2151047607</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676551</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>186</td>\n",
       "      <td>44</td>\n",
       "      <td>15</td>\n",
       "      <td>S/X</td>\n",
       "      <td>aGt/at</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676550_AC&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7676551</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8979</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676553</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BTR_01;TYPE=Deletion;DP=3250;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:3250:1:3249,1:0.0003:2007,1242:1,0:3261:32...</td>\n",
       "      <td>3261</td>\n",
       "      <td>1</td>\n",
       "      <td>3260</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>3250</td>\n",
       "      <td>1</td>\n",
       "      <td>3249</td>\n",
       "      <td>0.000308</td>\n",
       "      <td>6</td>\n",
       "      <td>0.001837</td>\n",
       "      <td>6652</td>\n",
       "      <td>1</td>\n",
       "      <td>6651</td>\n",
       "      <td>0.000150</td>\n",
       "      <td>9</td>\n",
       "      <td>0.001351</td>\n",
       "      <td>3391</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.490229</td>\n",
       "      <td>0.919015</td>\n",
       "      <td>chr17:7676553_CAG&gt;C</td>\n",
       "      <td>P19_0002_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676554-7676555</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>182-183</td>\n",
       "      <td>40-41</td>\n",
       "      <td>14</td>\n",
       "      <td>L/X</td>\n",
       "      <td>CTg/g</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676554-7676555</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>182-183</td>\n",
       "      <td>40-41</td>\n",
       "      <td>14</td>\n",
       "      <td>L/X</td>\n",
       "      <td>CTg/g</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676553_CAG&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7676554</td>\n",
       "      <td>AG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8980</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676568</td>\n",
       "      <td>GCTAGGATCTGA</td>\n",
       "      <td>G</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0012_BTR_01;TYPE=Deletion;DP=6936;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6936:1:6935,1:0.0001:4076,2859:0,1:6860:68...</td>\n",
       "      <td>6860</td>\n",
       "      <td>1</td>\n",
       "      <td>6859</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>6936</td>\n",
       "      <td>1</td>\n",
       "      <td>6935</td>\n",
       "      <td>0.000144</td>\n",
       "      <td>47</td>\n",
       "      <td>0.006805</td>\n",
       "      <td>17833</td>\n",
       "      <td>1</td>\n",
       "      <td>17832</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>77</td>\n",
       "      <td>0.004299</td>\n",
       "      <td>10973</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.384680</td>\n",
       "      <td>0.902405</td>\n",
       "      <td>chr17:7676568_GCTAGGATCTGA&gt;G</td>\n",
       "      <td>P19_0012_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676569-7676579</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>158-168</td>\n",
       "      <td>16-26</td>\n",
       "      <td>6-9</td>\n",
       "      <td>SDPS/X</td>\n",
       "      <td>TCAGATCCTAGc/c</td>\n",
       "      <td>-</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676569-7676579</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>158-168</td>\n",
       "      <td>16-26</td>\n",
       "      <td>6-9</td>\n",
       "      <td>SDPS/X</td>\n",
       "      <td>TCAGATCCTAGc/c</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676568_GCTAGGATCTGA&gt;G</td>\n",
       "      <td>17</td>\n",
       "      <td>7676569</td>\n",
       "      <td>CTAGGATCTGA</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows  149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM      POS           REF ALT    FILTER  \\\n",
       "8909  chr17  7669611         AGTCT   A      pSTD   \n",
       "8910  chr17  7669612           GTC   G      pSTD   \n",
       "8911  chr17  7669629        CTGTCT   C      pSTD   \n",
       "8912  chr17  7669655           CGG   C      pSTD   \n",
       "8913  chr17  7669672            CT   C      pSTD   \n",
       "...     ...      ...           ...  ..       ...   \n",
       "8976  chr17  7676266            AG   A      pSTD   \n",
       "8977  chr17  7676397            AG   A      pSTD   \n",
       "8978  chr17  7676550            AC   A  p10;pSTD   \n",
       "8979  chr17  7676553           CAG   C      pSTD   \n",
       "8980  chr17  7676568  GCTAGGATCTGA   G      pSTD   \n",
       "\n",
       "                                                   INFO  \\\n",
       "8909  SAMPLE=P19_0009_BDO_01;TYPE=Deletion;DP=3873;V...   \n",
       "8910  SAMPLE=P19_0048_BTR_01;TYPE=Deletion;DP=4416;V...   \n",
       "8911  SAMPLE=P19_0051_BTR_01;TYPE=Deletion;DP=10669;...   \n",
       "8912  SAMPLE=P19_0002_BDO_01;TYPE=Deletion;DP=4398;V...   \n",
       "8913  SAMPLE=P19_0008_BDO_01;TYPE=Deletion;DP=6204;V...   \n",
       "...                                                 ...   \n",
       "8976  SAMPLE=P19_0035_BDO_01;TYPE=Deletion;DP=6842;V...   \n",
       "8977  SAMPLE=P19_0035_BDO_01;TYPE=Deletion;DP=7758;V...   \n",
       "8978  SAMPLE=P19_0002_BDO_01;TYPE=Deletion;DP=3471;V...   \n",
       "8979  SAMPLE=P19_0002_BTR_01;TYPE=Deletion;DP=3250;V...   \n",
       "8980  SAMPLE=P19_0012_BTR_01;TYPE=Deletion;DP=6936;V...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "8909  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8910  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8911  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8912  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8913  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "...                                                 ...   \n",
       "8976  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8977  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8978  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8979  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "8980  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                 SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "8909  0/1:3873:1:3872,1:0.0003:2741,1131:1,0:3829:38...   3829          1   \n",
       "8910  0/1:4416:1:4415,1:0.0002:2387,2028:1,0:4434:44...   4434          1   \n",
       "8911  0/1:10669:1:10668,1:0.0001:5456,5212:1,0:10646...  10646          1   \n",
       "8912  0/1:4398:1:4397,1:0.0002:2481,1916:0,1:4365:43...   4365          1   \n",
       "8913  0/1:6204:1:6203,1:0.0002:2793,3410:0,1:6287:62...   6287          1   \n",
       "...                                                 ...    ...        ...   \n",
       "8976  0/1:6842:1:6841,1:0.0001:3536,3305:0,1:7003:70...   7003          1   \n",
       "8977  0/1:7758:1:7757,1:0.0001:3914,3843:0,1:7813:78...   7813          1   \n",
       "8978  0/1:3471:1:3470,1:0.0003:2318,1152:1,0:3452:34...   3452          1   \n",
       "8979  0/1:3250:1:3249,1:0.0003:2007,1242:1,0:3261:32...   3261          1   \n",
       "8980  0/1:6936:1:6935,1:0.0001:4076,2859:0,1:6860:68...   6860          1   \n",
       "\n",
       "      REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "8909       3828  0.000261      3873             1          3872  0.000258   \n",
       "8910       4433  0.000226      4416             1          4415  0.000226   \n",
       "8911      10645  0.000094     10669             1         10668  0.000094   \n",
       "8912       4364  0.000229      4398             1          4397  0.000227   \n",
       "8913       6286  0.000159      6204             1          6203  0.000161   \n",
       "...         ...       ...       ...           ...           ...       ...   \n",
       "8976       7002  0.000143      6842             1          6841  0.000146   \n",
       "8977       7812  0.000128      7758             1          7757  0.000129   \n",
       "8978       3451  0.000290      3471             1          3470  0.000288   \n",
       "8979       3260  0.000307      3250             1          3249  0.000308   \n",
       "8980       6859  0.000146      6936             1          6935  0.000144   \n",
       "\n",
       "      numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "8909     24  0.006229      7842             1          7841  0.000128   \n",
       "8910     13  0.002923      6854             1          6853  0.000146   \n",
       "8911     35  0.003277     22524             1         22523  0.000044   \n",
       "8912     39  0.008856      6790             1          6789  0.000147   \n",
       "8913     34  0.005379     11791            16         11773  0.001357   \n",
       "...     ...       ...       ...           ...           ...       ...   \n",
       "8976     28  0.003982     12582             1         12581  0.000079   \n",
       "8977     79  0.010010     14327             2         14324  0.000140   \n",
       "8978      5  0.001446      5909             1          5908  0.000169   \n",
       "8979      6  0.001837      6652             1          6651  0.000150   \n",
       "8980     47  0.006805     17833             1         17832  0.000056   \n",
       "\n",
       "      numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND    VAF_ND  \\\n",
       "8909        27   0.003431      4013             0  0.000000   \n",
       "8910         8   0.001166      2420             0  0.000000   \n",
       "8911        52   0.002303     11878             0  0.000000   \n",
       "8912        52   0.007600      2425             0  0.000000   \n",
       "8913        46   0.003886      5504            15  0.002725   \n",
       "...        ...        ...       ...           ...       ...   \n",
       "8976        42   0.003327      5579             0  0.000000   \n",
       "8977        95   0.006587      6514             1  0.000154   \n",
       "8978        11   0.001858      2457             0  0.000000   \n",
       "8979         9   0.001351      3391             0  0.000000   \n",
       "8980        77   0.004299     10973             0  0.000000   \n",
       "\n",
       "      VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "8909                   False                      False        0.488268   \n",
       "8910                   False                      False        0.646922   \n",
       "8911                   False                      False        0.472651   \n",
       "8912                   False                      False        0.642857   \n",
       "8913                    True                      False        8.531253   \n",
       "...                      ...                        ...             ...   \n",
       "8976                   False                      False        0.556589   \n",
       "8977                   False                      False        1.090668   \n",
       "8978                   False                      False        0.584194   \n",
       "8979                   False                      False        0.490229   \n",
       "8980                   False                      False        0.384680   \n",
       "\n",
       "      VAF_distortion_sq                        MUT_ID        SAMPLE_ID METHOD  \\\n",
       "8909           0.920055         chr17:7669611_AGTCT>A  P19_0009_BDO_01    med   \n",
       "8910           0.950691           chr17:7669612_GTC>G  P19_0048_BTR_01    med   \n",
       "8911           0.925227        chr17:7669629_CTGTCT>C  P19_0051_BTR_01    med   \n",
       "8912           0.949924           chr17:7669655_CGG>C  P19_0002_BDO_01    med   \n",
       "8913           1.324685            chr17:7669672_CT>C  P19_0008_BDO_01    med   \n",
       "...                 ...                           ...              ...    ...   \n",
       "8976           0.937931            chr17:7676266_AG>A  P19_0035_BDO_01    med   \n",
       "8977           1.009777            chr17:7676397_AG>A  P19_0035_BDO_01    med   \n",
       "8978           0.938104            chr17:7676550_AC>A  P19_0002_BDO_01    med   \n",
       "8979           0.919015           chr17:7676553_CAG>C  P19_0002_BTR_01    med   \n",
       "8980           0.902405  chr17:7676568_GCTAGGATCTGA>G  P19_0012_BTR_01    med   \n",
       "\n",
       "      PROJECT_NAME               Location Allele             Gene  \\\n",
       "8909           NaN  chr17:7669612-7669615      -  ENSG00000141510   \n",
       "8910           NaN  chr17:7669613-7669614      -  ENSG00000141510   \n",
       "8911           NaN  chr17:7669630-7669634      -  ENSG00000141510   \n",
       "8912           NaN  chr17:7669656-7669657      -  ENSG00000141510   \n",
       "8913           NaN          chr17:7669673      -  ENSG00000141510   \n",
       "...            ...                    ...    ...              ...   \n",
       "8976           NaN          chr17:7676267      -  ENSG00000141510   \n",
       "8977           NaN          chr17:7676398      -  ENSG00000141510   \n",
       "8978           NaN          chr17:7676551      -  ENSG00000141510   \n",
       "8979           NaN  chr17:7676554-7676555      -  ENSG00000141510   \n",
       "8980           NaN  chr17:7676569-7676579      -  ENSG00000141510   \n",
       "\n",
       "              Feature Feature_type         Consequence cDNA_position  \\\n",
       "8909  ENST00000269305   Transcript  frameshift_variant     1318-1321   \n",
       "8910  ENST00000269305   Transcript  frameshift_variant     1319-1320   \n",
       "8911  ENST00000269305   Transcript  frameshift_variant     1299-1303   \n",
       "8912  ENST00000269305   Transcript  frameshift_variant     1276-1277   \n",
       "8913  ENST00000269305   Transcript  frameshift_variant          1260   \n",
       "...               ...          ...                 ...           ...   \n",
       "8976  ENST00000269305   Transcript  frameshift_variant           244   \n",
       "8977  ENST00000269305   Transcript  frameshift_variant           222   \n",
       "8978  ENST00000269305   Transcript  frameshift_variant           186   \n",
       "8979  ENST00000269305   Transcript  frameshift_variant       182-183   \n",
       "8980  ENST00000269305   Transcript  frameshift_variant       158-168   \n",
       "\n",
       "     CDS_position Protein_position Amino_acids          Codons  \\\n",
       "8909    1176-1179          392-393        SD/X       tcAGAC/tc   \n",
       "8910    1177-1178              393         D/X           GAc/c   \n",
       "8911    1157-1161          386-387        KT/X        aAGACA/a   \n",
       "8912    1134-1135          378-379       SR/SX     tcCCgc/tcgc   \n",
       "8913         1118              373         K/X          aAg/ag   \n",
       "...           ...              ...         ...             ...   \n",
       "8976          102               34         P/X          ccC/cc   \n",
       "8977           80               27         P/X          cCt/ct   \n",
       "8978           44               15         S/X          aGt/at   \n",
       "8979        40-41               14         L/X           CTg/g   \n",
       "8980        16-26              6-9      SDPS/X  TCAGATCCTAGc/c   \n",
       "\n",
       "             Existing_variation IMPACT DISTANCE STRAND FLAGS SYMBOL  \\\n",
       "8909                          -   HIGH        -     -1     -   TP53   \n",
       "8910                          -   HIGH        -     -1     -   TP53   \n",
       "8911                          -   HIGH        -     -1     -   TP53   \n",
       "8912                          -   HIGH        -     -1     -   TP53   \n",
       "8913  rs2150988776,COSV52830683   HIGH        -     -1     -   TP53   \n",
       "...                         ...    ...      ...    ...   ...    ...   \n",
       "8976  rs2073488821,COSV52833841   HIGH        -     -1     -   TP53   \n",
       "8977  rs1192416464,COSV53662521   HIGH        -     -1     -   TP53   \n",
       "8978               rs2151047607   HIGH        -     -1     -   TP53   \n",
       "8979                          -   HIGH        -     -1     -   TP53   \n",
       "8980                          -   HIGH        -     -1     -   TP53   \n",
       "\n",
       "     SYMBOL_SOURCE     HGNC_ID CANONICAL             ENSP  gnomADe_AF  \\\n",
       "8909          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8910          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8911          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8912          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8913          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "...            ...         ...       ...              ...         ...   \n",
       "8976          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8977          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8978          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8979          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "8980          HGNC  HGNC:11998       YES  ENSP00000269305         0.0   \n",
       "\n",
       "      gnomADe_AFR_AF  gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  \\\n",
       "8909             0.0             0.0             0.0             0.0   \n",
       "8910             0.0             0.0             0.0             0.0   \n",
       "8911             0.0             0.0             0.0             0.0   \n",
       "8912             0.0             0.0             0.0             0.0   \n",
       "8913             0.0             0.0             0.0             0.0   \n",
       "...              ...             ...             ...             ...   \n",
       "8976             0.0             0.0             0.0             0.0   \n",
       "8977             0.0             0.0             0.0             0.0   \n",
       "8978             0.0             0.0             0.0             0.0   \n",
       "8979             0.0             0.0             0.0             0.0   \n",
       "8980             0.0             0.0             0.0             0.0   \n",
       "\n",
       "      gnomADe_FIN_AF  gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  \\\n",
       "8909             0.0             0.0             0.0             0.0   \n",
       "8910             0.0             0.0             0.0             0.0   \n",
       "8911             0.0             0.0             0.0             0.0   \n",
       "8912             0.0             0.0             0.0             0.0   \n",
       "8913             0.0             0.0             0.0             0.0   \n",
       "...              ...             ...             ...             ...   \n",
       "8976             0.0             0.0             0.0             0.0   \n",
       "8977             0.0             0.0             0.0             0.0   \n",
       "8978             0.0             0.0             0.0             0.0   \n",
       "8979             0.0             0.0             0.0             0.0   \n",
       "8980             0.0             0.0             0.0             0.0   \n",
       "\n",
       "      gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  \\\n",
       "8909         0.0             0.0             0.0             0.0   \n",
       "8910         0.0             0.0             0.0             0.0   \n",
       "8911         0.0             0.0             0.0             0.0   \n",
       "8912         0.0             0.0             0.0             0.0   \n",
       "8913         0.0             0.0             0.0             0.0   \n",
       "...          ...             ...             ...             ...   \n",
       "8976         0.0             0.0             0.0             0.0   \n",
       "8977         0.0             0.0             0.0             0.0   \n",
       "8978         0.0             0.0             0.0             0.0   \n",
       "8979         0.0             0.0             0.0             0.0   \n",
       "8980         0.0             0.0             0.0             0.0   \n",
       "\n",
       "      gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  \\\n",
       "8909             0.0             0.0             0.0             0.0   \n",
       "8910             0.0             0.0             0.0             0.0   \n",
       "8911             0.0             0.0             0.0             0.0   \n",
       "8912             0.0             0.0             0.0             0.0   \n",
       "8913             0.0             0.0             0.0             0.0   \n",
       "...              ...             ...             ...             ...   \n",
       "8976             0.0             0.0             0.0             0.0   \n",
       "8977             0.0             0.0             0.0             0.0   \n",
       "8978             0.0             0.0             0.0             0.0   \n",
       "8979             0.0             0.0             0.0             0.0   \n",
       "8980             0.0             0.0             0.0             0.0   \n",
       "\n",
       "      gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  \\\n",
       "8909             0.0             0.0             0.0   \n",
       "8910             0.0             0.0             0.0   \n",
       "8911             0.0             0.0             0.0   \n",
       "8912             0.0             0.0             0.0   \n",
       "8913             0.0             0.0             0.0   \n",
       "...              ...             ...             ...   \n",
       "8976             0.0             0.0             0.0   \n",
       "8977             0.0             0.0             0.0   \n",
       "8978             0.0             0.0             0.0   \n",
       "8979             0.0             0.0             0.0   \n",
       "8980             0.0             0.0             0.0   \n",
       "\n",
       "                               CLIN_SIG SOMATIC PHENO     canonical_Location  \\\n",
       "8909                                  -       -     -  chr17:7669612-7669615   \n",
       "8910                                  -       -     -  chr17:7669613-7669614   \n",
       "8911                                  -       -     -  chr17:7669630-7669634   \n",
       "8912                                  -       -     -  chr17:7669656-7669657   \n",
       "8913  uncertain_significance,pathogenic     0,1   1,1          chr17:7669673   \n",
       "...                                 ...     ...   ...                    ...   \n",
       "8976                         pathogenic     0,1   1,1          chr17:7676267   \n",
       "8977                         pathogenic     0,1   1,1          chr17:7676398   \n",
       "8978                                  -       -     -          chr17:7676551   \n",
       "8979                                  -       -     -  chr17:7676554-7676555   \n",
       "8980                                  -       -     -  chr17:7676569-7676579   \n",
       "\n",
       "     canonical_Allele   canonical_Gene canonical_Feature  \\\n",
       "8909                -  ENSG00000141510   ENST00000269305   \n",
       "8910                -  ENSG00000141510   ENST00000269305   \n",
       "8911                -  ENSG00000141510   ENST00000269305   \n",
       "8912                -  ENSG00000141510   ENST00000269305   \n",
       "8913                -  ENSG00000141510   ENST00000269305   \n",
       "...               ...              ...               ...   \n",
       "8976                -  ENSG00000141510   ENST00000269305   \n",
       "8977                -  ENSG00000141510   ENST00000269305   \n",
       "8978                -  ENSG00000141510   ENST00000269305   \n",
       "8979                -  ENSG00000141510   ENST00000269305   \n",
       "8980                -  ENSG00000141510   ENST00000269305   \n",
       "\n",
       "     canonical_Feature_type canonical_Consequence canonical_cDNA_position  \\\n",
       "8909             Transcript    frameshift_variant               1318-1321   \n",
       "8910             Transcript    frameshift_variant               1319-1320   \n",
       "8911             Transcript    frameshift_variant               1299-1303   \n",
       "8912             Transcript    frameshift_variant               1276-1277   \n",
       "8913             Transcript    frameshift_variant                    1260   \n",
       "...                     ...                   ...                     ...   \n",
       "8976             Transcript    frameshift_variant                     244   \n",
       "8977             Transcript    frameshift_variant                     222   \n",
       "8978             Transcript    frameshift_variant                     186   \n",
       "8979             Transcript    frameshift_variant                 182-183   \n",
       "8980             Transcript    frameshift_variant                 158-168   \n",
       "\n",
       "     canonical_CDS_position canonical_Protein_position canonical_Amino_acids  \\\n",
       "8909              1176-1179                    392-393                  SD/X   \n",
       "8910              1177-1178                        393                   D/X   \n",
       "8911              1157-1161                    386-387                  KT/X   \n",
       "8912              1134-1135                    378-379                 SR/SX   \n",
       "8913                   1118                        373                   K/X   \n",
       "...                     ...                        ...                   ...   \n",
       "8976                    102                         34                   P/X   \n",
       "8977                     80                         27                   P/X   \n",
       "8978                     44                         15                   S/X   \n",
       "8979                  40-41                         14                   L/X   \n",
       "8980                  16-26                        6-9                SDPS/X   \n",
       "\n",
       "     canonical_Codons canonical_IMPACT canonical_DISTANCE canonical_STRAND  \\\n",
       "8909        tcAGAC/tc             HIGH                  -             -1.0   \n",
       "8910            GAc/c             HIGH                  -             -1.0   \n",
       "8911         aAGACA/a             HIGH                  -             -1.0   \n",
       "8912      tcCCgc/tcgc             HIGH                  -             -1.0   \n",
       "8913           aAg/ag             HIGH                  -             -1.0   \n",
       "...               ...              ...                ...              ...   \n",
       "8976           ccC/cc             HIGH                  -             -1.0   \n",
       "8977           cCt/ct             HIGH                  -             -1.0   \n",
       "8978           aGt/at             HIGH                  -             -1.0   \n",
       "8979            CTg/g             HIGH                  -             -1.0   \n",
       "8980   TCAGATCCTAGc/c             HIGH                  -             -1.0   \n",
       "\n",
       "     canonical_FLAGS canonical_SYMBOL canonical_SYMBOL_SOURCE  \\\n",
       "8909               -             TP53                    HGNC   \n",
       "8910               -             TP53                    HGNC   \n",
       "8911               -             TP53                    HGNC   \n",
       "8912               -             TP53                    HGNC   \n",
       "8913               -             TP53                    HGNC   \n",
       "...              ...              ...                     ...   \n",
       "8976               -             TP53                    HGNC   \n",
       "8977               -             TP53                    HGNC   \n",
       "8978               -             TP53                    HGNC   \n",
       "8979               -             TP53                    HGNC   \n",
       "8980               -             TP53                    HGNC   \n",
       "\n",
       "     canonical_HGNC_ID canonical_CANONICAL   canonical_ENSP  \\\n",
       "8909        HGNC:11998                 YES  ENSP00000269305   \n",
       "8910        HGNC:11998                 YES  ENSP00000269305   \n",
       "8911        HGNC:11998                 YES  ENSP00000269305   \n",
       "8912        HGNC:11998                 YES  ENSP00000269305   \n",
       "8913        HGNC:11998                 YES  ENSP00000269305   \n",
       "...                ...                 ...              ...   \n",
       "8976        HGNC:11998                 YES  ENSP00000269305   \n",
       "8977        HGNC:11998                 YES  ENSP00000269305   \n",
       "8978        HGNC:11998                 YES  ENSP00000269305   \n",
       "8979        HGNC:11998                 YES  ENSP00000269305   \n",
       "8980        HGNC:11998                 YES  ENSP00000269305   \n",
       "\n",
       "                     canonical_CLIN_SIG canonical_Consequence_single  \\\n",
       "8909                                  -           frameshift_variant   \n",
       "8910                                  -           frameshift_variant   \n",
       "8911                                  -           frameshift_variant   \n",
       "8912                                  -           frameshift_variant   \n",
       "8913  uncertain_significance,pathogenic           frameshift_variant   \n",
       "...                                 ...                          ...   \n",
       "8976                         pathogenic           frameshift_variant   \n",
       "8977                         pathogenic           frameshift_variant   \n",
       "8978                                  -           frameshift_variant   \n",
       "8979                                  -           frameshift_variant   \n",
       "8980                                  -           frameshift_variant   \n",
       "\n",
       "     canonical_Consequence_broader canonical_Protein_affecting  \\\n",
       "8909                      nonsense           protein_affecting   \n",
       "8910                      nonsense           protein_affecting   \n",
       "8911                      nonsense           protein_affecting   \n",
       "8912                      nonsense           protein_affecting   \n",
       "8913                      nonsense           protein_affecting   \n",
       "...                            ...                         ...   \n",
       "8976                      nonsense           protein_affecting   \n",
       "8977                      nonsense           protein_affecting   \n",
       "8978                      nonsense           protein_affecting   \n",
       "8979                      nonsense           protein_affecting   \n",
       "8980                      nonsense           protein_affecting   \n",
       "\n",
       "      Consequence_single Consequence_broader  Protein_affecting CONTEXT_MUT  \\\n",
       "8909  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8910  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8911  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8912  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8913  frameshift_variant            nonsense  protein_affecting           -   \n",
       "...                  ...                 ...                ...         ...   \n",
       "8976  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8977  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8978  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8979  frameshift_variant            nonsense  protein_affecting           -   \n",
       "8980  frameshift_variant            nonsense  protein_affecting           -   \n",
       "\n",
       "     MUTTYPE CONTEXT_MUT_SIGPRO  gnomAD_SNP                  HOTSPOT_NAME  \\\n",
       "8909       -                  -       False                             -   \n",
       "8910       -                  -       False                             -   \n",
       "8911       -                  -       False                             -   \n",
       "8912       -                  -       False                             -   \n",
       "8913       -                  -       False                             -   \n",
       "...      ...                ...         ...                           ...   \n",
       "8976       -                  -       False  TP53_34-35_FrameshiftVariant   \n",
       "8977       -                  -       False                             -   \n",
       "8978       -                  -       False                             -   \n",
       "8979       -                  -       False                             -   \n",
       "8980       -                  -       False                             -   \n",
       "\n",
       "                        MUT_ID_pyr CHROM_ensembl  POS_ensembl  REF_ensembl  \\\n",
       "8909         chr17:7669611_AGTCT>A            17      7669612         GTCT   \n",
       "8910           chr17:7669612_GTC>G            17      7669613           TC   \n",
       "8911        chr17:7669629_CTGTCT>C            17      7669630        TGTCT   \n",
       "8912           chr17:7669655_CGG>C            17      7669656           GG   \n",
       "8913            chr17:7669672_CT>C            17      7669673            T   \n",
       "...                            ...           ...          ...          ...   \n",
       "8976            chr17:7676266_AG>A            17      7676267            G   \n",
       "8977            chr17:7676397_AG>A            17      7676398            G   \n",
       "8978            chr17:7676550_AC>A            17      7676551            C   \n",
       "8979           chr17:7676553_CAG>C            17      7676554           AG   \n",
       "8980  chr17:7676568_GCTAGGATCTGA>G            17      7676569  CTAGGATCTGA   \n",
       "\n",
       "     ALT_ensembl      TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  \\\n",
       "8909           -  DELETION             4          False            False   \n",
       "8910           -  DELETION             2          False            False   \n",
       "8911           -  DELETION             5          False            False   \n",
       "8912           -  DELETION             2          False            False   \n",
       "8913           -  DELETION             1          False            False   \n",
       "...          ...       ...           ...            ...              ...   \n",
       "8976           -  DELETION             1          False            False   \n",
       "8977           -  DELETION             1          False            False   \n",
       "8978           -  DELETION             1          False            False   \n",
       "8979           -  DELETION             2          False            False   \n",
       "8980           -  DELETION            11          False            False   \n",
       "\n",
       "      N_rich_frequency  VAF_Ns_threshold  FILTER.cohort_n_rich_uni  \\\n",
       "8909               0.0               1.1                     False   \n",
       "8910               0.0               1.1                     False   \n",
       "8911               0.0               1.1                     False   \n",
       "8912               0.0               1.1                     False   \n",
       "8913               0.0               1.1                     False   \n",
       "...                ...               ...                       ...   \n",
       "8976               0.0               1.1                     False   \n",
       "8977               0.0               1.1                     False   \n",
       "8978               0.0               1.1                     False   \n",
       "8979               0.0               1.1                     False   \n",
       "8980               0.0               1.1                     False   \n",
       "\n",
       "      FILTER.gnomAD_SNP  FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  \\\n",
       "8909              False               False                False       False   \n",
       "8910              False               False                False       False   \n",
       "8911              False               False                False       False   \n",
       "8912              False               False                False       False   \n",
       "8913              False               False                False       False   \n",
       "...                 ...                 ...                  ...         ...   \n",
       "8976              False               False                False       False   \n",
       "8977              False               False                False       False   \n",
       "8978              False               False                False        True   \n",
       "8979              False               False                False       False   \n",
       "8980              False               False                False       False   \n",
       "\n",
       "      FILTER.pSTD  FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "8909         True                      False                 False   \n",
       "8910         True                      False                 False   \n",
       "8911         True                      False                 False   \n",
       "8912         True                      False                 False   \n",
       "8913         True                      False                 False   \n",
       "...           ...                        ...                   ...   \n",
       "8976         True                      False                 False   \n",
       "8977         True                      False                 False   \n",
       "8978         True                      False                 False   \n",
       "8979         True                      False                 False   \n",
       "8980         True                      False                 False   \n",
       "\n",
       "      FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "8909                          False           False   \n",
       "8910                          False           False   \n",
       "8911                          False           False   \n",
       "8912                          False           False   \n",
       "8913                          False           False   \n",
       "...                             ...             ...   \n",
       "8976                          False           False   \n",
       "8977                          False           False   \n",
       "8978                          False           False   \n",
       "8979                          False           False   \n",
       "8980                          False           False   \n",
       "\n",
       "      FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "8909                           False                        False   \n",
       "8910                           False                        False   \n",
       "8911                           False                        False   \n",
       "8912                           False                        False   \n",
       "8913                           False                        False   \n",
       "...                              ...                          ...   \n",
       "8976                           False                        False   \n",
       "8977                           False                        False   \n",
       "8978                           False                        False   \n",
       "8979                           False                        False   \n",
       "8980                           False                        False   \n",
       "\n",
       "      FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "8909                     False          False                    False   \n",
       "8910                     False          False                    False   \n",
       "8911                     False          False                    False   \n",
       "8912                     False          False                    False   \n",
       "8913                     False          False                    False   \n",
       "...                        ...            ...                      ...   \n",
       "8976                     False          False                    False   \n",
       "8977                     False          False                    False   \n",
       "8978                     False          False                    False   \n",
       "8979                     False          False                    False   \n",
       "8980                     False          False                    False   \n",
       "\n",
       "      FILTER.PASS  FILTER.AM_not_searched_COMPLEX  \\\n",
       "8909        False                           False   \n",
       "8910        False                           False   \n",
       "8911        False                           False   \n",
       "8912        False                           False   \n",
       "8913        False                           False   \n",
       "...           ...                             ...   \n",
       "8976        False                           False   \n",
       "8977        False                           False   \n",
       "8978        False                           False   \n",
       "8979        False                           False   \n",
       "8980        False                           False   \n",
       "\n",
       "      FILTER.not_searched_COMPLEX  FILTER.MSI12  FILTER.AM_not_searched_SV  \\\n",
       "8909                        False         False                      False   \n",
       "8910                        False         False                      False   \n",
       "8911                        False         False                      False   \n",
       "8912                        False         False                      False   \n",
       "8913                        False         False                      False   \n",
       "...                           ...           ...                        ...   \n",
       "8976                        False         False                      False   \n",
       "8977                        False         False                      False   \n",
       "8978                        False         False                      False   \n",
       "8979                        False         False                      False   \n",
       "8980                        False         False                      False   \n",
       "\n",
       "      FILTER.not_searched_SV  FILTER.SN1.5  FILTER.q22.5  FILTER.v1  \\\n",
       "8909                   False         False         False      False   \n",
       "8910                   False         False         False      False   \n",
       "8911                   False         False         False      False   \n",
       "8912                   False         False         False      False   \n",
       "8913                   False         False         False      False   \n",
       "...                      ...           ...           ...        ...   \n",
       "8976                   False         False         False      False   \n",
       "8977                   False         False         False      False   \n",
       "8978                   False         False         False      False   \n",
       "8979                   False         False         False      False   \n",
       "8980                   False         False         False      False   \n",
       "\n",
       "      FILTER.low_mappability  FILTER.Bias  FILTER.d3  \n",
       "8909                   False        False      False  \n",
       "8910                   False        False      False  \n",
       "8911                   False        False      False  \n",
       "8912                   False        False      False  \n",
       "8913                   False        False      False  \n",
       "...                      ...          ...        ...  \n",
       "8976                   False        False      False  \n",
       "8977                   False        False      False  \n",
       "8978                   False        False      False  \n",
       "8979                   False        False      False  \n",
       "8980                   False        False      False  \n",
       "\n",
       "[72 rows x 149 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "frameshift_indels = indel_maf_df[(~indel_maf_df[\"INDEL_INFRAME\"]) &\n",
    "                                 (indel_maf_df[\"canonical_CDS_position\"] != '-' )].reset_index(drop = True)\n",
    "frameshift_indels[frameshift_indels[\"canonical_SYMBOL\"] == \"TP53\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a7be7d9-c2a0-4dc9-b86d-472228a25a41",
   "metadata": {},
   "source": [
    "___"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5da0879a-9131-49ad-902e-339266e7549d",
   "metadata": {},
   "source": [
    "Inframe indels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "65192cfc-6fbf-439d-a2c5-5c5941872eca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>ALT_DEPTH</th>\n",
       "      <th>REF_DEPTH</th>\n",
       "      <th>VAF</th>\n",
       "      <th>vd_DEPTH</th>\n",
       "      <th>vd_ALT_DEPTH</th>\n",
       "      <th>vd_REF_DEPTH</th>\n",
       "      <th>vd_VAF</th>\n",
       "      <th>numNs</th>\n",
       "      <th>VAF_Ns</th>\n",
       "      <th>DEPTH_AM</th>\n",
       "      <th>ALT_DEPTH_AM</th>\n",
       "      <th>REF_DEPTH_AM</th>\n",
       "      <th>VAF_AM</th>\n",
       "      <th>numNs_AM</th>\n",
       "      <th>VAF_Ns_AM</th>\n",
       "      <th>DEPTH_ND</th>\n",
       "      <th>ALT_DEPTH_ND</th>\n",
       "      <th>VAF_ND</th>\n",
       "      <th>VAF_distorted_expanded</th>\n",
       "      <th>VAF_distorted_expanded_sq</th>\n",
       "      <th>VAF_distortion</th>\n",
       "      <th>VAF_distortion_sq</th>\n",
       "      <th>MUT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>PROJECT_NAME</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>gnomADe_AF</th>\n",
       "      <th>gnomADe_AFR_AF</th>\n",
       "      <th>gnomADe_AMR_AF</th>\n",
       "      <th>gnomADe_ASJ_AF</th>\n",
       "      <th>gnomADe_EAS_AF</th>\n",
       "      <th>gnomADe_FIN_AF</th>\n",
       "      <th>gnomADe_NFE_AF</th>\n",
       "      <th>gnomADe_OTH_AF</th>\n",
       "      <th>gnomADe_SAS_AF</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>canonical_Location</th>\n",
       "      <th>canonical_Allele</th>\n",
       "      <th>canonical_Gene</th>\n",
       "      <th>canonical_Feature</th>\n",
       "      <th>canonical_Feature_type</th>\n",
       "      <th>canonical_Consequence</th>\n",
       "      <th>canonical_cDNA_position</th>\n",
       "      <th>canonical_CDS_position</th>\n",
       "      <th>canonical_Protein_position</th>\n",
       "      <th>canonical_Amino_acids</th>\n",
       "      <th>canonical_Codons</th>\n",
       "      <th>canonical_IMPACT</th>\n",
       "      <th>canonical_DISTANCE</th>\n",
       "      <th>canonical_STRAND</th>\n",
       "      <th>canonical_FLAGS</th>\n",
       "      <th>canonical_SYMBOL</th>\n",
       "      <th>canonical_SYMBOL_SOURCE</th>\n",
       "      <th>canonical_HGNC_ID</th>\n",
       "      <th>canonical_CANONICAL</th>\n",
       "      <th>canonical_ENSP</th>\n",
       "      <th>canonical_CLIN_SIG</th>\n",
       "      <th>canonical_Consequence_single</th>\n",
       "      <th>canonical_Consequence_broader</th>\n",
       "      <th>canonical_Protein_affecting</th>\n",
       "      <th>Consequence_single</th>\n",
       "      <th>Consequence_broader</th>\n",
       "      <th>Protein_affecting</th>\n",
       "      <th>CONTEXT_MUT</th>\n",
       "      <th>MUTTYPE</th>\n",
       "      <th>CONTEXT_MUT_SIGPRO</th>\n",
       "      <th>gnomAD_SNP</th>\n",
       "      <th>HOTSPOT_NAME</th>\n",
       "      <th>MUT_ID_pyr</th>\n",
       "      <th>CHROM_ensembl</th>\n",
       "      <th>POS_ensembl</th>\n",
       "      <th>REF_ensembl</th>\n",
       "      <th>ALT_ensembl</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>INDEL_LENGTH</th>\n",
       "      <th>INDEL_INFRAME</th>\n",
       "      <th>INDEL_MULTIPLE3</th>\n",
       "      <th>N_rich_frequency</th>\n",
       "      <th>VAF_Ns_threshold</th>\n",
       "      <th>FILTER.cohort_n_rich_uni</th>\n",
       "      <th>FILTER.gnomAD_SNP</th>\n",
       "      <th>FILTER.not_covered</th>\n",
       "      <th>FILTER.not_in_exons</th>\n",
       "      <th>FILTER.p10</th>\n",
       "      <th>FILTER.pSTD</th>\n",
       "      <th>FILTER.repetitive_variant</th>\n",
       "      <th>FILTER.cohort_n_rich</th>\n",
       "      <th>FILTER.low_complex_repetitive</th>\n",
       "      <th>FILTER.LongMSI</th>\n",
       "      <th>FILTER.cohort_n_rich_threshold</th>\n",
       "      <th>FILTER.AM_no_pileup_support</th>\n",
       "      <th>FILTER.no_pileup_support</th>\n",
       "      <th>FILTER.n_rich</th>\n",
       "      <th>FILTER.other_sample_SNP</th>\n",
       "      <th>FILTER.PASS</th>\n",
       "      <th>FILTER.AM_not_searched_COMPLEX</th>\n",
       "      <th>FILTER.not_searched_COMPLEX</th>\n",
       "      <th>FILTER.MSI12</th>\n",
       "      <th>FILTER.AM_not_searched_SV</th>\n",
       "      <th>FILTER.not_searched_SV</th>\n",
       "      <th>FILTER.SN1.5</th>\n",
       "      <th>FILTER.q22.5</th>\n",
       "      <th>FILTER.v1</th>\n",
       "      <th>FILTER.low_mappability</th>\n",
       "      <th>FILTER.Bias</th>\n",
       "      <th>FILTER.d3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7669639</td>\n",
       "      <td>CATG</td>\n",
       "      <td>C</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0034_BDO_01;TYPE=Deletion;DP=4948;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4948:1:4947,1:0.0002:2970,1977:0,1:4930:49...</td>\n",
       "      <td>4930</td>\n",
       "      <td>1</td>\n",
       "      <td>4929</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>4948</td>\n",
       "      <td>1</td>\n",
       "      <td>4947</td>\n",
       "      <td>0.000202</td>\n",
       "      <td>11</td>\n",
       "      <td>0.002226</td>\n",
       "      <td>8144</td>\n",
       "      <td>1</td>\n",
       "      <td>8143</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>15</td>\n",
       "      <td>0.001838</td>\n",
       "      <td>3214</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.605354</td>\n",
       "      <td>0.944260</td>\n",
       "      <td>chr17:7669639_CATG&gt;C</td>\n",
       "      <td>P19_0034_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7669640-7669642</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>1291-1293</td>\n",
       "      <td>1149-1151</td>\n",
       "      <td>383-384</td>\n",
       "      <td>LM/L</td>\n",
       "      <td>ctCATg/ctg</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669640-7669642</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>1291-1293</td>\n",
       "      <td>1149-1151</td>\n",
       "      <td>383-384</td>\n",
       "      <td>LM/L</td>\n",
       "      <td>ctCATg/ctg</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7669639_CATG&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7669640</td>\n",
       "      <td>ATG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7673832</td>\n",
       "      <td>TTAC</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0004_BTR_01;TYPE=Deletion;DP=10396;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:10396:1:10395,1:0.0001:4223,6172:1,0:10454...</td>\n",
       "      <td>10454</td>\n",
       "      <td>1</td>\n",
       "      <td>10453</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>10396</td>\n",
       "      <td>1</td>\n",
       "      <td>10395</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>35</td>\n",
       "      <td>0.003337</td>\n",
       "      <td>27268</td>\n",
       "      <td>1</td>\n",
       "      <td>27267</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>85</td>\n",
       "      <td>0.003108</td>\n",
       "      <td>16814</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.383380</td>\n",
       "      <td>0.906131</td>\n",
       "      <td>chr17:7673832_TTAC&gt;T</td>\n",
       "      <td>P19_0004_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7673833-7673835</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion,splice_region_variant</td>\n",
       "      <td>927-929</td>\n",
       "      <td>785-787</td>\n",
       "      <td>262-263</td>\n",
       "      <td>GN/D</td>\n",
       "      <td>gGTAat/gat</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7673833-7673835</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion,splice_region_variant</td>\n",
       "      <td>927-929</td>\n",
       "      <td>785-787</td>\n",
       "      <td>262-263</td>\n",
       "      <td>GN/D</td>\n",
       "      <td>gGTAat/gat</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7673832_TTAC&gt;T</td>\n",
       "      <td>17</td>\n",
       "      <td>7673833</td>\n",
       "      <td>TAC</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7674278</td>\n",
       "      <td>AGTCAGAGCC</td>\n",
       "      <td>A</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0028_BDO_01;TYPE=Deletion;DP=4937;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4937:1:4936,1:0.0002:1386,3550:0,1:4959:49...</td>\n",
       "      <td>4959</td>\n",
       "      <td>1</td>\n",
       "      <td>4958</td>\n",
       "      <td>0.000202</td>\n",
       "      <td>4937</td>\n",
       "      <td>1</td>\n",
       "      <td>4936</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>12</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>11094</td>\n",
       "      <td>1</td>\n",
       "      <td>11093</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>23</td>\n",
       "      <td>0.002069</td>\n",
       "      <td>6135</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.446998</td>\n",
       "      <td>0.913551</td>\n",
       "      <td>chr17:7674278_AGTCAGAGCC&gt;A</td>\n",
       "      <td>P19_0028_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7674279-7674287</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>818-826</td>\n",
       "      <td>676-684</td>\n",
       "      <td>226-228</td>\n",
       "      <td>GSD/-</td>\n",
       "      <td>GGCTCTGAC/-</td>\n",
       "      <td>COSV53313568</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr17:7674279-7674287</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>818-826</td>\n",
       "      <td>676-684</td>\n",
       "      <td>226-228</td>\n",
       "      <td>GSD/-</td>\n",
       "      <td>GGCTCTGAC/-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7674278_AGTCAGAGCC&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7674279</td>\n",
       "      <td>GTCAGAGCC</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>9</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>762</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7674865</td>\n",
       "      <td>C</td>\n",
       "      <td>CGGG</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0002_BTR_01;TYPE=Insertion;DP=4596;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:4596:1:4595,1:0.0002:2684,1911:0,1:4598:45...</td>\n",
       "      <td>4598</td>\n",
       "      <td>1</td>\n",
       "      <td>4597</td>\n",
       "      <td>0.000217</td>\n",
       "      <td>4596</td>\n",
       "      <td>1</td>\n",
       "      <td>4595</td>\n",
       "      <td>0.000218</td>\n",
       "      <td>56</td>\n",
       "      <td>0.012033</td>\n",
       "      <td>8513</td>\n",
       "      <td>1</td>\n",
       "      <td>8512</td>\n",
       "      <td>0.000117</td>\n",
       "      <td>75</td>\n",
       "      <td>0.008733</td>\n",
       "      <td>3915</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.540115</td>\n",
       "      <td>0.931932</td>\n",
       "      <td>chr17:7674865_C&gt;CGGG</td>\n",
       "      <td>P19_0002_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7674865-7674866</td>\n",
       "      <td>GGG</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>807-808</td>\n",
       "      <td>665-666</td>\n",
       "      <td>222</td>\n",
       "      <td>P/PP</td>\n",
       "      <td>ccg/ccCCCg</td>\n",
       "      <td>COSV99435175</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr17:7674865-7674866</td>\n",
       "      <td>GGG</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>807-808</td>\n",
       "      <td>665-666</td>\n",
       "      <td>222</td>\n",
       "      <td>P/PP</td>\n",
       "      <td>ccg/ccCCCg</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7674865_C&gt;CGGG</td>\n",
       "      <td>17</td>\n",
       "      <td>7674865</td>\n",
       "      <td>-</td>\n",
       "      <td>GGG</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>763</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7674918</td>\n",
       "      <td>ACTC</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=P19_0015_BDO_01;TYPE=Deletion;DP=6003;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:6003:2:6002,2:0.0003:2938,3064:0,2:5997:59...</td>\n",
       "      <td>5997</td>\n",
       "      <td>1</td>\n",
       "      <td>5996</td>\n",
       "      <td>0.000167</td>\n",
       "      <td>6004</td>\n",
       "      <td>2</td>\n",
       "      <td>6002</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>17</td>\n",
       "      <td>0.002827</td>\n",
       "      <td>9279</td>\n",
       "      <td>1</td>\n",
       "      <td>9278</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>29</td>\n",
       "      <td>0.003116</td>\n",
       "      <td>3282</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.646298</td>\n",
       "      <td>0.952220</td>\n",
       "      <td>chr17:7674918_ACTC&gt;A</td>\n",
       "      <td>P19_0015_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7674919-7674921</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>752-754</td>\n",
       "      <td>610-612</td>\n",
       "      <td>204</td>\n",
       "      <td>E/-</td>\n",
       "      <td>GAG/-</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7674919-7674921</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>752-754</td>\n",
       "      <td>610-612</td>\n",
       "      <td>204</td>\n",
       "      <td>E/-</td>\n",
       "      <td>GAG/-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7674918_ACTC&gt;A</td>\n",
       "      <td>17</td>\n",
       "      <td>7674919</td>\n",
       "      <td>CTC</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7674956</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0042_BDO_01;TYPE=Deletion;DP=8082;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:8082:1:8080,1:0.0001:3705,4375:0,1:8096:80...</td>\n",
       "      <td>8096</td>\n",
       "      <td>1</td>\n",
       "      <td>8095</td>\n",
       "      <td>0.000124</td>\n",
       "      <td>8082</td>\n",
       "      <td>1</td>\n",
       "      <td>8080</td>\n",
       "      <td>0.000124</td>\n",
       "      <td>23</td>\n",
       "      <td>0.002833</td>\n",
       "      <td>12936</td>\n",
       "      <td>1</td>\n",
       "      <td>12935</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>26</td>\n",
       "      <td>0.002006</td>\n",
       "      <td>4840</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.625850</td>\n",
       "      <td>0.950501</td>\n",
       "      <td>chr17:7674956_TGAG&gt;T</td>\n",
       "      <td>P19_0042_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7674957-7674959</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>714-716</td>\n",
       "      <td>572-574</td>\n",
       "      <td>191-192</td>\n",
       "      <td>PQ/Q</td>\n",
       "      <td>cCTCag/cag</td>\n",
       "      <td>rs1555525902</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>chr17:7674957-7674959</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>714-716</td>\n",
       "      <td>572-574</td>\n",
       "      <td>191-192</td>\n",
       "      <td>PQ/Q</td>\n",
       "      <td>cCTCag/cag</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7674956_TGAG&gt;T</td>\n",
       "      <td>17</td>\n",
       "      <td>7674957</td>\n",
       "      <td>GAG</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>765</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675129</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCC</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0015_BTR_01;TYPE=Insertion;DP=8798;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:8798:1:8797,1:0.0001:3086,5711:1,0:8800:87...</td>\n",
       "      <td>8800</td>\n",
       "      <td>1</td>\n",
       "      <td>8799</td>\n",
       "      <td>0.000114</td>\n",
       "      <td>8798</td>\n",
       "      <td>1</td>\n",
       "      <td>8797</td>\n",
       "      <td>0.000114</td>\n",
       "      <td>10</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>17555</td>\n",
       "      <td>1</td>\n",
       "      <td>17554</td>\n",
       "      <td>0.000057</td>\n",
       "      <td>15</td>\n",
       "      <td>0.000854</td>\n",
       "      <td>8755</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.501282</td>\n",
       "      <td>0.929338</td>\n",
       "      <td>chr17:7675129_G&gt;GGCC</td>\n",
       "      <td>P19_0015_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7675129-7675130</td>\n",
       "      <td>GCC</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>624-625</td>\n",
       "      <td>482-483</td>\n",
       "      <td>161</td>\n",
       "      <td>A/AA</td>\n",
       "      <td>gcc/gcGGCc</td>\n",
       "      <td>CI023564</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr17:7675129-7675130</td>\n",
       "      <td>GCC</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>624-625</td>\n",
       "      <td>482-483</td>\n",
       "      <td>161</td>\n",
       "      <td>A/AA</td>\n",
       "      <td>gcc/gcGGCc</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7675129_G&gt;GGCC</td>\n",
       "      <td>17</td>\n",
       "      <td>7675129</td>\n",
       "      <td>-</td>\n",
       "      <td>GCC</td>\n",
       "      <td>INSERTION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675143</td>\n",
       "      <td>CGCGGGT</td>\n",
       "      <td>C</td>\n",
       "      <td>pSTD</td>\n",
       "      <td>SAMPLE=P19_0009_BTR_01;TYPE=Deletion;DP=10814;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:10814:1:10813,1:0.0001:3653,7160:0,1:10844...</td>\n",
       "      <td>10844</td>\n",
       "      <td>1</td>\n",
       "      <td>10842</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>10814</td>\n",
       "      <td>1</td>\n",
       "      <td>10813</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>39</td>\n",
       "      <td>0.003584</td>\n",
       "      <td>22818</td>\n",
       "      <td>1</td>\n",
       "      <td>22816</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>66</td>\n",
       "      <td>0.002884</td>\n",
       "      <td>11974</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.475239</td>\n",
       "      <td>0.925868</td>\n",
       "      <td>chr17:7675143_CGCGGGT&gt;C</td>\n",
       "      <td>P19_0009_BTR_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7675144-7675149</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>605-610</td>\n",
       "      <td>463-468</td>\n",
       "      <td>155-156</td>\n",
       "      <td>TR/-</td>\n",
       "      <td>ACCCGC/-</td>\n",
       "      <td>rs1555526166</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>chr17:7675144-7675149</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>605-610</td>\n",
       "      <td>463-468</td>\n",
       "      <td>155-156</td>\n",
       "      <td>TR/-</td>\n",
       "      <td>ACCCGC/-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7675143_CGCGGGT&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7675144</td>\n",
       "      <td>GCGGGT</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>6</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>767</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676018</td>\n",
       "      <td>CCCA</td>\n",
       "      <td>C</td>\n",
       "      <td>p10;pSTD</td>\n",
       "      <td>SAMPLE=P19_0034_BDO_01;TYPE=Deletion;DP=8844;V...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...</td>\n",
       "      <td>0/1:8844:1:8843,1:0.0001:4162,4681:0,1:8715:87...</td>\n",
       "      <td>8715</td>\n",
       "      <td>1</td>\n",
       "      <td>8714</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>8844</td>\n",
       "      <td>1</td>\n",
       "      <td>8843</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>19</td>\n",
       "      <td>0.002175</td>\n",
       "      <td>12740</td>\n",
       "      <td>1</td>\n",
       "      <td>12739</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>24</td>\n",
       "      <td>0.001880</td>\n",
       "      <td>4025</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.684066</td>\n",
       "      <td>0.959831</td>\n",
       "      <td>chr17:7676018_CCCA&gt;C</td>\n",
       "      <td>P19_0034_BDO_01</td>\n",
       "      <td>med</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17:7676019-7676021</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>490-492</td>\n",
       "      <td>348-350</td>\n",
       "      <td>116-117</td>\n",
       "      <td>SG/S</td>\n",
       "      <td>tcTGGg/tcg</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17:7676019-7676021</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>490-492</td>\n",
       "      <td>348-350</td>\n",
       "      <td>116-117</td>\n",
       "      <td>SG/S</td>\n",
       "      <td>tcTGGg/tcg</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>missense</td>\n",
       "      <td>protein_affecting</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>TP53_117_FrameshiftVariant</td>\n",
       "      <td>chr17:7676018_CCCA&gt;C</td>\n",
       "      <td>17</td>\n",
       "      <td>7676019</td>\n",
       "      <td>CCA</td>\n",
       "      <td>-</td>\n",
       "      <td>DELETION</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM      POS         REF   ALT    FILTER  \\\n",
       "759  chr17  7669639        CATG     C  p10;pSTD   \n",
       "760  chr17  7673832        TTAC     T      pSTD   \n",
       "761  chr17  7674278  AGTCAGAGCC     A  p10;pSTD   \n",
       "762  chr17  7674865           C  CGGG      pSTD   \n",
       "763  chr17  7674918        ACTC     A      PASS   \n",
       "764  chr17  7674956        TGAG     T      pSTD   \n",
       "765  chr17  7675129           G  GGCC      pSTD   \n",
       "766  chr17  7675143     CGCGGGT     C      pSTD   \n",
       "767  chr17  7676018        CCCA     C  p10;pSTD   \n",
       "\n",
       "                                                  INFO  \\\n",
       "759  SAMPLE=P19_0034_BDO_01;TYPE=Deletion;DP=4948;V...   \n",
       "760  SAMPLE=P19_0004_BTR_01;TYPE=Deletion;DP=10396;...   \n",
       "761  SAMPLE=P19_0028_BDO_01;TYPE=Deletion;DP=4937;V...   \n",
       "762  SAMPLE=P19_0002_BTR_01;TYPE=Insertion;DP=4596;...   \n",
       "763  SAMPLE=P19_0015_BDO_01;TYPE=Deletion;DP=6003;V...   \n",
       "764  SAMPLE=P19_0042_BDO_01;TYPE=Deletion;DP=8082;V...   \n",
       "765  SAMPLE=P19_0015_BTR_01;TYPE=Insertion;DP=8798;...   \n",
       "766  SAMPLE=P19_0009_BTR_01;TYPE=Deletion;DP=10814;...   \n",
       "767  SAMPLE=P19_0034_BDO_01;TYPE=Deletion;DP=8844;V...   \n",
       "\n",
       "                                                FORMAT  \\\n",
       "759  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "760  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "761  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "762  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "763  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "764  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "765  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "766  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "767  GT:DP:VD:AD:AF:RD:ALD:CDP:CAD:NDP:CDPAM:CADAM:...   \n",
       "\n",
       "                                                SAMPLE  DEPTH  ALT_DEPTH  \\\n",
       "759  0/1:4948:1:4947,1:0.0002:2970,1977:0,1:4930:49...   4930          1   \n",
       "760  0/1:10396:1:10395,1:0.0001:4223,6172:1,0:10454...  10454          1   \n",
       "761  0/1:4937:1:4936,1:0.0002:1386,3550:0,1:4959:49...   4959          1   \n",
       "762  0/1:4596:1:4595,1:0.0002:2684,1911:0,1:4598:45...   4598          1   \n",
       "763  0/1:6003:2:6002,2:0.0003:2938,3064:0,2:5997:59...   5997          1   \n",
       "764  0/1:8082:1:8080,1:0.0001:3705,4375:0,1:8096:80...   8096          1   \n",
       "765  0/1:8798:1:8797,1:0.0001:3086,5711:1,0:8800:87...   8800          1   \n",
       "766  0/1:10814:1:10813,1:0.0001:3653,7160:0,1:10844...  10844          1   \n",
       "767  0/1:8844:1:8843,1:0.0001:4162,4681:0,1:8715:87...   8715          1   \n",
       "\n",
       "     REF_DEPTH       VAF  vd_DEPTH  vd_ALT_DEPTH  vd_REF_DEPTH    vd_VAF  \\\n",
       "759       4929  0.000203      4948             1          4947  0.000202   \n",
       "760      10453  0.000096     10396             1         10395  0.000096   \n",
       "761       4958  0.000202      4937             1          4936  0.000203   \n",
       "762       4597  0.000217      4596             1          4595  0.000218   \n",
       "763       5996  0.000167      6004             2          6002  0.000333   \n",
       "764       8095  0.000124      8082             1          8080  0.000124   \n",
       "765       8799  0.000114      8798             1          8797  0.000114   \n",
       "766      10842  0.000092     10814             1         10813  0.000092   \n",
       "767       8714  0.000115      8844             1          8843  0.000113   \n",
       "\n",
       "     numNs    VAF_Ns  DEPTH_AM  ALT_DEPTH_AM  REF_DEPTH_AM    VAF_AM  \\\n",
       "759     11  0.002226      8144             1          8143  0.000123   \n",
       "760     35  0.003337     27268             1         27267  0.000037   \n",
       "761     12  0.002414     11094             1         11093  0.000090   \n",
       "762     56  0.012033      8513             1          8512  0.000117   \n",
       "763     17  0.002827      9279             1          9278  0.000108   \n",
       "764     23  0.002833     12936             1         12935  0.000077   \n",
       "765     10  0.001135     17555             1         17554  0.000057   \n",
       "766     39  0.003584     22818             1         22816  0.000044   \n",
       "767     19  0.002175     12740             1         12739  0.000078   \n",
       "\n",
       "     numNs_AM  VAF_Ns_AM  DEPTH_ND  ALT_DEPTH_ND  VAF_ND  \\\n",
       "759        15   0.001838      3214             0     0.0   \n",
       "760        85   0.003108     16814             0     0.0   \n",
       "761        23   0.002069      6135             0     0.0   \n",
       "762        75   0.008733      3915             0     0.0   \n",
       "763        29   0.003116      3282             0     0.0   \n",
       "764        26   0.002006      4840             0     0.0   \n",
       "765        15   0.000854      8755             0     0.0   \n",
       "766        66   0.002884     11974             0     0.0   \n",
       "767        24   0.001880      4025             0     0.0   \n",
       "\n",
       "     VAF_distorted_expanded  VAF_distorted_expanded_sq  VAF_distortion  \\\n",
       "759                   False                      False        0.605354   \n",
       "760                   False                      False        0.383380   \n",
       "761                   False                      False        0.446998   \n",
       "762                   False                      False        0.540115   \n",
       "763                   False                      False        0.646298   \n",
       "764                   False                      False        0.625850   \n",
       "765                   False                      False        0.501282   \n",
       "766                   False                      False        0.475239   \n",
       "767                   False                      False        0.684066   \n",
       "\n",
       "     VAF_distortion_sq                      MUT_ID        SAMPLE_ID METHOD  \\\n",
       "759           0.944260        chr17:7669639_CATG>C  P19_0034_BDO_01    med   \n",
       "760           0.906131        chr17:7673832_TTAC>T  P19_0004_BTR_01    med   \n",
       "761           0.913551  chr17:7674278_AGTCAGAGCC>A  P19_0028_BDO_01    med   \n",
       "762           0.931932        chr17:7674865_C>CGGG  P19_0002_BTR_01    med   \n",
       "763           0.952220        chr17:7674918_ACTC>A  P19_0015_BDO_01    med   \n",
       "764           0.950501        chr17:7674956_TGAG>T  P19_0042_BDO_01    med   \n",
       "765           0.929338        chr17:7675129_G>GGCC  P19_0015_BTR_01    med   \n",
       "766           0.925868     chr17:7675143_CGCGGGT>C  P19_0009_BTR_01    med   \n",
       "767           0.959831        chr17:7676018_CCCA>C  P19_0034_BDO_01    med   \n",
       "\n",
       "     PROJECT_NAME               Location Allele             Gene  \\\n",
       "759           NaN  chr17:7669640-7669642      -  ENSG00000141510   \n",
       "760           NaN  chr17:7673833-7673835      -  ENSG00000141510   \n",
       "761           NaN  chr17:7674279-7674287      -  ENSG00000141510   \n",
       "762           NaN  chr17:7674865-7674866    GGG  ENSG00000141510   \n",
       "763           NaN  chr17:7674919-7674921      -  ENSG00000141510   \n",
       "764           NaN  chr17:7674957-7674959      -  ENSG00000141510   \n",
       "765           NaN  chr17:7675129-7675130    GCC  ENSG00000141510   \n",
       "766           NaN  chr17:7675144-7675149      -  ENSG00000141510   \n",
       "767           NaN  chr17:7676019-7676021      -  ENSG00000141510   \n",
       "\n",
       "             Feature Feature_type                             Consequence  \\\n",
       "759  ENST00000269305   Transcript                        inframe_deletion   \n",
       "760  ENST00000269305   Transcript  inframe_deletion,splice_region_variant   \n",
       "761  ENST00000269305   Transcript                        inframe_deletion   \n",
       "762  ENST00000269305   Transcript                       inframe_insertion   \n",
       "763  ENST00000269305   Transcript                        inframe_deletion   \n",
       "764  ENST00000269305   Transcript                        inframe_deletion   \n",
       "765  ENST00000269305   Transcript                       inframe_insertion   \n",
       "766  ENST00000269305   Transcript                        inframe_deletion   \n",
       "767  ENST00000269305   Transcript                        inframe_deletion   \n",
       "\n",
       "    cDNA_position CDS_position Protein_position Amino_acids       Codons  \\\n",
       "759     1291-1293    1149-1151          383-384        LM/L   ctCATg/ctg   \n",
       "760       927-929      785-787          262-263        GN/D   gGTAat/gat   \n",
       "761       818-826      676-684          226-228       GSD/-  GGCTCTGAC/-   \n",
       "762       807-808      665-666              222        P/PP   ccg/ccCCCg   \n",
       "763       752-754      610-612              204         E/-        GAG/-   \n",
       "764       714-716      572-574          191-192        PQ/Q   cCTCag/cag   \n",
       "765       624-625      482-483              161        A/AA   gcc/gcGGCc   \n",
       "766       605-610      463-468          155-156        TR/-     ACCCGC/-   \n",
       "767       490-492      348-350          116-117        SG/S   tcTGGg/tcg   \n",
       "\n",
       "    Existing_variation    IMPACT DISTANCE STRAND FLAGS SYMBOL SYMBOL_SOURCE  \\\n",
       "759                  -  MODERATE        -     -1     -   TP53          HGNC   \n",
       "760                  -  MODERATE        -     -1     -   TP53          HGNC   \n",
       "761       COSV53313568  MODERATE        -     -1     -   TP53          HGNC   \n",
       "762       COSV99435175  MODERATE        -     -1     -   TP53          HGNC   \n",
       "763                  -  MODERATE        -     -1     -   TP53          HGNC   \n",
       "764       rs1555525902  MODERATE        -     -1     -   TP53          HGNC   \n",
       "765           CI023564  MODERATE        -     -1     -   TP53          HGNC   \n",
       "766       rs1555526166  MODERATE        -     -1     -   TP53          HGNC   \n",
       "767                  -  MODERATE        -     -1     -   TP53          HGNC   \n",
       "\n",
       "        HGNC_ID CANONICAL             ENSP  gnomADe_AF  gnomADe_AFR_AF  \\\n",
       "759  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "760  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "761  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "762  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "763  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "764  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "765  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "766  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "767  HGNC:11998       YES  ENSP00000269305         0.0             0.0   \n",
       "\n",
       "     gnomADe_AMR_AF  gnomADe_ASJ_AF  gnomADe_EAS_AF  gnomADe_FIN_AF  \\\n",
       "759             0.0             0.0             0.0             0.0   \n",
       "760             0.0             0.0             0.0             0.0   \n",
       "761             0.0             0.0             0.0             0.0   \n",
       "762             0.0             0.0             0.0             0.0   \n",
       "763             0.0             0.0             0.0             0.0   \n",
       "764             0.0             0.0             0.0             0.0   \n",
       "765             0.0             0.0             0.0             0.0   \n",
       "766             0.0             0.0             0.0             0.0   \n",
       "767             0.0             0.0             0.0             0.0   \n",
       "\n",
       "     gnomADe_NFE_AF  gnomADe_OTH_AF  gnomADe_SAS_AF  gnomADg_AF  \\\n",
       "759             0.0             0.0             0.0         0.0   \n",
       "760             0.0             0.0             0.0         0.0   \n",
       "761             0.0             0.0             0.0         0.0   \n",
       "762             0.0             0.0             0.0         0.0   \n",
       "763             0.0             0.0             0.0         0.0   \n",
       "764             0.0             0.0             0.0         0.0   \n",
       "765             0.0             0.0             0.0         0.0   \n",
       "766             0.0             0.0             0.0         0.0   \n",
       "767             0.0             0.0             0.0         0.0   \n",
       "\n",
       "     gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  \\\n",
       "759             0.0             0.0             0.0             0.0   \n",
       "760             0.0             0.0             0.0             0.0   \n",
       "761             0.0             0.0             0.0             0.0   \n",
       "762             0.0             0.0             0.0             0.0   \n",
       "763             0.0             0.0             0.0             0.0   \n",
       "764             0.0             0.0             0.0             0.0   \n",
       "765             0.0             0.0             0.0             0.0   \n",
       "766             0.0             0.0             0.0             0.0   \n",
       "767             0.0             0.0             0.0             0.0   \n",
       "\n",
       "     gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  \\\n",
       "759             0.0             0.0             0.0             0.0   \n",
       "760             0.0             0.0             0.0             0.0   \n",
       "761             0.0             0.0             0.0             0.0   \n",
       "762             0.0             0.0             0.0             0.0   \n",
       "763             0.0             0.0             0.0             0.0   \n",
       "764             0.0             0.0             0.0             0.0   \n",
       "765             0.0             0.0             0.0             0.0   \n",
       "766             0.0             0.0             0.0             0.0   \n",
       "767             0.0             0.0             0.0             0.0   \n",
       "\n",
       "     gnomADg_OTH_AF  gnomADg_SAS_AF                CLIN_SIG SOMATIC PHENO  \\\n",
       "759             0.0             0.0                       -       -     -   \n",
       "760             0.0             0.0                       -       -     -   \n",
       "761             0.0             0.0                       -       1     1   \n",
       "762             0.0             0.0                       -       1     1   \n",
       "763             0.0             0.0                       -       -     -   \n",
       "764             0.0             0.0                       -       -     1   \n",
       "765             0.0             0.0                       -       1     1   \n",
       "766             0.0             0.0  uncertain_significance       -     1   \n",
       "767             0.0             0.0                       -       -     -   \n",
       "\n",
       "        canonical_Location canonical_Allele   canonical_Gene  \\\n",
       "759  chr17:7669640-7669642                -  ENSG00000141510   \n",
       "760  chr17:7673833-7673835                -  ENSG00000141510   \n",
       "761  chr17:7674279-7674287                -  ENSG00000141510   \n",
       "762  chr17:7674865-7674866              GGG  ENSG00000141510   \n",
       "763  chr17:7674919-7674921                -  ENSG00000141510   \n",
       "764  chr17:7674957-7674959                -  ENSG00000141510   \n",
       "765  chr17:7675129-7675130              GCC  ENSG00000141510   \n",
       "766  chr17:7675144-7675149                -  ENSG00000141510   \n",
       "767  chr17:7676019-7676021                -  ENSG00000141510   \n",
       "\n",
       "    canonical_Feature canonical_Feature_type  \\\n",
       "759   ENST00000269305             Transcript   \n",
       "760   ENST00000269305             Transcript   \n",
       "761   ENST00000269305             Transcript   \n",
       "762   ENST00000269305             Transcript   \n",
       "763   ENST00000269305             Transcript   \n",
       "764   ENST00000269305             Transcript   \n",
       "765   ENST00000269305             Transcript   \n",
       "766   ENST00000269305             Transcript   \n",
       "767   ENST00000269305             Transcript   \n",
       "\n",
       "                      canonical_Consequence canonical_cDNA_position  \\\n",
       "759                        inframe_deletion               1291-1293   \n",
       "760  inframe_deletion,splice_region_variant                 927-929   \n",
       "761                        inframe_deletion                 818-826   \n",
       "762                       inframe_insertion                 807-808   \n",
       "763                        inframe_deletion                 752-754   \n",
       "764                        inframe_deletion                 714-716   \n",
       "765                       inframe_insertion                 624-625   \n",
       "766                        inframe_deletion                 605-610   \n",
       "767                        inframe_deletion                 490-492   \n",
       "\n",
       "    canonical_CDS_position canonical_Protein_position canonical_Amino_acids  \\\n",
       "759              1149-1151                    383-384                  LM/L   \n",
       "760                785-787                    262-263                  GN/D   \n",
       "761                676-684                    226-228                 GSD/-   \n",
       "762                665-666                        222                  P/PP   \n",
       "763                610-612                        204                   E/-   \n",
       "764                572-574                    191-192                  PQ/Q   \n",
       "765                482-483                        161                  A/AA   \n",
       "766                463-468                    155-156                  TR/-   \n",
       "767                348-350                    116-117                  SG/S   \n",
       "\n",
       "    canonical_Codons canonical_IMPACT canonical_DISTANCE canonical_STRAND  \\\n",
       "759       ctCATg/ctg         MODERATE                  -             -1.0   \n",
       "760       gGTAat/gat         MODERATE                  -             -1.0   \n",
       "761      GGCTCTGAC/-         MODERATE                  -             -1.0   \n",
       "762       ccg/ccCCCg         MODERATE                  -             -1.0   \n",
       "763            GAG/-         MODERATE                  -             -1.0   \n",
       "764       cCTCag/cag         MODERATE                  -             -1.0   \n",
       "765       gcc/gcGGCc         MODERATE                  -             -1.0   \n",
       "766         ACCCGC/-         MODERATE                  -             -1.0   \n",
       "767       tcTGGg/tcg         MODERATE                  -             -1.0   \n",
       "\n",
       "    canonical_FLAGS canonical_SYMBOL canonical_SYMBOL_SOURCE  \\\n",
       "759               -             TP53                    HGNC   \n",
       "760               -             TP53                    HGNC   \n",
       "761               -             TP53                    HGNC   \n",
       "762               -             TP53                    HGNC   \n",
       "763               -             TP53                    HGNC   \n",
       "764               -             TP53                    HGNC   \n",
       "765               -             TP53                    HGNC   \n",
       "766               -             TP53                    HGNC   \n",
       "767               -             TP53                    HGNC   \n",
       "\n",
       "    canonical_HGNC_ID canonical_CANONICAL   canonical_ENSP  \\\n",
       "759        HGNC:11998                 YES  ENSP00000269305   \n",
       "760        HGNC:11998                 YES  ENSP00000269305   \n",
       "761        HGNC:11998                 YES  ENSP00000269305   \n",
       "762        HGNC:11998                 YES  ENSP00000269305   \n",
       "763        HGNC:11998                 YES  ENSP00000269305   \n",
       "764        HGNC:11998                 YES  ENSP00000269305   \n",
       "765        HGNC:11998                 YES  ENSP00000269305   \n",
       "766        HGNC:11998                 YES  ENSP00000269305   \n",
       "767        HGNC:11998                 YES  ENSP00000269305   \n",
       "\n",
       "         canonical_CLIN_SIG canonical_Consequence_single  \\\n",
       "759                       -             inframe_deletion   \n",
       "760                       -             inframe_deletion   \n",
       "761                       -             inframe_deletion   \n",
       "762                       -            inframe_insertion   \n",
       "763                       -             inframe_deletion   \n",
       "764                       -             inframe_deletion   \n",
       "765                       -            inframe_insertion   \n",
       "766  uncertain_significance             inframe_deletion   \n",
       "767                       -             inframe_deletion   \n",
       "\n",
       "    canonical_Consequence_broader canonical_Protein_affecting  \\\n",
       "759                      missense           protein_affecting   \n",
       "760                      missense           protein_affecting   \n",
       "761                      missense           protein_affecting   \n",
       "762                      missense           protein_affecting   \n",
       "763                      missense           protein_affecting   \n",
       "764                      missense           protein_affecting   \n",
       "765                      missense           protein_affecting   \n",
       "766                      missense           protein_affecting   \n",
       "767                      missense           protein_affecting   \n",
       "\n",
       "    Consequence_single Consequence_broader  Protein_affecting CONTEXT_MUT  \\\n",
       "759   inframe_deletion            missense  protein_affecting           -   \n",
       "760   inframe_deletion            missense  protein_affecting           -   \n",
       "761   inframe_deletion            missense  protein_affecting           -   \n",
       "762  inframe_insertion            missense  protein_affecting           -   \n",
       "763   inframe_deletion            missense  protein_affecting           -   \n",
       "764   inframe_deletion            missense  protein_affecting           -   \n",
       "765  inframe_insertion            missense  protein_affecting           -   \n",
       "766   inframe_deletion            missense  protein_affecting           -   \n",
       "767   inframe_deletion            missense  protein_affecting           -   \n",
       "\n",
       "    MUTTYPE CONTEXT_MUT_SIGPRO  gnomAD_SNP                HOTSPOT_NAME  \\\n",
       "759       -                  -       False                           -   \n",
       "760       -                  -       False                           -   \n",
       "761       -                  -       False                           -   \n",
       "762       -                  -       False                           -   \n",
       "763       -                  -       False                           -   \n",
       "764       -                  -       False                           -   \n",
       "765       -                  -       False                           -   \n",
       "766       -                  -       False                           -   \n",
       "767       -                  -       False  TP53_117_FrameshiftVariant   \n",
       "\n",
       "                     MUT_ID_pyr CHROM_ensembl  POS_ensembl REF_ensembl  \\\n",
       "759        chr17:7669639_CATG>C            17      7669640         ATG   \n",
       "760        chr17:7673832_TTAC>T            17      7673833         TAC   \n",
       "761  chr17:7674278_AGTCAGAGCC>A            17      7674279   GTCAGAGCC   \n",
       "762        chr17:7674865_C>CGGG            17      7674865           -   \n",
       "763        chr17:7674918_ACTC>A            17      7674919         CTC   \n",
       "764        chr17:7674956_TGAG>T            17      7674957         GAG   \n",
       "765        chr17:7675129_G>GGCC            17      7675129           -   \n",
       "766     chr17:7675143_CGCGGGT>C            17      7675144      GCGGGT   \n",
       "767        chr17:7676018_CCCA>C            17      7676019         CCA   \n",
       "\n",
       "    ALT_ensembl       TYPE  INDEL_LENGTH  INDEL_INFRAME  INDEL_MULTIPLE3  \\\n",
       "759           -   DELETION             3           True            False   \n",
       "760           -   DELETION             3           True            False   \n",
       "761           -   DELETION             9           True            False   \n",
       "762         GGG  INSERTION             3           True            False   \n",
       "763           -   DELETION             3           True            False   \n",
       "764           -   DELETION             3           True            False   \n",
       "765         GCC  INSERTION             3           True            False   \n",
       "766           -   DELETION             6           True            False   \n",
       "767           -   DELETION             3           True            False   \n",
       "\n",
       "     N_rich_frequency  VAF_Ns_threshold  FILTER.cohort_n_rich_uni  \\\n",
       "759               0.0               1.1                     False   \n",
       "760               0.0               1.1                     False   \n",
       "761               0.0               1.1                     False   \n",
       "762               0.0               1.1                     False   \n",
       "763               0.0               1.1                     False   \n",
       "764               0.0               1.1                     False   \n",
       "765               0.0               1.1                     False   \n",
       "766               0.0               1.1                     False   \n",
       "767               0.0               1.1                     False   \n",
       "\n",
       "     FILTER.gnomAD_SNP  FILTER.not_covered  FILTER.not_in_exons  FILTER.p10  \\\n",
       "759              False               False                False        True   \n",
       "760              False               False                False       False   \n",
       "761              False               False                False        True   \n",
       "762              False               False                False       False   \n",
       "763              False               False                False       False   \n",
       "764              False               False                False       False   \n",
       "765              False               False                False       False   \n",
       "766              False               False                False       False   \n",
       "767              False               False                False        True   \n",
       "\n",
       "     FILTER.pSTD  FILTER.repetitive_variant  FILTER.cohort_n_rich  \\\n",
       "759         True                      False                 False   \n",
       "760         True                      False                 False   \n",
       "761         True                      False                 False   \n",
       "762         True                      False                 False   \n",
       "763        False                      False                 False   \n",
       "764         True                      False                 False   \n",
       "765         True                      False                 False   \n",
       "766         True                      False                 False   \n",
       "767         True                      False                 False   \n",
       "\n",
       "     FILTER.low_complex_repetitive  FILTER.LongMSI  \\\n",
       "759                          False           False   \n",
       "760                          False           False   \n",
       "761                          False           False   \n",
       "762                          False           False   \n",
       "763                          False           False   \n",
       "764                          False           False   \n",
       "765                          False           False   \n",
       "766                          False           False   \n",
       "767                          False           False   \n",
       "\n",
       "     FILTER.cohort_n_rich_threshold  FILTER.AM_no_pileup_support  \\\n",
       "759                           False                        False   \n",
       "760                           False                        False   \n",
       "761                           False                        False   \n",
       "762                           False                        False   \n",
       "763                           False                        False   \n",
       "764                           False                        False   \n",
       "765                           False                        False   \n",
       "766                           False                        False   \n",
       "767                           False                        False   \n",
       "\n",
       "     FILTER.no_pileup_support  FILTER.n_rich  FILTER.other_sample_SNP  \\\n",
       "759                     False          False                    False   \n",
       "760                     False          False                    False   \n",
       "761                     False          False                    False   \n",
       "762                     False          False                    False   \n",
       "763                     False          False                    False   \n",
       "764                     False          False                    False   \n",
       "765                     False          False                    False   \n",
       "766                     False          False                    False   \n",
       "767                     False          False                    False   \n",
       "\n",
       "     FILTER.PASS  FILTER.AM_not_searched_COMPLEX  FILTER.not_searched_COMPLEX  \\\n",
       "759        False                           False                        False   \n",
       "760        False                           False                        False   \n",
       "761        False                           False                        False   \n",
       "762        False                           False                        False   \n",
       "763         True                           False                        False   \n",
       "764        False                           False                        False   \n",
       "765        False                           False                        False   \n",
       "766        False                           False                        False   \n",
       "767        False                           False                        False   \n",
       "\n",
       "     FILTER.MSI12  FILTER.AM_not_searched_SV  FILTER.not_searched_SV  \\\n",
       "759         False                      False                   False   \n",
       "760         False                      False                   False   \n",
       "761         False                      False                   False   \n",
       "762         False                      False                   False   \n",
       "763         False                      False                   False   \n",
       "764         False                      False                   False   \n",
       "765         False                      False                   False   \n",
       "766         False                      False                   False   \n",
       "767         False                      False                   False   \n",
       "\n",
       "     FILTER.SN1.5  FILTER.q22.5  FILTER.v1  FILTER.low_mappability  \\\n",
       "759         False         False      False                   False   \n",
       "760         False         False      False                   False   \n",
       "761         False         False      False                   False   \n",
       "762         False         False      False                   False   \n",
       "763         False         False      False                   False   \n",
       "764         False         False      False                   False   \n",
       "765         False         False      False                   False   \n",
       "766         False         False      False                   False   \n",
       "767         False         False      False                   False   \n",
       "\n",
       "     FILTER.Bias  FILTER.d3  \n",
       "759        False      False  \n",
       "760        False      False  \n",
       "761        False      False  \n",
       "762        False      False  \n",
       "763        False      False  \n",
       "764        False      False  \n",
       "765        False      False  \n",
       "766        False      False  \n",
       "767        False      False  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inframe_indels = indel_maf_df[(indel_maf_df[\"INDEL_INFRAME\"]) &\n",
    "                              (indel_maf_df[\"canonical_CDS_position\"] != '-' )].reset_index(drop = True)\n",
    "inframe_indels[inframe_indels[\"canonical_SYMBOL\"] == \"TP53\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77aa1972-951d-4e4f-a3ab-27148b95e860",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5b3556f-08eb-401d-9f44-cce9cfc5536b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "67214dae-96d5-4b75-a2b2-35a27206e901",
   "metadata": {},
   "source": [
    "# __SIDE FIGURES__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "78126b7c-819e-4bc7-a462-2d1de42ed24d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_all_side_figures(sample, \n",
    "                              run_dir,\n",
    "                              gene_list = None,\n",
    "                              store_plots_dir = None,\n",
    "                              tools = [\"oncodrivefml\", \"oncodrive3d\", \"omega_trunc\", \"omega_mis\", \"excess_indels\"]):\n",
    "    possible_genes = []\n",
    "    if \"oncodrivefml\" in tools:\n",
    "        oncodrivefml_data = pd.read_table(f\"{run_dir}/oncodrivefmlsnvs/{sample}.all/{sample}-oncodrivefml.tsv.gz\")\n",
    "        oncodrivefml_data = oncodrivefml_data[[\"GENE_ID\", \"Z-SCORE\", \"Q_VALUE\", \"AVG_SCORE_OBS\", \"POPULATION_MEAN\", \"STD_OF_MEANS\"]]\n",
    "        oncodrivefml_data.columns = [\"GENE\", \"OncodriveFML\", \"pvalue\", \"OBSERVED_MEAN\", \"BACKGROUND_MEAN\", \"BACKGROUND_STD\"]\n",
    "        possible_genes += list(pd.unique(oncodrivefml_data[\"GENE\"]))\n",
    "\n",
    "\n",
    "    if \"omega_trunc\" in tools or \"omega_mis\" in tools:\n",
    "        omega_data = pd.read_table(f\"{run_dir}/omega/output_mle.{sample}.tsv\")\n",
    "        omega_data = omega_data[omega_data[\"impact\"].isin(['missense', 'truncating'])]\n",
    "        if \"omega_trunc\" in tools :\n",
    "            omega_truncating = omega_data[omega_data[\"impact\"] == \"truncating\"].reset_index(drop = True)[[\"gene\", \"dnds\", \"pvalue\", \"lower\", \"upper\"]]\n",
    "            omega_truncating.columns = [\"GENE\", \"omega_trunc\", \"pvalue\", \"lower\", \"upper\"]\n",
    "            possible_genes += list(pd.unique(omega_truncating[\"GENE\"]))\n",
    "            \n",
    "        if \"omega_mis\" in tools :\n",
    "            omega_missense = omega_data[omega_data[\"impact\"] == \"missense\"].reset_index(drop = True)[[\"gene\", \"dnds\", \"pvalue\", \"lower\", \"upper\"]]\n",
    "            omega_missense.columns = [\"GENE\", \"omega_mis\", \"pvalue\", \"lower\", \"upper\"]\n",
    "            possible_genes += list(pd.unique(omega_missense[\"GENE\"]))\n",
    "\n",
    "            \n",
    "    if \"excess_indels\" in tools:\n",
    "        indels_data = pd.read_table(f\"{deepcsa_run_dir}/indels/{sample}.sample.indels.tsv\",\n",
    "                                    sep = '\\t',\n",
    "                                    header = 0)\n",
    "   \n",
    "    if gene_list is not None:\n",
    "        gene_list = list(set(gene_list))\n",
    "    else:\n",
    "        # gene_list = list(pd.unique(oncodrivefml_data[\"GENE\"]))\n",
    "        gene_list = list(set(possible_genes))\n",
    "        \n",
    "\n",
    "\n",
    "    for genee in gene_list:\n",
    "        print(genee)\n",
    "        \n",
    "        if \"oncodrivefml\" in tools:\n",
    "            # Select the specific gene data\n",
    "            oncodrivefml_gene_data = oncodrivefml_data[oncodrivefml_data[\"GENE\"] == genee].to_dict(orient='records')[0]\n",
    "            fig_gene_fml = plot_oncodrivefml_side(oncodrivefml_gene_data)\n",
    "    \n",
    "            if store_plots_dir is not None:\n",
    "                fig_gene_fml.savefig(f\"{store_plots_dir}/{genee}.oncodrivefml_side.png\", bbox_inches='tight')\n",
    "            plt.show()\n",
    "            plt.close()\n",
    "        \n",
    "        if \"omega_trunc\" in tools or \"omega_mis\" in tools:\n",
    "            omega_df = build_counts_from_df_complete(genee, snvs_maf, omega_truncating, omega_missense)\n",
    "            \n",
    "            fig_gene_omega = plot_omega_side_complete(omega_df)\n",
    "            if store_plots_dir is not None:\n",
    "                fig_gene_omega.savefig(f\"{store_plots_dir}/{genee}.omega_side.png\", bbox_inches='tight')\n",
    "            plt.show()\n",
    "            plt.close()\n",
    "        \n",
    "        \n",
    "        if \"excess_indels\" in tools:\n",
    "            indel_data_gene = indels_data[indels_data[\"SYMBOL\"] == genee].to_dict(orient='records')[0]\n",
    "            fig_gene_indel = plotting_indels_side(indel_data_gene)\n",
    "            if store_plots_dir is not None:\n",
    "                fig_gene_indel.savefig(f\"{store_plots_dir}/{genee}.indels_side.png\", bbox_inches='tight')\n",
    "            plt.show()\n",
    "            plt.close()\n",
    "        \n",
    "        \n",
    "        print(\"\\n\\n\\n\")\n",
    "        # break"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccb3d906-b421-47d2-8a2e-57814975fb02",
   "metadata": {},
   "source": [
    "### Omega missense and truncating"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "15d94538-41c8-447b-93f9-d125d49777e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_counts_from_df_complete(genee, snvs_maf, omega_truncating, omega_missense):\n",
    "\n",
    "    trunc_omega = float(omega_truncating[omega_truncating[\"GENE\"] == genee][\"omega_trunc\"].values[0])\n",
    "    trunc_pvalue = float(omega_truncating[omega_truncating[\"GENE\"] == genee][\"pvalue\"].values[0])\n",
    "\n",
    "    miss_omega = float(omega_missense[omega_missense[\"GENE\"] == genee][\"omega_mis\"].values[0])\n",
    "    miss_pvalue = float(omega_missense[omega_missense[\"GENE\"] == genee][\"pvalue\"].values[0])\n",
    "    snvs_gene = snvs_maf[snvs_maf[\"SYMBOL\"] == genee].reset_index(drop = True)\n",
    "\n",
    "\n",
    "    # Calculate counts based on canonical consequences\n",
    "    truncating_count = snvs_gene[snvs_gene[\"canonical_Consequence_broader\"].isin(['nonsense', \"essential_splice\"])].shape[0]\n",
    "    missense_count = snvs_gene[snvs_gene[\"canonical_Consequence_broader\"].isin(['missense'])].shape[0]\n",
    "    synonymous_count = snvs_gene[snvs_gene[\"canonical_Consequence_broader\"].isin([\"synonymous\"])].shape[0]\n",
    "\n",
    "    # Compute \n",
    "    expected_missense = (1 - ((miss_omega - 1) / miss_omega)) * missense_count\n",
    "    expected_truncating = (1 - ((trunc_omega - 1) / trunc_omega)) * truncating_count\n",
    "\n",
    "\n",
    "    # Create a dataframe from the counts and expected values\n",
    "    data = {\n",
    "        'type': ['truncating', 'synonymous', 'missense'],\n",
    "        'number_obs': [truncating_count, synonymous_count, missense_count],\n",
    "        'expected': [expected_truncating, None, expected_missense],\n",
    "        'omega': [trunc_omega, None, miss_omega],\n",
    "        'pvalue': [trunc_pvalue, None, miss_pvalue]\n",
    "    }\n",
    "    df = pd.DataFrame(data)\n",
    "\n",
    "    # Print the final dataframe\n",
    "    return df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "82046cff-c593-4ca2-baa5-48f93c035bf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_omega_side_complete(df):\n",
    "\n",
    "    consequence_order = ['truncating', \"synonymous\", \"missense\"]\n",
    "\n",
    "    # Define colors for each type\n",
    "    colors = {\n",
    "        'truncating': metrics_colors_dictionary[\"omega_trunc\"],\n",
    "        'missense': metrics_colors_dictionary[\"omega_miss\"],\n",
    "        'synonymous': metrics_colors_dictionary[\"omega_synon\"]\n",
    "    }\n",
    "    \n",
    "    t_omega = df[df['type'] == 'truncating']['omega'].item()\n",
    "    t_pvalue = df[df['type'] == 'truncating']['pvalue'].item()\n",
    "    m_omega = df[df['type'] == 'missense']['omega'].item()\n",
    "    m_pvalue = df[df['type'] == 'missense']['pvalue'].item()\n",
    "    \n",
    "    \n",
    "    # Plot for truncating, missense, and synonymous\n",
    "    fig, ax = plt.subplots(figsize=(4, 2))\n",
    "\n",
    "#     # Bar positions\n",
    "#     y_positions = np.arange(len(consequence_order)) * 2  # Add separation between bars\n",
    "\n",
    "#     # Plot the observed values as filled bars\n",
    "#     for i, cons in enumerate(consequence_order):\n",
    "#         obs_value = df[df['type'] == cons]['number_obs'].values[0]\n",
    "#         ax.barh(y_positions[i], obs_value, color=colors[cons], label=cons)\n",
    "\n",
    "    # Plot the observed values as filled bars\n",
    "    sns.barplot(data=df, y='type', x='number_obs',\n",
    "                hue = 'type', legend = False,\n",
    "                hue_order = consequence_order,\n",
    "                order=consequence_order, palette=[colors[x] for x in consequence_order],\n",
    "                ax = ax\n",
    "                )\n",
    "\n",
    "    # Overlay the expected values with shaded portions\n",
    "    leg = True\n",
    "    for i, row in df.iterrows():\n",
    "        if row['expected'] is not None:\n",
    "            # plt.barh(row['type'], row['expected'], color='gray', alpha=0.5, edgecolor='none')\n",
    "            # plt.barh(row['type'], row['expected'], color='none', edgecolor=colors[row['type']], hatch='//', linewidth=0)\n",
    "            if leg:\n",
    "                ax.barh(row['type'], row['expected'], color='none', edgecolor=\"gray\", hatch='////', linewidth=0, label = 'expected')\n",
    "                leg = False\n",
    "            else:\n",
    "                ax.barh(row['type'], row['expected'], color='none', edgecolor=\"gray\", hatch='////', linewidth=0)\n",
    "\n",
    "\n",
    "\n",
    "    # Remove the right and top spines\n",
    "    # ax = plt.gca()\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "\n",
    "    # Customize the ticks and labels\n",
    "    ax.xaxis.set_ticks([])\n",
    "    ax.set_xticklabels([])\n",
    "    ax.set_yticklabels([])\n",
    "\n",
    "\n",
    "    # Text annotations\n",
    "    text_separation = max(df['number_obs']) * 0.03\n",
    "    deviation_for_centering = text_separation * 12\n",
    "    \n",
    "    # text_of_line_separation = line_separation / 3\n",
    "    text_pos1 = max(df[df['type'].isin(consequence_order[:2])]['number_obs']) + text_separation + deviation_for_centering\n",
    "    text_pos2 = max(df[df['type'].isin(consequence_order[1:])]['number_obs']) + text_separation + deviation_for_centering\n",
    "    # print(text_pos1, text_pos2)\n",
    "    \n",
    "    ax.text(text_pos1, 0.05,\n",
    "            # r'$\\omega_{trunc}$ = ' + f\"{t_omega:.2f}\",\n",
    "            r'$dN/dS_{trunc}$ = ' + f\"{t_omega:.2f}\",\n",
    "            # transform=ax.transAxes,\n",
    "            fontsize=13, ha='center', va='bottom',\n",
    "            color=colors['truncating'])\n",
    "                   #f'$p$-value = {p_value:.2g}')\n",
    "\n",
    "    ax.text(text_pos1, 0.45, f'$p$-value < {max(t_pvalue, 1e-6):.1e}',\n",
    "            # transform=ax.transAxes,\n",
    "            fontsize=12, ha='center', va='bottom',\n",
    "            color=colors['truncating'])\n",
    "\n",
    "    ax.text(text_pos2, 1.95,\n",
    "            # r'$\\omega_{mis}$ = ' + f\"{m_omega:.2f}\",\n",
    "            r'$dN/dS_{mis}$ = ' + f\"{m_omega:.2f}\",\n",
    "            # transform=ax.transAxes,\n",
    "            fontsize=13, ha='center', va='bottom',\n",
    "            color=colors['missense'])\n",
    "    ax.text(text_pos2, 2.35, f'$p$-value < {max(m_pvalue, 1e-6):.1e}',\n",
    "            # transform=ax.transAxes,\n",
    "            fontsize=12, ha='center', va='bottom',\n",
    "            color=colors['missense'])\n",
    "\n",
    "    # Update labels\n",
    "    plt.xlabel('Number of Mutations')\n",
    "    # plt.xscale('log')\n",
    "    plt.ylabel('')\n",
    "    plt.legend(loc='right', frameon=False)\n",
    "    # plt.legend(loc='right', frameon=False, fontsize=12, handlelength=3, markerscale=2)  # Increase legend text and marker size\n",
    "    # plt.show()\n",
    "    return fig"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "391b2c70-af12-44f8-85e5-0c95244f6c10",
   "metadata": {},
   "source": [
    "### OncodriveFML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e1c08e9a-7f43-4e66-9edb-1bc14effa9bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_oncodrivefml_side(geneee_data):\n",
    "    legend_fontsize = 12\n",
    "    xaxis_fontsize = 12\n",
    "    \n",
    "    # Extract the necessary values\n",
    "    observed_mean = geneee_data['OBSERVED_MEAN']\n",
    "    background_mean = geneee_data['BACKGROUND_MEAN']\n",
    "    background_std = geneee_data['BACKGROUND_STD']\n",
    "    p_value = geneee_data['pvalue']\n",
    "\n",
    "    observed_color = metrics_colors_dictionary[\"ofml_score\"]\n",
    "    \n",
    "    deviation = abs(observed_mean - background_mean)\n",
    "\n",
    "    # Calculate the Z-score\n",
    "    z_score = (observed_mean - background_mean) / background_std\n",
    "        \n",
    "    # Generate a range of values for the x-axis\n",
    "    x = np.linspace(background_mean - 1*background_std - deviation, background_mean + 1*background_std + deviation, 1000)\n",
    "    # Generate the normal distribution based on background mean and std\n",
    "    y = norm.pdf(x, background_mean, background_std)\n",
    "\n",
    "    # Compute half of the maximum value\n",
    "    mid_y = max(y) / 2\n",
    "    \n",
    "    # Plot the normal distribution vertically\n",
    "    fig, ax = plt.subplots(figsize=(3, 1.75))\n",
    "\n",
    "    background_color = 'dimgrey'\n",
    "    background_color_line = 'dimgrey'\n",
    "    ax.plot(x, y, color = background_color)\n",
    "    ax.fill_betweenx(y, x, color = background_color, alpha=0.2) #, label = 'Randomized\\nmeans' )\n",
    "    \n",
    "    # Arrows with annotations\n",
    "    arrow_props = dict(facecolor= observed_color, edgecolor = observed_color, arrowstyle='<->')\n",
    "\n",
    "    if z_score > 0:\n",
    "        ax.set_xlim(background_mean - (1*background_std + deviation) / 2, background_mean + (1*background_std + deviation))\n",
    "        # ax.text(background_mean + background_std, mid_y * 1.75, f'Randomized\\nmeans', color=background_color, ha='left', va='top')\n",
    "        #ax.legend(frameon=False, loc='upper right', bbox_to_anchor=(1.2, 1.1), labelcolor = background_color)\n",
    "        \n",
    "        # # Set integer tick labels on the x-axis\n",
    "        # x_ticks = np.linspace(background_mean - (1*background_std + deviation) / 2, background_mean + (1*background_std + deviation), num=3)\n",
    "        # x_ticks_int = np.round(x_ticks).astype(int)\n",
    "        # ax.set_xticks(x_ticks_int)\n",
    "\n",
    "        # Arrow 1\n",
    "        ax.annotate('', xy=(background_mean, mid_y), xytext=(observed_mean, mid_y), arrowprops=arrow_props)\n",
    "        \n",
    "    else:\n",
    "        ax.set_xlim(background_mean - (1*background_std + deviation), background_mean + (1*background_std + deviation) / 2)\n",
    "        # ax.text(background_mean - background_std, mid_y * 1.75, f'Randomized\\nmeans', color=background_color, ha='left', va='top')\n",
    "        # Add legend without border\n",
    "        #ax.legend(frameon=False, loc='upper right', bbox_to_anchor=(1.2, 1.1), labelcolor = background_color)\n",
    "\n",
    "        # # Set integer tick labels on the x-axis\n",
    "        # x_ticks = np.linspace(background_mean - (1*background_std + deviation), background_mean + (1*background_std + deviation) / 2, num=3)\n",
    "        # x_ticks_int = np.round(x_ticks).astype(int)\n",
    "        # ax.set_xticks(x_ticks_int)\n",
    "\n",
    "        # Arrow\n",
    "        ax.annotate('', xy=(background_mean, mid_y), xytext=(observed_mean, mid_y), arrowprops=arrow_props)\n",
    "\n",
    "        \n",
    "        \n",
    "    # Legend\n",
    "    legend_handles = [Patch(facecolor=background_color_line, alpha=0.2, edgecolor='none', label='Randomized means'),\n",
    "                      Line2D([0], [0], color=\"black\", linestyle='--', label='Observed mean')]\n",
    "    legend = ax.legend(handles=legend_handles, frameon=False, loc='upper right', bbox_to_anchor=(1.6, 1.1), fontsize=legend_fontsize)\n",
    "\n",
    "    # Adjust the color of the text labels in the legend\n",
    "    for text, color in zip(legend.get_texts(), [\"black\", \"black\"]):\n",
    "        text.set_color(color)\n",
    "       \n",
    "        \n",
    "    ax.set_xticks([])\n",
    "        \n",
    "    # Add a vertical line for the observed mean\n",
    "    ax.axvline(observed_mean, color=\"black\", linestyle='--', ymin=0, ymax=0.5, label=\"Observed mean\")\n",
    "    \n",
    "    # Add a label for the observed mean\n",
    "    if p_value == 1e-6:\n",
    "        #ax.text(observed_score*1.1, max(y)/2, text, ha='left', va='center', fontsize=text_fontsize, color=observed_color)\n",
    "        #ax.text(observed_score*1.1, max(y)/2 - 0.35*(max(y)/2), fr'$\\mathit{{p}}$-value < {pvalue}', ha='left', va='center', fontsize=text_fontsize, color=observed_color)\n",
    "        ax.text(observed_mean * 1.03, mid_y, f'$Score$ = {z_score:.2f}',\n",
    "                color=observed_color, ha='left', va='center', fontsize = 13)\n",
    "        ax.text(observed_mean * 1.03, mid_y - 0.35*mid_y, f'$p$-value < {p_value:.2g}',\n",
    "                color=observed_color, ha='left', va='center', fontsize = 13)\n",
    "    else:\n",
    "        ax.text(observed_mean * 1.03, mid_y, f'$Score$ = {z_score:.2f}',\n",
    "                color=observed_color, ha='left', va='center', fontsize = 13)\n",
    "        ax.text(observed_mean * 1.03, mid_y - 0.35*mid_y, f'$p$-value = {p_value:.2g}',\n",
    "                color=observed_color, ha='left', va='center', fontsize = 13)\n",
    "\n",
    "    # Set labels and title\n",
    "    ax.set_xlabel('Impact score', fontsize = xaxis_fontsize)\n",
    "\n",
    "    # Hide the bottom, right, and top borders (spines) of the plot\n",
    "    for spine in ['left', 'right', 'top']:\n",
    "        ax.spines[spine].set_visible(False)\n",
    "\n",
    "    # Hide the entire y-axis\n",
    "    ax.yaxis.set_visible(False)\n",
    "\n",
    "    # Display the plot\n",
    "    plt.show()\n",
    "    \n",
    "    \n",
    "    \n",
    "#     # Annotate the Z-score, formula and p-value with LaTeX\n",
    "#     formula = (f'$Z = \\\\frac{{{observed_mean:.2f} - {background_mean:.2f}}}{{{background_std:.2f}}} = {z_score:.2f}$\\n'\n",
    "#                f'$p$-value = {p_value:.2g}')\n",
    "#     plt.text(observed_mean - 2, mid_y * 1.25, formula, \n",
    "#              color='black', ha='center', va='center')    \n",
    "\n",
    "    return fig\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a53e2a8-ff63-4c53-9391-8520fa960e69",
   "metadata": {},
   "source": [
    "### Excess of frameshift indels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "fe3ae9de-2601-4b73-8309-bce7a098b1b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plotting_indels_side(data_gene):\n",
    "\n",
    "    # Bar labels and values\n",
    "    labels = ['MULTIPLE_3.non_protein_affecting', 'NOT_MULTIPLE_3.non_protein_affecting',\n",
    "              'NON_TRUNCATING.protein_affecting', 'TRUNCATING.protein_affecting']\n",
    "    values = [ data_gene[x] for x in labels ]\n",
    "\n",
    "    # Colors:\n",
    "    truncating_indels_color = metrics_colors_dictionary[\"frameshift\"]\n",
    "    nontruncating_color = metrics_colors_dictionary[\"inframe\"]\n",
    "    colors = ['none', 'none', nontruncating_color, truncating_indels_color]\n",
    "    edgecolors = [nontruncating_color, truncating_indels_color, nontruncating_color, truncating_indels_color]\n",
    "\n",
    "    # Create the plot with a smaller height\n",
    "    fig, ax = plt.subplots(figsize=(3, 1.75))  # Adjust the height and width\n",
    "\n",
    "    # Plot the bars horizontally with the specified colors and hatches\n",
    "    bars = ax.barh(labels, values, color=colors, edgecolor=edgecolors, linewidth = 2)\n",
    "\n",
    "    # Remove the right and top spines\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "\n",
    "\n",
    "    max_syn = max(values) * 1.55\n",
    "\n",
    "    # Vertical lines\n",
    "    line_separation = max(values) * 0.03\n",
    "    text_of_line_separation = line_separation # / 2\n",
    "    line_pos1 = max(values[:2]) + line_separation\n",
    "    line_pos2 = max(values[2:]) + line_separation\n",
    "    line_margin = 0.04\n",
    "    ax.axvline(x=line_pos1, ymin = line_margin, ymax = 0.5 - line_margin, color='black',\n",
    "               # linestyle='--',\n",
    "               linewidth=1)\n",
    "    ax.axvline(x=line_pos2, ymin = 0.5 + line_margin, ymax = 1 - line_margin, color='black',\n",
    "               # linestyle='--',\n",
    "               linewidth=1)\n",
    "\n",
    "    # Annotation for \"PA\" and \"NPA\" next to vertical lines\n",
    "    ax.text(line_pos1 + 2.2*text_of_line_separation, 0.5, 'Non\\ncoding', ha='center', va='center', fontsize=9,\n",
    "            rotation = 270\n",
    "            # bbox=dict(boxstyle='round,pad=0.3', fc='white', alpha=0.5)\n",
    "           )\n",
    "    ax.text(line_pos2 + text_of_line_separation, 2.5, 'Coding', ha='center', va='center', fontsize=9,\n",
    "            rotation = 270\n",
    "            # bbox=dict(boxstyle='round,pad=0.3', fc='white', alpha=0.5)\n",
    "           )\n",
    "\n",
    "\n",
    "    # # Arrows with annotations\n",
    "    # arrow_props = dict(facecolor='black', arrowstyle='<-')\n",
    "\n",
    "    # # Arrow 1\n",
    "    # ax.annotate('', xy=(line_pos1 + 15, 0.5), xytext=(max_syn/1.25, 1.2),\n",
    "    #             arrowprops=arrow_props)\n",
    "\n",
    "    # # Arrow 2\n",
    "    # ax.annotate('', xy=(line_pos2, 2.5), xytext=(max_syn, 1.75),\n",
    "    #             arrowprops=arrow_props)\n",
    "\n",
    "    # formula_ratio = (f'$\\\\frac{{{data_gene[\"pa_TRUNC/NOTTRUNC\"]:.2f}}}{{{data_gene[\"Npa_NM3/M3\"]:.2f}}} = {data_gene[\"pa/Npa\"]:.2f}$')\n",
    "    # formula_ratio = (f'$\\\\frac{{{data_gene[\"pa_TRUNC/NOTTRUNC\"]:.2f}}}{{{data_gene[\"Npa_NM3/M3\"]:.2f}}} = {data_gene[\"pa/Npa\"]:.2f}$')\n",
    "    ax.text(max_syn, 1.5, f'$Score$ = {data_gene[\"pa/Npa\"]:.2f}',\n",
    "            color=truncating_indels_color, ha='center', va='center', fontsize = 13)\n",
    "\n",
    "    pvalue_tag = f'$p$-value = {data_gene[\"pvalue\"]:.2g}'\n",
    "    ax.text(max_syn, 0.8, pvalue_tag,\n",
    "            color=truncating_indels_color,\n",
    "            fontsize=12, ha='center', va='center'\n",
    "           )\n",
    "\n",
    "    # Add title and labels\n",
    "    ax.set_xlabel('Count')\n",
    "\n",
    "    # Customize the ticks and labels\n",
    "    ax.xaxis.set_ticks([])\n",
    "    ax.set_xticklabels([])\n",
    "    ax.set_yticklabels([])\n",
    "\n",
    "    # Update labels\n",
    "    plt.xlabel('Number of indels')\n",
    "    plt.ylabel('')\n",
    "   \n",
    "    return fig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3105e23b-1e4c-4a21-a5a6-d9bd486c18df",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a166b633-12d8-4384-b07f-417e0f51fa0a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "7952d410-2dfd-4478-ac70-f3b13d340822",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TP53\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYMAAAC0CAYAAAB2bECzAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAPaVJREFUeJzt3XlcFPX/B/DX7MFesJwCIiiI4n2kmEcqmrdp5tH3V15gpUWHmFflEWhWWh6VZ1ZqqWmmZXlfeKYWXnnjiSLIDbssuwt7fH5/bDOycsgN6vv5eOxDmP3MzGd2cd7zuTnGGAMhhJCnmqi6M0AIIaT6UTAghBBCwYAQQggFA0IIIaBgQAghBBQMCCGEgIIBIYQQUDAghBACCgaEEEJAwYAQQggoGBBCCAEFA0IIIaBgQAghBBQMCCGEgIIBIYQQAJLqzgAhpXH37t3qzgLq1q1b3VkgpMJRyYCQKtKtWzdwHGf3EolEcHZ2Rvv27bFx48YqzY+/vz/CwsKq5FzdunVDt27dquRcVXldTxIqGRBShZ555hksW7ZM+N1isSA+Ph6LFi3Cq6++ChcXF/Tt27cac1g58l8zqZkoGFQSg8GA27dvIyAgAAqForqzQ2oItVqNDh06FNjev39/eHp6YtWqVU9kMGjatGl1Z4E8AlUTVYKcnBzMmTMHS5YswSeffILs7OzqzhKp4eRyORwcHOy2GQwGfPTRR2jYsCFkMhnUajV69eqFc+fOCWnCwsLQs2dPrF69GkFBQZDJZGjVqhV27txpd6zz58+jV69ecHR0RL169bB+/foCeTAajfjkk0/QuHFjyOVyNGzYEPPmzYPVahXSdOvWDW+++SY+/fRT1KlTB0qlEv3790dycjJWr16NBg0awNHRET179kRcXJzdfnw1UVRUVIHqMv6Vv3rn6NGjCAkJgVKphJubG0JDQ5Gamlrq63rYmjVrIJfL8ddff6Fdu3aQy+Vo1KgRtm3bhtjYWPTo0QNKpRINGjQoUHV39+5dvPrqq3Bzc4NSqUSPHj1w9uxZuzRxcXEYPXo0fHx8IJVK4enpidGjRyM9PV1I4+/vj8jISEyZMgVeXl5QKBTo06cPrl27JqRJS0vDyJEj4e3tDblcjtatW2Pt2rWPvL6yopJBJdi6dSsyMzMBAFlZWdi2bRuGDx9ezbkiNQFjDGazWfjdbDbj3r17mD17NrKzszFq1CjhvdGjR+Pw4cOYO3cuAgMDce3aNcycOROvvPIKrly5Ao7jAACnTp1CYmIiZs+eDWdnZ8ycORPDhg1DQkICXF1dkZCQgK5duyIwMBDr16+HVqvFBx98gOTkZLt8DRw4ECdOnEBkZCRat26NgwcPYvr06bh58yZWrlwppN24cSOeeeYZ/PDDD7hz5w7effddhISEQKFQYP78+cjIyMD48ePxzjvvYMeOHQU+gzfeeKNA6Wfu3LnYsWOHEAyOHDmCnj17okePHti0aRMyMjIwc+ZMdO/eHTExMVAoFCW6rqKYTCa88soriIyMhJ+fH6ZOnYoRI0bAy8sL4eHhmDhxIubOnYvQ0FB07twZvr6+SEtLQ6dOnaBUKrFkyRKoVCp89dVX6Nq1K/755x80adIEer0e3bp1Q61atbB06VK4uLjg2LFjmDVrFuRyud3n+PXXX6Nz585Ys2YNMjIyEBERgdDQUJw4cQIAMGLECKSkpGDFihVQq9VYu3YtRo8eDT8/v8ppf2GkQun1ehYREcHCw8PZTz/9xMLDw9l7773HDAZDdWftiXDnzp1qf5VVSEgIA1DgxXEca9myJfv111+FtLm5uaxPnz5s48aNdsdYsGABA8ASExMZY4yFhoYyAOzGjRtCmsOHDzMAbPPmzYwxxiZPnsyUSiVLTk4W0pw8eZIBYKGhoYwxxnbu3MkAsHXr1tmd75NPPmEA2KVLl4RrkMvlLCMjQ0jTp08fBoDdvHlT2Pbuu+8yZ2dnu2sPCQkp9HPZuHEjA8CWLVsmbOvUqRNr3rw5M5vNwrbY2FgmFovZkiVLSnxdhVm9ejUDwJYvXy5s27BhAwPAZs6cKWw7deoUA8B+//13xhhj06ZNY3K5nMXFxQlpcnNzWf369dmwYcMYY4ydPXuWde7c2e77YIyxgQMHsqCgIOH3evXqMX9/f7vrmzVrFgPA0tLSGGOMyWQyNmfOHOF9i8XCJk2axI4ePVrktZUHVRNVsAsXLiAvLw8uLi7o2rUrnJ2dYTabceHCherOGqkB2rRpg5iYGMTExOD3339H8+bNERQUhI0bN2LYsGFCOgcHB+zevRv/93//h/v37+PIkSNYuXIltm/fDgDIy8sT0taqVQuBgYHC776+vgBs1ZWArbqlY8eO8PT0FNK0b9/erovsoUOHIBaL8X//9392+R05cqTwPq9JkyZwdXUVfvf29katWrVQv359YZu7uzs0Gs0jP48zZ85gzJgxePPNNxEeHg4A0Ov1OHnyJF544QWhJGU2m1G/fn00adIE+/btK/F1FadTp0521wDArj3H3d0dgK10DwAHDhxA69atUadOHSFPIpEI/fr1E/LUunVrHD16FAEBAbh16xb27NmDBQsW4MqVK3bfGQC0a9cOYrFY+P3h76179+6IjIzE//73P6xZswapqamYP38+OnfuXKLrKy0KBhXs4sWLAICGDRuC4zg0atQIgK1ukxAnJycEBwcjODgYL730Eg4cOACNRoPevXsXqA/fs2cPmjRpAh8fHwwYMAA//fQTZDIZAFu1Dk+pVNrtJxLZ/lvzdf0ZGRmoVatWgbzUrl1b+DkjIwMeHh6QSOxrjvmbJH9DBGyN4A97OA8lkZSUhEGDBiE4OBiLFy8WtmdmZsJqtWLevHmQSqV2r4sXLyIxMbHE11Wc0l5Heno6Tp48WSBPS5cuhUajgV6vBwAsXLgQXl5eCAwMRGhoKKKjo6FSqey+s8LO9fD3tnHjRkyaNAn//PMPxowZAx8fH/Tt2xe3b98u0fWVFgWDChYbGwsAwpMa/7R048aNAn8MhHh6emLp0qW4d+8eIiIihO03b97ESy+9hFatWuHGjRvQarU4duwYBg4cWOpzeHh4FFqPnr9B083NDWlpaXbtGQBw//594RgVKTc3F4MHD4ZIJMKWLVsglUqF99RqNTiOw8SJE4VSVP7XmjVrSnxdFcnFxQUhISGF5ikmJgYymQw///wzJk2ahMmTJyMlJQVJSUnYsWMHgoKCSn0+Z2dnzJs3D3Fxcbh69So+//xzHDt2DG+//XYlXB0FgwqVlZWF7OxscBwHHx8fAICPjw9EIhE0Gk2l/ZGSx9uQIUPQt29fbNiwQaiOOX36NIxGIz766CO7KqBdu3YBgF0Pn0fp0aMHjh8/joSEBGHb5cuXcevWLeH3kJAQWCwW/PLLL3b7rlu3DgAqvGpi3LhxOH/+PP74448CT/dOTk5o06YNrl69KpSigoOD0axZM0RFRQmfUUmuqyKFhIQgNjYWQUFBdvlat24dvv/+e4jFYhw7dgzOzs744IMPhOvS6XQ4duxYqb6zO3fuwM/PD5s3bwYANGrUCFOnTkWvXr0QHx9fKddHwaAC8cU3d3d3oZugg4OD8FR1586dassbqdm++uorSKVSvPfeezCbzWjTpg0kEgk++OAD7Nu3D9u3b8fQoUOF3jl8vXJJTJgwAW5ubujTpw+2bNmCTZs2YdCgQXZdWfv164fu3bvjzTffxJdffon9+/dj5syZmD17NkJDQyt0nMDChQvx008/YcaMGbBYLDh58qTw4rtpfvbZZ9izZw9GjBiBnTt3Ytu2bejbty/279+PNm3alPi6KtLEiRNhtVrRs2dPbNq0CQcOHMC4cePw9ddfo3HjxgCAZ599FhqNBpMmTcKhQ4fw888/o0uXLkhKSirVd1avXj34+vpi/PjxWLVqFQ4fPowFCxZg586ddm1LFYm6llYg/onk4TpLLy8vpKSk4O7du2jbtm11ZO2J8aTOC9SoUSNERERg/vz5WLJkCSZMmIANGzYgKioKL774Itzc3NChQwccOnQI3bp1w9GjR9GiRYsSHdvd3R3Hjh3DhAkTEBYWBkdHR0ydOtWuFMBxHLZv346PP/4YX3/9NVJTUxEQEIDPPvsMEydOrNBr/fPPPwEA06ZNw7Rp0+zeq1evHuLi4tC7d2/s2bMHs2bNwrBhw+Dg4IC2bdti//79QiNvSa6rIvn4+OD48eP46KOP8NZbb8FoNCIoKAg//PADXnvtNQBAaGgobt++jVWrVmHZsmWoU6cOXnjhBbz99tsYN24cLl++XOLA+vvvv+Ojjz7CzJkzkZaWBj8/P0RGRuLDDz+slOvjGFVkV5iFCxfixo0b6Nu3L1q3bi1sP3PmDPbu3Sv8hyeEkJqGqokqEN/Y5uXlZbed/z1/3SYhhNQkFAwqSHZ2tlAnyPdP5nl6eoLjOOh0uhL1vSaEkKpGwaCC8KUCtVpdoAFLKpUKfZpLMlSeEEKqGgWDCsIPhHm4VMBzc3MD8CBoEEJITULBoILwwaCowTl8MODTEUJITULBoIIkJSUBQKHD4/Nvp2oiQkhNRMGggvCji/kSwMP46qOUlJQqyxMhhJQUBYMKYDabhYm8XFxcCk2TfwbEh2cvJISQ6kbBoAJkZGSAMQaJRAKVSlVoGoVCIfQyojmKCCE1DQWDCpCWlgbgwWyLheE4TuheyqcnhJCagoJBBeDbAYqqIuLx7z88bz15uvz1118YOnSosLZt/fr1MXbsWFy5cqVAWo7jEBUVVfWZrAL8WsikZqBgUAFKGgycnZ0BUDB4ms2bNw9dunSBTqfDokWLsGfPHkyfPh1nzpxBmzZtCizATkhVoVlLKwBf7VNUTyIe/z5VEz2dtm/fjg8//FCYGpoXEhKC0aNH49VXX0VYWBhatGiBZs2aVWNOydOISgYVgG8Qzr8ubGH496kB+ek0a9YsNGrUCLNmzSrwnlQqxbfffguxWIx58+bZvafVajFy5Eg4OjrC09MT48ePF5ZYBGxTpw8aNAju7u5QKpXo2LGjsAgO7+LFixgwYADUajXUajUGDx5stwjMoUOHwHEcvv32W9SrVw9eXl746aefwHEc/v33X7tj7dq1CxzHISYmBoCtA8Wbb74JLy8vyOVydOjQAQcOHLDbx2g0YuLEifD29oajoyNee+01GI3GR35mHMdhxYoVCAsLg7OzM9zc3DB+/HgYDAZMmTIFtWrVgru7O9544w2741mtVsydOxcNGjSATCZDUFCQ3dKaAGCxWDBv3jw0b94cCoUCKpUKnTp1QnR0tJAmKioKDRo0wI4dO9CyZUvhWD/++OMj8/7YYaRcrFYri4iIYOHh4ezMmTPszp07Rb7Onj3LwsPDWUREBLNardWd9ceSTqcr8mUwGEqcVq/XlzltWaSmpjIAbPLkycWmGzRoEFOr1cLvAJhYLGYDBgxgO3fuZAsXLmQKhYINHz6cMcaYxWJhTZo0Yc8//zzbsWMH27t3L3vhhReYRCJh169fZ4wxFhsby5ycnFi7du3Yli1b2KZNm1jLli2Zt7c3S05OZowxdvDgQQaAubm5sV9//ZWtXbuWabVa5uTkxD788EO7PI4cOZI1atSIMcaYwWBgrVq1Yl5eXuy7775jO3bsYEOHDmUSiYQdOHBA2Gfo0KFMqVSyRYsWsZ07d7JBgwYxqVTKHnULAsDUajV788032YEDB1hERAQDwBo1asT+97//sd27d7Np06YxAOyLL74Q9hs3bhyTSqUsMjKS7dmzh02bNo2JRCI2e/ZsIc3kyZOZQqFg33zzDTt06BBbt24da9iwIXN1dWU6nY4xxlhkZCRTKpXM39+fff/992zfvn2sd+/eDAC7cuVKsXl/3FAwKCetVsvCw8NZeHg4u3nzZrHB4ObNm0JarVZb3Vl/LAEo8tW/f3+7tEqlssi0ISEhdmk9PDyKTBscHFzufMfExDAAbOnSpcWmmzRpEgPAMjIyhOtt2rQps1gsQpqvvvqKcRzHrly5wu7fv88AsHXr1gnvZ2Vlsffff59duHCBMcbY8OHDmaenJ9NoNEKa9PR05uzsLAQnPhhMnz7dLj9hYWEsICBA+F2v1zMnJyc2Z84cxhhjK1euZADYyZMnhTRWq5V17dpV+NwuXrzIALAlS5YIaSwWC2vatGmJgkH79u2F300mE1OpVCwgIICZTCZhe/PmzdmgQYMYY7bgx3Ecmzt3rt2xZsyYweRyOUtLSxM+l0WLFtml2bJlCwPAjh8/zhizBQMAbP/+/UKaO3fuMABs/vz5xeb9cUPVROXE1/87OjpCIim+CUYikcDR0dFuP/J0YP+tIZV/4ffC8H9DLN+aU8OGDYNI9OC/6pAhQ8AYQ3R0NLy8vNC0aVOMHTsWYWFh+OWXX8AYw8KFC9G8eXMAwIEDB9C9e3colUqYzWaYzWao1Wp06dIF+/btszv/w6unjRo1Crdv38bff/8NANi2bRt0Oh1GjBghHNvb2xtt27YVjm2xWDBw4ECcOnUKmZmZOHr0KABg0KBBwnFFIlGJl2/s1KmT3efj4eGB4OBgu/9v7u7uwsDP6OhoMMYwcOBAIU9msxkvvvgijEajkJ/169djwoQJSEtLw4kTJ7BmzRphzeeHB4Z27NhR+NnX1xdA6ZYefRxQA3I58fX//BiCR1Gr1dDpdEhPT0dAQEBlZu2JpNPpinxPLBbb/V7c1B/5b64AEBcXV+K0ZVGvXr1Hngew1f87OjradUbw9va2S+Pp6QkAyMzMBMdx2LdvH+bMmYPffvsNP/74I6RSKQYPHozly5fDzc0N6enp+OWXXwpdDvLhubQeXpipe/fu8PPzw8aNG9G+fXthTV9/f38Atr//pKSkIoPc/fv3kZGRUei5Hl4etiiF/d9SKpVFpuf/TxbVCM9PFnnq1Cm8/fbbiImJgUKhQLNmzYTviT20AGT+8/F/D6VZ4P5xQMGgnEobDJycnOz2I6VT1AjvqkxbFp6enujQoQN+++03zJkzp9D+9VqtFnv37rV7ggZsN/38+EkR+aDg4+ODZcuWYenSpfj333+xefNmzJ07F25ubli+fDlcXFzQs2dPTJo0qcA5H1Wa5TgOI0aMwNq1a/Hxxx9j165dWLp0qfC+i4sLGjZsiJ9//rnQ/QMCAoSZfJOTk+3WsK6s/wN8F+/o6Gjh/1t+devWhVarRd++fdGyZUtcvHgRTZo0gUgkws6dO7Fly5ZKyVdNR9VE5VTaYMCPNaBg8PSJiorC1atXMWPGjALvWSwWvPXWW0Ivmfx2795t9/vGjRvBcRy6deuGEydOwMvLCzExMeA4Dq1bt8acOXPQokULxMfHA7B1Xb18+TJat26N4OBgBAcHo23btli0aBF+//33R+Z71KhRSEhIQGRkJDiOw8svvyy8FxISgvj4eHh6egrHDg4Oxv79+/HFF19AIpHg+eefBwD8+uuvdsfdtm1byT64UgoJCQFgq4rNn6f09HTMmDED6enpuHr1KtLT0xEREYFmzZoJT/t8L6wn7am/JKhkUE78Uxt/k38U/qnl4ac98uTr06cPFixYgClTpuDs2bMICwtD7dq1cfv2bSxfvhznzp3DDz/8gFatWtntd+rUKbzxxhsYPnw4/vnnH0RGRuL1119Hw4YN4efnB6VSiVGjRiEqKgre3t7Yv38/zp07h4iICADAxx9/jI4dO2LAgAEIDw+HXC7Ht99+i61bt2Lz5s2PzHfTpk3Rtm1bLFu2DEOGDLH7Wx8zZgyWLFmCXr16Ydq0aahbty727duHefPm4b333oNUKkWDBg0wbtw4TJ8+HSaTCc888wzWrl2L8+fPV+wH/J/mzZtj5MiRGDt2LOLi4hAcHIzY2FhMmzYNAQEBCAoKgk6ng1qtxqeffgqJRAKpVIrNmzfjhx9+APDktQeUBJUMyom/qZemzSD/fuTpMnHiRPz1119Qq9WYNGkSevXqhcjISLRo0QKnT5/GyJEjC+wzY8YMZGVlYcCAAfj6668xdepULF++HAAgl8uxd+9eNGvWDBEREejTpw+2bt2Kb7/9FmFhYQCAli1b4ujRo+A4DqNGjcKwYcNw//59bN26FUOGDClRvkeNGgWLxVIgfyqVCkeOHEHnzp0xdepU9OvXD7/99hvmzp2LhQsXCumWLVuGDz74AEuWLMHgwYOh1+sxffr0Mn6Kj7Z69WpMmjQJK1asQJ8+ffDpp5/ilVdewb59+yAWi+Hs7Iw//vgDjDG8/PLLGDVqFO7evYsjR47AyclJaGR+mnDs4ZYSUipTpkxBTk4OxowZU6DxrTApKSlYtWoVlEol5s+fXwU5JISQR6OSQTnk5eUJxcnSlgz0ej2ta0AIqTEoGJQD369ZIpFALpeXaB+ZTCZ0w6OqIkJITUHBoBz4m7mTk1OJp+LlOE7o7kbBgBBSU1AwKAe+e2hhfZmLQ2MNCCE1DQWDcihrMKAVzwghNQ0Fg3Io7RgDHp+eqokIITUFBYNyoGBACHlSUDAoh9IOOOPRwDNCSE1DwaAcNBoNgLIHg6ysrAKzIxJCSHWgYFBGBoMBubm5AMrem8hsNhc7JTMhhFQVCgZlxFfxyGQyODg4lGpfiUQChUJhdxxCCKlOFAzKiF+wo7SlAh4NPCOE1CQUDMqotOsYPIzfjw8qhBBSnSgYlFFFlQxoFDIhpCagYFBGfDAo7RgDHr8flQwIITUBBYMyKuuAM17+7qWEEFLdKBiUEX8TL2ubAR9EKBgQQmoCCgZlYLVahQFnZW0z4IOIVqt9KhffJoTULBQMykCn08FisQAoezBQqVQQiUR2gYUQQqoLBYMy4NsLVCoVxGJxmY4hEomgVCrtjkcIIdWFgkEZ8D2AHB0dy3UcGnhGCKkpJNWdgccRHwzK2njMU6vVuH//Po01IKSGWT8nHal3TZiw0lvYlp5oxscvJhS7X8QKLwQFF1wPvbT73jhnxJ9Ls3D3ch6UahFadVPghTdd4OhStpqIkqBgUAYVFQz4kgGNNSBVSZ9tRfTPWlw4rEdKvBnMCji5iVC7vgNadFWgy9CytYM9KY5vzcbxrTo0bCOz2+7oKkLobPcC6U25DJu+zICTqxi+QdJCj1mafa+dMmLpe8lQOInQZ4waIjGHgxu0iI0xYvIqbyjVlRMQKBiUQUUFA1rkhlS1+7fysOTdFORorAjuo0SHFx0hEnFIjTfh/BEDADy1wcBqYdi9SoOdKwvv0CFTiPBs/4JVw7/Oz4DFDITN8SjyRl2afX/9MgOciMOkVd6o5WsLEK26K/HZK4nYvUqLIRNcy3qJxaJgUAblHXDGo2BAqhJjDD98mAZjjhVTf/SGTwP72XaHTmTQpFmqPF9WK4PVAkikXJWfm2fKZfgy7D4SrpvQ/gUVYmOMJdrv3rU8HP4lGx0GqtDgmYLVQ6XdNz3RjMSbJjw32FEIBADg7S9Fiy5K/L1dR8GgJinvgDOeWq0GjC5IjS1YfCRPvlyDFWcP6NEyRAmlU+X35Ui4bsL9WyY800NZIBAAAMdxcKllf0vIzrTgwDotLhw1ID3BDIkDh9r1pej+qhPa9FQJ6VLvmbD7Bw2u/m2ELtMCF08JOg91RI+RaohED27yW7/JxL6ftJj5qw+O/ZaNswf00KRaELHCCw3bypGRZMa+HzW49JcBmjQLXGpJ0Pp5JfqPc4ZMUfRn9E7wnWKvvf9YZ7zwpkuR75vyGIw5Vrz2uQfa9lJh5sB7xR6P9+fSLEjlHAa+XfSxS7NvVooZAODToGB1Uy0/Cc4dtCIzyQxX74q/dVMwKCWLxYLs7GwA5QsGmUkM/2x0B86PRa4sC7nGPMjkpVsXgTzespIt2Ph5BjYvyMTzw53Q7RV1pQaFXL1tcGNqvAlGvRVyZfHnun8rD4vfToE+24qOLzrCb7gU2ZlWXD5hwP1bJiHdlZMGrJycCudaYnQZ5gSVWoRLxw3Y+k0WsjOsdk+y8bF5kMo4fDspBR6+EvQcrYbJyFAnyAF3LuVi8bspkKs4dBzoCBdPMe5czkP0z1qk3TNj7Je1isxrYfXx+dVpWPz/LbmKQ+RvdSCWlLx0cvdqLi79ZUCPkWo4e5TuVlrUvg7/BbxcfcEVEHM0tu9Pm26hYFATaDQaMMYgEomgUqkevcNDMpMYTmwGLh4ClGoJRP7RsHqchTZ7JmrJi/5jJ08eL38pZv3hg30/arF3jRbRP2dXalDwa+wAdx8x7l0zYXq/e2jaSYGgYDmadJDDo479k6hBZ8XS91IgkgDTNtaGp9+D9/uMcYYp13azSkswYeWUVPg1dsA7iz2Fp/cuw5zw1bgkHNygRb83nKFwtG2/F5sHUy5D+wGO6Pvag2pWXZYFy99PgV8jKcK/8oSD3Jb+ucGAk6sIu1dpkXbPBA/fwhtoC6uPLw2RiCt1R/ujm3UQiYGQ/yt9G0tR+9auL4VcxeFctB69w9TgOFtwMuUyXD5pa9Mx5VXOUrkUDEop/xgD/osqqV/nMNw6A4glgJsPoK4FxMcHwppZF2umZcNRReshP63qt5Ih7Z4JO7/TYNf3GgwId0afMJcKPYeDXIRJq7yxf60WZw/ocWaf7QUAjdvLMXKmu/DEufsHDTKTLZiw0ssuEPCkMtvf/p5VWpiMDMNnuBeoxmnYVo7rZ3KRds8Mv8YOyEw2Q5dlRUBLGfqMsS9V71mtQY7GiiET3JBnZMgzPmi7qF3f9lSfEm8uMhjosopv63CQc0KAqQh5RitO7clBi64KuNcu3W20uH0lUg7Pj1Bj50oN1sxIQ+8xzrBagO3Ls5BnsN0fROLKaVuhYFBKfGNvWaahEIkA/Pc9cpztd4lUBJPZCrPFBFHldSEmNRzH36eY7We+nt1gMKBfv34AgF27dgnLpZaVs4cEQ993w9D33ZCWYMKVE0Yc3pSNq38b8dOsdEQs9wJjDDG7clC3qQMatim6UdRqZTh3UI/AZ2Tw9i94k+YfbRwUtmuJv5oHAGj/gsruQYoxhtN79LBagLkj7xd5PkUxpaUPehZfx/+oNoPSio0xIs/A0KZH6WsHHrVvvzecYci24tAv2Ti1xxasW3RRoNdoNf5YkgWVunKqEikYlFJ5FrUZOo2DJoXhxBbgQjSg1wDqIC0MbA9aDeqD/v37V3R2SQ2XmWzG3jVaHN+aDZlShAHhLgj5n5NQraJQKHDo0KFKObdHHSm6DJPimZ5KfNjrHm6eNYIxBm26FZo0C1o/ryx2/6wUC/RaK7wDCn9av3/DBJmSQy1f220mPtYWDBq1sw8w2Rm287V/QYVn+xd9c/UNKrre/72lnsXm1aNOxd7qLh83QCIFmnUufXB+1L4iEYdhk9zQO8wZKXdNcPWWwL22BH8uzYRIDLiVsiRSUhQMSqm8y106e3LoGw50HGoLCucPNAGkPkh/7mJFZpM8BpJum/D5iETIFCL0G2sfBKqSRMqBEwFylQgcx8GY898suo+ojeDfLqxLqCbNgkvHDWj9vFKo1rgXmwe5ikMtP/vbjkFnO5+LpxiN25et5FPW/crq1r+5qNtUVqbv61H7ntqdA7WHGEHBcqjdH1QX3DibC7/GDkIVXUWjuYlKqcLGGHhy6BvOofM7VwG/g8jMolHITxu32mKM/NgDs/6og76vOVdqILhxzijcdB+263sNrBYIT+WuXmJIpMDVvw2wmO3bsRhjwjYXLzEUjhxunLGVKHimPIafItPAGEPf1x/8P4mPzYNvI4cCbW2uXmJIHIB/DxmEhun8dFmWAvmoThYzQ9JtE/walb73X0n2jf5Zi01fZNhd88Vjetw8l4uQlytvQCCVDEqpooIBz7ueAnC7jqws70cnJk8UB7kI7foWX+eck5MDf39/AEBcXFyZerABwI4VWbh7JQ8tuipRr6kDZEoRNKlm/HvQgPjYPDR+Vi70d3eQi9D1ZSdE/5yN+a8loW1vFeRKDinxZpyL1mPqj95wdBGD4zj0ec0ZW7/JwrKIFLTsqoQxx4oT23RITzBjzGe1hLaEHI0FmUkWtO5esOrJQS5C91fV2PejFp+PSET7Fxzh6CqCJtWCxBsm3DhrxOd7fMt03ZUhI8kMswlw9S66kS/tngm3zueifkuZXaN3SfbtFarG9x+kYfn7KWjdXYn0+2ZEr9eiSUc52vUr2/dfEhQMSqmiBpzxaPlL8ihpaWnlPsbzw9U4f9iAWxdycfm4AQadFUq1CH6NHRD6iTva9bVv1B0c4QovfymO/abDru+yANjqqtv1VdlNltZzlO3v99gWHX6dnwGVsxgN28rw+txaqJNvYBvfXuDXuPAn4kHvusAnUIojv2bjwDot8owMancRfIMcMGyiW6l77lWmnCxbCUuuKrokd/1sLtbNSsfISHe7YFCSfZ/pocKYT4G9P2qwZWEmnNzF6DnKWZinqLJwLH/5jhQrLy8PEyZMAABMmDABcnnphp8XJjc3F4sWLQIALFy4sEKOSZ4cOTk5wlTpOp2uzCUDQh6F2gxKga8ikkqlkMlkj0hdMjKZDA4ODnbHJ4SQqkbBoBT4m3VZBpwVh3/yo6msCSHVhYJBKfB1txXVeMzj2w0oGBBCqgsFg1JITU0FUPHBgBa5IYRUN+pNVAp8ycDFxaVCj8sfryJ6jZAni0gkQnBwsPAzIZWFgkEp8KOPXV0rdnEJPhjQWsjkYQqFAjExMdWdDfIUoGBQCnwDckWXDPjgQtVE5Gkyc+A9uNeW2C06/zi4fMKA3T9ocPdKHjgRENBChoHhLghoUTE9DHlpCSb89lUmrp/OBQA076zAkPdd4eRqP2AtO9OCP5dm4cIRPUy5DL6NHDDoHRfUb1W6bupU7iyhvLw8YVGbim4z4IOLVquFyWQqPjEhpNpcO2XEsvEpMGRb8eLbLug/1hlp90z4alwSbl/MrbDz6LIs+PqtZMRdyEOv0Wr0GKHGhSN6LH4nGWbTg6FhxhwrFo1Nwpl9Oegy1AkD3nKBJtWCb95OQcKNvFKdk4JBCfFVOA4ODhU+MEyhUEAqldqdhxAA0Ov18Pf3h7+/P/R6fXVn56m3eUEGXLzEmPKjN54foUav0c6YvLo2HBQibFuWVWHniV6vRVaKBeOXe6F3mDP6vu6MN+bVQsI1E/7erhPS7V2jQcodM95a5IkX3nRB91fVeP87W0lr34/aUp2TgkEJ5Z+6uqKHxnMcJ3QvpWBA8mOM4c6dO7hz5w5osoDqpdfa5kpq00tlt1CO2l2Mhm1kuH2+4koGp/fq0bCt3G568MbtFfCqJ8GpvbaHAsYY/t6Rg2adFXbrTjh7iDFkgisatildtRW1GZRQZXUr5Tk7OyM9PR0pKSlo1qxZpZyDkF3fZ2H7Cg3eW+qJv7bqcP20EcYchtr1pRgc4Yqg4KJLvRs+T8fxrTp8tssXTm4P6q1NuQwf9IpHq25KhM72AGMMx7bocOJPHZLiTLCYGdxrS9BhoCN6haqLfJgqqg2hsO03zxmxY6UGcf9VzfD19v7Ni78BvhN8p9j3i1sER64S4eMtPsJiPfnpsqwFFqcqax71WgvSEsx4pkfBSf38Gjvg4jHb8pfpiWZkpVjQc7TtO2OMIdfAIFfaJhosLQoGJcQHg4puPOZR91JSFe7FmsCJgNUz0hDYSoYBb7kgM9mCgxu0WBaRgsgtPkUutv5sPxWObdHhbLQeXYc9uNlcPKZHrp4JM2puX56F3au0aD9AhecGOyJXz/D3Dh3+WJIFtYcYHQaUb73iyycMWPF+CnyDHDDgLReY8xhObtNh0bgkvLfUCw2eKTqghc52L/bYdRoWPbW0SMzBs27BhXwSrufh1r+5aNLxwXnLk8esFNsSns6eBWc2VXuIYcxhMOisSI0327a5ibH1m0wc+y0bBh2Dh68Ewya6okXX4hcnehgFgxLiq28qKxjwPYr4oENIZYi/lgdmBXqHOqPHyAcz73rVk+DHj9MRszsHvcMKL/3WbyWDu48YZ/bl2AWD0/v0cHITofGzcljMDId+yUbb3kqMjvIQ0nQc5IgPe8fj3AF9uYKB1cqwcW4G6jWT4f2VXsIsniH/54TPh9/Hr19m4KOffYrc/9n+5QtEDzPqrfgx0vYA12eMc4Xk0ai3VQc6yAuWQBz+W9gm12CFPts2A+r2FVkQiTkMneQGkQjYv1aLbyen4t3FnqVa9IeCQQlV1hgDHn9cajMglcWgsyIj0YyAljK7QAAAjZ61Pamm3zcXuT/HcQjuo8LeH7XQpFng7CFGrsGKS8cM6PSSo3DTm7vXr8BiNDlZFshVIuQaCl9gp6TuxeYhPcGMrsMchZshr0UXBaJ/zkZmshmuXoXf2nRZlmKP7yDn7NoDipNntGLF+ylIuGZC39edhaf98uaRWW2fXXEtkxzHwZxnS6fXWhH5mw+UaltJokVXJaJeSsAfS7MoGFQ0xphQfVNZwYAvcWRkZIAxVqPmbydPhnuxeWAM6PRiwWmw2X/3LJlShDyjFQad/c1cpuQgV4rQrp8Ke1ZrcS5aj5D/OeHCYQPyjMxukR6xFLh4zIDzhw1IvmNCarwZeq3V7jxlxVeN/P51Fn7/OqvQNJnJliJvtB/0vFfs8YtrM8hPr7Vg2YRU3D6fi06DHDEw/ME+Jc2jyllU+Of831oHeYWs+sZvk6s4yBS2dK26K4VAAABKJxFadFXg7+05MOZYi107IT8KBiWQnZ2N3FxbI1BlVRPxDdO5ubl2c9iTpxvHcWjatKnwc3ncu2brd163acEGzLiL/y0+08gBp/fpsW6WfQmVv0nWru8A3yApTu/LQcj/nHB6Xw48fCVCoyhjDD/OTMOpPXoEtpahfksZugx1QoM2Mnz9VnKZ8m3N9zDPB5MBbzkXOcjL27/o29p7Sz2LPZdHnUffErMzLFj8TjISrpvQeYgjXvnIze79kuaxqM+5+3BbqU2bVrAUo0m1QOEkgkwhEtoU8jfm85xcxWAMtgblEi6BQcGgBFJSUgDYupVKJJXzkUmlUqhUKuTk5CAlJYWCAQEAKJVKXLp0qUKOxa829nCvFwCI3qCFXMWh2XMKmHKtBW6a+W+S7fo7Yus3mUi9Z8LlEwb0Cn3QxnDjbC5O7dGj3xvOGPCWi7DdamHI0VjhUXRVOUQizm5AFWBbMzhHY0Wt/1a9dPOx5UOmFBWoArlzKRc5WiuksqKfhEtTbVIYY44Vi9+1BYLnhzth6ES3AmlKmsemHeWFfs5KJxHc60iE7yu/e7F5qNfE1sjtEyiFxAG4f6tgurREM6QyDk6uJR89QOMMSiA52fZEU1mlAp6bm+0PKykpqVLPQ55O9/67udw4Y98f/p+dOtw8m4s+Y5yhdBLB2UOCxu0Vdq/8SzcG97H1Uvn1y0yY84B2fR48euZobI/F+fvHA8DxP3TIMzBYiqmyV7uLkXzHjDzjg7qkC0cMMOWrLqnX1AHOHmIc+iUbRv2DdAadFT98lIp1s9ILDXYVZePcDCRcM6H7q4UHgtLksbjPufXzSlz924ikuAczElz924DkO2a07W37/GUKEVp0VeLiUQMSbz4ICGkJJlw4okfzzopSLZNJJYMS4IMBf7OuLG5uboiPj8f9+/cr9Tzk6WM2MSTdNsGvsQN++yoTmclmePhKcPNsLv7ZmYPW3ZXoFVqydb1daknQsK0cl/4yoG5TB3j5P7jx128pg1zFYcvCTGTcN0OpFuH6aSNO79VDKuOQqy+60aBtHyV+/TITS8en4Nl+KqTGm3Hsdx3caj+4u4slHF6e4oofPkrDvJH30WmQIyQyDsd/1yHjvgVhn3hALKmc9rbEG3mI2ZUDhSOHOkEO+GenrkCaZ/s7Vkgee41W458dOnwTnoweI9Qw5THs/0kDv0YOaJevR9Tg8a64ftqIr99KRvdX1BBLgUMbsyF14PDiOy6luj4KBiXABwN39+L7KJeXh4eH3fkI0ev1aNeuHQAgJiYGSmXp+o7zEm/mwWIGnh/uBH22FdHrtdCkWuDuI8FLEa54/tXSjaxv11eFazFGu1IBYHu6f/trT2xdnIXdqzSQSDl41pNizGceiLuYi0Mbs4WeSA/r+rIT9Forjm/VYdOXGajT0AHjvqyFA+u0dkHkmR4qvLtEhD2rNNj1gwYcB/gEOuDNhbXQokvZPp+SuHHWVqIy6FiBun4e33W1vHl0chXj/ZXe2LwwA9u/zYKDnEPLECVeGu8KqcOD78ndR4Ipq72xdXEW9q/VgDEgsLUcL413KXRMRHE4RmPcH2n27NlISkrCyy+/jMDAwEo7z+3bt/HLL7/A09MTUVFRlXYe8vjI35lAp9NBpSpha+BDjv+hw/pP0jFtQ+1iB1aRpxe1GTyC1Wqt9DEGPL4aKj09HZbiKlcJKaV7sXkQS2BXpUNIfhQMHiEzMxMmkwkikajS5iXiqdVqSCQSWCwWGnxGKtS9a3nwqieFRErjV0jhKBg8QmJiIgBbqUAsrsRuCrD1I+d7LFGPIlJRGGNIuJ4HnwZUKiBFowbkR0hISABQ+Y3HPDc3N6SlpSEpKQktW7asknOSJxvHcVhwuG51Z4PUcFQyeAQ+GHh6Fj9ysaLwQYe6lxJCqhKVDB6BvylXVTDgz8NXT5GnG8dxqFevnvAzIZWFgkExrFarMBVFrVq1quScXl5eAGxtBlarFSIRFd6eZkqlEnFxcdWdDfIUoDtNMVJTU2E2myGRSCq9JxHPxcUFEokEJpNJCESEEFLZKBgU494923S3bm5uVfaELhKJhJHI8fHxVXJOQgihYFAMvnjOV91UFb7dgIIBMRgMaNeuHdq1aweDwVDd2SFPMGozKMadO7bFs318ipl3txJ4e3vj/PnzFAwIrFYrTp06JfxMSGWhkkERrFarUE1U1cGAL4ncu3cPNHUUIaQqUDAoQlpaGoxGI8RisVCHX1W8vLwgEomQk5ND01IQQqoEBYMi8FVEHh4elT4NxcMkEonQbnDz5s0qPTch5OlEwaAIN27cAADUrl27Ws5fp04du3wQQkhlomBQhOvXrwOAMPqzqvn5+QEAbt26VS3nJ4Q8Xag3USGys7OFWUPr1q2eCb58fW0rgCclJcFgMEChKN9C3uTxVdVtVuTpRCWDQly7dg2AbbBZWVeWKi9HR0c4OzuDMSbkhzx9VCoVUlNTkZqaWm1/i+TpQMGgELGxsQCqr1TACwgIAABcuHChWvNBCHnyUTB4CGMMV65cAfDgZlxdGjRoAAC4fPkyjTcghFQqCgYPuX//PtLT0yEWi+Hv71+tealXrx7EYjGysrJo5bOnlMFgQLdu3dCtWzeajoJUKgoGDzl79iwAW28emUxWrXmRSqVCQ/KZM2eqNS+kelitVhw+fBiHDx+m6ShIpaJgkA9jTJgHplGjRtWcG5umTZsCAP7++2+qKiKEVBoKBvncuXMHycnJEIvFaNKkSXVnBwDQuHFjSCQSpKWl0SInhJBKQ8EgnyNHjgCwNdzK5fJqzo2NTCYTGpKPHj1azbkhhDypKBj8R6PRICYmBgDQtm3bas6NvTZt2gAAYmJioNFoqjk3hJAnEQWD/+zatQsWiwW1a9eu9vEFD/Pz80Pt2rVhsViwa9eu6s4OIeQJRMEAQGJiolAF07Vr12rOTUEcxwn5Onr0KBITE6s5R6QqKZVKKJXK6s4GecI99cEgLy8P33//PRhjCAwMrPaBZkUJCAhAYGAgGGP4/vvvkZeXV91ZIlVApVIhJycHOTk5NB0FqVRPdTDIy8vDihUrkJSUBIVCgX79+lV3lorVr18/KBQKJCUlYdmyZRQQCCEVhmNPaef1uLg4rFu3DomJiRCLxXjllVeEaaNrsvj4ePzyyy8wm82oXbs2QkNDa1wbByHk8fNUBAOr1YqcnBykpKQgLi4OZ8+eFdYJkMvlGDx4cLWtW1AW8fHx2LJlC4xGIwAgMDAQrVu3hr+/P7y8vKBUKiESPdWFvieG0WjE0KFDAQBbtmypMV2eyZOnxMEgMjISAAqMgi1s9+IOWZL0JT1mSfazWCxFzunSsGFDdO7cGY6OjkXmt6bS6/U4fPgwbt68WeC6OY6DTCaDg4MDRCIRRCIROI4T/uU4rppyXbmexOvKy8vDp59+CgCYPn06HBwcCk3HcRxmzpxZlVkjT5gSL26TmppamfmoEiqVCp6envD19UXjxo2hVqurO0tl5uDggEGDBkGj0eDatWu4e/cu0tPTkZ2dDcYYjEajUHIgjy+TyST8nJycDKlUWo25IU+yEpcMrl69+mCnh57ACnsi47cV915Z0+f//VF5EYvFcHR0hEqlqvKF7auDyWSCXq+HXq+H0WiE1WoFYwxWq1V4PYme1NpOvV6PZ599FgDwzz//FNvFtFmzZlWVLfIEeiraDAh5XOXk5AjVmDqdjrqXkkpDrYyEEEIoGBBCCClhAzJjDNnZ2ZWdF0LIQ3JycoSftVotLBZLsemdnJyeyF5VpPKVqM1Aq9XC2dm5KvJDCCkHjUbzWPeSI9WnRMGASgaEPB6oZEDKinoTEUIIoQZkQgghFAwIIYSAggEhhBBQMHhqUVMRISQ/CgblFBUV9dj13rh8+TKee+656s4GIaQGoWDwFNq0aRNOnDhR3dkghNQgFAwIIYRQMKhoa9asgVwux19//YV27dpBLpejUaNG2LZtG2JjY9GjRw8olUo0aNAAGzdutNuP4zj8/fffaNOmDRQKBVq0aIFNmzbZHT8uLg6jR4+Gj48PpFIpPD09MXr0aKSnpwtpGGNYunQpmjVrBoVCgQYNGuCLL74AYwxRUVGYNWsWANt031FRUYVeh9Vqxccff4yAgADIZDIEBARg2rRpdvPr63Q6TJw4Eb6+vlCpVAgODsa2bduE9y0WC5YtW4YWLVpAoVCgbt26+PDDD+3WWQgLC0OPHj0QHh4OFxcXtGnTBmazGVarFXPnzkWDBg0gk8kQFBSExYsXl+u7IYQUg5FyiYyMZPk/xtWrVzORSMR8fX3Zd999x3bv3s1atmzJnJycWIMGDdiCBQvY9u3bWefOnZmDgwOLj48X9gPAPDw82KxZs9iuXbvYsGHDGMdx7I8//mCMMZaTk8Pq1avHgoOD2W+//caio6PZ7NmzmVgsZmPHjhXy8OGHHzKxWMymTJnC9u7dy+bOncvEYjGbPXs2i4+PZ6+//joDwE6cOCGc/2GfffYZc3V1ZatWrWKHDh1i8+bNY2KxmEVGRjLGGLNYLOy5555jrq6ubPHixWz//v0sNDSUicVidvDgQcYYY6+//jqTSCRs+vTpbO/evWzevHlMqVSy3r17M6vVyhhjLDQ0lEkkEtajRw924MAB9vvvvzPGGBs3bhyTSqUsMjKS7dmzh02bNo2JRCI2e/bsivz6CCH/oWBQToUFAwBs+fLlwrYNGzYwAGzmzJnCtlOnTjEAws2P32/WrFlCGqvVylq3bs2Cg4MZY4ydPXuWde7cmd24ccMuDwMHDmRBQUGMMcYyMzOZVCpl77//vl2aiRMnsl69ehWa58L06dOH9ezZ027b4sWL2U8//cQYY2z79u0MgBCo+Pw+99xzbObMmezSpUsMAJszZ47dMdauXcsAsJ07dzLGbMEAALt+/bqQJjY2lnEcx+bOnWu374wZM5hcLmdpaWnF5p0QUnpUTVRJOnXqJPzs7e0NAOjQoYOwzd3dHQCQlZVlt9+oUaOEnzmOw5AhQ3D69Gno9Xq0bt0aR48eRUBAAG7duoU9e/ZgwYIFuHLlCvLy8gAAJ0+ehMlkwuDBg+2Ou2DBAuzdu7fE+e/evTv279+PLl26YOHChbh69SreffddIX9Hjx6FVCrFgAED7PJ77NgxzJ49G4cPHwYAjBgxwu64r7zyCsRiMQ4ePChsUygUCAwMFH6Pjo4GYwwDBw6E2WwWXi+++CKMRiOOHj1a4usghJQMBYNKUtjMkcUtWcjz8fGx+93T0xOMMWg0GgDAwoUL4eXlhcDAQISGhiI6OhoqlUoYN8C3HXh6epYr/1OmTMGSJUug1+sxefJkNGnSBC1atMCBAweE87i7u0MkKvxPKCMjA8CDQMiTSCTw8PCwC4Kenp523XP5a2jWrBmkUqnw4pd/TExMLNe1EUIKomBQw+RvCAZsi6CLxWK4ubnh559/xqRJkzB58mSkpKQgKSkJO3bsQFBQkJDexcUFAJCammp3nPj4eBw8eNCuAbg4IpEI77zzDk6fPo2kpCSsXr0aRqMRQ4cORV5eHlxcXJCenl5gTeVz587h1KlTcHNzAwAkJSXZvW8ymZCWlgYPD48iz81fQ3R0NGJiYgq8hg0bVqJrIISUHAWDGubPP/8UfmaMYcuWLejcuTNkMhmOHTsGZ2dnfPDBB6hVqxYAW4+eY8eOCTfl9u3bQyqVYuvWrXbH/eqrr/Dyyy+D4ziIxeJH5qNTp06IiIgAYHtyDwsLw7vvvguNRgOtVosuXbrAZDJh586ddvl94403MGfOHISEhAAA1q9fb3fcjRs3wmKxoHPnzkWem983LS0NwcHBwis9PR0zZswoEDAJIeVXopXOSNWZOnUqcnNz0ahRI3z33Xe4fPkyoqOjAQDPPvssli9fjkmTJmHgwIFITEzEl19+iaSkJLi6ugIAPDw8MGHCBCxatAhyuRzdu3dHTEwMFi9ejM8//xwSiUR48t6wYQM6dOiAgICAAvkICQnB/Pnz4eXlhU6dOiEhIQELFixASEgIPDw88MILL6Bjx44YM2YMPvnkEzRo0ADr16/H+fPnsXjxYjRt2hShoaGIioqCwWBAt27dcO7cOURFRaF79+7o27dvkZ9B8+bNMXLkSIwdOxZxcXEIDg5GbGwspk2bhoCAALuSECGkglRr8/UToKjeRLdv3xa2HTx4kAEQulwyxtjt27cZALZ69Wq7/datW8eaNm3K5HI569ixo90+VquVffzxx8zX15fJ5XIWGBjIxo8fz1auXMkAsEuXLgnp5s+fzwIDA5lMJmONGzdmy5YtE46TkJDA2rVrx6RSKQsPDy/0ukwmE4uMjBSO4enpyd544w27njxZWVksPDyceXp6MqVSyTp27Miio6OF981mM5szZw6rX78+k0qlzN/fn3300UfMYDAIaUJDQ1m9evUKPf/s2bOFfX19fVl4eDhLT08v9vsghJQNLW5TQ6xZswZjxozB7du34e/vX93ZIYQ8ZajNgBBCCAUDQgghtAYyIYQQUMmAEEIIKBgQQggBBQNCCCGgYEAIIQQUDAghhICCASGEEFAwIIQQAgoGhBBCAPw/ZOeajf+j0xMAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 300x175 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAfUAAAC8CAYAAABhaDcBAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAMl5JREFUeJzt3Xd8FHX+x/HXd2bTO0kIxdDBWJDeQQUUbIflTuTsnvXwbIinnnqAnsrpWU49u4hwYhd/1rNiAcWCgIj0EAIJJYH0ujvz/f0xySZLCpvshrJ8no9HHmGnfL/f2QXeO9/5znyV1lojhBBCiEOecaAbIIQQQojgkFAXQgghQoSEuhBCCBEiJNSFEEKIECGhLoQQQoQICXUhhBAiREioCyGEECGi1aGutaa4uBi5zV0IIYQ4OLQ61EtKSkhISKCkpCSY7RFCCCFEK0n3uxBCCBEiJNSFEEKIECGhLoQQQoQICXUhhBAiREioCyGEECFCQl0IIYQIEa5AC/jXff+go8tmQvF6wrQdcIO6PP5+wGUIIYQQh6OAQz3RqmBC+ZagBLoQQgghWi/g7vdxJRsl0IUQQoiDQMChLoEuhBBCHByCOlDOrWTcnRBCCHGgBC2F3crgk/g+wSouJOkt63BfMgjro/mHVd2HCp29AfefhmEtfMa/7fNy8Dw2HffUsbinjsXzzN/RxQUNtysuwDPnHtzXnYz76uPx3Hcl9oaVwW6+EEIEPlAO6gK9wIwKRnEhS2/dAIDqeqTzuqIUz59PBK1Rx5+J6/K/N9jHfcd5ULSbsCc+81luzZuN/ePnuP79McrY93ezvesG0GUl2B+/jL38a9iZDbYNCe1QnXugBpyAOfb3rT3UQ462PHienwGWx7/tSwvxzL4aPG6M0y8By8L+aB6ebRtwzZiHcoU521WU4bnvCijMw5h4AcTEYX/2OtY//4ya8RIqvXdbHpYQ4jATcKi7lcHXNYE+sXh9MNoUsrzB2qUm1LPWgtZgmOifv0Rf+jeUWfeR6Ooq2L4ZdfRQ33K0xv75K9SAE/wK9EbrzsnE8+C1UFaEGn4KaszvwDBh51bs5V85Ox1GoW6//yLkZPq//f9ehj27cN37GqpTdwBUz2OxHpiKXvwe6sRznO0+mAs7tmDe9ixGxkAAjKET8NwyCevDebiuvifYhyKEOIwFHOpfxPWisibQUz1lwWhTyNJbN0BKR1RsgvN6y1oA1MhT0YvfR6/9GXXM0HrbrwfLQnU7yreczNVQsAtj0NhW1a21xvOf26CyzDmrPKKXz7bG+dOgML+1h3nI0Vs3YL/7AsakK7DffsqvfeylH6MyBnkDHcA4ZhhWx67Y33+CceI5zpevxe+j+o32BjqASkzBmHKjzxc4IYQIhoD/Vyk0ozhTAt2H3roBa+Ez6DU/AqAGj8c8/2b01g2oXsfVbZe1BgBz0hV4vv8E/eNnUD/Us2pCv2uGb/nLFkFkjM8XgBbVvXUD5GxCDRnfINABlFKQlBrAO7B/uC8Z1Ox646yrMM++utlttOXBen4W6pihGCNP8yvUdVkx5OWghpzUYJ3qmoFeudh5kZ8LBbtQp13s7Kc1VFWgIqMxx0/eZz1CCNFSAYf6+OKNpJrVwWhLSLDX/Ij18A3QrgPGGZdBeBR6yftYj9wIxXt8r2lnrYV2aai0dNQxw7B//hLj4tu8Xeq1ob/3mbr98yJUv9He67YtrVtXlju/d25DV5ajIqPb6N1oW+ZVzXdd+3O92v7gJfTObFw3PASW5V/FBbuc34188VGJKeiKMnR5CXpHtrMsIRnrtcewv3wLykuh/RGY59+MMeB4/+oTQgg/BRzqKVYZmGH73vAwoIv3YD1xK6pzL8zbn0FFOAMH9fGT8EyfBNSddevKctiRjRowBgBj0DisFd+gN6xEHTnA2SZrLcQkoFI71dWRuxm2b8E4+5pW1626HgkpnSB7HZ4bT0H1HYk6egjGscNRqZ3b6u0JOmPUaQHtr7dtwv6/5zAv+iuqXRo6L9e//Wq+FKnwyIYrw2qWVVVCeQkA1ltPgWlinn8zGCbWh/Ow/n0z3PIExjHDAjoGIYSoT24sDyL7g7lQWoRxwc3eUAVQEVGoXn2dP3dxbvvT2etA23VBO/AEME2nCx7Q7mqni7ybb9e7vWwRhIWjjhvV6rpVRBSuu+ZgnHIBRMehf/gUe+59eKZPwvPAVPTuHcF7U9qQLilo/qeqoul9bcvpdu/TH6NmUJvf7JoHLinV9DZKgcft/Lm8GNedczDGTMIYdTquvz0P0XHYbzzRsnqFEGIfgjpSZ0VUR/pXbA9mkYcUe+knqJ59MXr3a2SlDbEJqOQOQP2u9ZpQj01AZQzGXrYI44Lp6G0bwfI0cj39S9TRQ1BRMa2uG0AlpmL+cRrmH6eh83KwV32H/dnr6NXfYz0/E9etT3u39cy6GOPkKRgjAzszDjbPXxpe066vuWvq9ofz0FvXY97xArqk5t7y8mLnd3WlsywqtsElDgAV6bz3urqqYcHuSud3VAzUfLkyBo1DxcTX7R8ThxpwPHrJ++iKsgafpRBCtFbQQn1FVEeWxxxx2Ia6LsyHwjzUkPEN19kWevNvPgFdNwiu7nq5GjwO/dL96Mxf0dnO7YH1z9T1np3orN8wL70joLr3plI7Y477A8bQk/D85ST0uuVorVFKoW0bvW2T91a4Ro/d8hyQkdzmX59sdn1zlxL0qu/A48aadXGDdfaH87A/nId52zOoowY33Ln2y1FRwzsEdEE+RMehIqLQtdfc45Mati2+nXfgHBLqQoggCcr/xLWBPqBsWzCKOzR5u3p1g1X624+gaDdq1Ol1y7LWQkIyqt5gK2PQWOx5/0T/9IUzoArfQXL2si8B5XTVB1B3k8wwUAZExTiBXpiH55YzwV2N5+5LnE2mPQYVpVivPuqMFv/ybVSHrrhufQrrf/9Fr1mG66ZHvEW6/34B5umXYAybgP3TF1gLn8YYdQb2F29CSQHG+MmYk6+rO8bVP2C/8yw6JxMM5axv4mw7kOvR5pSb0LVn5rWKdmM9cxdq5OkYo0/3Xq7Ym4qJg9TO3i9m9enstajuRzvbHdELwsKdY9l7u7wcCItoNPCFEKK1Ag71VZEdWFcT6IfrWTrgnL2ZJvbqHzFsu24Ee2E+1lvOGaX3wS/VlbA9C9V3uE8RKiEZ1fs47B8/d7pro2Oh/RHe9frnRaje/VDx7Vpdt71+OSq9NyoqtsEh2O8+D7bl7WZXiamYl8/A/mg+rln/9W5nLXwW8rZBdCyuB96pa1/WWt/R/R435GbWDQ7MXgc7t6Fi4nHNfgudvR5r1sUYE89HJSRjf/cR1iuPYF4xE3X0ECgvQe/Y4ucH0DKq+1HsfUW8dqCcat95n18YjMHjsD9ZgM7d7L1X3V79vTOI8VTn7F9FRKH6H+88WGjbJtQRPWvqyUGv+BrVfwzKMIN7YEKIw1rAob4yuhMjDvdAB5QrDDXqDPTX/4f14FTnHuaiPdiL3nSeGke9x8NmrwfbarRLXA0ej17wEDo/F3XkQOeecZx7o/W6nzEmXx9Q3fbbT6M3r0ENPMEJtsgYdMEuZwDelnWoY4Zh/OFab9l6y1rY+7r+lrWo4adgnjylwXJjaL3r3DmZYLogLb1m/TqM0WdgnHCWs75zj7p9K8ux5j+AedUsjONGOgvj2zX8AnMA6F3b0Bt+QfU+DlXzJcs47RLsJR/g+eefMU69ENzV2B++BF2PRNUbe2CedwOetcvwzL4aY8IfwRWG/ckrEBaO+Ye/HKhDEkKEqIBDvV95Lv3t3cFoyyHPvPAWbNOF/fOX6A0PQ8euGGdeiV67DL3iG+jQFWj6oTIAxuCx2Asecp4HX/96+vKvwbKafIqcv3UbEy9wttm4Cr1yCVSWQnQ8qlsGxtX3oEac6v0iUdvWva/V6y1rMS/9m++yqgrYvsV33MCWdagufeq+mGxZh1HvoSs6e71zCSIhGfvnryAqFqP/wXfvtl73M9bzszCvmOENdRWfhOtvz2EteBj77achPNJ5Xv55N6DCwr37qtROuP4+F+v1x7A/mud8rn36Y06+AdWhy4E6JCFEiFJa64YXYhtRVVVFVVXdaN/i4mLS09P59aoJxIUH7z71Lo+/H7SyQonn3zej83MJu+eV/Vqv+9pxmNMew+h5LODcRub5y0m4nvgMFVd3Pdje9CvWQ9cR9uSiujY/cSsqMRnzwr86M5Vd57uf9elr6JWLcU1/HPurd7A+f52wuxfs1+MTQohQ4vd96vfffz8JCQnen/T09LZsl9iL6nUcZr1u8f1B27Z3wJ53WdZa5xnycXsN8NIaLMv7YBZ76cfon75Adam5nr5lLSR38NlPb1lbd0tftwzIycRe8yNaa3TRbuy1P7fh0QkhROjxO9Rvv/12ioqKvD9bt25tsE2eS27NaSvm6Zdg9Bu9X+tUhoFx6oVY/7oW91Wj0Tu3orPXNXhsLYDqcQyq32g8fzsXz31XosuKIDq23jiCdd5R4bX0lnXe9aprBsaFt2C9cA+eq0bjmXWxMx2sEEIIv/nd/b634uJiEhISvN3vea4YPo7vw4V7lgfUIOl+F0IIIVonKI+JrQ30JKvpx3IKIYQQom0FPPo934xhSU2gTyheH4w2CSGEEKIVAj5T/zy+lzfQw7QdjDYJIYQQohUCPlNPtCqYUL5FAl0IIYQ4wAIO9XElGwkLk0ddHszcN5+ByhiE68pZB7opftO7d+C58zzM6x/CaGxSlXrsX77FfutJdG4mxCVhjP09xhmX+TxEJxD+lm9vXIX9xuPozNUQGY3qOwJzyo0HxVPxhBCHh4C73+UMXfhL2/79XdH52/E8OLXBPfKNsTesxHr0JujUHfO6f2GMPB37rSex35sTaHNbVL7evAZr9tUQEYV5/b8wJ1+H/nUp1r9vDko7hBDCH0GdL9OtDAl50YDetc2Zqz1vG64bHm56O9tGL34f69VH/S7bfudZVJcjcV19j7PguJFgebA/mItxygWo8MiA2u5v+dZrj6K69MG88eG6SVqiYrFe/hc6L6fZaWCFECJYgnJLGziB/kl841NVHu48j9yIZ8aF2F++jfvOKbivGIn7ptOw3n662bNXz5x/4P7LeLTl8VluvfYY7qlj0R43uroS6/XHcf/1LNyXD8d99fF4HpiK3rKuyXLdN5+B57kZPsvsb97Ffckg70xlAPaXC3Hffq5T7k2nYS18pkFbmmOv/gHPo9Pw/PVs7O8/QR07ovkdtm7Amnc/xugzMK++e5/la3c1eu0y1GDf5+GrIeOhshy9zveZCS09Hn/L16WF6LXLMMaf6zPrmjF4HGGPfCiBLoTYb4Jypl4b6AVmVDCKCzk6aw1UlGF9vADzjD9BQjvsr9/F/r/nUO3SUCee3eh+xujTsb5aiP7tB1RfZ+YyrTX29x9jDD0Z5QrD88Qd6LU/O3OStz8CdmzBevtpPE/djuv+t1p9Xdl6bw72W09inHQe6vxp6Oz12AufQe/Zievyvzd9rFUV6G8/wvr0VcjZhMoYhDn1PtSgsShzH3/dkjvgeuAdVLs07DU/7buReTngcaNqJquppWpnhduRDX1HtP54/Cxfb93gPCY3PgnP03c4k+9ojRp4IuZFf3Wm0RVCiP0g4FB3K4OvawJ9otyn3oDeswsK8yG1M667XkRFxwGgMgbhWfsT9vKvMJoIddW7P6R2xl76CUZtqK9fAbt3oEae5sxXXlmOedFfMYZNcHbKGISuLMd+5REoyofE1Ja3ubwE+90XME48B/PCW5yFfUegYhOw5tyDnniBd25wn/2KduO5/Q/ObHKjTsO4djaq3vSq+6JiE4CEFrUTgKi9Hk8cGe38riwL7Hj8LJ/iAgCs5+9GHTcS8/qH0Duzsd94AmvXNsw753jnuBdCiLYUcKh/EdeLyppAT/WUBaNNIUVvXg2AefbV3kCHmjnQ09KhpNDpgt9rLIIyXSilMEaeiv3pq2jPHShXGHrp/yC1M0af/gC4pj/h1FOYh96xFbZnOVOtAnj87yr3afPGVVBdiRpwgk/3tBrgTItqr/4es5EQRCnnB0AZdX9uK95LF03UUzvlqx/HY3Tq3uAz8Lf82vdZdTuq7qz/mKGo6Disp/6GXr3U29MihBBtKeBQLzSjOFMCvUl68xowXaghJzVcV5CH6nYU9v89h/3Osz7rwl5aBoAx6gzs/3se/csS6Dca+8fPMMad693OXvUt1ssPwfYsiIxBpfeCyJozy9Y91h9KCwGwHr6+8fWFeY0uVvHtcD3yIXrJh1ifvor9+euoo4dinDQZ1f/4oJ+ter8kVew1Sr5mpjiiYp3ffhxPY5+B697X/Ss/yjlzN/qP8W1fbe9K9nqQUBdC7AcBh/r44o2kmtXBaEtI0lm/QVwiKjzCZ7m9aRXk5WCceQXq2BGovQKhlkpLR/Xsi/3DpxhmGJQUYow8zSl751asf09HDTwB86ZHvdd6rc9fR6/6tplWqXpnoTXtrKz33P6asDSv+QfsdT0ZaPa+axUeiRp7DsbYc7B/XYr9yQKsx6ZDcgfMSVdgnHBWM+1qofZHgGGid23zWax3OjMIerv+/Tkew2z4GfhZvkrr4iz37PXvoLZXIMz3sxdCiLYS8KlTiiVn6M3Rm3+DkgJ0WUndMtvCfv1xSOmEGnEqKikVo/vRPj/1qVGnoVcuwV76P1TPvqgONSGStQbcVZhnXOYNdAD9y7e1FTXeqKgY2LPTt50bVtbV16svuMLQBXk+bVJmGPbrj/uMkG+OcexwXNMewzX7LYx+o7GXLfJrP3+p8AjUkQPQP31B/ckG9Y+fQ3Qcqscxfh9PY5+Bv+XTqTukdEIv/cSnfXr5V877cOSAoB63EEI0Jaj3qQtfOi8HSosguQPWf27FOOVCcFc518iz1uC67VmUK2yf5RjDJmIveBi99GOMC6d7l6tuR4FpYr3+mFO2pxr7m/fQKxc7G1Q1Pmue0X8M9vsvYr03B9WrL3rFN+jffqgrNzYR47SLsd9+CipKURmDoWAX1ttPgVKoLr1b9D6oDl0xL76tRbfDNUXv2YneswvV9UhUWDjGpMuxHpjqvL9jzkRvXIn90TyMydd77yEP5Hj8Kl8pzCk3YP3nNjz/uQ3jhLPQ27Ow3/wPavB4VNeMgI9bCCH8EdRQXxHVkf4V24NZ5CFNb/4NAHPq/difv4n15G2gQR09GNeMeahO3f0qR8UmoI4bhV65uG6UO07XvPnn+7AWPov16DSIiUf16ot527NYs6/CXr8cM71hYBm/+xO6uAD7o/lgeVD9RmNe/nfnyWk1zN9PhcQU7M/egA/nOWUfPRTz3Gt9Bvy1xD5vafOD/dU72O88i+tf70FqJ4yjh8JfHsBa+AzWYzdDUnuM827APPUin/1aezz+lm8MOQlufBj7need9zEm3nmc7O+nBnzMQgjhL6V160ZTFRcXk5CQwK9XTSAuPIwVUR1ZHnMEl+X/GFCDujz+fkD7H0ys1/6N/fkbuJ7+yuehJEIIIURbCPjUKf3BN4iPj6cLMCkIDQolevNvqK4ZEuhCCCH2C3kiRhvRWqO3rEXtNehNCCGEaCsyUK6NKKUIe+qrA90MIYQQhxE5UxdCCCFChIS6EEIIESIk1IUQQogQIaEuhBBChAgJdSGEECJESKgLIYQQIUJCXQghhAgRAd+n/uzGN4mKjQ5GW3xc22dK0MsUQgghQpmcqQshhBAhQkJdCCGECBES6kIIIUSIkFAXQgghQoSEuhBCCBEiJNSFEEKIECGhLoQQQoQICXUhhBAiREioCyGEECFCQl0IIYQIERLqQgghRIiQUBdCCCFChIS6EEIIESIk1IUQQogQIaEuhBBChAgJdSGEECJESKgLIUJGbm7ugW6CEAeUhLoQotVycnKYPXs2c+bMoaqqqsH6qqoq5syZw+zZs8nJyWnz9txxxx3Nrr/00kvbvA1CHEgufzesqqry+UdbXFzcJg0SQhwacnJymD9/Pu3bt+eCCy4gIiLCZ31VVRUvv/wyu3bt4qKLLqJz585NlrV06VIWLFhASUkJ2dnZPP3009x+++3ce++9XH755Zx11lnEx8fz+9//nhkzZgAwcuRIhgwZwoMPPkhxcTFTp07l559/5quvvmLZsmVs3boVj8fDo48+yvTp0zFNk40bN7bpeyLEgeZ3qN9///3MmjWrLdsihDiEBCvQAZ588kk6deqEYRhMmDCBSZMm8eKLLxIREcG4ceOYPn0611xzDdXV1VRWVpKSksIPP/zAihUrmD17NoZhUF5ezoABAxg4cCC33norJ554Ijt37mTx4sWkp6czbdo0zjvvvLZ8S4Q44Pzufr/99tspKiry/mzdurUt2yWEOMgFK9AB3G43d955J9OmTWP48OEkJyeTnZ0NgG3b3m1s2+a8887j3nvvZcSIEXg8HrTWuN1utm3b5t2+V69ezJ49m/POO4/U1FS01gCYphnMt0CIg47fZ+oREREN/vEKIQ5fwQp0gKlTp3LllVdSWVmJy+Xigw8+4JprrqF3797873//o7y8nBEjRjBp0iSuvfZa5s6dy5QpU7jqqquYPn06FRUV3HPPPcTGxvL555+Tnp7ONddcg2maTJo0iRdeeIFp06axY8eOtngrhDhoKF37FbaFiouLSUhI4MFlLxAVGx3sdnFtnylBL1MIcWjJyspi7ty5zJw580A3RYhDgox+F0IctLp16yaBLkQLSKgLIYQQIUJCXQghhAgREupCCCFEiJBQF0IIIUKEhLoQQggRIiTUhRBCiBAhoS6EEEKECAl1IYQQIkRIqAshhBAhQkJdCCGECBES6kIIIUSIkFAXQgghQoSEuhBCCBEiJNSFEEKIECGhLoQQQoQIpbXWrdmxuLiYhIQEioqKiI+PD3a7hBBCCNFCcqYuhBBChAgJdSGEECJESKgLIYQQIUJCXQghhAgREupCCCFEiJBQF0IIIUKEhLoQQggRIiTUhRBCiBDhOtANEEIIERhdsgnPT9dj9Lwcs8s5h03d+5v7x+uhdFOD5Sp1FK5j/9bsvvbuZdhbXkOXbASlUPFHYnS/GCMho8l9PGsfg4pcXANm+93GgEP9Xw/cR8ckxSkZOwk3fR9OV20p/rc2jYKKcE7N2EH72OoG+y/PSWDZtiQGHVHAgM5FDdbvKg3no7UdSIqqPmB1CCEOLZ1PefFAN2G/0mVZAKi4ns5rTzmebyYDGtXxZFwZNzbYx/3DtVBdQNjoBT7LrfVPYu9agmvUfJTad2fu3nUDaHcp9rZ3sPO/h4pc0DaEJ6Kiu6BShmN2PrVVx3kgaW1D+VZU6kiMlBG+KyPbN7uvXfAL1i8zIKYLRo+LQNvYOR9gLb8VBvyz0WC3cz9Gb/8Yldi3Re0MONQTo6o5JaNQAl0IIQ4QXboZABXbw3ldshHQoAx0/vdo20IZZt32VrUTUEn9fMvRGjt/KSplmF+B3mjdZdl4Vt4J7lJU2vGojicDBlTkYucvhXzgEAx1KraDXY2RMgKjw7gW7WpteBYiUnANehhlRgJgdBiH5/trsDfPw+h/n3dbrS3srNewsxY0VVyzAg71k/rsItwM81kWKoFebSkJeSHEQU+XbobI9qiwOOd1idNFrNLGond8ji5chWrXv972maAtVFwv33JK1kPVbozUka2qW2uNZ/Vs8FQ4ARbbzWdbo9eVUL2ndQd5gHl7JGLSW7afuwTKsjDSz/YGOoAKT0Il9kXv+bluW6saz7JpULYZ1WE8umBFi9sZcKiHG6F5hl5bx6Rjdvj1PgghRFvTpVlYm/+LLvwFAJU6ErPXVejSLFTCUfW22wiA2e2PeHZ9jc5bAj6hXhP6sXuFet53YEY1OIP3u+6yzVC2BZU6qkGgAyilICK51ccfCPei05tdb3Q7H7P7BU2u16VbAAXR6Witwa7yCekmuaJxDXsGGtvWXQyqrgcFuxqscsxjbsNoPwb3d5ftu/y9q2vxHs0ItUAvqAhv6VsghBBtwi5YifXLLIhIwehyLpiR6B2fO8vchd7ub6jpfo9IQUV1RCUNwM7/FqPPn71d6k73PA3O1O3871DJQ1CGb++rv3VrT6Xzu2I72lOBckW12fvRUuZRNze7XsV2b3a9LtsCrmis9U+j8xaDVQGRHTB7XIyRdkLT5SoTojs3LK90M7roN1S7QXULXdG4hj3nc6mkpYIW6qEY6KdmyFm6EOLA09WFWKtno2K6Yg6433uGqDuejOe7y4H6g+QqoDwXlTIUACN1JNbuH5wASTzW2aZkI7jiUFFpdXWUZUP5NoxuvmerLalbxfWAyDQozcTz7cWo5EGoxH4Y7Qagojq01dvjl5ZeB9+bLtsCnjKwq5wvCJ4SrK3vYv32AGirReVrTwWeNQ857ep6rne5UgaogJoZnFAP1UBvrA4hhNjf7Ow3wV2M0fcu3+uyZiQqIQOdv7TubLk0E7BRsTVBmzIclOl0wScei7bdUJbtDXhvHflLwQhDJQ9udd3KjMQ18F/YW9/G3rUYvesb9K5vsAGVNAAz4wZUZGobvEP7pqsbZoIPM6LZ7nSj8+mgTMzOp3mXqfYn4PlhKtamF1BpJzhn5ftqh1WJtWoWlG7G6DoFY6/PIVABh3q1rVh0EIStBLoQIlTZO79GxWdgJBzdcKW2ISzeG5Z7d62rsDhUYl/svG8xaq6Boz0+t6CBcz1dJfZDuaJbXTeAimiH2esKzF5XoCt2YO/5GTvnfXTBcqw1j+AacF/DclrA89NNGN3+iFHTE+H3fkvOb3b9vq6pm0f8rsEyZUZgdBjnjFQv2wqNjCOoT7tLsH6ZiS5ei+o4AbPHRf40vUUCDvXP1renUkugCyFEW9BVe6B6N6r9qIbrtIUu2eB7Pb12EFy90Fapo9Dr/4MuXld3C1pc77p9KvPRJRswj7wuoLr3pqI6YHY+DaP9aDyLz0cXrUZr7QyYayXX4EdatZ/Z7x/Nrm/15YHwBAC0VdFsz7muLsSz4k4o24zR6VSMPte2rr59CDjUCyvCObu/BLoQQrQJu8r5rRveXqt3fgnVBai0uuu5umQjhCeh6o0yN1JHYK9/Cp33LVhlAN7ueXAGyIFCpQwLqO4mKRcoBa5ob6Db+d9jbZqDkTYOO/cDsCoxelyGkZCBtf4pdGkmKq4XZt8Z3gF3zj4vEjbsaXR1AdaG59CFK8GqgohUXMfejorp0mgTjHYD9t3OJuiKHXh+mYmRdiJmtym+K8u2OYcYmdZwx9r9PeXOvftlmzGOOAuz95Wtbsu+BBzqJ/fZSfvYhssl0IUQIggiUkGZ2AUrMbRdN4K9ag9W5nygZoAaoK2qmofKDPQpQoUnoRKOws5bggqLBVcMRHX0rtd536ESjkKFJ7a6brtwNSq2e4PuewA761XQNkba2Lo6SzZB5S5UZCqu4S9ib/8Ee9ML6OQhmMfcCkYEnp+uR+d/i+ow3rtPbQ+Ete5xVEw3zIwbwTCd8iJSWvMO71tke3CXYG//GOOISd5j1JW7sHd8hko8DhXRrsndrfVPOtfQjzizTQMdghDqqTHVgO+tX6ES6MtzEhotWwgh9hdluJwHkWz/BGvFnaj2o6G60Dm7rflvrfas23mojN3gVjVw7ivXG59DVypU4rHeM2btLkEX/YrRo+E90S2p2978X6c7PmU4Kq43yhWFrtqNnfcdlG5CJfXH6HGxt2xduhGj4wTvqHEV2x1sN2afP6PC4p1lke3B9vjsoxJqRvCX56CiOjvjA1Q4Kr5PIG9zs5QyMPv8GWv1/Xh+vgWj00TwlGHnvO8Mnuvz57o2VmxHF61xviRFdUSXZqF3LgJXDCq2O/aOLxqUH+jI/PqCPqFLKAX6sm1JEupCiAPO7H01tnJh53+H3vgcRHfG6HY+unAVOv8H733Q3ifJ7TUIDpxb2+yNzwHaJ/T17h+cW7JSRzTYpyV1G+lnYecvRRetRe/+CaxycMWi4npiHDUdlXaiz7V0XZKJ0aneA2HKNjtBWBPoALpsK8R09d3niDOddh01DXvzAjzfXoJKOBqzx8WNHnewGO1Hg7oDe8sb2JvmgBGOSjwOs8clPk+Z04W/Yq19FDPjRlRUR+yiX50VnjKstY82XnYQQ11p3cjFEj8UFxeTkJDAmjfPJy7GOVMPtUBvqg4hxMHtcJvQJRCeVf9AV+4gbMgT+61OXV2EZ8n5uEYtQNUMNLPWPQFmFGYv5953XZmPZ+lluMa8iTIjava5ANfoV51LCLVlWZVO93ZVAa7+9+y3YzhYBW0+9YMhbCXQhRCiZVR8BmaPS/Zrnbp0k/O8+JpAh9rr5fV6EEo3QnQ6yozw3ScsFnvXEnRZtjNzmlUF1YWo+IaXHA5HQel+PxjCVgJdCCFazuz6h/1eZ4MA1xa6LMt3WUnmXq/r9tHFa7A2PgvuEgiLx2h/fIMn4R2uAu5+f+3xa1izu4MEuhDioCHd7+JwFXD3+/KcRAl0IYQQ4iAQcPf7gM6FDOhc2WC5BLoQQgixf/kd6lVVVVRVVXlfFxcXA3Bcx2L2vk89VAJ9V2m4PJBGCCHEIcPv7vf777+fhIQE7096enqj24VSoH+09sBOFSiEEEK0hN8D5Ro7U09PT/e5Tz3UAj0pqppJx8ic6kIcamSgXHC5v7sMldgX11HTDnRT/KYr8/D8eC3msXdiJB3X7Lb27p+wN893HnYTFo/R+TSMLucGNPFMa8q3i9ZiZ85FF68HMwqVPAiz558aPr63GX53v0dERBAREdHk+lAM9FMydjZ5vEIIIfYvXe/5881uV7kLz8q7wFO2z23tot+wVt2Daj8Gs/tF6KLfsDPngbYbTt7SCv6Wr0s2YK24HZXUD7PvnVC1GyvzJazyHFyDHvK7vqDcpx6qgb53HUIIIfY/XbEde9v76ModuPre1fR22kbv+Bxr4wt+l21vfgUV2wPX0dOdBcmDQXuws9/ESD/b+/Cb1vK3fGvjHFRsD8y+d6GU6Wzrisba8Cy6YoffU8MGHOp5ZeF8ky2BLoQQbcXzyyyo3oPR6VSsbe9DRY7TjdvxZIxu5zd59upZ+xg6/ztcI/+LMkzvcmvTHOzcj3GN+i9oGzvrFey8JVC5C4wwVPyRTrdvE89Sb6w73t7+KdbaR3ENn4OKcqYhtXM/xtr6DlTkQniS096uU3za0hx7zwrsnHfR+T86+3c7r/kdSjdjrf8PRqfTUe36Y/0ys9nNte1GF/6C0d33wTUqdTRkv4UuWo1qVzfjXUuPx9/ytbsYXbgK86hpdYEOGKmjMFIbzmXfnIBD/dP1aXRMkkAXQoi2oks2glWOtfUdzK6TITzJmao06xVURCqq08RG9zM6jMfa/jG6YAUqeZBTltbYO7/GaD8GZYTh+fU+dOGvmD0vhciOUJGDtXk+nt8ewDX06VZfV7a2vI6dOQ/jiN+hkq9El2RiZ72MrsrDlXFj08dqVaJ3LsLa9h6UbUEl9sU85q+olJH7/jIQ2R7XsOdRkSnYBb/su5EV251Z3mompamlaqal1eU5UBPqrToeP8vXpZsBDWEJeH57EJ3/PaBRKcMxe1+DCovb97HUCDjUE6OqOSWjUAJdCCHagK7Kh+o9ENkB16CHUK4YAFRiXzyFq7DzlzpTgTZCJRwNkWnYu77GqA31otVQlYdKG4e23WBVYPa+BiPteGenpL5oTwX2puehugCamSe8yTZ7yrCzXsXodCpm76udhe0GosLisNY9hk4/G1Vv9jXvftUFeL6/xpk1rsM4jGNuQ8V08bteFRYHLQhAXXvN3dxrDvja157ywI7Hz/JxO7eIW2v/7QyOO/ZOdEUuduZcrIodmAMf9GssAQQh1E/qs4twM8xnWagEerWlJOSFEAeULt4AgNn9Am+gQ81c51GdwF3sTGyy141MyjBRSmGkjcXOeQ9t/wVlhKF3fgWRHTASjwbA1c+Z2UxX7UFX5EL5Vmc6VgDtoTV00Rqwq1Apw9C2VdemlGGwDuw9yzEbCUFQoAzQlvNngjP6vOmG1r5nTdRTO+e8H8djRKc3+Az8LR/b7byM64kr44aalf1Rrhis3x5A71nu7WnZl4BDPdwIzTP02jrkljYhxIGkSzaAcjnXYfdeV7UbFdfLuSaetcBnXdjYDwCnC97e8qozx3nyUOy8xRid6+Yxt3cvcyZHKd/m3EYV2x3MqJoKWnlS4y4BwPplRuPrq/Y0uliFJ+IaMRe98wusbe9h53yASuqH0fkMVMowv89W/aXCar4kWeW+K2pf136J8uN4GvsMXEOf9K/8mvfbSBnq2752Nb0rpZmwv0K9vlAL9IKK8Ab7CiHE/qRLNkBYPMr0/f/ILloLlTswuk1BtRuISh7a6P4quhMqPsMJcyMM3MUYaWOdsiu2Y/36D+fa7XEzvdd6rZz30XuWNdMqBdr2badV73HhNWFlHn0LRPleTwaave9amRGoTqdidDoVe89y7K3vYP16L0SmYnad0uSlhlaJ7AjKQFds91lc+1pF13T9+3M8ymj4GfhZfu01d11zxl63YU2vQAtG4Afta08oBvqpGXKWLoQ4sHTJRnAXod2ldcu0hZ05FyLTUGknoiKSMeJ7+/zUpzqMRe/+EXvnl6j4jLoQKdkIdjVm13O9gQ6gd9cGum9we5nRUJXv286i3+rqS8gA5UJX7fZpkzJc2Jkvoiv9ewaI0W4Arn6zcA17BiN5CHb+d37t5y9lhqMSjkXnfUv957DpvMXgikHF9/H7eBr7DPwtn+h0iExD7/zap306f6nzPiQc4/cxBeVMPVQDXZ77LoQ4kHTFDmcQVUQq1ur7MdLPBtuNve1ddMlGXAPuRxlh+yzHaH8C9obn0Lu+wqgd6AWo2J6gTKxNL2Kkn+OUveNT9O4fnQ2sqsbLSxmCveUNrKzXUAkZ6N0/ogtW1pUbFo/R5Q/Ym+eDpxyV2Beqd2NlzgelnC7+FlDRnTH7TPW5nt1aujIfXZWPiuuJMsIwuk3BWnGH8/52nIAuWoOd/TZGz8u895AHcjx+la8UZs8/Ya2ejWf1bIyOE9HlW7Ez56FSRzV5a2FjAg71alux6CAIWwl0IUSo0SU1g+SOuRU750Os1f8ENCqpH65BD/s9MlyFxaGSh6B3/4jR/vi65dGdMI/+K1bWAqxVd4MrDpVwJOaA2VjLb8Mu+hUztluD8oyu56Gri7C3vg3ZFip5CGbGDU4ZNcweF0FEEva2DyD7TQiLRSX1x+xxic+Av5bw9/725tjbP8bOWoBr+ByISsNI6gfH/g1r88tYq+6BiGSMnn/C7HKOz36tPR5/yzfajwbj79hZr2CtmgWuOIxOp2L0uLhFx+f3s9/3VlxcTEJCAg/94yYqdZwEuhDioBEqz363Ns7Bzv0A15jXfR5KIkRTAr6mXihn6EII0SZ0yQZUbE8JdOG3gEP95D47JdCFECLItNbo0k2ouN773liIGgFfU0+NqQZ8b7UIlUBfnpPQaNlCCNHWlFKEjXn9QDdDHGKCeyc/oRXoy7YltfTwhRBCiAMmqKEeaoE+6IiClr4FQgghxAETtFAPxUCXrnchhBCHkoBvaSsqKiI+Pj7Y7RJCCCFECwX9mroQQgghDgwJdSGEECJESKgLIYQQIUJCXQghhAgREupCCCFEiGj1E+VqB80XFxcHrTFCCHEwiIuLQyl1oJshRIu1OtRLSkoASE9PD1pjhBDiYCC36opDVavvU7dtm9zcXPlGK4QIOfL/mjhUtTrUhRBCCHFwkYFyQgghRIiQUBdCCCFChIS6EEIIESIk1IUQQogQIaEuvObOncvEiRN9Xs+dO7dFZWRlZXHppZcGrU2TJ0/mggsu8GlTYmIiHo/Hu+yoo45qsp2vv/56k2W/9tprANx4441BaasQQhxoEurCx08//cS77757oJvhtXPnTl5++WWfZcnJySxZsgSAzMxMSktLm9z/ySefbHLdU089BcCjjz4aeEOFEOIgIKEufEyfPp2ZM2dSXV3tXTZz5ky+/PJLAE488UQAJkyYwOTJkxk+fDiPPPIII0eOZMaMGQCsWrWK8ePHM3HiRPbs2UNubi4TJ05kzJgxPPHEEwCccMIJjB07lvfee89bz+eff87w4cMZMWIEixYt4t5772XlypU8/PDDPm08/fTT+fDDDwF47733+N3vftdoO3/44QdWrFjBzTffzOLFixk/fjwjRozggQce4N1332XFihU8+OCD3mN6+eWXGTZsGGPGjGHVqlXMnTuXKVOmMG7cOKZMmQLAP//5T8aMGcP48ePJz88P3hsvhBBB0OonyonQ1LFjRyZPnswjjzxCWlpak9tt3bqV1157jTlz5lBYWMjixYsZM2YMl112GREREXz22We88cYbPPfcc+Tm5jJz5kxGjBjB2WefzR//+Ed27NjB8uXLiY6O9pY5a9YsPv30U8AJ7q+//ppPP/2UadOm+dQ9cOBAXn31VcDpWRg3blyjbRw6dCj9+/fnoYce4qWXXuLtt98mJiaGMWPG8N1339G/f39uueUWPvjgA2zb5vHHH2fJkiXk5uZy3XXXcfbZZ9OtWzdeffVVJk6cSG5uLp999hkLFy4kMzOTwsJCUlJSAn3LhRAiaCTURQM33XQTI0aMYPLkyXTo0MFnXe2zitq3b09SUhLJyckkJydjGAZhYWEADBgwAKUU/fr1Y9GiRWRlZXHbbbehlKKgoIDs7GzS0tJ8Ah3ANE3i4uIA9vk0r65du/Lrr782+eSvvZ+p1KlTJ66++moSEhKoqqpqsH1eXh7du3fH5XLRpUsXCgsLATjyyCMB58tOdXU199xzD1deeSW2bUu3vRDioCPd76KBiIgIZsyYwYMPPuh9nZeXR1VVFZmZmUDzobtq1Sq01vz000/07t2bHj168PTTT/Pll19y6aWX0q1bNwyj4V89j8dDSUkJJSUljQZvfaeffjo333yzz8C+ptqpteaOO+7gpZde4u6776asrKxBeSkpKWRmZuLxeNiyZYv3C8fex/nee++xcOFCrrzySubPn99sG4UQYn+TM3XRqDPPPJNBgwYBcM455zBlyhTmz5/v1wQ+lmUxfvx4oqKieOWVVygqKuKKK66gsLCQwYMHk5SU1Oh+d911FyeffDKWZfHQQw81W8dJJ53EJZdcwsKFC70j3Btr5+DBg7nqqqs47bTTGDp0KMnJycTGxuLxeOjYsSN333034PQSXHvttYwePRqtNXPnzuX7779vUG+7du0YOHAgiYmJ3oF2QghxsJBnvwshhBAhQrrfhRBCiBAhoS6EEEKECAl1IYQQIkRIqAshhBAhQkJdCCGECBES6kIIIUSIkFAXQgghQoSEuhBCCBEiJNSFEEKIECGhLoQQQoSI/wcqyU5LcPlvwwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 400x200 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAasAAACpCAYAAACVrhLSAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAIU5JREFUeJzt3Xl0VPXdP/D3vXf2JDNZSAgmRBaRUHGDygNxC1CWUlOrzwP1aOuCYpejVsVjjdoDPtpGjqhQu4iVivFYitXjafuAFLFqBf3ZgoJRCUEIEsQA2WYmme0u398fQyYZMgmTyYQZkvfrnBzP3Jl75zODue98l3u/khBCgIiIKI3JqS6AiIjoVBhWRESU9hhWRESU9hhWRESU9hhWRESU9hhWRESU9hhWRESU9hhWRESU9hIOKyEEPB4PeE0xERENtoTDyuv1wuVywev1JrMeIiKiHtgNSEREaY9hRUREaY9hRUREaY9hRTQE+Hw+fPTRR/D5fKkuhWhQmAZ6gMYtPnQ4BnwYIoqTYpVQMNseta22thZTp07Fzp07MWXKlBRVRjR4Bpwyhl/AkDh9nYiIBg+7AYmIKO0xrIiIKO0xrIiIKO0xrIiIKO0xrIiIKO0xrIiIKO0xrIiIKO0xrIiIKO0xrIiIKO3xPklEw0AgEMAHH3yAo0ePIi8vD2VlZcjIyEh1WURxY1gRDXGrVq3CsmXLoGkacnNz0draCkmSUFVVhTvuuCPV5RHFJe6wCgaDCAaDkccej2dQCiKi5HnhhRfw1FNPYcOGDZg7dy5kWYZhGNiyZQtuu+02FBcX43vf+16qyyQ6pbjHrKqqquByuSI/o0ePHsy6iCgJfv3rX+PFF1/E/PnzIcvhX3dZljF//nxUV1dj5cqVKa6QKD5xh1VlZSXcbnfkp6GhYTDrIqIkOHjwIMrLy2M+V15ejs8+++z0FkSUoLi7Aa1WK6xW62DWQkRJZrfbcfz4cRQUFPR4rqOjAzabLQVVEfUfp64TDWHz5s3D0qVLoet61HbDMPDzn/8cc+bMSVFlRP3DsCIawqqqqrBr1y5MmDABzc3Nke033ngj3n77baxYsSKF1RHFj1PXiYawwsJCrF+/Hq+//joeeugh5OfnY9GiRXj44Ycxfvx4mM3mVJdIFBe2rIiGsE2bNqGsrAy1tbXIy8tDfX09ZsyYgX379jGo6IzClhXREFZZWYnXXnstamxq69atWLp0KSoqKlJYGVH/sGVFNITV19dj1qxZUdtmzZqFQ4cOpagiosQwrIiGsEmTJqG6ujpqW3V1NSZPnpyiiogSw25AoiFs9erVmD9/PtasWYOxY8fi0KFDqKurw+bNm1NdGlG/MKyIhrDp06ejrq4OmzZtwrFjx1BRUYEFCxYgOzs71aUR9Qu7AYmGuOPHj6O+vh779+9HbW0tDhw4kOqSiPqNYUU0hL3yyisoKyvDgQMHkJ+fj4aGBsycORNr165NdWlE/TLgbkDZLkG2S8mohYjioFjj/327//778cYbb6CsrCyy7c4778TVV1+NW2+9dTDKIxoUAw6rwrkOOJ2OZNRCREnm9Xpx8cUXR22bNGlS1Np0RGcCdgMSDWF33nknbrnlFtTW1sLj8aCmpgbXX389lixZgoMHD+LLL79MdYlEcZGEECKRHT0eD1wuF9xuN5xOZ7LrIqJ++OijjzB16lTs3LkTU6ZMiWxXFKXP/YQQMAxjsMsjGrABdwP+8Z4fwW6xJKOWPmVk5+AHVasG/X2IhpKTlwYhOlMNOKw6Wlug84aYRGnp3XffjbldCIHy8nJs27YNl1122Wmuiqj/eFEw0RB2yy23xNwuhEB9fT1+8pOfoKam5jRXRdR/DCuiIexUFwAzqOhMwbAiGsJefPHFXp+76aabTmMlRAPDsCIawk4Oq+PHj2PPnj246qqrGFZ0RmFYEQ1h//znP3ts27hxI/74xz+moBqixPGiYKJhZu7cudiyZUuqyyDqF7asiIawk6eua5qGv/zlL5gwYUKKKiJKDMOKaAg7eeq6LMsYP3481q1bl5qCiBLEsCIawrh2FQ0VHLMiGmJ8Ph8ee+wx7N69G0IIrFu3Dt/61rdQWlqK2bNncy0rOiOxZUU0xNx+++1oa2vD7bffjvvvvx+bN2/GPffcg9GjR+Pw4cNYuXIl6uvr8dhjj6W6VKK4xR1WwWAwag0cj8czKAUR0cBs3rwZDQ0NsNvteP7551FTU4Pi4uLI83PnzsWFF17IsKIzStzdgFVVVXC5XJGf0aNHD2ZdRJQgSZKgqioAIDc3Fw5H9OKoNpsN2dnZKaiMKHFxh1VlZSXcbnfkp6GhYTDrIqIELVy4ENdddx3+8Y9/4JprrsG1116LrVu3Yu/evdiyZQsqKiqwZMmSVJdJ1C8DXnzxsWvmwnYalgjJzM3Dj37f+33OiIaz7osvnnfeefjVr36FN998Ey0tLQiFQj1e33nXdaIzBSdYEA0xVqsVjzzyCB555JFUl0KUNJy6TkREaY9hRUREaY9hRUREaY9jVkREQ1ygUYO3VoXaZgAALLkynOdZYMlT+nWc1p1BaO0G8q+093hODwp4Pg0h8LUGoQPmbBnOyRZY+/kevWFYEdGQYYQE2r9QETiiQ2s3IASg2CSYnDLsoxRkjBv8mcvpJnhcR/P2IExOCc7zLBCGQMcBDcf/FUD+lTZYcuMLk456Fb6DGiwjenbIGapA07t+6H6BzAlmyGYJ7ftVNP0rgIJZdphdA+/EY1gR0ZCgegw0bQtAhATso01wjDEBEqC1Gwgc0eEHhmVYte0OQXFIyJ9ph2ySAACOs804usUHz6chjLiiZyupOyEEvLUqvJ+rvb7Gu1eF5hUYcYUN1vxw+NlHKzi62Q9vXQi5l9gG/DkYVkR0xhNCoOXDAIQqkD+z51/y4gIBI5DQJaUDrgsCkGTptL83EG5pam4j3NoxddWg2CRY8xUEj+p97i90gWNvB6C5DdhLTAge7/l6IQR8hzTYCpVIUIXfQ4bzfAukJM2MYFgR0RlPcxvQPAK2IiVml5MkSVDs0YGhBwXa61QEvtagdQhICmB2ysg8xwx7cdepUWs34K1VETimwwgKKHYJGWNNyDzXDEnqOqa7JoT2OhUFc+zoqFfh/0qH4e9qbWg+A+17VQQadeiB8HHsRSZkTYoOkpN99VpHn589a5IZzm9YYj4nmYGRc+2QYpzpjaAATpGhQgeEKpDzX1Y4ik1ofMPX4zW6T8DwC1jPDQeVEAJCB2SThMzxyWvJMqyIhoDS0lLs3LkTpaWlqS4lJQwt/F+9XcDQRJ8nf+BEl+F7ARghgYwxJmROkGEEBQJHdageA50dY4GjGlo+CEK2ScgcZ4JskRBo1OH5VIURFHBdYO06ZpsOyEDLBwEoGTKyzjVD6AJml4xQi46mbQHIZgmOs01Q7BJCrQba96nQ2g3kzei9myznEmuvzwHoczxIkiSYsnp+F6rbQKjZgHVk3+NVkhkYOc/eZ8tQaw9P2pCtEtw1IXTUqxAqoGRIcF1ggf2s5MQMw4poCHA4HJgyZUqqy0gZc7YMxSFBdRto3OiLdElZRyowZUSfzA1VoHlbAJIMjJxjhymz6/ms0nDXFwBoHQZaPgjCnC0j7zJbJAAzxplx/F0/2r/QkDXJAtkc3q62GYABOM42Iau0q6WjBwWa3w+Ej1PW7ThjAcUqwVsbDqzudXTnKEnuadrQBFr/Ezzxeftu+UiSdMrWl3FiKMvzeQiSBLgusECSAG+dipYPgsi7TILtFKEYjwF/Cxk5ubBbYjdBkykjO2fQ34OIzkyySUJ+uQ3t+8Ldb/7D4R8AsBbIyJ5qhckRDgNvrQr9RPdcrICQFCnyOqED2VOsPVpq1nwFoSYDWocBS7YC3WfACIWnhGdOjA6A9r0hGKHwSVzogK53jZ2ZnOH37yus9GDfY22SglO2JDsZWjg4VbeBrFIzrCOSMK38xOcRIYGCeQ7IlnAttlEmNP7DB89nIdhG9j2JIx4DDqvFT6+B0+kccCFERAOh2GW4LrDCdUG4VRQ8qqN9v4rgMQNtO4IYcYU9MhnAnCNHTQY4mRAC/q/C07TNzj662U4EW+jE9UuOs01R41hCCPgadEAAx98K9HqczhN8LI3/13OcqLu+xqy6M0ICzdsDCLUYcIwxwXlechoZ0omgtBWZoj6HbJFgH2WC70sNhioiLdBEsRuQiIYcU4YM0zgZtiITGv/Ph2CTASEEjGB4VqC9qO8Whe4XECpgzoodVKrbgGQCTJndugCBHgFoBBF+vxJTn915fY075V3W97RvU8apQ0APCDRtC8/qc4w1Ifvi5PWGdU5cUaw965BPbBO6ABhWRESxSTIACZDN4fEXQ+3n9PUYGaIHDAQaddiLlEgrSm0Lh5eSGX1C7nw/xZ74uM1Ax3s6x+g0t4GMc0zIvrDvCRv9ZXLKgByetHIyvcMA5K7QGtD7DPQAjVt86HAw8+j0U6wSCmYPvC+czmzBJh1mlxyzm8lbqwICsJ9o1Sh2CZCB4DEdwhBRs9y6XxOl2CVIZiB0okXWGUpC7z45oat1oroNmLPlqC7A7u8XOKLBOckc6TbspAdFOEgH8Tqstl1BqIMUVEB4vMw+SoH/SHgmZWe3qdZhwP+1Dtsopcf3kogBp4zhFzCk03+xHREREJ6FprYasJ2lwJKtQDKHu/ECR3SobQasBXJkfEY2ScgYZ0LHFxqOvxOAvdgE2QRo7eExqvxZdijWcCssa6IZnk9VNG8Pwn6WAkMV8H0ZviYrd5o1clI2QgK6T8B2Vs8WkGySkHmOGe11Ko695YejxATZKkEPiPD08SYdhd9xDNp3o7oN+A/pkMzhrkbfIa3Hazq7J7V2A6EWA5ZcudfJHr1xnm9BsMmPpn/5kXmOGZCA9v0aJAVwJWlsLClNItk+eH8VEJ3M8Pf+x5HP50NtbS1KS0vhcAzeSYDSR+Y5ZgS+1hFq1hFo1CFUQLYA5mwFOZdYYR8d/Ze96wILzFkyOuo1ePeEV1FWHBIcJaaocZfMc8Oz+jrqNbTt0iFbJVhHyMj9L0vUGFPn5ApLduwTvHOyGWaXjPb9Ktr3hWcYylYJ5mwZrgutSWl19CbYFJ4RKVSgbWfPFaOBrrAKNulo2xlC9lRLv8PKlCEjv9wO96cheOvCrVnLCAWu8y0w9TLu118DXtZ+71++xrn/U5iUYoji8fUmHwy/gGyXMGpBdCB1X959OF93RDTUcD0rIiJKewwrIiJKewwrIiJKewwrIiJKewwrIiJKewwrIiJKewwrIiJKewwrIiJKewwrIiJKe3HfbikYDCIYDEYeezyeQSmIaDA1NTXhjTfewFdffYWCggIsWLAAhYW8Awv1T+MbPljzFeR8M/k3hk22QKMGz2cqNK8B2SohY6wJmRPNfd7mqb/7aD4Dx7b6kTfD1mOZlFCzDvdnIagt4TvTW0ea4DrfAsXWv9tMxd2yqqqqgsvlivyMHj26X29ElGrvvfceJk6ciLVr16KmpgbV1dU499xzsXXr1lSXRjQogs06mt8PwuSUkDvdCkeJCZ7PVLTvVZO2j9ZhoPm9AESMp0OtOo7/KwBJkZA7wwrnZAuCx3S0fND7QpS9ibtlVVlZiXvvvTfy2OPxMLDojHL33Xejuroa3/nOdyLb/vrXv+K+++7Drl27UlcY0SDxfq7CnC0j95LwAo62QkAYgHeviswJPZcs6c8+QoTvQu+uiX2DXABw14RgzpaRV9Z1w17ZLMG9OwStw4ApI/6RqLjDymq1wmpN/yYvUW8OHDiABQsWRG27+uqrceONN6aoIkpU8/YA9IBAxjgT2vdr0LwGFKsExxgTsib13l3VujOIwBENhd9xRK0h5a4JoaNexairHIAAPHtU+A9r0P0CkgxYcmU4z7fAkh17IcRY3YIdB1W07Qxh5Hx75KTcUR++87rWIbrqLTX3up6V1mHg6GZ/r9+DZYSM/Ctjr+kmdIFgkw7nN8xR2+1FCtrrVASbdNhGmhLeR3UbaPs4hIxxJtgKFDS/H4zaRw8KhI4byPmmJerfw15kgr2o/wt+cNVEGjYuvvhiPPXUU7j77ruhKAo0TcPq1asxefLkVJdG/RRqMyBUgfZ9KrJKLZCtEnwHVXj3qFDsEjLGmmPu5zjbBN9BDcFjOmyF4dOfEAL+wxrsxSZIsoTm/xdAqEmHc7IFpgwZWrsBz2cqWj8MomCuPeElPby1IXg+U5Ex3gTXKAVqmwHP5yp0v0DO1NgNAcUmIb+892XtpT6Witc6BGCgx3IfnY81rwBGJr6PySGjcJ4dikNG8Lje8/3d4aVTZKuEln8HEPhaBwRgO0tB9kVWyJb+fY8MKxo2nn/+eSxcuBCbNm3CkiVL0NDQgBdeeAGvvvpqqkujftD9BoyAgJIhIX+mPbJCsDVfRrDJj8ARvdewsuTJUBwS/Ie7wirUbED3CThKTBCGgNAA14VWOEabThxXgaECnppQ+H0TWL/PUAW8tSoyxpqQfVE4mGwjAdkioe2jEDInmCOLOXYnKRIseYkta2+o4dWfJFN0vdKJs77Qeq4O1Z99ZIsE9BE4eij82tadIdgKFeTOsEFvN+D+NISm7QHkl9v6FfwMKxo2NE1De3s7Dh06hGeeeQYNDQ2QJAmyzCs4ziShlvBf7M5J5qil7CVZgilDgh4SkSXqu5NkCZIUXmSxfb+KbMMCSZbgb9CgZEiwjgiHwojLwi0Z3W9AaxfQvAYCjeEVdoWRYM3NOoQeblUIo6sw2ygTgBCCR/WYYRV+zz6WHJTQ+wlfdL2mt32Tsk9vTnxXlhy5q+VYoEAyS2j9dxDBo11/MMSDYUXDxk9/+lPcdNNNePDBByPbHn/8cdx1113YsmVLr/spioKT1yiVJClqm2EkeBajflPbDEAC7MU9T196QMCSLcO7J9wl2F3Rf2cACHcFemtVBBp12AoV+A9ryBjf1RILNGpwfxKC5hWQTOHl4E9uafSXcWIOQvP2YMzn9UDsQBrImFVnN5tQo48tTqxsH+szJbJPbzpbY7bC6JahbWT4seo2YOvHVSMMKxo2duzYgc2bN0dtu++++7BixYo+96uvr++xTVVVVFVV4aWXXsLSpUuTWif1LdQavvbn5JlsoRYdeoeAtVSBbaQC26jY3WemTBnmXBn+wxokORwknV1+WruB5g+CsJ+lIK+sa3n39v0qgkd7jst0d/IfNJ0neACQT2RhziVWmLJihIQ1dggodgn5s/oYs+ojPEwZEiCFP1N3nY9jteQS2afX9z/x3Z3cGu38mmLNROzzeP16NdEZbMSIEfj4449xySWXRLZ9/PHHGDVqVJ/7lZSURD3etWsXFi9eDFVV8f7772Pq1KmDUi/FprbqMFTACImuloAQcNeEoDjC3XySLEGJ3eAAgBPXDoUACTDnyjBlhU+soTYDMICsieaoSQbBxhNB1UuPnGSWoPuinww1d4WbOVcB5HDXoqPE0vVZ3Abcu4PImmSByRHjuLIES05iY1aSIsEyQob/iI7Mc0Wku9D/lQ7JHJ7hmIx9emPKkk6MD2rIPKdby/VIOMUtI/rX/c6womHjwQcfxPz587FkyRKMHTsWX375JZ577jmsXr06rv1VVcWjjz6KJ598Evfeey+WLVsGk4m/QqeT1mHACAGKQ0LLhwFkTjBD6OGWj9pmYMQVtl6ngXfnGG2C+5MQ/A06XBd2hYclWwYkwP2piswJ4fEi30ENgRNhJXppXNkKFbTvVeGtDcGSqyDQqEXNkFOsErLONcPzuQqhAZZ8BYY/PBsQAMzZgzNu6iy1oOm9AFo+DCJjjAmhZgPtdSqck7uul9J9BnS/gDlbhqRIce0TD0mS4DrfgpYPg2j5MADHGDM0rwHPZyHYipReLwPoDX/TaNhYvHgxzjrrLPz5z3/GJ598guLiYrz++uu4/PLLT7nvjh07sHjxYiiKgu3bt+Oiiy4a/IKpB7U13KeUM82KjgMaWj4MjwFZ8xXkz7TH3U0lWyTYChUEvtajxr5MmTJyp1nh2RNC8/sByBYJllwZI66woelfAQSbdJhdPd8jq9QMIyjgrVMBocJWqCB7ihUtH3SNUTnPs0C2SejYr8Jbp0I2S7AWyHBOtkRNFEkma4GC3OlWeD4PofmDIBSbBOf5FmSd29XS6TiowbtHPXE9mBTXPvGyF5uQqwDePWrk+8wYa4bzvP4fi2FFw8qsWbNgtVrR2NiIkSNHRnUJ9qayshIrV67ED37wAzz00EMwm8348ssvo15z9tlnD1bJ1E2o1YCkhLujrHlWAInfqCBvRuyxIHuxKebkjc4JGgBQ+O3oPjvZJCFnqrXH9VJF/x19nMzxZmSO7/+JeiBOdRGu8xsWOL9hidrW3wt3rflK1PcTdaxRJthHDTxqBnwEwy/w9SbfgAshipfh72Mqbx9qampQUVEBwzBQUlKCI0eOwO/34/XXX8f06dN73e+JJ56AEALV1dWorq7u8bwQgrMBT5NQqw5zjpzwhbl05kpKyyrRkwfR6bR48WLcc889+NnPfhbZtnbtWvz4xz/u896Amqb1+hydPkIIqG0GHGezQ2g4ksTJ8y3j5PF44HK5sPcvXyPL4Ux2XUSnpFglFMyOnvL10UcfYerUqdi5cyemTJkS9VxWVhZaW1ujJkXouo6cnJxTLnmzb98+bNiwAYcPH0Z+fj4WLlyICy64IHkfhoj6NOA/UQrnOuB0xphzSZRmKioq8Nvf/jbSshJCYOXKlbjhhhv63G/Tpk247rrrUFFRgTFjxqC+vh5lZWVYv349KioqTkfpRMMe29M0bOzZswcbNmzAihUrUFhYiMOHD6O5uRnFxcUYO3YsgNgXAFdWVuK1117DnDlzItu2bt2KpUuXMqyIThOGFQ0b8V5PdbL6+nrMmjUratusWbNw6NChZJRFRHFgWNGwccUVVyS036RJk1BdXY1bbrklsq26uppLixCdRgMOqz/e8yPYLZZTvzCFMrJz8IOqVakug1KsrxvSGoYBp9MZc6LF6tWrMX/+fKxZsyZy54t9+/b1uM8gEQ2eAYdVR2sLdPPpvciNKBGxxqO6+/zzz2Nunz59Ourq6rBp0yYcO3YMFRUVWLBgAbKzswehSiKKZcBhJUkyMnPzklFL0nW0tkIkugANDTmnukqjuLg45vbOu1XMnDkzss3tdsPtdvPOFUSnyYDDypGdjR/9/sVk1JJ0a35yE9pbmlNdBqWJcePGQQgRdfcDIQSKiorQ0NCQ9P2IKHk4wYKGDV2PvmV2S0sLnnjiiVPeOT3R/YgoebieNw1bubm5eOyxx/D73//+tOxHRIljWNGwpes6/vSnPyErK+u07EdEiWM/Bg0bnVPXu489ZWZm4ne/+11c+3WXlZV1yv2IKHkYVjRsnDx1XZZlFBQUwHKK6wQT3Y+IkodhRUPetm3bUFJSgpKSEuzfvx/r16/H4cOHUVRUhEWLFmHixIl97l9SUnKaKiWi3nDMioa0xx9/HNdffz38fj9efvllTJkyBXv37kVeXh6++OILTJs2Da+88kqqyySiU2DLioa0Z555Bv/+979RVFSEOXPm4I033kBZWVnk+e3bt+OHP/whFi1alMIqiehU4g6rYDCIYDAYeXyqxeqI0kFHRwdGjhwJIPz/8CWXXBL1/LRp0+Dz+VJRGhH1Q9zdgFVVVXC5XJGf0aNHD2ZdREkxffp0/O///i8CgQBuu+02PPDAA/D7/QAAn8+Hu+++GzfeeGOKqySiU4k7rCorKyP3Q3O73bzNDJ0RnnvuObz77rvIysrCk08+iVWrVsHlcmHUqFHIycnBs88+i1dffTXVZRLRKcTdDWi1WmG1WgezFqKkKykpwbvvvotAIIC2tjaEQqFUl0RECeAECxoWbDYbCgsLU10GESWIU9eJiCjtMayIiCjtMayIiCjtMayIiCjtMayIiCjtMayIiCjtMayIiCjtMayIiCjtDfiiYF9bG9b85KZk1JJ0Ha2tqS6BiIiSYMBhJYSB9pbmZNRCREQU04DDKiMnF/Y0X947Izsn1SUQEdEASEIIkciOHo8HLpcLbrcbTqcz2XURJcTn86G2thalpaVwOBypLoeIkoRhRUREaY+zAYmIKO0lPGbV2SDj8vZE6SsrKwuSJKW6DKIBSzisvF4vAHB5e6I0xm56GioSHrMyDANHjhzhX25EaYy/nzRUJBxWREREpwsnWBARUdpjWBERUdpjWBERUdpjWKWZdevWYd68eVGP161b169jHDx4EDfffHPSalq0aBFuuOGGqJp27drV5z7vvPMOli9fHvO58vLypNVGRMMDwyoN7dixA3/7299SXUbE0aNH8fLLL0ce33zzzbjoootSVxARDTsMqzR03333Yfny5QiFQpFty5cvxzvvvAOgq2Uyd+5cLFq0CNOnT8fTTz+NsrIyLFu2DABQU1OD2bNnY968eWhpacGRI0cwb948XH755fjNb34DALjyyisxc+ZM/P3vf4+8z1tvvYXp06djxowZePvtt/HLX/4Su3fvxlNPPdWjlptvvhl33HEHpk2bhtWrVwMAHn74YVx66aWR13u9XlxzzTUoLy9HZWVl1Oe86667cOWVV+K73/0ugsFgcr9EIhpSGFZpaNSoUVi0aBGefvrpPl/X0NCANWvWYOHChWhra8O2bduwdetWAIDVasXWrVtx66234g9/+ANWrFiB5cuX47333sNbb72F5uZmNDY2YuPGjaioqIgc85FHHsGbb76JLVu2YNmyZXjooYdw0UUX4d57741Zw1VXXYVt27ahuroaTU1N+OSTT7B9+3Z8+9vfBgA8//zzWLRoEd555x243W7U1NRE9v3444+xceNG3HXXXXC73QP92ohoCGNYpal77rkHGzZsQGNjY4/nOi+NKygoQE5ODvLy8jB27FjIsgyz2QwAuPjiiyFJEi688EIcPHgQdXV1eOCBB1BeXo4DBw7g0KFDGDlyZI87kyuKgqysrLgvJp04cSIsFguysrJQX1+PyZMnAwCmTJkCAKirq8OqVatQXl6ODz/8EAcPHozsW1lZie9///tYu3YtFEVJ6HsiouGBYZWmrFYrli1bhieeeCLy+Pjx4wgGgzhw4AAA9BkmNTU1EEJgx44dmDBhAsaNG4dnn3020n03ZswYyHLPf35N0+D1euH1euPqmutew/jx47F79+7I+wPAuHHjIt2Gd955J8477zwAQDAYxH/+8x9s3LgRl156KTZu3BjnN0NEw9GAF1+kwXP11Vdj6tSpAIBrr70W1113HV566aW47seo6zpmz54Nu92O9evXw+1247bbbkNbWxu++c1vIicn9oKUv/jFLzBnzhzouo4nn3yyX/Xm5uaivLwcM2bMQFFRESZPnozbb78dN910Ex599FEUFxdHZhVarVa0tbVh2rRpyMvLw4svvtiv9yKi4YW3WyIiorTHbkAiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7DCsiIkp7/x/3AvwiSHF1LAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 300x175 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "generate_all_side_figures(\"all_samples\", deepcsa_run_dir,\n",
    "                          gene_list = ['TP53'],\n",
    "                          store_plots_dir = f\"{plots_dir}\",\n",
    "                          tools = [\"oncodrivefml\", \"oncodrive3d\", \"omega_trunc\", \"omega_mis\", \"excess_indels\"]\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d917798-cbeb-479a-bfdc-3313dbbe4030",
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate_all_side_figures(\"all_samples\", deepcsa_run_dir,\n",
    "#                           gene_list = ['TP53', \"RBM10\", \"FGFR3\"],\n",
    "#                           store_plots_dir = f\"{plots_dir}\",\n",
    "#                           tools = [\"excess_indels\"]\n",
    "#                          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba483108-fde0-448c-a537-7009b0dca604",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fa8bc3d-c7ad-4f73-9dd6-fec72fe69ee1",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import lognorm\n",
    "from matplotlib.lines import Line2D\n",
    "\n",
    "\n",
    "def get_toy_sim_vector(sim_vec, scale=0.002):\n",
    "    \n",
    "    shape, loc, _ = lognorm.fit(sim_vec, floc=0)\n",
    "    toy_vec = lognorm.rvs(shape, loc, scale, size=(1, 10000)).flatten()\n",
    "    \n",
    "    return toy_vec\n",
    "\n",
    "\n",
    "def plot_o3d_side_plot(sim_score_vec,\n",
    "                       observed_score,\n",
    "                       o3d_top_pos_score_gene,\n",
    "                       pvalue,\n",
    "                       scale=0.017,\n",
    "                       figsize=(3, 1.75),\n",
    "                       label_fontsize=12,\n",
    "                       legend_fontsize=12,\n",
    "                       text_fontsize=12,\n",
    "                       background_color_fill=\"darkgray\",\n",
    "                       background_color_line=\"dimgrey\",\n",
    "                       observed_color=\"#3b99aa\",\n",
    "                       output_path=None):\n",
    "    \n",
    "    \n",
    "    toy_sim_vec = get_toy_sim_vector(sim_score_vec, scale=0.017)\n",
    "\n",
    "    fig, ax = plt.subplots(figsize=figsize)\n",
    "\n",
    "    # Plot DKE and extract data\n",
    "    kde = sns.kdeplot(x=toy_sim_vec, ax=ax, color=background_color_fill, linewidth=0.1)\n",
    "    x, y = kde.get_lines()[0].get_data()\n",
    "\n",
    "    # Plot the fill and extended line\n",
    "    x_min, x_max = 0, 0.075\n",
    "    x_extended = np.concatenate(([x_min - (x_max - x_min) * 0.05], x, [x_max + (x_max - x_min) * 0.05]))\n",
    "    y_extended = np.concatenate(([0], y, [0]))\n",
    "    ax.fill_betweenx(y, x, color = background_color_fill, alpha=0.2, label=\"Randomized mut densities\")\n",
    "    ax.plot(x_extended, y_extended, color=background_color_line)\n",
    "    ax.vlines(observed_score, ymin=0, ymax=max(y)/2, color=\"black\", zorder=1, linestyle='--', label=\"Observed mut density\")\n",
    "\n",
    "    # Details\n",
    "    ax.set_ylim(-0.05 * max(y), max(y) * 1.05)\n",
    "    ax.set_xlim(0.0005, x_max)\n",
    "    ax.set_xticklabels([])\n",
    "    ax.set_xticks([])\n",
    "    ax.yaxis.set_visible(False)\n",
    "    plt.ylabel(None)\n",
    "    plt.xlabel(\"Clustering score\", fontsize=label_fontsize)\n",
    "\n",
    "    for spine in ['left', 'right', 'top']:\n",
    "        ax.spines[spine].set_visible(False)\n",
    "\n",
    "    # Arrow\n",
    "    arrow_props = dict(facecolor= observed_color, edgecolor = observed_color, arrowstyle='<->')\n",
    "    ax.annotate('', xy=(np.mean(toy_sim_vec), max(y)/2), xytext=(observed_score, max(y)/2), arrowprops=arrow_props)\n",
    "\n",
    "    # Legend\n",
    "    legend_handles = [Patch(facecolor=background_color_line, alpha=0.2, edgecolor='none', label='Randomized mut densities'),\n",
    "                      Line2D([0], [0], color=\"black\", linestyle='--', label='Observed mut density')]\n",
    "    legend = ax.legend(handles=legend_handles, frameon=False, loc='upper right', bbox_to_anchor=(1.677, 1.1), fontsize=legend_fontsize)\n",
    "\n",
    "    # Adjust the color of the text labels in the legend\n",
    "    for text, color in zip(legend.get_texts(), [\"black\", \"black\"]):\n",
    "        text.set_color(color)\n",
    "\n",
    "    text = fr'$\\mathit{{Score}} = {np.round(o3d_top_pos_score_gene, 2)}$'\n",
    "    ax.text(observed_score*1.1, max(y)/2, text, ha='left', va='center', fontsize=text_fontsize+1, color=observed_color)\n",
    "    pvalue = \"{:.0e}\".format(pvalue)\n",
    "    ax.text(observed_score*1.1, max(y)/2 - 0.35*(max(y)/2), fr'$\\mathit{{p}}$-value < {pvalue}', ha='left', va='center', \n",
    "            fontsize=text_fontsize+1, color=observed_color)\n",
    "\n",
    "    if output_path is not None:\n",
    "        print(f\"Saving {output_path}\")\n",
    "        plt.savefig(output_path, bbox_inches='tight')\n",
    "        \n",
    "    plt.show()\n",
    "    \n",
    "o3d_temp_data = \"/home/spellegrini/TEMP_DATA\"\n",
    "sim_score = pd.read_csv(f\"{o3d_temp_data}/TP53_simulated_scores.csv\")\n",
    "sim_score = sim_score.drop(columns=[\"index\"])[:1].values[0]\n",
    "\n",
    "plot_o3d_side_plot(sim_score,\n",
    "                   o3d_top_score_gene,\n",
    "                   o3d_top_pos_score_gene,\n",
    "                   o3d_pvalue,\n",
    "                   scale=0.017,\n",
    "                   figsize=(3, 1.75),\n",
    "                   label_fontsize=12,\n",
    "                   legend_fontsize=12,\n",
    "                   text_fontsize=12,\n",
    "                   output_path=f\"{plots_dir}/side_plot_toy_tp53.png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c6c2f94-b69e-4e19-874a-e4cc72b592b4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf156095-540c-4c1b-905f-2df602b78deb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "notebooks_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
