{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4c17f924-1b27-498b-aa7c-c76cb965c08e",
   "metadata": {},
   "source": [
    "# Figure 5a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bf065ef5-b980-4828-baa2-f32c360ebd76",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d174c92c-14ee-493b-9161-db3e5f3f8083",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append('../../')\n",
    "from consensus_variables import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6729d064-73b5-428b-9c8c-01f4c37c1107",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Init\n",
    "# ====\n",
    "\n",
    "\n",
    "# Cancer\n",
    "path_intogen = \"/data/bbg/datasets/intogen/output/runs/v2024/20240409_ALL\"\n",
    "cancer_cohort_df = pd.read_table(f\"{path_intogen}/cohorts.tsv\")\n",
    "path_cancer_maf_all = f\"{path_intogen}/steps/vep/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52f74667-285e-4443-8b89-edf53b0e86f7",
   "metadata": {},
   "source": [
    "# Get data\n",
    "## Cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2ba6eb41-2f21-4bc2-95da-9d532573bde7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_broad_consequence(list_of_annotations):\n",
    "    \"\"\"\n",
    "    Group variants into broader consequence types.\n",
    "    \"\"\"\n",
    "        \n",
    "    CONSEQUENCES_LIST = [\n",
    "        'transcript_ablation',\n",
    "        'splice_acceptor_variant',\n",
    "        'splice_donor_variant',\n",
    "        'stop_gained',\n",
    "        'frameshift_variant',\n",
    "        'stop_lost',\n",
    "        'start_lost',\n",
    "        'transcript_amplification',\n",
    "        'inframe_insertion',\n",
    "        'inframe_deletion',\n",
    "        'missense_variant',\n",
    "        'protein_altering_variant',\n",
    "        'splice_region_variant',\n",
    "        'splice_donor_5th_base_variant',\n",
    "        'splice_donor_region_variant',\n",
    "        'splice_polypyrimidine_tract_variant',\n",
    "        'incomplete_terminal_codon_variant',\n",
    "        'start_retained_variant',\n",
    "        'stop_retained_variant',\n",
    "        'synonymous_variant',\n",
    "        'coding_sequence_variant',\n",
    "        'mature_miRNA_variant',\n",
    "        '5_prime_UTR_variant',\n",
    "        '3_prime_UTR_variant',\n",
    "        'non_coding_transcript_exon_variant',\n",
    "        'intron_variant',\n",
    "        'NMD_transcript_variant',\n",
    "        'non_coding_transcript_variant',\n",
    "        'upstream_gene_variant',\n",
    "        'downstream_gene_variant',\n",
    "        'TFBS_ablation',\n",
    "        'TFBS_amplification',\n",
    "        'TF_binding_site_variant',\n",
    "        'regulatory_region_ablation',\n",
    "        'regulatory_region_amplification',\n",
    "        'feature_elongation',\n",
    "        'regulatory_region_variant',\n",
    "        'feature_truncation',\n",
    "        'intergenic_variant'\n",
    "    ]\n",
    "    \n",
    "    GROUPING_DICT = {\n",
    "        'transcript_ablation': 'nonsense',\n",
    "        'splice_acceptor_variant': 'nonsense',\n",
    "        'splice_donor_variant': 'nonsense',\n",
    "        'stop_gained': 'nonsense',\n",
    "        'frameshift_variant': 'nonsense',\n",
    "        'stop_lost': 'nonsense',\n",
    "        'start_lost': 'nonsense',\n",
    "\n",
    "        'missense_variant': 'missense',\n",
    "\n",
    "        'inframe_insertion': 'indel',\n",
    "        'inframe_deletion': 'indel',\n",
    "\n",
    "        'splice_donor_variant': 'splicing',\n",
    "        'splice_acceptor_variant': 'splicing',\n",
    "        'splice_region_variant': 'splicing',\n",
    "        'splice_donor_5th_base_variant': 'splicing',\n",
    "        'splice_donor_region_variant': 'splicing',\n",
    "        'splice_polypyrimidine_tract_variant': 'splicing',\n",
    "\n",
    "        'synonymous_variant': 'synonymous',\n",
    "        'incomplete_terminal_codon_variant': 'synonymous',\n",
    "        'start_retained_variant': 'synonymous',\n",
    "        'stop_retained_variant': 'synonymous',\n",
    "\n",
    "        'protein_altering_variant' : 'protein_altering_variant',\n",
    "        'transcript_amplification' : 'transcript_amplification', \n",
    "        'coding_sequence_variant': 'coding_sequence_variant', \n",
    "\n",
    "        'mature_miRNA_variant': 'non_coding_exon_region',\n",
    "        '5_prime_UTR_variant': 'non_coding_exon_region',\n",
    "        '3_prime_UTR_variant': 'non_coding_exon_region',\n",
    "        'non_coding_transcript_exon_variant': 'non_coding_exon_region',\n",
    "        'NMD_transcript_variant': 'non_coding_exon_region',\n",
    "\n",
    "        'intron_variant': 'intron_variant',\n",
    "\n",
    "        'non_coding_transcript_variant' : 'non_coding_transcript_variant',\n",
    "\n",
    "        'upstream_gene_variant': 'non_genic_variant',\n",
    "        'downstream_gene_variant': 'non_genic_variant',\n",
    "        'TFBS_ablation': 'non_genic_variant',\n",
    "        'TFBS_amplification': 'non_genic_variant',\n",
    "        'TF_binding_site_variant': 'non_genic_variant',\n",
    "        'regulatory_region_ablation': 'non_genic_variant',\n",
    "        'regulatory_region_amplification': 'non_genic_variant',\n",
    "        'feature_elongation': 'non_genic_variant',\n",
    "        'regulatory_region_variant': 'non_genic_variant',\n",
    "        'feature_truncation': 'non_genic_variant',\n",
    "        'intergenic_variant': 'non_genic_variant',\n",
    "        '-'  : '-'\n",
    "    }\n",
    "    \n",
    "    consequence_rank_dict = { consequence : rank for rank, consequence in enumerate(CONSEQUENCES_LIST) }\n",
    "    rank_consequence_dict = { rank : consequence for rank, consequence in enumerate(CONSEQUENCES_LIST) }\n",
    "    \n",
    "    list_of_single_annotations = []\n",
    "    list_of_broad_annotations = []\n",
    "    for x in list_of_annotations:\n",
    "        all_consequences = x.split(\",\")\n",
    "        all_consequences_ranks = map(lambda x: consequence_rank_dict[x], all_consequences)\n",
    "        single_consequence = rank_consequence_dict[min(all_consequences_ranks)]\n",
    "        list_of_single_annotations.append(single_consequence)\n",
    "        if single_consequence in GROUPING_DICT:\n",
    "            list_of_broad_annotations.append(GROUPING_DICT[single_consequence])\n",
    "        else:\n",
    "            list_of_broad_annotations.append(single_consequence)\n",
    "\n",
    "    return list_of_broad_annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54d7d6cb-036c-4ea5-8032-cdd8568af32a",
   "metadata": {},
   "source": [
    "### Functions to load tumors data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cd261e2a-dea0-4f35-b1b2-ec71385916bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_cancer_maf_cohort(path_vep_out, lst_genes=None, only_protein_pos=True):\n",
    "    \n",
    "    columns_to_load = [\"SYMBOL\", \"Location\", \"Protein_position\", \"Feature\", \"Consequence\"]\n",
    "    vep_cohort = pd.read_table(path_vep_out, usecols=columns_to_load, low_memory=False)\n",
    "    if lst_genes is not None:\n",
    "        vep_cohort = vep_cohort[vep_cohort[\"SYMBOL\"].isin(lst_genes)]\n",
    "    vep_cohort = vep_cohort[vep_cohort[\"SYMBOL\"] != \"-\"].reset_index(drop=True)\n",
    "    \n",
    "    \n",
    "    # here we are removing the mutations that are in splice sites\n",
    "    if only_protein_pos:\n",
    "        vep_cohort = vep_cohort[vep_cohort[\"Protein_position\"] != \"-\"].reset_index(drop=True)\n",
    "        \n",
    "    cols = [\"SYMBOL\", \"Feature\", \"Location\", \"Protein_position\", \"Consequence\"]\n",
    "    vep_cohort = vep_cohort[cols].rename(columns={\"SYMBOL\" : \"Gene\", \n",
    "                                                    \"Feature\" : \"Ens_transcript_ID\",\n",
    "                                                    \"Protein_position\" : \"Pos\"})\n",
    "    \n",
    "    return vep_cohort\n",
    "\n",
    "\n",
    "def get_cancer_maf_all(cohort_df, path_all_vep_out, lst_genes=None, only_protein_pos=True):\n",
    "    \n",
    "    lst_vep_out = []\n",
    "    for cohort, ttype in cohort_df[[\"COHORT\", \"CANCER_TYPE\"]].values:\n",
    "        print(cohort)\n",
    "        path_vep = f\"{path_all_vep_out}/{cohort}.tsv.gz\"\n",
    "        if os.path.exists(path_vep):\n",
    "\n",
    "            vep_cohort = get_cancer_maf_cohort(path_vep, lst_genes, only_protein_pos)\n",
    "            vep_cohort[\"Cohort\"] = cohort\n",
    "            vep_cohort[\"Cancer_type\"] = ttype\n",
    "            lst_vep_out.append(vep_cohort)\n",
    "        else:\n",
    "            print(f\"Path {path_vep} doesn't exist: Skipping..\")\n",
    "    \n",
    "    df = pd.concat(lst_vep_out).reset_index(drop=True)\n",
    "    df.Pos = df.Pos.apply(lambda x: \n",
    "                            int(x) if len(x.split(\"-\")) == 1 else \n",
    "                            np.nan if (x.split(\"-\")[0] == \"?\" or x.split(\"-\")[0] == \"\") \n",
    "                            else int(x.split(\"-\")[0]))\n",
    "    if only_protein_pos:\n",
    "        df = df.dropna(subset=\"Pos\").reset_index(drop=True)\n",
    "        df.Pos= df.Pos.astype(int)\n",
    "    \n",
    "    # Ensure that all indels (including frameshift) are mapped to indels and not nonsense\n",
    "    df[\"INDEL_INFRAME\"] = False\n",
    "    df.loc[(df[\"Consequence\"] == 'inframe_insertion') | (df[\"Consequence\"] == 'inframe_deletion'), \"INDEL_INFRAME\"] = True\n",
    "    \n",
    "    # FIXME\n",
    "    # this redefinition here looks dangerous to me\n",
    "    df[\"Consequence\"] = df[\"Consequence\"].replace(\"frameshift_variant\", \"inframe_insertion\") \n",
    "    df[\"Consequence\"] = get_broad_consequence(df[\"Consequence\"])\n",
    "    \n",
    "    # Parse DNA location\n",
    "    df[\"Chr\"] = df.Location.apply(lambda x: f'chr{x.split(\":\")[0]}')\n",
    "    df[\"Location\"] = df.Location.apply(lambda x: x.split(\":\")[1])\n",
    "    df[\"DNA_pos\"] = df.Location.apply(lambda x: int(x.split(\"-\")[0]))\n",
    "    \n",
    "    return df.drop(columns=[\"Location\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6b34d8bd-163c-4719-b9f3-faf3c1e17c2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_cancer_type = \"BLCA\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dbfbbfe0-af89-4660-8229-9fffe60bddbe",
   "metadata": {},
   "source": [
    "### Covered DNA and DNA to protein mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ac841aa2-44c0-40e5-b190-c3236d32cdc8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>DNA_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>Protein_pos</th>\n",
       "      <th>REVERSE_STRAND</th>\n",
       "      <th>EXON</th>\n",
       "      <th>COVERED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696404</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696405</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696406</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696407</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26696408</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140207</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6751207</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>601.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140208</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6751208</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>601.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140209</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6751209</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>602.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140210</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6751210</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>602.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140211</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6751211</td>\n",
       "      <td>TRIP10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>602.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140212 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr   DNA_pos    Gene CONTEXT  Protein_pos  REVERSE_STRAND  EXON  \\\n",
       "0        chr1  26696404  ARID1A     NaN          1.0             1.0   0.0   \n",
       "1        chr1  26696405  ARID1A     NaN          1.0             1.0   0.0   \n",
       "2        chr1  26696406  ARID1A     NaN          1.0             1.0   0.0   \n",
       "3        chr1  26696407  ARID1A     NaN          2.0             1.0   0.0   \n",
       "4        chr1  26696408  ARID1A     NaN          2.0             1.0   0.0   \n",
       "...       ...       ...     ...     ...          ...             ...   ...   \n",
       "140207  chr19   6751207  TRIP10     NaN        601.0             1.0  14.0   \n",
       "140208  chr19   6751208  TRIP10     NaN        601.0             1.0  14.0   \n",
       "140209  chr19   6751209  TRIP10     NaN        602.0             1.0  14.0   \n",
       "140210  chr19   6751210  TRIP10     NaN        602.0             1.0  14.0   \n",
       "140211  chr19   6751211  TRIP10     NaN        602.0             1.0  14.0   \n",
       "\n",
       "        COVERED  \n",
       "0             0  \n",
       "1             0  \n",
       "2             0  \n",
       "3             0  \n",
       "4             0  \n",
       "...         ...  \n",
       "140207        0  \n",
       "140208        0  \n",
       "140209        0  \n",
       "140210        0  \n",
       "140211        0  \n",
       "\n",
       "[140212 rows x 8 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add depth info\n",
    "dna_prot_df = pd.read_table(f\"{deepcsa_run_dir}/dna2proteinmapping/exons_splice_sites.mapping.tsv\",\n",
    "                           sep=\"\\t\",\n",
    "                           header = 0)\n",
    "#dna_prot_df = dna_prot_df.merge(depth_df, how=\"left\", on=[\"CHR\", \"DNA_POS\"])\n",
    "#dna_prot_df.loc[dna_prot_df[\"COVERED\"] == 0, \"DEPTH\"] = 0\n",
    "dna_prot_df = dna_prot_df.rename(columns={\"CHR\" : \"Chr\",\n",
    "                                          \"DNA_POS\" : \"DNA_pos\",\n",
    "                                          \"GENE\" : \"Gene\",\n",
    "                                          \"PROT_POS\" : \"Protein_pos\"})\n",
    "\n",
    "dna_prot_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92a23617-464e-48f0-b222-ad997fdcb783",
   "metadata": {},
   "source": [
    "# Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bfc2e412-c825-4452-ad0c-89942ff49b40",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_pars = {\"fsize\"                     : (12,9),\n",
    "             \"hspace\"                    : 0.1,                    # General space between all tracks\n",
    "             \"track_title_x_coord\"       : 0.83,                   # x-coordinate (respect to protein len) for track txt title\n",
    "             \"score_txt_x_coord\"         : 1.13,                   # as track title but for track score txt\n",
    "             \"track_title_fontsize\"      : 6,\n",
    "             \"ylabel_fontsize\"           : 6,\n",
    "             \"xlabel_fontsize\"           : 6,\n",
    "             \"ylabel_pad\"                : 38,\n",
    "             \"ticksize\"                  : 10.25,\n",
    "             \"legend_fontsize\"           : 5,\n",
    "             \"legend_frameon\"            : False,\n",
    "             \"y_labels_coord\"            : [-0.05, 0.5],\n",
    "             \"cnsq_bbox_to_anchor\"       : [1.1123, 0.9],\n",
    "             \"o3d_bbox_to_anchor\"        : [1.27, 1.35],\n",
    "             \"depth_bbox_to_anchor\"      : [1.086, -2.2, 1.4, 5.05],\n",
    "             \"depth_0_bbox_to_anchor\"    : [1, -2.2, 0.35, 4.31],\n",
    "             \"legend_depth_fontsize\"     : 10.5,\n",
    "             \"txt_fontsize\"              : 5,\n",
    "             \"sse_bbox_to_anchor\"        : [1.2645, 1],\n",
    "             \"domain_y_bbox_to_anchor\"   : -3,\n",
    "             \"domain_x_bbox_to_anchor\"   : {\"ARID1A\" : 1.25,\n",
    "                                            \"KDM6A\"  : 1.245,\n",
    "                                            \"KMT2D\"  : 1.255,\n",
    "                                            \"EP300\"  : 1.267,\n",
    "                                            \"CREBBP\" : 1.267,\n",
    "                                            \"NOTCH2\" : 1.241,\n",
    "                                            \"KMT2C\"  : 1.255\n",
    "                                            },\n",
    "             \"sse_lw\"                    : {\"KMT2D\"  : 1,\n",
    "                                             \"EP300\"  : 1,\n",
    "                                             \"ARID1A\" : 1,\n",
    "                                             \"CREBBP\" : 1,\n",
    "                                             \"NOTCH2\" : 1,\n",
    "                                             \"KMT2C\"  : 1,\n",
    "                                             \"STAG2\"  : 1,\n",
    "                                             \"RB1\"    : 1,\n",
    "                                             \"RBM10\"  : 1,\n",
    "                                             \"KDM6A\"  : 1,\n",
    "                                             \"TP53\"   : 2.2,\n",
    "                                             \"FGFR3\"  : 1,\n",
    "                                             \"CDKN1A\" : 4,\n",
    "                                             \"FOXQ1\"  : 2.2,\n",
    "                                             \"PIK3CA\" : 1,\n",
    "                                             \"TERT\"   : 1,\n",
    "                                             },\n",
    "             \"covered_region_y_shift\"    : {\"KMT2D\"  : (-0.5,0),\n",
    "                                             \"EP300\"  : (0.1,0),\n",
    "                                             \"ARID1A\" : (0.2,-0.095),\n",
    "                                             \"CREBBP\" : (-0.65,0),\n",
    "                                             \"NOTCH2\" : (0,0.03),\n",
    "                                             \"KMT2C\"  : (0,0),\n",
    "                                             \"STAG2\"  : (-0.1,0.1),\n",
    "                                             \"RB1\"    : (0.04,0.03),\n",
    "                                             \"RBM10\"  : (0,0),\n",
    "                                             \"KDM6A\"  : (-0.3,0.18),\n",
    "                                             \"TP53\"   : (0,0),\n",
    "                                             \"FGFR3\"  : (0,0),\n",
    "                                             \"CDKN1A\" : (-0.4,0),\n",
    "                                             \"FOXQ1\"  : (0.15,0),\n",
    "                                             \"PIK3CA\" : (0,0),\n",
    "                                             \"TERT\"   : (0,0)\n",
    "                                            },\n",
    "                                             \n",
    "             \"dpi\"                       : 300,\n",
    "             \"colors\"                    : {\"nonsense\"    : \"#FB8E6F\",  \n",
    "                                             \"synonymous\"  : \"#ACECBD\", \n",
    "                                             \"missense\"    : \"#FBD180\", \n",
    "                                             \"o3d_score\"   : \"#6DBDCC\",   \n",
    "                                             \"o3d_cluster\" : \"#DAF0F8\", # \"skyblue\",  \n",
    "                                             \"indel\"       : \"#ECC4F7\", \n",
    "                                             \"splicing\"    : \"#A1C5DF\", \n",
    "                                             \"vlines\"      : \"lightgray\",\n",
    "                                             \"plddt_pacc\"  : \"#f7f3ff\"\n",
    "                                           },\n",
    "             \"sse_colors\"                : {'Coil'   : \"#d1d1d1\", \n",
    "                                             'Helix'  : \"#DDDD90\", \n",
    "                                             'Ladder' : \"#8ADEE6\"\n",
    "                                           }\n",
    "            }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d1f5530-4589-41fc-b4a3-7f0d268f254e",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Proportions of mutations by consequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5dfb8d66-49d5-43e4-be7d-86c5b6af28f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Individual gene\n",
    "\n",
    "def plot_proportion_mutation(gene_df, plot_pars, figsize=(1, 2), save=False, filename=\"gene_proportion.png\"):\n",
    "    \n",
    "    cnsq_gene_df = gene_df.groupby(\"Consequence\").Count.sum()\n",
    "    total_count = cnsq_gene_df.sum()\n",
    "    cnsq_gene_df = cnsq_gene_df / cnsq_gene_df.sum() * 100\n",
    "    df = pd.DataFrame(cnsq_gene_df).T\n",
    "    #cols = [\"synonymous\", \"splicing\", \"missense\", \"nonsense\", \"indel\"]\n",
    "    cols = [\"synonymous\", \"splicing\", \"missense\", \"nonsense\", \"indel\"]\n",
    "    #cols = [\"indel\", \"nonsense\", \"missense\", \"splicing\", \"synonymous\"]\n",
    "    df = df[[col for col in cols if col in df]]\n",
    "    ax = df.plot(kind='bar', stacked=True, figsize=figsize, legend=False,\n",
    "                    color = [plot_pars[\"colors\"][cnsq] for cnsq in df.columns])\n",
    "    # ax.legend(bbox_to_anchor=(1.677, 1.1))\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.set_ylabel(\"% mutations\")\n",
    "    ax.set_xticks([])\n",
    "\n",
    "    ax.bar_label(ax.containers[len(ax.containers)-1], labels=[total_count], label_type=\"edge\", fontsize=10, color='black', padding=4)\n",
    "    if save:\n",
    "        plt.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "    plt.show()\n",
    "                \n",
    "\n",
    "## All genes\n",
    "\n",
    "def get_proportion_mutation(gene_df):\n",
    "    \n",
    "    cnsq_gene_df = gene_df.groupby(\"Consequence\").Count.sum()\n",
    "    total_count = cnsq_gene_df.sum()\n",
    "    cnsq_gene_df = cnsq_gene_df / cnsq_gene_df.sum() * 100\n",
    "    df = pd.DataFrame(cnsq_gene_df).T\n",
    "    \n",
    "    return df\n",
    "\n",
    "    \n",
    "def add_edgecolor_for_tissue(df, count_lst, ax, shift=0, bar_width=0.5, lw=1):\n",
    "    \n",
    "    height = 100\n",
    "    df_dummy = df[[\"Type\"]]\n",
    "    df_dummy[\"Dummy\"] = height\n",
    "    df_dummy[df_dummy['Type'] == 'Normal'].rename(columns={\"Dummy\" : \"Normal\"}).plot(\n",
    "        kind='bar', ax=ax, position=1+shift, width=bar_width, color=['red'], lw=lw, edgecolor=\"black\", fill=False)\n",
    "    df_dummy[df_dummy['Type'] == 'Cancer'].rename(columns={\"Dummy\" : \"Cancer\"}).plot(\n",
    "        kind='bar', ax=ax, position=0-shift, width=bar_width, color=['red'], lw=lw, edgecolor=\"black\", fill=False)\n",
    "    \n",
    "    # Group the patches by their x position\n",
    "    bar_positions = {}\n",
    "    for patch in ax.patches:\n",
    "        x_position = patch.get_x()\n",
    "        if x_position not in bar_positions:\n",
    "            bar_positions[x_position] = []\n",
    "        bar_positions[x_position].append(patch)\n",
    "\n",
    "    # Sort the x positions to ensure \"Normal\" comes before \"Cancer\" for each gene\n",
    "    sorted_positions = sorted(bar_positions.keys())\n",
    "\n",
    "    # Iterate through the sorted positions in pairs, applying styles\n",
    "    x_txt_offset = -0.1\n",
    "    for i in range(0, len(sorted_positions), 2):\n",
    "\n",
    "        normal_patches = bar_positions[sorted_positions[i]]\n",
    "        x_pos = normal_patches[0].get_x() + normal_patches[0].get_width() / 2 + x_txt_offset\n",
    "        ax.text(x_pos, height + 2, count_lst[i], ha='left', va='bottom', fontsize=10, rotation=45)\n",
    "        for patch in normal_patches:\n",
    "            patch.set_linestyle('-')\n",
    "        \n",
    "        j = i + 1\n",
    "        if j < len(sorted_positions): \n",
    "            cancer_patches = bar_positions[sorted_positions[j]]\n",
    "            x_pos = cancer_patches[0].get_x() + cancer_patches[0].get_width() / 2 + x_txt_offset\n",
    "            ax.text(x_pos, height + 2, count_lst[j], ha='left', va='bottom', fontsize=10, rotation=45)\n",
    "            for patch in cancer_patches:\n",
    "                patch.set_linestyle('--') \n",
    "                \n",
    "\n",
    "def plot_proportion_mut_by_tissue(df, figsize=(10.5, 3), bar_width=0.35, shift=0.07, \n",
    "                                    title='Proportion of mutations', save=False, filename='proportion.png'):\n",
    "    \n",
    "    \n",
    "    total_count = df.Count.values\n",
    "    df = df.drop(columns=\"Count\")\n",
    "    \n",
    "    # Create the plot\n",
    "    fig, ax = plt.subplots(figsize=figsize)\n",
    "\n",
    "    genes = df.index.get_level_values('Gene').unique()\n",
    "    positions = np.arange(len(genes))\n",
    "\n",
    "    df = df.copy()\n",
    "    df_normal = df[df['Type'] == 'Normal']\n",
    "    df_cancer = df[df['Type'] == 'Cancer']\n",
    "    normal_cnsq = df_normal.drop(columns=[\"Type\"]).columns\n",
    "    cancer_cnsq = df_cancer.drop(columns=[\"Type\"]).columns\n",
    "    df_normal.columns = [col.capitalize() for col in df_normal.columns]\n",
    "    df_cancer.columns = [col.capitalize() for col in df_cancer.columns]\n",
    "\n",
    "    # Plot the normal and cancer bars\n",
    "    normal_bars = df_normal.plot(kind='bar', stacked=True, ax=ax, position=1+shift,\n",
    "                                 width=bar_width, color=[plot_pars[\"colors\"][cnsq] for cnsq in normal_cnsq], \n",
    "                                 label=[cnsq.capitalize() for cnsq in normal_cnsq])\n",
    "    cancer_bars = df_cancer.plot(kind='bar', stacked=True, ax=ax, position=0-shift,\n",
    "                                 width=bar_width, color=[plot_pars[\"colors\"][cnsq] for cnsq in cancer_cnsq], \n",
    "                                 label=[cnsq.capitalize() for cnsq in cancer_cnsq])\n",
    "\n",
    "    # Add edges for tissue type\n",
    "    add_edgecolor_for_tissue(df, total_count, ax, shift, bar_width, lw=1)\n",
    "\n",
    "    ax.set_xticks(positions)\n",
    "    ax.set_xticklabels(genes, rotation=45)\n",
    "    ax.set_xlim(-0.55, len(genes) - 0.5)\n",
    "    ax.set_xlabel(None)\n",
    "    ax.set_ylabel('% of mutation')\n",
    "    ax.set_title(title)\n",
    "\n",
    "    # Legend\n",
    "    handles, labels = ax.get_legend_handles_labels()\n",
    "    labels_dict = dict(zip(labels, handles)) \n",
    "    tissue_labels_dict = {key: labels_dict[key] for key in ['Normal', 'Cancer']}\n",
    "    del labels_dict['Normal']\n",
    "    del labels_dict['Cancer']\n",
    "    legend_mut = ax.legend(labels_dict.values(), labels_dict.keys(), title='Mutation', \n",
    "                                bbox_to_anchor=(1.03, 0.68), frameon=False,\n",
    "                                # fontsize=11, title_fontsize=11\n",
    "                                )\n",
    "    legend_tissue = ax.legend(tissue_labels_dict.values(), tissue_labels_dict.keys(), title='Tissue', \n",
    "                                bbox_to_anchor=(1.03, 1.23), frameon=False,\n",
    "                            #   fontsize=11, title_fontsize=11\n",
    "                                )\n",
    "    ax.add_artist(legend_mut)\n",
    "\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    plt.tight_layout()\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight', bbox_extra_artists=(legend_tissue,legend_mut))\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c42be7ed-3c8d-420a-a5ba-f2fee152a59a",
   "metadata": {},
   "source": [
    "## The data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2692d800-f0a4-4cb4-b775-6e374f6aa271",
   "metadata": {},
   "source": [
    "# Table of protein affecting mutations count (Fig6C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "af3278d3-869a-4133-b72e-29f67dfc3ef7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein pos\n",
    "labels_prot_dict = {\"Ratio mutated protein positions (Mut > 2)\"  : \"3+\",\n",
    "                   \"Ratio mutated protein positions (Mut == 2)\" : \"2\",\n",
    "                   \"Ratio mutated protein positions (Mut == 1)\" : \"1\", \n",
    "                   \"Ratio non mutated protein positions\"        : \"0\"}\n",
    "mut_count_colors_prot_dict = {\"Mutated protein positions (Mut > 2)\"        : \"#A63602\",\n",
    "                             \"Mutated protein positions (Mut == 2)\"       : \"#E6550E\",\n",
    "                             \"Mutated protein positions (Mut == 1)\"       : \"#FD8D3D\", \n",
    "                             \"Non mutated protein positions\"              : \"#D6D6D6\",\n",
    "\n",
    "                             \"Ratio mutated protein positions (Mut > 2)\"  : \"#A63602\",\n",
    "                             \"Ratio mutated protein positions (Mut == 2)\" : \"#E6550E\",\n",
    "                             \"Ratio mutated protein positions (Mut == 1)\" : \"#FD8D3D\", \n",
    "                             \"Ratio non mutated protein positions\"        : \"#D6D6D6\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "206b95ff-624e-48bc-9609-5201de3ec77c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_normal_mut_info(path_maf, indel_prot_pos=False):\n",
    "    \n",
    "    maf_df = pd.read_table(path_maf)\n",
    "    maf_df_f = maf_df.loc[(maf_df[\"canonical_SYMBOL\"].isin(gene_order))].reset_index(drop = True)\n",
    "    \n",
    "    \n",
    "    cols = [\"canonical_SYMBOL\", \"CHROM\", \"POS\", \"canonical_Protein_position\", \"canonical_Consequence_broader\", \"ALT_DEPTH\", \"TYPE\", \"INDEL_INFRAME\"]\n",
    "    rename_map = {\"canonical_SYMBOL\" : \"Gene\", \n",
    "                    \"CHROM\" : \"Chr\",\n",
    "                    \"POS\" : \"DNA_pos\",\n",
    "                    \"canonical_Protein_position\" : \"Protein_pos\",\n",
    "                    \"canonical_Consequence_broader\" : \"Consequence\",\n",
    "                    \"ALT_DEPTH\" : \"Alt_depth\"}\n",
    "\n",
    "    # SNV\n",
    "    # ---\n",
    "    maf_df_snv = maf_df_f[(maf_df_f[\"TYPE\"].isin([\"SNV\"]))].reset_index(drop = True)\n",
    "\n",
    "    # Parse    \n",
    "    maf_df_snv = maf_df_snv[[col for col in cols if col in maf_df_snv.columns]].rename(columns=rename_map)\n",
    "    maf_df_snv[\"INDEL_INFRAME\"] = np.nan\n",
    "\n",
    "    # Indels\n",
    "    # ------\n",
    "\n",
    "    maf_df_indel = maf_df_f.loc[(maf_df_f[\"TYPE\"].isin([\"INSERTION\", \"DELETION\"]))].reset_index(drop = True)\n",
    "    maf_df_indel[\"canonical_Consequence_broader\"] = \"indel\"\n",
    "    maf_df_indel = maf_df_indel[[col for col in cols if col in maf_df_indel.columns]].rename(columns=rename_map)\n",
    "\n",
    "    if indel_prot_pos == False:\n",
    "        maf_df_indel[\"Protein_pos\"] = np.nan\n",
    "\n",
    "    # Parse\n",
    "    # -----\n",
    "\n",
    "    maf_final = pd.concat((maf_df_snv, maf_df_indel)).reset_index(drop=True)\n",
    "    maf_final.Consequence = maf_final.Consequence.replace(\"splice_region_variant\", \"splicing\")\n",
    "    maf_final.Consequence = maf_final.Consequence.replace(\"essential_splice\", \"splicing\")\n",
    "    maf_final.Protein_pos = maf_final.Protein_pos.apply(lambda x: \n",
    "                                                        np.nan if (pd.isnull(x) or x.split(\"-\")[0] == \"?\" or x.split(\"-\")[0] == \"-\" or x.split(\"-\")[0] == \"\") else \n",
    "                                                        int(x) if len(x.split(\"-\")) == 1 else \n",
    "                                                        int(x.split(\"-\")[0]) if len(x.split(\"-\")) > 1 else np.nan).astype(float)\n",
    "    maf_final.DNA_pos = maf_final.DNA_pos.astype(int)\n",
    "\n",
    "    return maf_final\n",
    "\n",
    "\n",
    "\n",
    "# Get table\n",
    "# ---------\n",
    "\n",
    "def get_mut_info_count(mut_df, gene, dna_prot_df, non_masked_mut_only=True):\n",
    "    \n",
    "    gene_mut_df = mut_df[mut_df[\"Gene\"] == gene].reset_index(drop=True)\n",
    "    gene_dna_prot_df = dna_prot_df[dna_prot_df[\"Gene\"] == gene]\n",
    "    \n",
    "    \n",
    "    # Positions covered (where we can call mutations from)\n",
    "    # ----------------------------------------------------\n",
    "    \n",
    "    gene_dna_prot_df = gene_dna_prot_df[gene_dna_prot_df[\"COVERED\"] == 1].reset_index(drop=True)\n",
    "    len_res = len(gene_dna_prot_df.Protein_pos.dropna().unique()) \n",
    "    len_dna = len(gene_dna_prot_df.DNA_pos.dropna().unique())  \n",
    "    if non_masked_mut_only:\n",
    "        gene_mut_df = gene_mut_df[gene_mut_df.DNA_pos.isin(gene_dna_prot_df.DNA_pos)]\n",
    "    \n",
    "    \n",
    "    # Total counts\n",
    "    # ------------\n",
    "    \n",
    "    tot_mut = len(gene_mut_df)\n",
    "    if \"TYPE\" in gene_mut_df.columns:\n",
    "        tot_snv = len(gene_mut_df[gene_mut_df[\"TYPE\"] == \"SNV\"]) \n",
    "    else:\n",
    "        tot_snv = len(gene_mut_df[gene_mut_df[\"Consequence\"] != \"Indel\"]) \n",
    "        \n",
    "    tot_mut_missense = len(gene_mut_df[gene_mut_df[\"Consequence\"] == \"missense\"])\n",
    "    tot_mut_nonsense = len(gene_mut_df[gene_mut_df[\"Consequence\"] == \"nonsense\"])\n",
    "    tot_mut_splicing = len(gene_mut_df[gene_mut_df[\"Consequence\"] == \"splicing\"])\n",
    "    tot_mut_synonymous = len(gene_mut_df[gene_mut_df[\"Consequence\"] == \"synonymous\"])\n",
    "\n",
    "    tot_mut_indel = len(gene_mut_df[gene_mut_df[\"Consequence\"] == \"indel\"])\n",
    "    tot_mut_indel_frameshift = len(gene_mut_df[(gene_mut_df[\"Consequence\"] == \"indel\") & (gene_mut_df[\"INDEL_INFRAME\"] == False)])\n",
    "    tot_mut_indel_inframe = len(gene_mut_df[(gene_mut_df[\"Consequence\"] == \"indel\") & (gene_mut_df[\"INDEL_INFRAME\"] == True)])                      \n",
    "    \n",
    "    \n",
    "    # Mutations count\n",
    "    # ---------------\n",
    "    \n",
    "    # Number of DNA and protein positions mutated at least once\n",
    "    mut_dna = len(gene_mut_df.DNA_pos.dropna().unique())      # CDS + splicing regions\n",
    "    mut_res = len(gene_mut_df.Protein_pos.dropna().unique())  # Residues + stop codon\n",
    "    \n",
    "    gene_dna_mut_df = gene_mut_df.value_counts(\"DNA_pos\").reset_index().rename(columns={'count' : \"Mut_count\"})\n",
    "    gene_res_mut_df = gene_mut_df.value_counts(\"Protein_pos\").reset_index().rename(columns={'count' : \"Mut_count\"})\n",
    "\n",
    "    print(gene_res_mut_df)\n",
    "    print(gene_dna_mut_df)\n",
    "    \n",
    "    # Number of DNA and protein positions mutated once (counting each mutation only once per sample)\n",
    "    mut_dna_once = len(gene_dna_mut_df[gene_dna_mut_df[\"Mut_count\"] == 1])\n",
    "    mut_res_once = len(gene_res_mut_df[gene_res_mut_df[\"Mut_count\"] == 1])\n",
    "    \n",
    "    # Number of DNA and protein positions mutated twice (counting each mutation only once per sample)\n",
    "    mut_dna_twice = len(gene_dna_mut_df[gene_dna_mut_df[\"Mut_count\"] == 2])\n",
    "    mut_res_twice = len(gene_res_mut_df[gene_res_mut_df[\"Mut_count\"] == 2])\n",
    "    \n",
    "    # Number of DNA and protein positions mutated more than twice (counting each mutation only once per sample)\n",
    "    mut_dna_more_twice = len(gene_dna_mut_df[gene_dna_mut_df[\"Mut_count\"] > 2])\n",
    "    mut_res_more_twice = len(gene_res_mut_df[gene_res_mut_df[\"Mut_count\"] > 2])\n",
    "    \n",
    "    # Maximum number of times a DNA and protein position is mutated (counting each mutation only once per sample)\n",
    "    try:\n",
    "        mut_dna_max = int(gene_dna_mut_df[\"Mut_count\"].max())\n",
    "    except:\n",
    "        mut_dna_max = 0\n",
    "        \n",
    "    try:\n",
    "        mut_res_max = int(gene_res_mut_df[\"Mut_count\"].max())\n",
    "    except:\n",
    "        mut_res_max = 0\n",
    "    \n",
    "    # Ratios\n",
    "    # ------\n",
    "    \n",
    "    ratio_mut_dna = mut_dna / len_dna\n",
    "    ratio_mut_res = mut_res / len_res\n",
    "    ratio_mut_dna_once = mut_dna_once / len_dna\n",
    "    ratio_mut_res_once = mut_res_once / len_res\n",
    "    ratio_mut_dna_twice = mut_dna_twice / len_dna\n",
    "    ratio_mut_res_twice = mut_res_twice / len_res\n",
    "    ratio_mut_dna_more_twice = mut_dna_more_twice / len_dna\n",
    "    ratio_mut_res_more_twice = mut_res_more_twice / len_res\n",
    "    mut_dna_zero = len_dna - mut_dna\n",
    "    mut_res_zero = len_res - mut_res\n",
    "    ratio_mut_dna_zero = 1 - ratio_mut_dna\n",
    "    ratio_mut_res_zero = 1 - ratio_mut_res\n",
    "    \n",
    "    return [gene, \n",
    "            tot_mut, \n",
    "            tot_snv, \n",
    "            tot_mut_missense, \n",
    "            tot_mut_nonsense, \n",
    "            tot_mut_splicing, \n",
    "            tot_mut_synonymous, \n",
    "            tot_mut_indel, \n",
    "            tot_mut_indel_frameshift, \n",
    "            tot_mut_indel_inframe,\n",
    "            \n",
    "            len_dna, \n",
    "            mut_dna, \n",
    "            ratio_mut_dna, \n",
    "            \n",
    "            mut_dna_zero,\n",
    "            ratio_mut_dna_zero,\n",
    "            \n",
    "            mut_dna_once, \n",
    "            ratio_mut_dna_once, \n",
    "            \n",
    "            mut_dna_twice, \n",
    "            ratio_mut_dna_twice, \n",
    "            \n",
    "            mut_dna_more_twice, \n",
    "            ratio_mut_dna_more_twice,\n",
    "            \n",
    "            len_res, \n",
    "            mut_res, \n",
    "            ratio_mut_res, \n",
    "            \n",
    "            mut_res_zero,\n",
    "            ratio_mut_res_zero,\n",
    "            \n",
    "            mut_res_once, \n",
    "            ratio_mut_res_once, \n",
    "            \n",
    "            mut_res_twice, \n",
    "            ratio_mut_res_twice, \n",
    "            \n",
    "            mut_res_more_twice, \n",
    "            ratio_mut_res_more_twice,\n",
    "            \n",
    "            mut_dna_max, \n",
    "            mut_res_max]\n",
    "\n",
    "\n",
    "final_cols = [\"Gene\", \n",
    "              \"Total mutations\",\n",
    "              \"SNVs\",\n",
    "              \"Missense\",\n",
    "              \"Nonsense\",\n",
    "              \"Splicing\",\n",
    "              \"Synonymous\",\n",
    "              \"Indel\",\n",
    "              \"Indel frameshift\",\n",
    "              \"Indel inframe\",\n",
    "\n",
    "              \"DNA sites (CDS + splicing)\", \n",
    "              \"Mutated DNA sites\", \n",
    "              \"Ratio mutated DNA sites\", \n",
    "              \n",
    "              \"Non mutated DNA sites\",\n",
    "              \"Ratio non mutated DNA sites\",\n",
    "\n",
    "              \"Mutated DNA sites (Mut == 1)\", \n",
    "              \"Ratio mutated DNA sites (Mut == 1)\", \n",
    "\n",
    "              \"Mutated DNA sites (Mut == 2)\", \n",
    "              \"Ratio mutated DNA sites (Mut == 2)\", \n",
    "              \n",
    "              \"Mutated DNA sites (Mut > 2)\", \n",
    "              \"Ratio mutated DNA sites (Mut > 2)\", \n",
    "\n",
    "              \"Protein positions (Res + stop codon)\",\n",
    "              \"Mutated protein positions\", \n",
    "              \"Ratio mutated protein positions\",\n",
    "              \n",
    "              \"Non mutated protein positions\",\n",
    "              \"Ratio non mutated protein positions\",\n",
    "\n",
    "              \"Mutated protein positions (Mut == 1)\", \n",
    "              \"Ratio mutated protein positions (Mut == 1)\",\n",
    "\n",
    "              \"Mutated protein positions (Mut == 2)\", \n",
    "              \"Ratio mutated protein positions (Mut == 2)\",\n",
    "              \n",
    "              \"Mutated protein positions (Mut > 2)\", \n",
    "              \"Ratio mutated protein positions (Mut > 2)\",\n",
    "\n",
    "              \"Max number mutated DNA site\",\n",
    "              \"Max number mutated protein position\"]\n",
    "\n",
    "final_cols_sub = [\"Gene\", \n",
    "                  \"DNA sites (CDS + splicing)\", \n",
    "                  \"Mutated DNA sites\", \n",
    "                  \"Ratio mutated DNA sites\", \n",
    "                  \"Protein positions (Res + stop codon)\",\n",
    "                  \n",
    "                  \"Mutated protein positions\", \n",
    "                  \"Ratio mutated protein positions\",\n",
    "                  \n",
    "                  \"Non mutated protein positions\",\n",
    "                  \"Ratio non mutated protein positions\",\n",
    "                  \n",
    "                  \"Mutated protein positions (Mut == 1)\", \n",
    "                  \"Ratio mutated protein positions (Mut == 1)\",\n",
    "                  \n",
    "                  \"Mutated protein positions (Mut == 2)\", \n",
    "                  \"Ratio mutated protein positions (Mut == 2)\",\n",
    "                  \n",
    "                  \"Mutated protein positions (Mut > 2)\", \n",
    "                  \"Ratio mutated protein positions (Mut > 2)\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a1cd61d-dd53-430e-beec-ef88b3f37951",
   "metadata": {},
   "source": [
    "### Obtain the table with updated coordinates"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17543c09-f2cf-4d0e-b28c-246aa06c80a9",
   "metadata": {},
   "source": [
    "#### Table 2: do not include indels pos as affected protein position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce7aabcb-0e6f-4fd5-9215-24d3ad6bba65",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b4e7217d-7bc6-4d69-a0e7-7a2d6577bbc8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3217904/3467031355.py:3: DtypeWarning: Columns (49,92,112) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  maf_df = pd.read_table(path_maf)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Protein_pos  Mut_count\n",
      "0          1721.0         32\n",
      "1           633.0         27\n",
      "2           581.0         22\n",
      "3          2087.0         20\n",
      "4           944.0         20\n",
      "...           ...        ...\n",
      "1156       1454.0          1\n",
      "1157       1456.0          1\n",
      "1158       1428.0          1\n",
      "1159       1430.0          1\n",
      "1160       1466.0          1\n",
      "\n",
      "[1161 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0     26779439         53\n",
      "1     26780644         42\n",
      "2     26773569         42\n",
      "3     26779060         28\n",
      "4     26762296         28\n",
      "...        ...        ...\n",
      "2741  26780664          1\n",
      "2742  26780666          1\n",
      "2743  26780669          1\n",
      "2744  26780674          1\n",
      "2745  26780676          1\n",
      "\n",
      "[2746 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0            1.0         44\n",
      "1           10.0         33\n",
      "2           65.0         26\n",
      "3           48.0         24\n",
      "4           49.0         21\n",
      "..           ...        ...\n",
      "110        125.0          1\n",
      "111        122.0          1\n",
      "112        150.0          1\n",
      "113        162.0          1\n",
      "114        161.0          1\n",
      "\n",
      "[115 rows x 2 columns]\n",
      "      DNA_pos  Mut_count\n",
      "0    36684421         31\n",
      "1    36684129         31\n",
      "2    36684348         29\n",
      "3    36684102         28\n",
      "4    36684257         28\n",
      "..        ...        ...\n",
      "350  36684526          1\n",
      "351  36684517          1\n",
      "352  36684541          1\n",
      "353  36684540          1\n",
      "354  36685794          1\n",
      "\n",
      "[355 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0         1446.0         71\n",
      "1         1435.0         58\n",
      "2         1472.0         40\n",
      "3         1491.0         31\n",
      "4         1543.0         30\n",
      "..           ...        ...\n",
      "988         60.0          1\n",
      "989         63.0          1\n",
      "990         68.0          1\n",
      "991         71.0          1\n",
      "992         75.0          1\n",
      "\n",
      "[993 rows x 2 columns]\n",
      "      DNA_pos  Mut_count\n",
      "0     3738617         70\n",
      "1     3738650         33\n",
      "2     3736739         25\n",
      "3     3738649         25\n",
      "4     3731322         24\n",
      "...       ...        ...\n",
      "1708  3739662          1\n",
      "1709  3739647          1\n",
      "1710  3739646          1\n",
      "1711  3739645          1\n",
      "1712  3739696          1\n",
      "\n",
      "[1713 rows x 2 columns]\n",
      "      Protein_pos  Mut_count\n",
      "0          1399.0         74\n",
      "1          1414.0         55\n",
      "2          1514.0         30\n",
      "3          1466.0         29\n",
      "4          1446.0         28\n",
      "...           ...        ...\n",
      "1037        426.0          1\n",
      "1038        433.0          1\n",
      "1039        435.0          1\n",
      "1040        440.0          1\n",
      "1041       1360.0          1\n",
      "\n",
      "[1042 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0     41169571         50\n",
      "1     41169525         47\n",
      "2     41172664         40\n",
      "3     41160723         35\n",
      "4     41169526         28\n",
      "...        ...        ...\n",
      "1962  41178790          1\n",
      "1963  41178797          1\n",
      "1964  41178803          1\n",
      "1965  41178804          1\n",
      "1966  41178810          1\n",
      "\n",
      "[1967 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          380.0         15\n",
      "1          395.0          4\n",
      "2          722.0          4\n",
      "3          249.0          3\n",
      "4          362.0          3\n",
      "..           ...        ...\n",
      "115        694.0          1\n",
      "116        684.0          1\n",
      "117        707.0          1\n",
      "118        715.0          1\n",
      "119        723.0          1\n",
      "\n",
      "[120 rows x 2 columns]\n",
      "     DNA_pos  Mut_count\n",
      "0    1804392         15\n",
      "1    1806679          4\n",
      "2    1801959          3\n",
      "3    1804338          3\n",
      "4    1806163          3\n",
      "..       ...        ...\n",
      "182  1806623          1\n",
      "183  1806667          1\n",
      "184  1806659          1\n",
      "185  1806682          1\n",
      "186  1806815          1\n",
      "\n",
      "[187 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          123.0         19\n",
      "1          171.0         18\n",
      "2          124.0         15\n",
      "3          166.0         15\n",
      "4          172.0         14\n",
      "..           ...        ...\n",
      "226        389.0          1\n",
      "227        400.0          1\n",
      "228        398.0          1\n",
      "229        403.0          1\n",
      "230        404.0          1\n",
      "\n",
      "[231 rows x 2 columns]\n",
      "     DNA_pos  Mut_count\n",
      "0    1313061         65\n",
      "1    1313138         38\n",
      "2    1313215         29\n",
      "3    1313241         26\n",
      "4    1313240         23\n",
      "..       ...        ...\n",
      "581  1313837          1\n",
      "582  1313838          1\n",
      "583  1313839          1\n",
      "584  1313840          1\n",
      "585  1313844          1\n",
      "\n",
      "[586 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          607.0         46\n",
      "1         1281.0         26\n",
      "2          831.0         26\n",
      "3          729.0         22\n",
      "4          191.0         22\n",
      "..           ...        ...\n",
      "658       1393.0          1\n",
      "659       1396.0          1\n",
      "660       1400.0          1\n",
      "661       1409.0          1\n",
      "662       1417.0          1\n",
      "\n",
      "[663 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0     45063557         48\n",
      "1     45063799         35\n",
      "2     45110078         32\n",
      "3     45069990         27\n",
      "4     45063815         25\n",
      "...        ...        ...\n",
      "2097  45110211          1\n",
      "2098  45110214          1\n",
      "2099  45110217          1\n",
      "2100  45110219          1\n",
      "2101  45110220          1\n",
      "\n",
      "[2102 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          254.0          6\n",
      "1         4484.0          5\n",
      "2         4580.0          5\n",
      "3         2309.0          5\n",
      "4         4357.0          5\n",
      "..           ...        ...\n",
      "574        157.0          1\n",
      "575        161.0          1\n",
      "576        136.0          1\n",
      "577        138.0          1\n",
      "578        139.0          1\n",
      "\n",
      "[579 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0    152182508         11\n",
      "1    152311940          5\n",
      "2    152310055          5\n",
      "3    152148477          5\n",
      "4    152148188          5\n",
      "..         ...        ...\n",
      "967  152330680          1\n",
      "968  152330683          1\n",
      "969  152330695          1\n",
      "970  152330696          1\n",
      "971  152330706          1\n",
      "\n",
      "[972 rows x 2 columns]\n",
      "      Protein_pos  Mut_count\n",
      "0          5048.0         53\n",
      "1          5404.0         42\n",
      "2          1402.0         39\n",
      "3          2540.0         36\n",
      "4          5179.0         32\n",
      "...           ...        ...\n",
      "3216       5520.0          1\n",
      "3217       3138.0          1\n",
      "3218       3141.0          1\n",
      "3219       3142.0          1\n",
      "3220       3143.0          1\n",
      "\n",
      "[3221 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0     49038625         42\n",
      "1     49022717         40\n",
      "2     49047997         36\n",
      "3     49040152         35\n",
      "4     49053302         30\n",
      "...        ...        ...\n",
      "7128  49021852          1\n",
      "7129  49054936          1\n",
      "7130  49054937          1\n",
      "7131  49054941          1\n",
      "7132  49054954          1\n",
      "\n",
      "[7133 rows x 2 columns]\n",
      "      Protein_pos  Mut_count\n",
      "0           360.0         13\n",
      "1           526.0         10\n",
      "2           313.0         10\n",
      "3           339.0          9\n",
      "4           452.0          9\n",
      "...           ...        ...\n",
      "1043       1461.0          1\n",
      "1044        703.0          1\n",
      "1045        700.0          1\n",
      "1046       1465.0          1\n",
      "1047        712.0          1\n",
      "\n",
      "[1048 rows x 2 columns]\n",
      "        DNA_pos  Mut_count\n",
      "0     119986988         16\n",
      "1     119969540         11\n",
      "2     119969603          9\n",
      "3     119969553          8\n",
      "4     119966376          8\n",
      "...         ...        ...\n",
      "1809  119923962          1\n",
      "1810  119923958          1\n",
      "1811  119923957          1\n",
      "1812  119923953          1\n",
      "1813  119925302          1\n",
      "\n",
      "[1814 rows x 2 columns]\n",
      "    Protein_pos  Mut_count\n",
      "0        1047.0         29\n",
      "1        1043.0         16\n",
      "2        1049.0          5\n",
      "3        1021.0          4\n",
      "4        1048.0          4\n",
      "5        1044.0          4\n",
      "6        1069.0          4\n",
      "7        1040.0          3\n",
      "8        1050.0          2\n",
      "9        1011.0          2\n",
      "10       1058.0          2\n",
      "11        995.0          2\n",
      "12       1036.0          2\n",
      "13       1017.0          2\n",
      "14       1065.0          2\n",
      "15       1004.0          1\n",
      "16        992.0          1\n",
      "17        991.0          1\n",
      "18        989.0          1\n",
      "19        997.0          1\n",
      "20       1008.0          1\n",
      "21       1015.0          1\n",
      "22       1035.0          1\n",
      "23       1029.0          1\n",
      "24       1025.0          1\n",
      "25       1024.0          1\n",
      "26       1020.0          1\n",
      "27       1023.0          1\n",
      "28       1018.0          1\n",
      "29       1019.0          1\n",
      "30       1005.0          1\n",
      "31       1007.0          1\n",
      "32       1045.0          1\n",
      "33       1038.0          1\n",
      "34       1055.0          1\n",
      "35       1052.0          1\n",
      "36       1064.0          1\n",
      "37       1059.0          1\n",
      "      DNA_pos  Mut_count\n",
      "0   179234297         28\n",
      "1   179234284         10\n",
      "2   179234286          5\n",
      "3   179234302          5\n",
      "4   179234300          3\n",
      "5   179234275          3\n",
      "6   179234219          3\n",
      "7   179234140          2\n",
      "8   179234288          2\n",
      "9   179234307          2\n",
      "10  179234363          2\n",
      "11  179234351          2\n",
      "12  179234329          2\n",
      "13  179234201          1\n",
      "14  179234178          1\n",
      "15  179234189          1\n",
      "16  179234180          1\n",
      "17  179234172          1\n",
      "18  179234110          1\n",
      "19  179234129          1\n",
      "20  179234122          1\n",
      "21  179234148          1\n",
      "22  179234165          1\n",
      "23  179234169          1\n",
      "24  179234131          1\n",
      "25  179234238          1\n",
      "26  179234232          1\n",
      "27  179234229          1\n",
      "28  179234224          1\n",
      "29  179234218          1\n",
      "30  179234215          1\n",
      "31  179234212          1\n",
      "32  179234211          1\n",
      "33  179234190          1\n",
      "34  179234207          1\n",
      "35  179234206          1\n",
      "36  179234269          1\n",
      "37  179234265          1\n",
      "38  179234263          1\n",
      "39  179234240          1\n",
      "40  179234242          1\n",
      "41  179234296          1\n",
      "42  179234290          1\n",
      "43  179234289          1\n",
      "44  179234287          1\n",
      "45  179234285          1\n",
      "46  179234261          1\n",
      "47  179234301          1\n",
      "48  179234298          1\n",
      "49  179234322          1\n",
      "50  179234312          1\n",
      "51  179234305          1\n",
      "52  179234334          1\n",
      "53  179234348          1\n",
      "54  179234347          1\n",
      "55  179234362          1\n",
      "56  179234364          1\n",
      "     Protein_pos  Mut_count\n",
      "0          654.0         10\n",
      "1          775.0          5\n",
      "2          320.0          5\n",
      "3          621.0          4\n",
      "4          805.0          4\n",
      "..           ...        ...\n",
      "234        916.0          1\n",
      "235        917.0          1\n",
      "236        918.0          1\n",
      "237        920.0          1\n",
      "238        924.0          1\n",
      "\n",
      "[239 rows x 2 columns]\n",
      "      DNA_pos  Mut_count\n",
      "0    48360004         27\n",
      "1    48456349         10\n",
      "2    48376911          5\n",
      "3    48345184          5\n",
      "4    48479988          4\n",
      "..        ...        ...\n",
      "440  48367577          1\n",
      "441  48367574          1\n",
      "442  48367565          1\n",
      "443  48367547          1\n",
      "444  48368550          1\n",
      "\n",
      "[445 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          416.0         39\n",
      "1          761.0         35\n",
      "2          784.0         31\n",
      "3          776.0         30\n",
      "4          726.0         30\n",
      "..           ...        ...\n",
      "589        920.0          1\n",
      "590        923.0          1\n",
      "591          9.0          1\n",
      "592          8.0          1\n",
      "593          2.0          1\n",
      "\n",
      "[594 rows x 2 columns]\n",
      "       DNA_pos  Mut_count\n",
      "0     47181867         52\n",
      "1     47186056         45\n",
      "2     47171161         36\n",
      "3     47181540         32\n",
      "4     47185557         31\n",
      "...        ...        ...\n",
      "1579  47186337          1\n",
      "1580  47186346          1\n",
      "1581  47186347          1\n",
      "1582  47186353          1\n",
      "1583  47186355          1\n",
      "\n",
      "[1584 rows x 2 columns]\n",
      "     Protein_pos  Mut_count\n",
      "0          216.0         38\n",
      "1         1029.0         32\n",
      "2          593.0         22\n",
      "3          210.0         21\n",
      "4          573.0         20\n",
      "..           ...        ...\n",
      "568       1211.0          1\n",
      "569       1212.0          1\n",
      "570        149.0          1\n",
      "571        151.0          1\n",
      "572        126.0          1\n",
      "\n",
      "[573 rows x 2 columns]\n",
      "        DNA_pos  Mut_count\n",
      "0     124045347         34\n",
      "1     124086578         31\n",
      "2     124042561         29\n",
      "3     124045330         22\n",
      "4     124063161         22\n",
      "...         ...        ...\n",
      "1423  124061334          1\n",
      "1424  124061329          1\n",
      "1425  124061328          1\n",
      "1426  124061327          1\n",
      "1427  124061800          1\n",
      "\n",
      "[1428 rows x 2 columns]\n",
      "   Protein_pos  Mut_count\n",
      "0         17.0          1\n",
      "1         19.0          1\n",
      "2         29.0          1\n",
      "3         30.0          1\n",
      "4         32.0          1\n",
      "5         35.0          1\n",
      "6         38.0          1\n",
      "7         49.0          1\n",
      "   DNA_pos  Mut_count\n",
      "0  1294845          1\n",
      "1  1294876          1\n",
      "2  1294885          1\n",
      "3  1294896          1\n",
      "4  1294901          1\n",
      "5  1294904          1\n",
      "6  1294931          1\n",
      "7  1294935          1\n",
      "8  1294940          1\n",
      "     Protein_pos  Mut_count\n",
      "0          248.0         25\n",
      "1          158.0         22\n",
      "2          273.0         19\n",
      "3          125.0         18\n",
      "4          179.0         15\n",
      "..           ...        ...\n",
      "225        383.0          1\n",
      "226        386.0          1\n",
      "227        384.0          1\n",
      "228        391.0          1\n",
      "229        388.0          1\n",
      "\n",
      "[230 rows x 2 columns]\n",
      "     DNA_pos  Mut_count\n",
      "0    7673802         19\n",
      "1    7675139         18\n",
      "2    7674220         14\n",
      "3    7674221         11\n",
      "4    7675994         11\n",
      "..       ...        ...\n",
      "406  7676579          1\n",
      "407  7676116          1\n",
      "408  7676620          1\n",
      "409  7676621          1\n",
      "410  7669697          1\n",
      "\n",
      "[411 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3217904/12754042.py:4: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  normal_count_df = normal_df.groupby(\"Gene\").apply(lambda x: get_mut_info_count(x, x.name, dna_prot_df, non_masked_mut_only=True)).tolist()\n"
     ]
    }
   ],
   "source": [
    "# Do not include indels pos as affected protein position\n",
    "normal_df = get_normal_mut_info(somatic_maf_file, indel_prot_pos=False)\n",
    "\n",
    "normal_count_df = normal_df.groupby(\"Gene\").apply(lambda x: get_mut_info_count(x, x.name, dna_prot_df, non_masked_mut_only=True)).tolist()\n",
    "normal_count_df = pd.DataFrame(normal_count_df, \n",
    "                               columns=final_cols)\n",
    "\n",
    "normal_count_df.to_csv(\"data/normal_mutation_count.corrected.filtered.no_indels_for_prot.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cabca001-0d38-45e4-9d08-f30a9e8b8d3c",
   "metadata": {},
   "source": [
    "## Plot of proportion and number of mutations by mutations count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c49a2a0-e79c-4651-9971-549a680100a1",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Check that calculations are correct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fcfa26c0-658f-429e-bb03-ffe76ebdad86",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_mut_pos = [#\"Non mutated protein positions\",\n",
    "                \"Mutated protein positions (Mut == 1)\", \n",
    "                \"Mutated protein positions (Mut == 2)\",\n",
    "                \"Mutated protein positions (Mut > 2)\"]\n",
    "cols_ratio_mut_pos = [#\"Ratio non mutated protein positions\",\n",
    "                      \"Ratio mutated protein positions (Mut == 1)\",\n",
    "                      \"Ratio mutated protein positions (Mut == 2)\",\n",
    "                      \"Ratio mutated protein positions (Mut > 2)\"]\n",
    "\n",
    "cols_mut_dna = [col.replace(\"protein positions\", \"DNA sites\") for col in cols_mut_pos]\n",
    "cols_ratio_mut_dna = [col.replace(\"protein positions\", \"DNA sites\") for col in cols_ratio_mut_pos]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a0126a37-ff7b-4413-90b6-b9c5e2a77549",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein positions (Res + stop codon)</th>\n",
       "      <th>Ratio mutated protein positions</th>\n",
       "      <th>Ratio non mutated protein positions</th>\n",
       "      <th>Ratio mut prot pos sum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>1907</td>\n",
       "      <td>0.608810</td>\n",
       "      <td>0.391190</td>\n",
       "      <td>0.608810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>165</td>\n",
       "      <td>0.696970</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>0.696970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>2435</td>\n",
       "      <td>0.407803</td>\n",
       "      <td>0.592197</td>\n",
       "      <td>0.407803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EP300</td>\n",
       "      <td>2415</td>\n",
       "      <td>0.431470</td>\n",
       "      <td>0.568530</td>\n",
       "      <td>0.431470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>313</td>\n",
       "      <td>0.383387</td>\n",
       "      <td>0.616613</td>\n",
       "      <td>0.383387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>369</td>\n",
       "      <td>0.626016</td>\n",
       "      <td>0.373984</td>\n",
       "      <td>0.626016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>1454</td>\n",
       "      <td>0.455983</td>\n",
       "      <td>0.544017</td>\n",
       "      <td>0.455983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KMT2C</td>\n",
       "      <td>1413</td>\n",
       "      <td>0.409766</td>\n",
       "      <td>0.590234</td>\n",
       "      <td>0.409766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>5520</td>\n",
       "      <td>0.583514</td>\n",
       "      <td>0.416486</td>\n",
       "      <td>0.583514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>2171</td>\n",
       "      <td>0.482727</td>\n",
       "      <td>0.517273</td>\n",
       "      <td>0.482727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>91</td>\n",
       "      <td>0.417582</td>\n",
       "      <td>0.582418</td>\n",
       "      <td>0.417582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>RB1</td>\n",
       "      <td>929</td>\n",
       "      <td>0.257266</td>\n",
       "      <td>0.742734</td>\n",
       "      <td>0.257266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>931</td>\n",
       "      <td>0.638024</td>\n",
       "      <td>0.361976</td>\n",
       "      <td>0.638024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>1259</td>\n",
       "      <td>0.455123</td>\n",
       "      <td>0.544877</td>\n",
       "      <td>0.455123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TERT</td>\n",
       "      <td>40</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.200000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53</td>\n",
       "      <td>394</td>\n",
       "      <td>0.583756</td>\n",
       "      <td>0.416244</td>\n",
       "      <td>0.583756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  Protein positions (Res + stop codon)  \\\n",
       "0   ARID1A                                  1907   \n",
       "1   CDKN1A                                   165   \n",
       "2   CREBBP                                  2435   \n",
       "3    EP300                                  2415   \n",
       "4    FGFR3                                   313   \n",
       "5    FOXQ1                                   369   \n",
       "6    KDM6A                                  1454   \n",
       "7    KMT2C                                  1413   \n",
       "8    KMT2D                                  5520   \n",
       "9   NOTCH2                                  2171   \n",
       "10  PIK3CA                                    91   \n",
       "11     RB1                                   929   \n",
       "12   RBM10                                   931   \n",
       "13   STAG2                                  1259   \n",
       "14    TERT                                    40   \n",
       "15    TP53                                   394   \n",
       "\n",
       "    Ratio mutated protein positions  Ratio non mutated protein positions  \\\n",
       "0                          0.608810                             0.391190   \n",
       "1                          0.696970                             0.303030   \n",
       "2                          0.407803                             0.592197   \n",
       "3                          0.431470                             0.568530   \n",
       "4                          0.383387                             0.616613   \n",
       "5                          0.626016                             0.373984   \n",
       "6                          0.455983                             0.544017   \n",
       "7                          0.409766                             0.590234   \n",
       "8                          0.583514                             0.416486   \n",
       "9                          0.482727                             0.517273   \n",
       "10                         0.417582                             0.582418   \n",
       "11                         0.257266                             0.742734   \n",
       "12                         0.638024                             0.361976   \n",
       "13                         0.455123                             0.544877   \n",
       "14                         0.200000                             0.800000   \n",
       "15                         0.583756                             0.416244   \n",
       "\n",
       "    Ratio mut prot pos sum  \n",
       "0                 0.608810  \n",
       "1                 0.696970  \n",
       "2                 0.407803  \n",
       "3                 0.431470  \n",
       "4                 0.383387  \n",
       "5                 0.626016  \n",
       "6                 0.455983  \n",
       "7                 0.409766  \n",
       "8                 0.583514  \n",
       "9                 0.482727  \n",
       "10                0.417582  \n",
       "11                0.257266  \n",
       "12                0.638024  \n",
       "13                0.455123  \n",
       "14                0.200000  \n",
       "15                0.583756  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mut = normal_count_df[[\"Gene\", \"Protein positions (Res + stop codon)\", \"Mutated protein positions\", \"Non mutated protein positions\"]].copy()\n",
    "df_mut[\"Mut prot pos sum\"] = normal_count_df.apply(lambda x: sum(x[cols_mut_pos]), axis=1)\n",
    "# display(df_mut)\n",
    "\n",
    "df_ratio = normal_count_df[[\"Gene\", \"Protein positions (Res + stop codon)\", \"Ratio mutated protein positions\", \"Ratio non mutated protein positions\",]].copy()\n",
    "df_ratio[\"Ratio mut prot pos sum\"] = normal_count_df.apply(lambda x: sum(x[cols_ratio_mut_pos]), axis=1)\n",
    "df_ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "342d0465-2372-4f1d-922f-40e91bdf58d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>DNA sites (CDS + splicing)</th>\n",
       "      <th>Ratio mutated DNA sites</th>\n",
       "      <th>Ratio non mutated DNA sites</th>\n",
       "      <th>Ratio mut dna sum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>6188</td>\n",
       "      <td>0.443762</td>\n",
       "      <td>0.556238</td>\n",
       "      <td>0.443762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>540</td>\n",
       "      <td>0.657407</td>\n",
       "      <td>0.342593</td>\n",
       "      <td>0.657407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>8053</td>\n",
       "      <td>0.212716</td>\n",
       "      <td>0.787284</td>\n",
       "      <td>0.212716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EP300</td>\n",
       "      <td>7995</td>\n",
       "      <td>0.246029</td>\n",
       "      <td>0.753971</td>\n",
       "      <td>0.246029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>1112</td>\n",
       "      <td>0.168165</td>\n",
       "      <td>0.831835</td>\n",
       "      <td>0.168165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FOXQ1</td>\n",
       "      <td>1100</td>\n",
       "      <td>0.532727</td>\n",
       "      <td>0.467273</td>\n",
       "      <td>0.532727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KDM6A</td>\n",
       "      <td>5086</td>\n",
       "      <td>0.413291</td>\n",
       "      <td>0.586709</td>\n",
       "      <td>0.413291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KMT2C</td>\n",
       "      <td>4507</td>\n",
       "      <td>0.215665</td>\n",
       "      <td>0.784335</td>\n",
       "      <td>0.215665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>17886</td>\n",
       "      <td>0.398804</td>\n",
       "      <td>0.601196</td>\n",
       "      <td>0.398804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>7226</td>\n",
       "      <td>0.251038</td>\n",
       "      <td>0.748962</td>\n",
       "      <td>0.251038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>288</td>\n",
       "      <td>0.197917</td>\n",
       "      <td>0.802083</td>\n",
       "      <td>0.197917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>RB1</td>\n",
       "      <td>3415</td>\n",
       "      <td>0.130307</td>\n",
       "      <td>0.869693</td>\n",
       "      <td>0.130307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>RBM10</td>\n",
       "      <td>3363</td>\n",
       "      <td>0.471008</td>\n",
       "      <td>0.528992</td>\n",
       "      <td>0.471008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>STAG2</td>\n",
       "      <td>4541</td>\n",
       "      <td>0.314468</td>\n",
       "      <td>0.685532</td>\n",
       "      <td>0.314468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TERT</td>\n",
       "      <td>118</td>\n",
       "      <td>0.076271</td>\n",
       "      <td>0.923729</td>\n",
       "      <td>0.076271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53</td>\n",
       "      <td>1427</td>\n",
       "      <td>0.288017</td>\n",
       "      <td>0.711983</td>\n",
       "      <td>0.288017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  DNA sites (CDS + splicing)  Ratio mutated DNA sites  \\\n",
       "0   ARID1A                        6188                 0.443762   \n",
       "1   CDKN1A                         540                 0.657407   \n",
       "2   CREBBP                        8053                 0.212716   \n",
       "3    EP300                        7995                 0.246029   \n",
       "4    FGFR3                        1112                 0.168165   \n",
       "5    FOXQ1                        1100                 0.532727   \n",
       "6    KDM6A                        5086                 0.413291   \n",
       "7    KMT2C                        4507                 0.215665   \n",
       "8    KMT2D                       17886                 0.398804   \n",
       "9   NOTCH2                        7226                 0.251038   \n",
       "10  PIK3CA                         288                 0.197917   \n",
       "11     RB1                        3415                 0.130307   \n",
       "12   RBM10                        3363                 0.471008   \n",
       "13   STAG2                        4541                 0.314468   \n",
       "14    TERT                         118                 0.076271   \n",
       "15    TP53                        1427                 0.288017   \n",
       "\n",
       "    Ratio non mutated DNA sites  Ratio mut dna sum  \n",
       "0                      0.556238           0.443762  \n",
       "1                      0.342593           0.657407  \n",
       "2                      0.787284           0.212716  \n",
       "3                      0.753971           0.246029  \n",
       "4                      0.831835           0.168165  \n",
       "5                      0.467273           0.532727  \n",
       "6                      0.586709           0.413291  \n",
       "7                      0.784335           0.215665  \n",
       "8                      0.601196           0.398804  \n",
       "9                      0.748962           0.251038  \n",
       "10                     0.802083           0.197917  \n",
       "11                     0.869693           0.130307  \n",
       "12                     0.528992           0.471008  \n",
       "13                     0.685532           0.314468  \n",
       "14                     0.923729           0.076271  \n",
       "15                     0.711983           0.288017  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mut = normal_count_df[[\"Gene\", \"DNA sites (CDS + splicing)\", \"Mutated DNA sites\",  \"Non mutated DNA sites\"]].copy()\n",
    "df_mut[\"Mut dna sum\"] = normal_count_df.apply(lambda x: sum(x[cols_mut_dna]), axis=1)\n",
    "# display(df_mut)\n",
    "\n",
    "df_ratio = normal_count_df[[\"Gene\", \"DNA sites (CDS + splicing)\", \"Ratio mutated DNA sites\", \"Ratio non mutated DNA sites\"]].copy()\n",
    "df_ratio[\"Ratio mut dna sum\"] = normal_count_df.apply(lambda x: sum(x[cols_ratio_mut_dna]), axis=1)\n",
    "df_ratio"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54cbb49f-470d-4611-95e9-35ec2fbbdd5e",
   "metadata": {},
   "source": [
    "### Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "06ccc4c7-ca14-4989-ba3f-5bba20235ecc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutated DNA sites</th>\n",
       "      <th>Non mutated DNA sites</th>\n",
       "      <th>Mutated DNA sites (Mut == 1)</th>\n",
       "      <th>Mutated DNA sites (Mut == 2)</th>\n",
       "      <th>Mutated DNA sites (Mut &gt; 2)</th>\n",
       "      <th>Ratio mutated DNA sites</th>\n",
       "      <th>Ratio non mutated DNA sites</th>\n",
       "      <th>Ratio mutated DNA sites (Mut == 1)</th>\n",
       "      <th>Ratio mutated DNA sites (Mut == 2)</th>\n",
       "      <th>Ratio mutated DNA sites (Mut &gt; 2)</th>\n",
       "      <th>...</th>\n",
       "      <th>Non mutated protein positions</th>\n",
       "      <th>Mutated protein positions (Mut == 1)</th>\n",
       "      <th>Mutated protein positions (Mut == 2)</th>\n",
       "      <th>Mutated protein positions (Mut &gt; 2)</th>\n",
       "      <th>Ratio mutated protein positions</th>\n",
       "      <th>Ratio non mutated protein positions</th>\n",
       "      <th>Ratio mutated protein positions (Mut == 1)</th>\n",
       "      <th>Ratio mutated protein positions (Mut == 2)</th>\n",
       "      <th>Ratio mutated protein positions (Mut &gt; 2)</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>2746</td>\n",
       "      <td>3442</td>\n",
       "      <td>1492</td>\n",
       "      <td>582</td>\n",
       "      <td>672</td>\n",
       "      <td>0.443762</td>\n",
       "      <td>0.556238</td>\n",
       "      <td>0.241112</td>\n",
       "      <td>0.094053</td>\n",
       "      <td>0.108597</td>\n",
       "      <td>...</td>\n",
       "      <td>746</td>\n",
       "      <td>528</td>\n",
       "      <td>271</td>\n",
       "      <td>362</td>\n",
       "      <td>0.608810</td>\n",
       "      <td>0.391190</td>\n",
       "      <td>0.276875</td>\n",
       "      <td>0.142108</td>\n",
       "      <td>0.189827</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN1A</th>\n",
       "      <td>355</td>\n",
       "      <td>185</td>\n",
       "      <td>101</td>\n",
       "      <td>75</td>\n",
       "      <td>179</td>\n",
       "      <td>0.657407</td>\n",
       "      <td>0.342593</td>\n",
       "      <td>0.187037</td>\n",
       "      <td>0.138889</td>\n",
       "      <td>0.331481</td>\n",
       "      <td>...</td>\n",
       "      <td>50</td>\n",
       "      <td>44</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>0.696970</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>0.266667</td>\n",
       "      <td>0.157576</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREBBP</th>\n",
       "      <td>1713</td>\n",
       "      <td>6340</td>\n",
       "      <td>1237</td>\n",
       "      <td>281</td>\n",
       "      <td>195</td>\n",
       "      <td>0.212716</td>\n",
       "      <td>0.787284</td>\n",
       "      <td>0.153607</td>\n",
       "      <td>0.034894</td>\n",
       "      <td>0.024215</td>\n",
       "      <td>...</td>\n",
       "      <td>1442</td>\n",
       "      <td>636</td>\n",
       "      <td>206</td>\n",
       "      <td>151</td>\n",
       "      <td>0.407803</td>\n",
       "      <td>0.592197</td>\n",
       "      <td>0.261191</td>\n",
       "      <td>0.084600</td>\n",
       "      <td>0.062012</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EP300</th>\n",
       "      <td>1967</td>\n",
       "      <td>6028</td>\n",
       "      <td>1333</td>\n",
       "      <td>332</td>\n",
       "      <td>302</td>\n",
       "      <td>0.246029</td>\n",
       "      <td>0.753971</td>\n",
       "      <td>0.166729</td>\n",
       "      <td>0.041526</td>\n",
       "      <td>0.037774</td>\n",
       "      <td>...</td>\n",
       "      <td>1373</td>\n",
       "      <td>583</td>\n",
       "      <td>232</td>\n",
       "      <td>227</td>\n",
       "      <td>0.431470</td>\n",
       "      <td>0.568530</td>\n",
       "      <td>0.241408</td>\n",
       "      <td>0.096066</td>\n",
       "      <td>0.093996</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR3</th>\n",
       "      <td>187</td>\n",
       "      <td>925</td>\n",
       "      <td>142</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>0.168165</td>\n",
       "      <td>0.831835</td>\n",
       "      <td>0.127698</td>\n",
       "      <td>0.028777</td>\n",
       "      <td>0.011691</td>\n",
       "      <td>...</td>\n",
       "      <td>193</td>\n",
       "      <td>70</td>\n",
       "      <td>35</td>\n",
       "      <td>15</td>\n",
       "      <td>0.383387</td>\n",
       "      <td>0.616613</td>\n",
       "      <td>0.223642</td>\n",
       "      <td>0.111821</td>\n",
       "      <td>0.047923</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXQ1</th>\n",
       "      <td>586</td>\n",
       "      <td>514</td>\n",
       "      <td>264</td>\n",
       "      <td>120</td>\n",
       "      <td>202</td>\n",
       "      <td>0.532727</td>\n",
       "      <td>0.467273</td>\n",
       "      <td>0.240000</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.183636</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>97</td>\n",
       "      <td>55</td>\n",
       "      <td>79</td>\n",
       "      <td>0.626016</td>\n",
       "      <td>0.373984</td>\n",
       "      <td>0.262873</td>\n",
       "      <td>0.149051</td>\n",
       "      <td>0.214092</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM6A</th>\n",
       "      <td>2102</td>\n",
       "      <td>2984</td>\n",
       "      <td>1053</td>\n",
       "      <td>403</td>\n",
       "      <td>646</td>\n",
       "      <td>0.413291</td>\n",
       "      <td>0.586709</td>\n",
       "      <td>0.207039</td>\n",
       "      <td>0.079237</td>\n",
       "      <td>0.127015</td>\n",
       "      <td>...</td>\n",
       "      <td>791</td>\n",
       "      <td>304</td>\n",
       "      <td>120</td>\n",
       "      <td>239</td>\n",
       "      <td>0.455983</td>\n",
       "      <td>0.544017</td>\n",
       "      <td>0.209078</td>\n",
       "      <td>0.082531</td>\n",
       "      <td>0.164374</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2C</th>\n",
       "      <td>972</td>\n",
       "      <td>3535</td>\n",
       "      <td>754</td>\n",
       "      <td>151</td>\n",
       "      <td>67</td>\n",
       "      <td>0.215665</td>\n",
       "      <td>0.784335</td>\n",
       "      <td>0.167295</td>\n",
       "      <td>0.033503</td>\n",
       "      <td>0.014866</td>\n",
       "      <td>...</td>\n",
       "      <td>834</td>\n",
       "      <td>406</td>\n",
       "      <td>116</td>\n",
       "      <td>57</td>\n",
       "      <td>0.409766</td>\n",
       "      <td>0.590234</td>\n",
       "      <td>0.287332</td>\n",
       "      <td>0.082095</td>\n",
       "      <td>0.040340</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>7133</td>\n",
       "      <td>10753</td>\n",
       "      <td>4046</td>\n",
       "      <td>1373</td>\n",
       "      <td>1714</td>\n",
       "      <td>0.398804</td>\n",
       "      <td>0.601196</td>\n",
       "      <td>0.226210</td>\n",
       "      <td>0.076764</td>\n",
       "      <td>0.095829</td>\n",
       "      <td>...</td>\n",
       "      <td>2299</td>\n",
       "      <td>1399</td>\n",
       "      <td>721</td>\n",
       "      <td>1101</td>\n",
       "      <td>0.583514</td>\n",
       "      <td>0.416486</td>\n",
       "      <td>0.253442</td>\n",
       "      <td>0.130616</td>\n",
       "      <td>0.199457</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH2</th>\n",
       "      <td>1814</td>\n",
       "      <td>5412</td>\n",
       "      <td>1332</td>\n",
       "      <td>302</td>\n",
       "      <td>180</td>\n",
       "      <td>0.251038</td>\n",
       "      <td>0.748962</td>\n",
       "      <td>0.184334</td>\n",
       "      <td>0.041794</td>\n",
       "      <td>0.024910</td>\n",
       "      <td>...</td>\n",
       "      <td>1123</td>\n",
       "      <td>607</td>\n",
       "      <td>236</td>\n",
       "      <td>205</td>\n",
       "      <td>0.482727</td>\n",
       "      <td>0.517273</td>\n",
       "      <td>0.279595</td>\n",
       "      <td>0.108706</td>\n",
       "      <td>0.094427</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>57</td>\n",
       "      <td>231</td>\n",
       "      <td>44</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>0.197917</td>\n",
       "      <td>0.802083</td>\n",
       "      <td>0.152778</td>\n",
       "      <td>0.020833</td>\n",
       "      <td>0.024306</td>\n",
       "      <td>...</td>\n",
       "      <td>53</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>0.417582</td>\n",
       "      <td>0.582418</td>\n",
       "      <td>0.252747</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.087912</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1</th>\n",
       "      <td>445</td>\n",
       "      <td>2970</td>\n",
       "      <td>373</td>\n",
       "      <td>58</td>\n",
       "      <td>14</td>\n",
       "      <td>0.130307</td>\n",
       "      <td>0.869693</td>\n",
       "      <td>0.109224</td>\n",
       "      <td>0.016984</td>\n",
       "      <td>0.004100</td>\n",
       "      <td>...</td>\n",
       "      <td>690</td>\n",
       "      <td>185</td>\n",
       "      <td>41</td>\n",
       "      <td>13</td>\n",
       "      <td>0.257266</td>\n",
       "      <td>0.742734</td>\n",
       "      <td>0.199139</td>\n",
       "      <td>0.044133</td>\n",
       "      <td>0.013994</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBM10</th>\n",
       "      <td>1584</td>\n",
       "      <td>1779</td>\n",
       "      <td>767</td>\n",
       "      <td>299</td>\n",
       "      <td>518</td>\n",
       "      <td>0.471008</td>\n",
       "      <td>0.528992</td>\n",
       "      <td>0.228070</td>\n",
       "      <td>0.088909</td>\n",
       "      <td>0.154029</td>\n",
       "      <td>...</td>\n",
       "      <td>337</td>\n",
       "      <td>222</td>\n",
       "      <td>125</td>\n",
       "      <td>247</td>\n",
       "      <td>0.638024</td>\n",
       "      <td>0.361976</td>\n",
       "      <td>0.238453</td>\n",
       "      <td>0.134264</td>\n",
       "      <td>0.265306</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAG2</th>\n",
       "      <td>1428</td>\n",
       "      <td>3113</td>\n",
       "      <td>898</td>\n",
       "      <td>281</td>\n",
       "      <td>249</td>\n",
       "      <td>0.314468</td>\n",
       "      <td>0.685532</td>\n",
       "      <td>0.197754</td>\n",
       "      <td>0.061881</td>\n",
       "      <td>0.054834</td>\n",
       "      <td>...</td>\n",
       "      <td>686</td>\n",
       "      <td>316</td>\n",
       "      <td>113</td>\n",
       "      <td>144</td>\n",
       "      <td>0.455123</td>\n",
       "      <td>0.544877</td>\n",
       "      <td>0.250993</td>\n",
       "      <td>0.089754</td>\n",
       "      <td>0.114376</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TERT</th>\n",
       "      <td>9</td>\n",
       "      <td>109</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.076271</td>\n",
       "      <td>0.923729</td>\n",
       "      <td>0.076271</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>32</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>411</td>\n",
       "      <td>1016</td>\n",
       "      <td>256</td>\n",
       "      <td>76</td>\n",
       "      <td>79</td>\n",
       "      <td>0.288017</td>\n",
       "      <td>0.711983</td>\n",
       "      <td>0.179397</td>\n",
       "      <td>0.053259</td>\n",
       "      <td>0.055361</td>\n",
       "      <td>...</td>\n",
       "      <td>164</td>\n",
       "      <td>105</td>\n",
       "      <td>46</td>\n",
       "      <td>79</td>\n",
       "      <td>0.583756</td>\n",
       "      <td>0.416244</td>\n",
       "      <td>0.266497</td>\n",
       "      <td>0.116751</td>\n",
       "      <td>0.200508</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Mutated DNA sites  Non mutated DNA sites  \\\n",
       "Gene                                               \n",
       "ARID1A               2746                   3442   \n",
       "CDKN1A                355                    185   \n",
       "CREBBP               1713                   6340   \n",
       "EP300                1967                   6028   \n",
       "FGFR3                 187                    925   \n",
       "FOXQ1                 586                    514   \n",
       "KDM6A                2102                   2984   \n",
       "KMT2C                 972                   3535   \n",
       "KMT2D                7133                  10753   \n",
       "NOTCH2               1814                   5412   \n",
       "PIK3CA                 57                    231   \n",
       "RB1                   445                   2970   \n",
       "RBM10                1584                   1779   \n",
       "STAG2                1428                   3113   \n",
       "TERT                    9                    109   \n",
       "TP53                  411                   1016   \n",
       "\n",
       "        Mutated DNA sites (Mut == 1)  Mutated DNA sites (Mut == 2)  \\\n",
       "Gene                                                                 \n",
       "ARID1A                          1492                           582   \n",
       "CDKN1A                           101                            75   \n",
       "CREBBP                          1237                           281   \n",
       "EP300                           1333                           332   \n",
       "FGFR3                            142                            32   \n",
       "FOXQ1                            264                           120   \n",
       "KDM6A                           1053                           403   \n",
       "KMT2C                            754                           151   \n",
       "KMT2D                           4046                          1373   \n",
       "NOTCH2                          1332                           302   \n",
       "PIK3CA                            44                             6   \n",
       "RB1                              373                            58   \n",
       "RBM10                            767                           299   \n",
       "STAG2                            898                           281   \n",
       "TERT                               9                             0   \n",
       "TP53                             256                            76   \n",
       "\n",
       "        Mutated DNA sites (Mut > 2)  Ratio mutated DNA sites  \\\n",
       "Gene                                                           \n",
       "ARID1A                          672                 0.443762   \n",
       "CDKN1A                          179                 0.657407   \n",
       "CREBBP                          195                 0.212716   \n",
       "EP300                           302                 0.246029   \n",
       "FGFR3                            13                 0.168165   \n",
       "FOXQ1                           202                 0.532727   \n",
       "KDM6A                           646                 0.413291   \n",
       "KMT2C                            67                 0.215665   \n",
       "KMT2D                          1714                 0.398804   \n",
       "NOTCH2                          180                 0.251038   \n",
       "PIK3CA                            7                 0.197917   \n",
       "RB1                              14                 0.130307   \n",
       "RBM10                           518                 0.471008   \n",
       "STAG2                           249                 0.314468   \n",
       "TERT                              0                 0.076271   \n",
       "TP53                             79                 0.288017   \n",
       "\n",
       "        Ratio non mutated DNA sites  Ratio mutated DNA sites (Mut == 1)  \\\n",
       "Gene                                                                      \n",
       "ARID1A                     0.556238                            0.241112   \n",
       "CDKN1A                     0.342593                            0.187037   \n",
       "CREBBP                     0.787284                            0.153607   \n",
       "EP300                      0.753971                            0.166729   \n",
       "FGFR3                      0.831835                            0.127698   \n",
       "FOXQ1                      0.467273                            0.240000   \n",
       "KDM6A                      0.586709                            0.207039   \n",
       "KMT2C                      0.784335                            0.167295   \n",
       "KMT2D                      0.601196                            0.226210   \n",
       "NOTCH2                     0.748962                            0.184334   \n",
       "PIK3CA                     0.802083                            0.152778   \n",
       "RB1                        0.869693                            0.109224   \n",
       "RBM10                      0.528992                            0.228070   \n",
       "STAG2                      0.685532                            0.197754   \n",
       "TERT                       0.923729                            0.076271   \n",
       "TP53                       0.711983                            0.179397   \n",
       "\n",
       "        Ratio mutated DNA sites (Mut == 2)  Ratio mutated DNA sites (Mut > 2)  \\\n",
       "Gene                                                                            \n",
       "ARID1A                            0.094053                           0.108597   \n",
       "CDKN1A                            0.138889                           0.331481   \n",
       "CREBBP                            0.034894                           0.024215   \n",
       "EP300                             0.041526                           0.037774   \n",
       "FGFR3                             0.028777                           0.011691   \n",
       "FOXQ1                             0.109091                           0.183636   \n",
       "KDM6A                             0.079237                           0.127015   \n",
       "KMT2C                             0.033503                           0.014866   \n",
       "KMT2D                             0.076764                           0.095829   \n",
       "NOTCH2                            0.041794                           0.024910   \n",
       "PIK3CA                            0.020833                           0.024306   \n",
       "RB1                               0.016984                           0.004100   \n",
       "RBM10                             0.088909                           0.154029   \n",
       "STAG2                             0.061881                           0.054834   \n",
       "TERT                              0.000000                           0.000000   \n",
       "TP53                              0.053259                           0.055361   \n",
       "\n",
       "        ...  Non mutated protein positions  \\\n",
       "Gene    ...                                  \n",
       "ARID1A  ...                            746   \n",
       "CDKN1A  ...                             50   \n",
       "CREBBP  ...                           1442   \n",
       "EP300   ...                           1373   \n",
       "FGFR3   ...                            193   \n",
       "FOXQ1   ...                            138   \n",
       "KDM6A   ...                            791   \n",
       "KMT2C   ...                            834   \n",
       "KMT2D   ...                           2299   \n",
       "NOTCH2  ...                           1123   \n",
       "PIK3CA  ...                             53   \n",
       "RB1     ...                            690   \n",
       "RBM10   ...                            337   \n",
       "STAG2   ...                            686   \n",
       "TERT    ...                             32   \n",
       "TP53    ...                            164   \n",
       "\n",
       "        Mutated protein positions (Mut == 1)  \\\n",
       "Gene                                           \n",
       "ARID1A                                   528   \n",
       "CDKN1A                                    44   \n",
       "CREBBP                                   636   \n",
       "EP300                                    583   \n",
       "FGFR3                                     70   \n",
       "FOXQ1                                     97   \n",
       "KDM6A                                    304   \n",
       "KMT2C                                    406   \n",
       "KMT2D                                   1399   \n",
       "NOTCH2                                   607   \n",
       "PIK3CA                                    23   \n",
       "RB1                                      185   \n",
       "RBM10                                    222   \n",
       "STAG2                                    316   \n",
       "TERT                                       8   \n",
       "TP53                                     105   \n",
       "\n",
       "        Mutated protein positions (Mut == 2)  \\\n",
       "Gene                                           \n",
       "ARID1A                                   271   \n",
       "CDKN1A                                    26   \n",
       "CREBBP                                   206   \n",
       "EP300                                    232   \n",
       "FGFR3                                     35   \n",
       "FOXQ1                                     55   \n",
       "KDM6A                                    120   \n",
       "KMT2C                                    116   \n",
       "KMT2D                                    721   \n",
       "NOTCH2                                   236   \n",
       "PIK3CA                                     7   \n",
       "RB1                                       41   \n",
       "RBM10                                    125   \n",
       "STAG2                                    113   \n",
       "TERT                                       0   \n",
       "TP53                                      46   \n",
       "\n",
       "        Mutated protein positions (Mut > 2)  Ratio mutated protein positions  \\\n",
       "Gene                                                                           \n",
       "ARID1A                                  362                         0.608810   \n",
       "CDKN1A                                   45                         0.696970   \n",
       "CREBBP                                  151                         0.407803   \n",
       "EP300                                   227                         0.431470   \n",
       "FGFR3                                    15                         0.383387   \n",
       "FOXQ1                                    79                         0.626016   \n",
       "KDM6A                                   239                         0.455983   \n",
       "KMT2C                                    57                         0.409766   \n",
       "KMT2D                                  1101                         0.583514   \n",
       "NOTCH2                                  205                         0.482727   \n",
       "PIK3CA                                    8                         0.417582   \n",
       "RB1                                      13                         0.257266   \n",
       "RBM10                                   247                         0.638024   \n",
       "STAG2                                   144                         0.455123   \n",
       "TERT                                      0                         0.200000   \n",
       "TP53                                     79                         0.583756   \n",
       "\n",
       "        Ratio non mutated protein positions  \\\n",
       "Gene                                          \n",
       "ARID1A                             0.391190   \n",
       "CDKN1A                             0.303030   \n",
       "CREBBP                             0.592197   \n",
       "EP300                              0.568530   \n",
       "FGFR3                              0.616613   \n",
       "FOXQ1                              0.373984   \n",
       "KDM6A                              0.544017   \n",
       "KMT2C                              0.590234   \n",
       "KMT2D                              0.416486   \n",
       "NOTCH2                             0.517273   \n",
       "PIK3CA                             0.582418   \n",
       "RB1                                0.742734   \n",
       "RBM10                              0.361976   \n",
       "STAG2                              0.544877   \n",
       "TERT                               0.800000   \n",
       "TP53                               0.416244   \n",
       "\n",
       "        Ratio mutated protein positions (Mut == 1)  \\\n",
       "Gene                                                 \n",
       "ARID1A                                    0.276875   \n",
       "CDKN1A                                    0.266667   \n",
       "CREBBP                                    0.261191   \n",
       "EP300                                     0.241408   \n",
       "FGFR3                                     0.223642   \n",
       "FOXQ1                                     0.262873   \n",
       "KDM6A                                     0.209078   \n",
       "KMT2C                                     0.287332   \n",
       "KMT2D                                     0.253442   \n",
       "NOTCH2                                    0.279595   \n",
       "PIK3CA                                    0.252747   \n",
       "RB1                                       0.199139   \n",
       "RBM10                                     0.238453   \n",
       "STAG2                                     0.250993   \n",
       "TERT                                      0.200000   \n",
       "TP53                                      0.266497   \n",
       "\n",
       "        Ratio mutated protein positions (Mut == 2)  \\\n",
       "Gene                                                 \n",
       "ARID1A                                    0.142108   \n",
       "CDKN1A                                    0.157576   \n",
       "CREBBP                                    0.084600   \n",
       "EP300                                     0.096066   \n",
       "FGFR3                                     0.111821   \n",
       "FOXQ1                                     0.149051   \n",
       "KDM6A                                     0.082531   \n",
       "KMT2C                                     0.082095   \n",
       "KMT2D                                     0.130616   \n",
       "NOTCH2                                    0.108706   \n",
       "PIK3CA                                    0.076923   \n",
       "RB1                                       0.044133   \n",
       "RBM10                                     0.134264   \n",
       "STAG2                                     0.089754   \n",
       "TERT                                      0.000000   \n",
       "TP53                                      0.116751   \n",
       "\n",
       "        Ratio mutated protein positions (Mut > 2)    Type  \n",
       "Gene                                                       \n",
       "ARID1A                                   0.189827  Normal  \n",
       "CDKN1A                                   0.272727  Normal  \n",
       "CREBBP                                   0.062012  Normal  \n",
       "EP300                                    0.093996  Normal  \n",
       "FGFR3                                    0.047923  Normal  \n",
       "FOXQ1                                    0.214092  Normal  \n",
       "KDM6A                                    0.164374  Normal  \n",
       "KMT2C                                    0.040340  Normal  \n",
       "KMT2D                                    0.199457  Normal  \n",
       "NOTCH2                                   0.094427  Normal  \n",
       "PIK3CA                                   0.087912  Normal  \n",
       "RB1                                      0.013994  Normal  \n",
       "RBM10                                    0.265306  Normal  \n",
       "STAG2                                    0.114376  Normal  \n",
       "TERT                                     0.000000  Normal  \n",
       "TP53                                     0.200508  Normal  \n",
       "\n",
       "[16 rows x 21 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols_mut_pos = [\"Mutated protein positions\",\n",
    "                \"Non mutated protein positions\",\n",
    "                \"Mutated protein positions (Mut == 1)\", \n",
    "                \"Mutated protein positions (Mut == 2)\",\n",
    "                \"Mutated protein positions (Mut > 2)\"]\n",
    "cols_ratio_mut_pos = [\"Ratio mutated protein positions\",\n",
    "                      \"Ratio non mutated protein positions\",\n",
    "                      \"Ratio mutated protein positions (Mut == 1)\",\n",
    "                      \"Ratio mutated protein positions (Mut == 2)\",\n",
    "                      \"Ratio mutated protein positions (Mut > 2)\"]\n",
    "cols_mut_dna = [col.replace(\"protein positions\", \"DNA sites\") for col in cols_mut_pos]\n",
    "cols_ratio_mut_dna = [col.replace(\"protein positions\", \"DNA sites\") for col in cols_ratio_mut_pos]\n",
    "\n",
    "normal_mut_plot_df = normal_count_df[[\"Gene\"] + cols_mut_dna + cols_ratio_mut_dna + cols_mut_pos + cols_ratio_mut_pos].copy()\n",
    "normal_mut_plot_df[\"Type\"] = \"Normal\"\n",
    "\n",
    "mut_plot_df = normal_mut_plot_df.sort_values([\"Gene\", \"Type\"], ascending=[True, False]).set_index(\"Gene\")\n",
    "mut_plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d28e1096-70fb-41ab-82d4-740f0a2bfc14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total mutations</th>\n",
       "      <th>SNVs</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDKN1A</th>\n",
       "      <td>1688</td>\n",
       "      <td>492</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXQ1</th>\n",
       "      <td>1709</td>\n",
       "      <td>672</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM6A</th>\n",
       "      <td>5694</td>\n",
       "      <td>2942</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBM10</th>\n",
       "      <td>5017</td>\n",
       "      <td>3373</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2D</th>\n",
       "      <td>16893</td>\n",
       "      <td>11204</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAG2</th>\n",
       "      <td>2779</td>\n",
       "      <td>1912</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>6406</td>\n",
       "      <td>3589</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>801</td>\n",
       "      <td>710</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EP300</th>\n",
       "      <td>3735</td>\n",
       "      <td>2894</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREBBP</th>\n",
       "      <td>3037</td>\n",
       "      <td>2416</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH2</th>\n",
       "      <td>2650</td>\n",
       "      <td>2106</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1</th>\n",
       "      <td>569</td>\n",
       "      <td>374</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR3</th>\n",
       "      <td>258</td>\n",
       "      <td>229</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Total mutations   SNVs    Type\n",
       "Gene                                  \n",
       "CDKN1A             1688    492  Normal\n",
       "FOXQ1              1709    672  Normal\n",
       "KDM6A              5694   2942  Normal\n",
       "RBM10              5017   3373  Normal\n",
       "KMT2D             16893  11204  Normal\n",
       "STAG2              2779   1912  Normal\n",
       "ARID1A             6406   3589  Normal\n",
       "TP53                801    710  Normal\n",
       "EP300              3735   2894  Normal\n",
       "CREBBP             3037   2416  Normal\n",
       "NOTCH2             2650   2106  Normal\n",
       "RB1                 569    374  Normal\n",
       "FGFR3               258    229  Normal"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_total_count = normal_count_df[[\"Gene\", \"Total mutations\", \"SNVs\"]].set_index(\"Gene\").reindex(gene_order_mutation_rate_normal)\n",
    "# blca_cancer_total_count = blca_cancer_count_df[[\"Gene\", \"Total mutations\", \"SNVs\"]].set_index(\"Gene\").reindex(gene_order_mutation_rate_normal)\n",
    "normal_total_count[\"Type\"] = \"Normal\"\n",
    "# blca_cancer_total_count[\"Type\"] = \"Cancer\"\n",
    "total_count_df = normal_total_count# pd.concat((normal_total_count, blca_cancer_total_count))\n",
    "total_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ac68a91-2abd-45b2-accb-87bad8bde7b8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "459cd54d-615b-42fb-827e-c1ebd063b924",
   "metadata": {},
   "source": [
    "#### Plot only one tissue "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "cf2585cf-d973-4b7f-94e0-bb15379ce2d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_edgecolor_for_tissue(df, cols_mut, txt_lst, n, ax, tissue, alt_txt=None, \n",
    "                             shift=0, bar_width=0.5, y_shift=0, lw=1, txt_fontsize=9):\n",
    "    \n",
    "    height = 100 #if n == 1 else df[cols_mut].sum(axis=1).values\n",
    "    df_dummy = df[[\"Type\"]].copy()\n",
    "    df_dummy[\"Dummy\"] = height\n",
    "    \n",
    "    df_dummy[df_dummy['Type'] == tissue].rename(columns={\"Dummy\" : tissue}).plot(\n",
    "        kind='bar', ax=ax, position=0.5+shift, width=bar_width, color=['red'], lw=lw, edgecolor=\"white\", fill=False)\n",
    "    \n",
    "    # Group the patches by their x position\n",
    "    bar_positions = {}\n",
    "    for patch in ax.patches:\n",
    "        x_position = patch.get_x()\n",
    "        if x_position not in bar_positions:\n",
    "            bar_positions[x_position] = []\n",
    "        bar_positions[x_position].append(patch)\n",
    "\n",
    "    # Iterate through the sorted positions\n",
    "    sorted_positions = sorted(bar_positions.keys())\n",
    "    x_txt_offset = -0.25, -0.25\n",
    "    \n",
    "    for i in range(0, len(df_dummy)):\n",
    "        normal_patches = bar_positions[sorted_positions[i]]\n",
    "        if alt_txt is not None:\n",
    "            txt = f\"{alt_txt[i]}\"\n",
    "            x_pos = normal_patches[0].get_x() + normal_patches[0].get_width() / 2\n",
    "            ax.text(x_pos, height[i]+y_shift, txt, ha='center', va='bottom', fontsize=txt_fontsize, rotation=0)\n",
    "        else:\n",
    "            txt = f\"{txt_lst[i]}%\"\n",
    "            ix_txt_offset = x_txt_offset[0] if len(txt) == 2 else x_txt_offset[1]\n",
    "            x_pos = normal_patches[0].get_x() + normal_patches[0].get_width() / 2 + ix_txt_offset\n",
    "            ax.text(x_pos, txt_lst[i]+y_shift, txt, ha='left', va='bottom', fontsize=txt_fontsize, rotation=0)\n",
    "\n",
    "\n",
    "def plot_proportion_mut_count_one_tissue(df, \n",
    "                                            count_df,\n",
    "                                            plot_pars, \n",
    "                                            mut_count_colors_dict, \n",
    "                                            in_labels_dict, \n",
    "                                            tissue,\n",
    "                                            ylabel_top='Number of protein\\npositions', \n",
    "                                            ylabel_bottom='Proportion of mutated\\nresidues (%)',\n",
    "                                            figsize=(8, 4.7),\n",
    "                                            bar_width=0.7, \n",
    "                                            shift=0.07, \n",
    "                                            y_shift=0,\n",
    "                                            title=None, \n",
    "                                            save=False, \n",
    "                                            filename='proportion.png'):\n",
    "    \n",
    "    df = df[df[\"Type\"] == tissue].copy()\n",
    "    count_df = count_df[count_df[\"Type\"] == tissue].copy()\n",
    "    cols_mut = list(mut_count_colors_dict.keys())[:4]\n",
    "    cols_mut_ratio = list(mut_count_colors_dict.keys())[4:]\n",
    "    \n",
    "    text_cols = [list(mut_count_colors_dict.keys())[0].split(\" (\")[0], list(mut_count_colors_dict.keys())[4].split(\" (\")[0]]\n",
    "    mut_total, ratio_mut_total = df[text_cols[0]].values, df[text_cols[1]].values # Count and ratio of utated DNA sites or protein positions\n",
    "    mut_total_count = count_df[\"Total mutations\"].values # Count of mutations\n",
    "    df = df.drop(columns = text_cols)\n",
    "    \n",
    "    # Create the plot\n",
    "    fig, axes = plt.subplots(1, 1, \n",
    "                                figsize=figsize, \n",
    "                                # sharex=True, \n",
    "                                # gridspec_kw={'hspace': plot_pars[\"hspace\"], \n",
    "                                #              'height_ratios': [0.5]}\n",
    "                                )\n",
    "\n",
    "    genes = df.index.get_level_values('Gene').unique()\n",
    "    x_genes = np.arange(len(genes))\n",
    "    \n",
    "    df_normal_mut = df.loc[df['Type'] == tissue, cols_mut]\n",
    "    df_normal_mut_ratio = df.loc[df['Type'] == tissue, cols_mut_ratio]\n",
    "\n",
    "    normal_mut_cols = df_normal_mut.columns              \n",
    "    normal_mut_ratio_cols = df_normal_mut_ratio.columns\n",
    "    \n",
    "    mut_lst = [df_normal_mut, normal_mut_cols, mut_total]\n",
    "    mut_ratio_lst = [df_normal_mut_ratio, normal_mut_ratio_cols, ratio_mut_total]\n",
    "        \n",
    "    # for i, track in enumerate([mut_lst, mut_ratio_lst]):\n",
    "    for i, track in enumerate([mut_ratio_lst]):\n",
    "        df_normal, normal_cols, mtotal = track\n",
    "        \n",
    "        df_normal *= 100\n",
    "        mtotal *= 100\n",
    "        mtotal = [int(i) for i in mtotal.round()]\n",
    "        alt_txt = None\n",
    "\n",
    "        \n",
    "        # Plot the normal and cancer bars\n",
    "        normal_bars = df_normal.plot(kind='bar', stacked=True, ax=axes, position=0.5+shift,\n",
    "                                        width=bar_width, color=[mut_count_colors_dict[col] for col in normal_cols])\n",
    "\n",
    "        # Add edges for tissue type\n",
    "        add_edgecolor_for_tissue(df, cols_mut, mtotal, 0, axes, tissue, alt_txt, \n",
    "                                    shift, bar_width, y_shift, lw=1, txt_fontsize=plot_pars[\"txt_fontsize\"])\n",
    "        \n",
    "        axes.spines['top'].set_visible(False)\n",
    "        axes.spines['right'].set_visible(False)\n",
    "\n",
    "    axes.set_xticks(x_genes)\n",
    "    axes.set_xticklabels(genes, rotation=45)\n",
    "    axes.set_xlim(-0.55, len(genes) - 0.5)\n",
    "    axes.set_ylim(0,100)\n",
    "    axes.set_xlabel(None)\n",
    "    axes.set_ylabel(ylabel_bottom)\n",
    "    # axes[0].set_ylabel(ylabel_top, fontsize=12)\n",
    "    axes.legend().remove()\n",
    "\n",
    "    # # Legend\n",
    "    # # axes[0].legend().remove()\n",
    "    # handles, labels = axes.get_legend_handles_labels()\n",
    "    # labels_dict = dict(zip(labels, handles)) \n",
    "    # tissue_labels_dict = {key: labels_dict[key] for key in [tissue]}\n",
    "    # del labels_dict[tissue]\n",
    "    # labels_dict = {in_labels_dict[key]: value for key, value in labels_dict.items()}\n",
    "    # legend_mut = axes.legend(labels_dict.values(), labels_dict.keys(), #title='Mutation', \n",
    "    #                        bbox_to_anchor=(1.03, 1), frameon=False, fontsize=11, title_fontsize=11)\n",
    "    # # legend_tissue = axes[1].legend(tissue_labels_dict.values(), tissue_labels_dict.keys(), title='Tissue', \n",
    "    # #                           bbox_to_anchor=(1.03, 1.23), frameon=False, fontsize=11, title_fontsize=11)\n",
    "    # #axes[1].add_artist(legend_mut)\n",
    "    \n",
    "    if title is not None:\n",
    "        fig.suptitle(title,\n",
    "        y=0.83\n",
    "        )\n",
    "\n",
    "    plt.tight_layout()\n",
    "    if save:\n",
    "        fig.savefig(filename, dpi=300, bbox_inches='tight')\n",
    "\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d0c60334-b883-45d4-92e4-bf4abb70af54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CDKN1A', 'RBM10', 'FOXQ1', 'ARID1A', 'TP53', 'KMT2D', 'NOTCH2',\n",
       "       'KDM6A', 'STAG2', 'EP300', 'PIK3CA', 'KMT2C', 'CREBBP', 'FGFR3',\n",
       "       'RB1', 'TERT'], dtype=object)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sort genes by mutrate in normal tissue\n",
    "genessss_order = mut_plot_df[mut_plot_df[\"Type\"] == \"Normal\"][\"Ratio mutated protein positions\"].sort_values(ascending= False).index.values\n",
    "genessss_order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "64d29820-25a3-452f-a854-9330db9aba87",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sort genes by mutrate in normal tissue\n",
    "genessss_order = mut_plot_df[mut_plot_df[\"Type\"] == \"Normal\"][\"Ratio mutated protein positions\"].sort_values(ascending= False).index.values\n",
    "genessss_order_custom = [x for x in genessss_order if x in gene_list_fig5]\n",
    "mut_plot_df = pd.concat([mut_plot_df[mut_plot_df.index == gene] for gene in genessss_order_custom])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "6acc0a68-4044-4f2d-9e79-224295895c78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXcAAACICAYAAAAcc6uGAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAOKRJREFUeJzt3XdUFNf7+PH3giIqKio2NIq9148xGlGwd2NiCVawYkGjgooVNCL2gg1jr7HEEjXWqIjdgBKMaMSCIkgvSl3K/f3hj/2iYmQLKnhf53AOuzs8++ww+8zMnXvvKIQQAkmSJClP0fvUCUiSJEm6J4u7JElSHiSLuyRJUh4ki7skSVIeJIu7JElSHiSLuyRJUh4ki7skSVIeJIu7JElSHiSLuyRJUh4ki/sXzMPDg5IlSxISEqJ6zsbGBg8Pjxx7z4CAACwtLd953tDQEEtLSywsLGjatClXr15VOx9LS0sCAgLeeM7DwwMbGxuNcj179ixbt25VPQ4NDaV48eKqxwsWLKBhw4ZYWlpiaWlJXFzce2Pl9HrNbP78+dy9e/ejvJf0+cr3qROQPi19fX1Gjx7NkSNHPmkeZcuWVRU/f39/+vfvj5eX1yfLJz09nSVLlnDy5EnVc9OmTUOpVKoe37p1i0OHDlG1atVPkeJ7TZgwgYEDB3Ls2LFPnYr0Ccni/oXr2rUrERER7Nq1i0GDBqmeT0lJYejQoTx9+pS0tDTmz59P27ZtadCgAZUqVcLU1JTk5GQKFCjAo0ePMDQ0xMLCglOnTpGSksLZs2cJDQ1lzJgxKJVKoqOjcXNzw9TU9IM5RUZG8tVXX73xXExMDCNGjODly5dEREQwe/Zsvv/+e9zc3Ni2bRsVKlRQnYEEBwczYMAA9PT0KFasGMWKFQNg7969rFq1Cn19ffr06cPEiROxsbEhPDycuLg4zpw5Q4ECBQD4888/adSoEfr6+gCcP38eExMTSpUqpcrJx8eHqVOnEhoayvDhwxk6dOgbOe/fvx9XV1dKly5NdHT0f67Xxo0b06JFC3x9fSlevDiHDx8mLi6OQYMG8erVKwDc3Nxo2LAhZmZmVK1alVatWlGmTBm2b98OQO/evZkyZQpFixbF0NAQPz8/6tSpk/2NQcpbhPTFunDhgrC2thZBQUGiSpUqIjg4WFhbW4sLFy6I1atXixkzZgghhAgJCRHVqlUTaWlpolKlSsLPz08IIYS1tbVYtmyZEEKInj17ivXr16t+v3btmjh9+rS4cuWKEEKIgwcPihEjRognT54ICwuLd3IpUKCAsLCwEObm5qJYsWLi119/Vb3HhQsXxF9//SWOHj0qhBDC29tbtG/fXsTExIh69eoJpVIp4uLiRPny5cWTJ0+Evb292LVrlxBCiJUrVwpra2sREREhGjRoIJKSkkR6erro3LmzuH//vrC2thbr1q17J5/p06eLHTt2CCGESE5OFm3bthWvXr0SlSpVEkIIkZaWJpydnUV0dLR49eqVaNq0qWq9ZKhRo4aIjY0VKSkpomnTph9cr7dv3xZCCNGuXTtx7do1YW9vL3755RchhBB3794VX3/9tRBCiHz58onY2FghhBBNmzYVPj4+IiUlRbi7u6vee+7cuW88lr488shdwtTUFGdnZ0aNGkXJkiUB8PPzo2fPngCUKVOGEiVKEBYWBkDNmjVVf9u4cWMAihcvTq1atVS/JyUlUaFCBebPn8/GjRuJi4ujcOHC780hc7PMq1evaNmyJS1atHgjx9WrV3Pw4EEUCgUpKSncv3+fevXqkT9/fvLnz0+jRo0AuHv3LiNHjgSgZcuW3L59m0ePHhESEkKnTp0AiI6Oxt/f/53PkyEiIoKyZcsCsHDhQkaPHo2RkdEby0yaNImiRYsC0KZNG/755x9q164NvG6fL1GihOr1b7755oPrtWHDhgBUqFCBpKQk/Pz8GDFiBAB16tQhNDQUgHLlyqnibt68maVLlxIQEEC7du3eWJ8RERHvXd9S3icvqEoADB48GD09PU6cOAFArVq1uHLlCvC6UIWGhqoKv57e/202CoXivTFnzZrFkCFD2Lp1K19//TUim7NLGxkZUaRIEVJSUlTPLV26lBYtWrBt2za6dOmCEAIzMzN8fX1RKpUkJyerLiJWr16da9euAeDt7Q1A5cqVqVKlCufOncPDw4MRI0ZQr169dz5PhlKlShETEwO8bqJZu3YtlpaWhISE0L17d6KiomjcuDFJSUmkpKRw6dIlVXEGKFmyJOHh4URHR5Oeno6Pj88H1+vb6zLzsn5+fqrmpcz5btq0CXd3dy5evMhvv/1GUFAQ8HrnVbp06WytbylvkkfuksqGDRuoW7cuALa2tgwbNoxWrVqRmJjImjVryJ8/v1rxevfuzfjx4ylTpgwVKlQgKirqvcuGhIRgaWmJnp4eSqWSLl26UK1aNdXr3bp1Y+zYsezatUsVq0yZMkyfPp3mzZtTrlw5VZGcP38+ffv2ZceOHZQtWxYDAwNKlSqFra0trVu3RqlU0qRJE8aOHfvefCwsLDhz5gx9+/bF09NT9byZmRnHjx8HwNHREXNzcwwMDLCxsaFGjRqq5fLly4e7uzvt2rXDxMREte7UWa8zZ85k6NChbN++HaVSycaNG99Zpm7dulhaWlK4cGEsLCwoX748AH/99ReLFy9+7+eT8j6FyO7hlCR9QdLS0ujUqROnT59WXVTNLWJiYhg6dCiHDx/+1KlIn5BslpGkLOjr6+Pg4MC2bds+dSpqc3NzY968eZ86DekTk0fukiRJeVC22tyHDh363gtnW7Zs0WlCkiRJkvaydeT+9OlT4PUIPRsbG5o0aYKvry87d+5UDaCQJEmSPh9qNct888033LhxA3g9PLt58+bcvHlT50lduHCBffv20atXL3777TcUCgXu7u7s3LmTGzduULhwYZYuXarz95UkScor1LqgamNjQ+vWrRk0aBBt2rRh0qRJOk/o8ePHeHt7k5SUxK5du9i0aROtWrXi/Pnz/Pnnn6xfv56iRYuqBqBIkiRJ71Krn/uYMWP48ccfefz4MdWqVcPY2FjnCVWpUgUHBwdsbGxU/X/LlStHaGjoO4+rV6/+xt8mJyeTnJyseiyEQKlUYmJi8p+DbSRJkvIatY7c161bx4gRIxg5ciTbt29n/PjxOZUXgKp/8YsXLyhbtuw7j9/m6uqqmiiqWLFiGBsbU7p0adXES/C6OSkn5WR8mXvejJ+bc8/p+Lk5948R/7+o1eZuYWHBxYsXadOmDRcuXKB58+Zcv349RxKzsbGhb9++HD16lLS0NH755Re2bdvGrVu3MDQ0zLLN/e0j95cvX/LVV18RGxurmosDICgo6I2pW7PLwMBANQLwv+RkfE1j53T83Jx7TsfPzbnndPzcnHtOx89uvXkftZplChYsqLqg+u+//+ZIs0yGjMEj3bp1Uz03bNgwhg0b9t6/KVCggGrK1v+iVCpJSkrSOsdPEV/mnjfj5+bcczp+bs79Y8R/H7WaZbZu3cqvv/5KwYIFWbNmDZs3b86pvCRJkiQtZOvIPTIykri4OBwcHFi2bBnwui1p5MiRqlkEJUmSpM9Htor7nTt32L59O5cvX8bZ2RkhBHp6enTt2jWn85MkSZI0kK3innED4MDAQPT19UlNTSU9PV11kwFJkiTp86LWBVV7e3vS09N5+PAh+fPnp1ChQly8eDGncpMkSZI0pNYF1cjISH777Tf+97//cePGjU/ah1OSJEl6P7WKe+HChfH29kapVHL58mUiIyNzKi9JkiRJC2oV940bN5KamsrPP//MoUOHWLRoUU7lJUmSJGkhW8Xd39+fixcv0r17d5KTk3n27Bm9evVi7ty5WS4fERHB6dOn2bt3L2fOnCE6OlqnSUuSJEn/LVsXVBMTE/Hw8CAiIgIPDw9VV8hZs2a9sdyNGzdYv349+fLlo1q1apiYmBAUFMTevXtJT09n9OjRNG/ePEc+iCRJkvR/slXcGzRoQIMGDfj222//c3j/kydP2Lx5c5Y3FE5LS2P//v2yuEuSJH0EanWFvHr1KvB6Kl0/Pz/S0tJo3bq16nUrKyvV76mpqfz5558kJibSvn17ihQpQv/+/dVKbtmyZdy+fRt/f38GDBjAhQsXMDIyYsmSJZQrV06tWFnZunUr9+7d4+nTp3Tv3p0HDx6gUChwcnJi7dq1JCYmMmjQIP7++2+qV69O/fr1tX5PSZKkj0Gt4u7k5PTG4w4dOrzxOD09HT29183406dPx9zcHD09PXr16sW5c+fUTs7e3h6lUomdnR0+Pj6UK1eOokWLZjndL2Q9K+R/GTp0KEqlEhcXF3x9fVmyZAm///47N27coFKlSkRERGBoaMjTp0/laFxJknIVtYr73LlzVTe9yGp0qr29PbVq1cLGxoayZcvi6emJQqHQ6ij7119/5YcffqBYsWI0b96cjRs3cvbsWTp27PjOsq6uru+9yPs+J06coH379pw8eRKAUqVKERERwXfffad6/+rVq+Pm5sbAgQOpXLmyxp9FkiTpY1GrK2TGNASWlpYMHjxYVRAzrFixgtatWzN79myEEEycOJHp06eza9cujRP09PSkU6dO+Pj4oFAoMDY2JiUlJctlp0+fTmxsrOonMDDwg/G9vLwwNzdXXScIDw/HxMQEeH3kn5CQwL///outrS3Hjx/X+HNIkiR9TGoduT969Ij9+/ermj4UCgXnz59Xvf748WNOnDhBlSpVaNmyJRs2bMDAwIBRo0a9tynlQ4QQKBQKFAoF48aNIz09HTc3tyyXze587lnFb9++PU5OTqSnp6uO/vfv30+/fv24cuUKq1atokuXLtmO+/jxY7Zt24ZCoWDQoEHky5ePRYsW4e7ujpub2zvt+W/fMlCSJEkbahX3devW4eHhgZGRUZavDxkyBHd3d+Lj41m0aBG7du0iIiICd3f3d7pNZteWLVsAGD16tEZ//yEuLi7A67tMWVhYvPHaiBEjAOjSpQtdunTB0NAw23EPHjxI6dKliYyMpGzZsri7u1O8eHEAnbTnZ955lC5dmpCQEAoWLMiMGTPkzkOSJPWKe58+fTh48OAb7c6Ze8u0aNGCJUuWoKenR9u2bQEwMTHRuLDnlEpXXCDsgfp/WLoGVN6UrUX9/f2ZN28eAQEB7Nu3jwEDBrB27VqALNvze/fuTdWqVbOdSuadR61atRg3bhzz588nLi5O653H22cd9+/fJzAwkLFjx8odhyTlEmoV9z///JM6deoQEBAAvG6WyVzc586dS6FChd779wkJCf/5el5SvHhxChUqRNGiRVmxYgXPnz/Hx8eHmzdv0qxZs3fa83ft2sW4ceOyHT/zzuPZs2ccOnSIwMBADAwMtN55ZN5xxMTE8OzZMzJutSt7EUlS7qBWcS9UqBDLly8nX76s/2zBggUIIejcuTM1a9akZMmSvHz5El9fX06ePIm+vr6qGSSvGzx4MM7OzhgZGXHlyhWMjY2ZMWMGzZo1A95tz2/Xrp1a8TPvPKKiohg9ejRGRkZcu3YNCwsLrXYemXccT548oVevXhw+fBjQzVmHh4cHO3fupGTJktSvX5+nT58SGRmJo6Mj+/bt0/rMIKfjS1JuoFZxDwsLo1q1aqpmmbcvqM6fP5/g4GB+/fVX1qxZQ0xMDCVLluSbb75h/PjxWt3JO7epV68ey5cvf+O5BQsWqH5/uz1fXZl3HpUrV8bZ2Zn4+HhmzpwJaLfzyLzjyGpaZ23POu7cuUOZMmUoVKgQYWFhBAcHk5KSQvHixXVyZpDT8SUpN1CruF+/fv2Dy5iammJvb69xQnnBx2jTz2rnkZk2O4/MO47JkycTHx//xuvannW0bduW6tWr89dff3Hv3j3Wrl3Lvn37uHPnjk7ODHI6viTlBmoVd+nT03jHAdneeby94zA2NsbOzk71WNuzDj8/P2rWrEmRIkXYtGkTI0aMoESJEsTGxgLanxnkdHxJyg2yVdwvXryIhYUFMTExGBsbq/UGGf3IpdzhY5x1lCxZEicnJxQKBTY2NsyYMQN9fX1mz54NaH9mkNPxJSk3yFZxnzhxIsuWLcPBwYGVK1e+8Vrm3jIZ1q9fT/HixQkJCeHAgQNYWlp+MRdSpQ9r27atqqtsVrQ9M8jp+JKUG2SruK9evRoPDw+io6Px8PAA/u+IPKvivnv3bi5dukT79u25cuXKf37RpC9HTjcpfYwmK0nKLbJV3M3NzTE3N2fYsGFcvXqVhw8fYmZmxo8//ph10Hz5mD9/Pi1btuTcuXNvzNQoSbldVFQUY8eOZcyYMVy6dEl2s5Q+S2pNHObo6MjDhw9p0qQJjx8/xtraOsvl9u7dS6NGjXByckJPT49Dhw7pJFlJ+hxs27YNU1NTvL29CQ4OJi4uTtXN0sTERNXNUtPCHhUVhZWVFV5eXjg7OzNnzhwiIiJwc3Nj0aJFBAUFceLECfz9/T/L+NLnQa3i/uzZMxwdHenYsSMzZsxQjVR926tXr9i9ezfdunXj3r17+Pr6apTc8ePH6dChA4MGDcLNzY0RI0YwcuRI0tLSNIonSdo6cuQI7dq1w9DQkHbt2rF27VratWun6mY5fPhwzpw5o+pmGRQUpPZ7ZOw8Tpw4QZEiRShcuDAlSpTQ2c4jp+NLnwe1irupqSmurq6cPXuWBQsWvHdQkq2tLStWrCA5OZm+ffvi6OioUXI3b96kfPnyGBsbc/HiRTZt2kSrVq3eGDiVWXJyMi9fvnzjR5J06dq1axw+fBgfHx/69++PQqH4z26Wp06dUit+5p3H3bt3GTNmDNWqVePatWs62XnkdPyMs4IjR44we/Zs1cA9eVbw8alV3Hft2kWNGjW4ffs21atXZ+fOnVkul5ycTNmyZVEoFJQqVYrChQtrlNx3333Hhg0b6Nmzp6ppp1y5coSGhma5vKurK8WKFVP9fPXVVxq9ryS9z6JFi3B2dqZRo0bMmjWLGTNmcOnSJczNzYHX3Sz79u1LlSpVWLVqFY0aNVIrfuadx6NHjyhQoMAbI4W13XnkdPyMswJjY2N+/vln0tPTVZPZ6bLJCuDYsWOsW7cOkDuPrKg1iElPT4/evXt/cLmJEyfSunVr/P396dy5M2PGjNEouVu3btGwYcM3+ta/ePECU1PTLJefPn06kydPVj1++fKlLPCSzmTujXOwFcBFaPX/Xzw3HoAl5YErf/E1QH0g7BFc8ch2T5xFixYBMGPGDDp16sSMGTMwMjJi2rRpgPZ99HMyfsZZwb59+7C0tNTpZHYZMnYef//9t5zQ7gNyZIRq37596dOnD+Hh4ZQqVUrjQUxlypRh1KhRKBQKdu3aha2tLWlpafzyyy9ZLq/JzTok6XOU0ZzxoXsMfE7xr127ho+PDz4+Pqxdu5Zx48bpbDI7eHPn0bBhQ0xMTHQ6od2VK1c4evQoBQoUwMLCgmvXruXqnlBqFffU1FS8vLxQKpWq57Lq596mTRsUCgXp6ek8efKEihUrcunSJbWT69mzJz179lQ9HjhwoNoxJCk3yAtjADKfFRgZGel0Mjt4c+dx5swZ6tat+8br2u48hBC4uLiwfft2jh49SkpKSq6ecE6t4t6tWzcqV66suuH1+wYxXbhwQfV7cnIy/fr10zJNSZJyi8yzn2am7VlH5p1Hx44d37nYq+3Ow9zcnIsXL3LhwgV++uknmjZtmqsnnFOruAshcHd3/+Bynp6eqt/DwsKydaPqjynyzgNSn/yt9t/lqwylciAfScrtPsacRBkydh7ly5fX6YR2t27dwsLCAhMTE4YPH87169dz9YRzahX3woULM2DAAGrWrKlqR58zZ847y124cAGFQoEQAkNDQ/bu3aubbCVJ+uJ8rGklQkJCmDlzJsnJyYwbNy7XTzinVnGfOHHiGxdHM65UZ9ixYweA6mYeGfPPXL9+nRo1amiba64hzwwkKffp2rXrf7al57YJ59RulpkzZw4KhQKlUsnixYvfeR1eTxz2v//9j8aNG3Pnzh18fHwYMmSI7rKWJEnSkbw64ZxaxX3mzJmcPn0aIyMjXr16Rfv27blx44bq9Yy5ZrZv346rqysA/fr1e6e7laQ5Tc8KQJ4ZSNKXRK3inp6ejp7e60Gtenp6qt/fVqlSJSZMmECDBg3w9vamcePG2mcqSZIkZZtaxd3FxYUuXbqgr69PWloaP//8c5bLbd26lVu3bvHkyRPGjBlDgwYNdJKsJElSbuTl5cW+fftQKpUMGjSIP/74g/T0dCZMmMCePXtyZIBUtop7WFgYpUuXplq1aqqLpsA7I08nT57M8uXLVYOYMtrgFQrFeyf7kj4vuflisGyykj5XcXFxLFy4kAcPHjBgwAAGDRpEamqqajbOnBggla3ivm3bNqZOnaq6L2Xmor1lyxbVchk3Vc48iCkwMFDO7yIBsvhKXy5LS0uSkpLYs2cPJiYmjBkzhpMnT6pm4wTdD5DKVnGfOnUqACYmJixZskT1/KhRo7JcftasWdStWxdfX18ePnxIsWLF2LTp87yiLEmSlNNiY2NZvnw5Y8eOJTo6+oOzcepigFS2ivuBAwdYt24df//9N97e3sDro/amTZtmubynpyc///wzW7Zs4ezZsxr3lvH09MTd3Z3k5GRat27NhQsXMDIyYsmSJaopECRJkj53ixcvJjw8nBUrVtC+fXudz/aZlWwV9759+9K3b1/mzZuX5YjUtxUuXBhbW1u6du3K3r17MTQ01Ci5ly9fsnPnTu7cuUPjxo0ZPXo0RYsWpWzZslkun5yc/Mb9Wt93sw7vFwlEP41TO5/ihgl0zsZyOR1fkqTcxcXF5Y3HvXr1euNxTgyQUqu3jK+vL+fPn6dKlSqqbpAVK1Z8Z7lDhw7x8OFD6tevzz///MORI0c0Sq579+4kJiayZs0aPDw8aN26NRs3buTs2bN07NjxneVdXV2ZO3euRu8lSZ+SvB6Rd33MeXcyU6u4FylShF27dr3xXOYLqhl8fHyYN28eL168oH///piamjJ48GC1k4uOjmbatGk4OTlx/PhxLCwsMDY2JiUlJcvl5c06JEmSXlPrNntubm7UqVMHAwMDatSowdq1a7NcburUqRw8eJDixYvz008/sWLFCo2Ss7e359mzZ0yfPp3IyEjGjRvHhQsXsjxqB1QXKTL/SJIkfYnUOnIfPHgwVlZW9OzZEy8vL/r168exY8feu7xCoSBfvnwUKVJEo+SyOiv40mnang+fvk0/N+cOcgyAlLuoVdwz35y2Ro0arF+/PsvlXFxc6NGjB35+fvTs2RMnJyftM5U+itx8MTi37zxys9y848ur1CruTZo0wdbWlqZNm/LXX3+9d86Yffv2ce7cOZ0kKEmSJKlPreK+cuVKvL29efz4MaNGjXpvP/eAgAC2bNnyRq+arG7HJ0nSa/Ks49PJq01WahX3Y8eOsXLlSvT19VEoFEyZMoX27du/s1yzZs0IDAxU3V7vffdalSRJ+pC8WnxzmlrFfd68eXh6elKwYEHi4uJo165dlsVdtrFLkiR9WmoV9/r16/P06VNq1apFeHj4Z33nb0mS/o9s9vnyqFXcIyIiGDduHAULFiQxMZH09HTV9L5ySl9JkqTPh1rF/ciRI6qpBczMzPjxxx/fmdNdkvKq3NxNNKfJdfP5UWuEqo2NDQ8fPqRJkyY8fvxYdc9USZIk6fOi1pH7s2fPVHdi6tixo+wBI0lSjsvt1ws+1QAvtYq7qakprq6uqkFM5cuX1+KtJUmSPr3cvvN4H7WaZRwcHKhRowa3b9+mevXq7Ny5M6fyypKPjw9DhgzB2tqa2NjYj/rekiRJuYlaR+4TJ07E09Mzp3L5oNWrV7Nx40Zu3LjB/v37GTly5Buvv32zjowdwNs37dAvVxUDZZra769frup7bwDyseJrGjun4+fm3HM6fm7OPafj5+bcsxvfh3K81HulduyilOOb98QuUqTIhzuzCDXUqVNHVKpUSVhYWAhLS0vRpk0bdf5cazY2NkIIIR48eCB+/vnnd153cnISgPyRP/JH/uTpn9jY2A/Wy2wfuSckJHD79m0MDAyy+yc6V7BgQVJSUnjx4kWWt9p7+2Yd6enpREVFUbJkyWx32cy4wUdgYGCOzAefm+Pn5txzOn5uzj2n4+fm3HM6vqaxszONeraK+4IFCzhz5gwA48aNo2/fvtlOQpeGDx+Ora0tycnJ/PLLL++8XqBAAQoUKPDGc8bGxhq9V07f7CM3x8/Nued0/Nyce07Hz82553T8nIidreJ+4sQJLl++jFKppFu3bp+suP/vf/+TN/CQJEnKhmz1lsk4GjYwMCA9PT1HE5IkSZK0l60j9+DgYHbs2IEQQvV7hiFDhuRYcp9CgQIFcHJyeqd5R8bP3bnndPzcnHtOx8/Nued0/JyMrRBCiA8ttH379ve+JqcgkCRJ+vxkq7hLkiRJuYtaI1QlSZKk3EEWd0n6SORJ8qfzJa57WdwlndL1lygvfCkzpsRQKBR54vPkJjm17nNDr0FZ3LMpJSWFpKSkN57T1caiVCrfia1L8fHx/PPPPzkSOyQkBE9PTx49egSgGgmszbqJj49/I15OFcSkpCRevVJ/zg91hIeHM2fOHJYuXQqgs5vbZJ5DCXRbbKKjozl79iz379/XWczMYmJi8PPzy5HYmYWFheHo6MiCBQsA3a378PBwXFxc8Pb2BrSvA0lJSTkyEaIs7tkQFxfHsGHDmDFjBlevXlV9kXRReEJDQ5k4cSJr165l2bJlukhXJSO3nTt3sn//fm7duqXT+ABnz55lxowZzJs3D3d3d548eUJaWprGX6Tw8HAcHBw4cOAAq1atAnKmwIeFhWFra8uyZctUhVfXlEolM2bMwMLCgpSUFH7//XfVa5p+HiEESUlJfPvttyxdupQ9e/YAoKf3+qusbZGPiopi6NChBAcHs27dOuLj43W67l++fMmsWbNwd3dn9+7dPHz4UGexMwsNDWXWrFn07t2buLg4rl27ppO4ycnJuLm5UaFCBdauXYuXlxfBwcEaxRJCkJqayvz58zl27BgRERE6yTGDLO7ZYGRkRJMmTShVqhSbNm1i2rRpzJ8/H9D+aODMmTNUrVoVe3t7IiMjdVpoMnIzMTGhcOHCeHt7c/36dZ3FB+jbty92dnYsXryYHTt24OzszOjRozWOd//+fSpXrsyAAQPw8/NTTTOhy9s5hoWFMWzYMAYOHIizszNRUVG4urrqLD68npE0NTWVIUOG0LRpU1q0aEGtWrV49uwZoPnnUSgUGBoaMmDAAJo0acK1a9ewsbFh5cqVhISEqIq8Nnm3bdsWa2trypYtS+HChXV6Vvno0SPMzMxwc3NDT0+PGzdu6Cx2ZqtWrcLY2Bhzc3NCQ0MxMTHROmZCQgLp6elUrlyZgQMHUqJECW7fvs2VK1c0iqdQKMiXLx+FChXi/v37XL9+nZCQEK3zzCCL+39ISEhQrexWrVpRs2ZNKlWqRMWKFYmKiuLFixcax85oDmjUqBFlypRBCMGCBQsQQpCYmKh17tHR0Rw7doxnz55RrVo1Jk2aRP369fnrr7+4efOm1rF37dpFSkoKhoaGeHl5MXDgQIYOHcr27dtZuHCh2jGjoqJYtGgRjRo1YuTIkSQmJjJp0iQCAgI4ffq0VvlmlpKSQmhoKAULFqRMmTLA67mToqKiCA8P19n7XLp0CTs7O7766itKly4NwJo1a5g6dSphYWEaxXz16hUHDhwgPT2dqlWrcurUKapXr46ZmRnFixdHqVRqnG9YWBj79+8nISGBPn36AFCmTBmOHDnCypUrSUlJ0Tg2vD4jO3DggOog4/fff6d///54e3vj6+urVezMIiIiOH/+PFOmTCElJYVOnTphZ2dH9erVtTqrCQ8PZ8iQIWzbtg0zMzMMDAyYPn06VapU4fTp0+80k31IVFQU+/btQwhB37596d+/P1FRUVy5ckXj7eNtas3n/qXIOA1duHAhZcuWpU+fPtSqVYtZs2bx1Vdf4eTkpFVshUKBnZ0d9erVY/LkyZw7d479+/fz6tUr7t27p/VpcHh4OJMmTaJy5crcunWLatWqUadOHWrXrk1aWho1atTQKn5iYiKbN2/GwMCAfv36sXjxYmxtbenatSsAxYsXVyvey5cvmTNnDmFhYRw+fJimTZtSp04ddu7cSVRUFA0bNtQq3wwREREsX76c6tWrs3DhQvbu3Yu+vj7x8fE8fvxYJ6MEX758SdGiRenevTsPHjzAysqKEydOEBUVxalTp7hw4YKq2Ksbd8WKFSQkJFCxYkV69OjB1atXCQgIYPny5VrlHB4ezvTp0xk4cCD//vsvycnJ9O/fn8OHDyOEYP369eTPn1/j+ElJSSxYsIDixYujp6dHp06dOHToEM+fPycpKYlSpbS5mdz/CQkJYdasWQA8ffqUBQsW8NNPP/HixQsaN26s8VlNZGQkrq6uTJw4kcjISNWB2cmTJ3n06BEzZ85Ua9sJDQ3F0dERY2NjfHx8VGeNenp6eHt7U6hQIY3yfJss7u+hUCgoXbo0z58/5+LFi/Ts2ZO5c+dy9+5dhBAIITTaWJKSkihYsCDffvstR44coVKlSkyYMIG///6bv//+m5kzZ2r1z01LS2P9+vX07t2b77//noCAANXFsQYNGtCyZUuNY6empqJUKjExMaFr1648fPiQAwcOYGlpSb169VQzcKqzXpRKJTExMVy+fJmuXbtSvnx5li9fztSpUzE2NmbKlClZTu+srri4OGbPnk2XLl24evUqFSpUoEuXLri4uBAREcGGDRt0MivfH3/8gY+PD9bW1gQGBtK1a1emTp3KtGnT8PT0pFy5cmrHVCqV6Onp4evrS61atQgMDMTd3R1zc3OKFSsG/N9Bg7qEECxatIh27drRtm1bBg8erGp+6NSpE926daNixYpqx32bsbExc+bMITAwkL/++gtHR0diYmLo06eP6gxKG8nJybi4uGBubk6bNm2wsrLCwMCA77//Hg8PD8zNzTX6/8bHx7NkyRIqV67MN998Q+fOndHX18ff3x99fX3s7OwoWbJktuPFxsYyd+5crKysaNasGf3792f+/Pn079+f2rVrU7VqVZ1Nqy5HqL4lMTERFxcXHBwcSEhIICUlhatXrwJQsWJF6tatq/E0wsnJybRp0wYnJydat25Neno6c+fOpWrVqtja2mqde1RUFIUKFWL37t2UKVOG7t27k5CQQP/+/TE1NWXWrFmULVsWfX19tWOHhoayYsUK9PX1sba2pkaNGrx69YqFCxfSoUMHmjdvjqGhoVoxw8LCGDt2LFZWViiVSipVqkTLli3ZunUrenp6OpvaIjo6Gj8/P+7evcuAAQO4dOkS9evXR19fn/DwcIyMjKhSpYpW7xEeHs7BgwcZOHAgGzZsYOfOnWzYsIHmzZtz7NgxmjZtqlFhT05OxtLSEgcHB9LS0nj69ClTpkxh9+7dFClShKZNm2JqaqpV7lu2bKFbt26qInv58mXu3LnDkCFDKFy4sMZxQ0ND+eWXX+jduzenTp0iODiYpUuXsnTpUkxNTRkwYIBWeWeIi4sjPj6eY8eOkT9/fvbt28eSJUv4/fffMTU1xdraWqMdX2hoKDt37qRu3bocOXIEpVLJxIkTqVWrFt7e3tSuXVuts9Tw8HAmT57M5cuXad68OSkpKSxYsIBff/0VPT09Zs+erfFOOiuyzf0tBQsWpEmTJsybNw8jIyMqVapEs2bNUCgUNGrUSOPCDq8nCbKwsGD79u3cuXOHwoUL4+TkxNOnT4mJidG6Oeb333/n4MGDNGzYkKioKM6dO8fly5dJTEzE3t6e8uXLa1TYg4ODmT59Oh06dKBnz55s3boVLy8vihQpwsSJE6lfv77ahT0kJARHR0eGDx9OSkoKFStWpFq1asyZMwcvLy86dOigdp7ve58+ffqgUCgYNWoURkZG6Ovrs2rVKn766ScqVKigk8I+efJkoqOjefHiBQ4ODvTr14+jR48C0KNHD40KO7zeZtq0acORI0eIiorC1NQUe3t7Tp06RatWrTQu7FFRUarfS5YsibOzM0FBQezbt4+VK1fSsWNHrQq7UqnE3d2dokWLsmfPHvr27Ut6ejqLFy8mMDCQ5s2baxw7s7CwMIYMGcLBgwcpXLgwRYsWxcrKirt37/LgwQNatmypUbEMCwtjxowZfP3113Tp0oXx48dToEABEhISKFCgAN9++63ahX3u3LmMGjWKDRs2ULJkSbp3705wcDABAQEMHDgQ0G3HAdks8/9FR0czefJkGjRowHfffce5c+eYOXMmc+bMoWrVqlSqVIl8+TRbXREREURERFC5cmWsrKxYvnw57u7uREdH06lTJ1U/XE3Fxsaip6eHgYEBxsbGNG3alNTUVP7++2+eP3/Ohg0bqFy5ssbxz58/j6+vr2oufYVCwdatW6lVq5ZG7aUhISF07tyZTp06UbZsWQ4ePMiYMWMoU6YMw4YNo0CBAhoXw8zi4uJUZ2AxMTGq52/dusWpU6c4efIkJUqU0Oo9hBDcunWL7t278+OPP5KSkoKXlxdDhgzhwoULhIaGatTskHmb6devH6tXr8bLy4vOnTszYcIEFAqF2tc2MrOzsyMlJYXatWvz008/ER0dzZ9//smtW7dYtGgRVatW1Tj2q1ev8PPzo06dOvTt25fx48dz48YNvv76aywsLChSpEi27iT0IbGxsXTp0oWVK1cSHx/PhQsXMDAwoEyZMty4cYOZM2dSvXp1jWIfOHAAgAYNGmBlZcXatWtVXZYbNGhAoUKFsl2I09LSWLNmDc2aNaNixYpMmDCB8PBwIiMjMTAwYPbs2VofYGRFNsvw+oJJUlISixYt4vbt2zg4OODn58dvv/0GwM2bN9HT09N4r2pnZ4eHhwc9e/Zk5syZnDt3jqJFi3LkyBFcXFzU2lCysnr1apKSkjh37hyGhoYMHjyYoKAgfvzxR4yNjTW+UJiSkoKzszOOjo54eHhw9uxZ3Nzc8Pb2ZsuWLSxdupSCBQuqFTMsLIzRo0djaGiIlZUV+/fvp0ePHvz4448a5fg+ISEh7NmzBysrK0xNTbGzs6NFixYMHDiQsLAwEhISMDMz0+o9IiMj8fDwICYmBn9/f1UvoWXLlmFgYMD48eM1jj127FguX75Mt27dVNuMsbExhw4dYsmSJRq3y0ZFRfHvv/+SlpZGWFgYRYoUYcqUKXTr1o1mzZrx3XffaZwzvP7/Tpo0iRo1apCQkMCiRYtITEzk2rVr/Pbbb6xatUqri7MZYmJiMDQ0xNHRkUKFCqnOmE6ePEmTJk1o1qwZRkZGaseNiIjg8uXL9OrVC2dnZ65fv868efNo1qwZ8Po7oUn+/v7+7Nu3D39/f4YPH06lSpWIjIzEzMxM6wOM98r23anzsHXr1onDhw8LIYRYvHixWLJkieq18PBwjeNGRkaKkydPCiGE2LJli2jbtq3YvHmz+OGHH4S7u7tWOQshRHR0tLhx44aIiIgQQgjx9OlTYW1tLY4dOybc3NzEs2fPtIodGRkpTp06JcaNGydiY2PFsWPHRMuWLcWgQYOEv7+/2jFTUlLE8OHDxezZs8Xx48fFnj17xLZt28Tp06fF06dPNc41K/v37xe1a9cW9+7dE0IIERERIYYMGSIOHDigk/gvXrwQnTp1EpcvXxZCCDF37lwxe/ZsceTIETFgwAAREBCgUdzo6Gjh6+srUlJSxKpVq0TXrl11ts28ePFCDBo0SFy9elU8e/ZMdO/eXUydOlUsXbpU+Pv7a7W9CCFEcHCw6Ny5s/D29hYxMTFi9erVIiEhQRw8eFBMmzZNPHjwQKv4GUJDQ8X3338vfH19hRBCTJo0SYwfP17cuXNHfPfdd+LFixcax7W1tRW7du0Sly9fFrdu3RKTJk0Sq1at0kneT58+FU5OTiIwMFAn8T7kiz5yj4uLIzU1lbNnz/Ly5UuGDx8OwKxZsyhZsiSTJk0iPT1do14xGd2djIyMKF68OPPmzWPx4sUkJSUxceJEwsPDtTr1DQsLw97eXjXQpHbt2vzwww8sW7YMR0dHjePC66OXadOmERMTw6JFi3j69CmHDh1i4cKFHDhwgBYtWlC7dm2NYsfExGBsbMz169cJCAggMjJS1f6oyZHW2zL+X0FBQUyZMgUDAwN++uknGjduTExMDElJSVr3vnn16hVjxozh0aNHODo6qo52M+570LJlS6pVq6Z23PDwcKZMmYKJiQmmpqa0bt2a8+fPEx8fj729vVbbTEREBCNGjCAkJIQrV66gr6/PwYMHOXfuHOvWrdMo5tu5T5s2jRcvXmBhYcHjx48ZMmQI5ubmxMbGolAodNIbKSoqCjs7O0aMGEGNGjU4c+YMw4YNY9KkSZw9e5ajR49q1MQRHx/P+vXrqVWrFu3ataN79+706NGDtm3bcvr0aYYNG6ZWr5j3uX//Pn/88Qd2dnY5dnMRlY+yC/lMHTlyRGzatEls3rxZtGnTRqxYsUJMnz5dXL16VYSGhmocNyYmRowZM0acOnVKREVFifbt24sVK1aIx48fi8WLF4vY2Fit8o6PjxcrV64UR48eFUII4efnJ5YuXSqCgoLEtGnTxIsXL0R6erpGsUNCQsTIkSNFYGCgOHXqlFi5cqVITU0VBw8eFFOnTtUq77c/w8WLF8XBgwdFTEyMTmI+f/5c2Nvbi927d4v09HRx6dIl4eTkJHr37i2uX7+uk/dITU0V9+7dE8+fPxdCCGFnZyd27Nihddzg4GDRs2dP1Zne6dOnxeLFi0ViYqJwdXXVapsJDw8Xw4cPF3v27BGXL18Wtra2IiwsTDx+/FjMmjVLCCE03l6EeH3EO2jQIHHz5k0RFBQkmjRpotNtJUN6err4448/hJOTk7h9+7awtrYWfn5+wt/fXzx//lwEBQVpFDcsLExMnDhR2NvbC09PT5GYmChiYmLE+vXrxebNm3X8KYRITk7WecysfJG9ZcLCwtiwYQMlSpTgwYMH3Lt3j+HDh2NlZUXjxo0xMzPTaKAJvD6CsbOz4+TJk2zbto2RI0eydu1aIiMjOXLkCFOmTNHqCCY8PJyZM2cSGhqqilO7dm3i4uJ49OiRauCVpn2eHz16xOPHj3n16hWrV6/m3r17jB49moIFC2p9RpBZoUKFaNasGV27dlX11dZGWloaDg4OXL9+nQ0bNuDq6srLly8pV64ca9eu1Ulf7ZCQEKZPn87GjRtVk2rNnTuXP//8U3V9RtPcJ0+e/Mao544dO/LixQu8vLxwdHTUeJuJjIzEwcGBrl27UqRIEZo3b863337LtGnTKFeuHA4ODoDmvTTS0tJYvXo1bdu2xcTEhCVLlrBv3z4CAgLYuHGjRjGzEhoayty5c4mMjERfXx83NzdGjRpFQkKC6jNo0nMoLCwMOzs7evTogbW1NSEhIXh5eeHj48OlS5do06aNzj5DBl31Y/+QL664h4WFMXPmTBISEsiXLx/x8fHo6emRmJhIgQIF6Nu3r8Y9Nf6ru9Pz58/p2bOn1rlnbIgDBw4kOjqaq1ev4unpycOHDzVqCsgQFxeHlZUVJUuWxNzcnHHjxjFt2jTc3d2ZNGkSjRs31qp3RlYMDQ3V7kKZlbS0NPT19TE3N+fy5ct8//33hIeHY2xszK5duzA0NNS69010dDQLFy6kVatWDBs2jFOnTnHmzBlKlCiBm5sb5ubmWud+6dIlAgICOHToENeuXSM4OFirXk4Ajx8/plq1ajRp0oRTp06xevVqGjduzIABA4iNjdV6x6qvr8+QIUMICgrC0dGRYcOGkZ6ejoGBgc4KY3h4OK6urlSvXh0TExNGjx5Nr169OHfuHK6urixbtozy5curHVcIwYEDB2jYsCFly5Zl3759FC1alNOnT/P777/j7Oys9fr/lL6orpAxMTHMnz+fli1bYmNjA7xuo42LiyM2NlaruSey091Jmzb2tzfE3bt307p1ay5fvkx8fDxz587VqoAlJiaSmprKnTt3KFasGE+ePMHFxYWNGzdSp04djePmtLS0NNatW0f37t3x8PBAoVBgZmbG3r176dChA56enlr3HU5MTGTnzp2YmprSo0cPAEaNGsXatWtVo0Q1KZJv566vr0/jxo1xcHCgSJEiHDp0SKOilZmZmRlVq1bl5MmTzJ8/n9OnT7N9+3adTlBXvXp1rK2t2bp1K/fu3ePw4cO4uLjopHtfTEwMLi4udOnShbp16zJ06FBq1qypmupizZo1Gl9DUSgU/PDDDxw/fhxHR0dGjRpFp06d6NSpE4mJiWr3BPvcfDHFPTw8nAkTJvDs2TNMTEyIioqiRIkS+Pr6kpiYyOTJk7WKr6+vz6BBg9i3bx9z5szB3t5ep92d3t4QbW1tVRtiUlKS1kfApUqVomXLlty5c4eCBQtiaWlJXFwcCQkJWsXNafr6+rRv3549e/agVCpJT09HCEHDhg2pWbOmTgaF+Pj4YGJi8saUrOHh4ejp6Wk89iGr3FNTU9HT06Ndu3ZMmDBBJ0eNpUqVon///pw5c4aLFy+yf//+HJni+KuvvqJ///5s3LiR+fPn66SwZ3xnAwIC0NfXZ8eOHTg6OhIZGUlqaio2NjZaXxwvV64cP/zwA6mpqRQrVoyIiAhMTExyfWEHvowLqtHR0cLW1lYcPXpUJCcni7Vr14oJEyaII0eOiCFDhmjUre99crq7U0REhFi3bp3w9PTUqptmZhkX0xITE8WAAQPEpk2bdBL3Y3r27JlwdnZWdalMTU3VOmZcXJwIDAwUSUlJ4sqVK6Jq1api9erVYtWqVWLUqFE669qXE7m/LTg4WHh4eIhHjx7pPHZmKSkpOonz9nd227Ztonfv3jqJnZXnz5+L7du3i7i4uBx7j4/ti+gKGRYWxubNm5k+fTrweiBCzZo1adeuHY6Ojlo1l2Qlp7s7BQUFce7cOXr37q3VEPG3KZVKxo8fT69evejSpYtO57n4GO7fv8/x48cZP348BgYGWuWelpbGggUL8PT0pHv37gQFBZGQkEC5cuUoW7YsXbp00XpOl5zKPS/I6jtrZmaGi4sLgwcP1mgajQ9JTU3V6kzsc5N3Psl/KFSoEElJSapTroyh0D///LNOZhx8W61atahSpUqOXRUvX748AwYM0PmGaGBggIODgyrv3FZgdLneMy4URkdHY2FhQUhICBEREVy/fp0xY8bofFRhTm8zuc3b39mbN2/SsmVLOnbsmCOFHchThR2+oOkH/vnnHw4dOkS+fPn4559/dNYuKOVt/v7+HD9+nA4dOlCvXj2Nh59L6nv7O+vi4pKre698bF9McYfXIwujo6PR19fXuheC9OX4999/OXbsGBMmTCB//vy57owmN5PfWc19UcVdkjSlVCplk4mUq8jiLkmSlAd9cSNUJUmSvgSyuEuSJOVBsrhLkiTlQbK4S5Ik5UGyuEuSJOVBsrhLkiTlQbK4S5Ik5UGyuEuSJOVBsrhLkiTlQf8PZTijmaJIyykAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 385x158 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_proportion_mut_count_one_tissue(mut_plot_df,\n",
    "                                    total_count_df,\n",
    "                                    plot_pars, \n",
    "                                    mut_count_colors_prot_dict, \n",
    "                                    labels_prot_dict, \n",
    "                                    tissue = \"Normal\",\n",
    "                                    title='Normal Bladder (45 donors)', \n",
    "                                    figsize=(3.85, 1.58),\n",
    "                                    bar_width=0.7,\n",
    "                                    save=True, \n",
    "                                    filename=\"plots/Fig5a_proportion_and_count_mut_prot.normal.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "354f64a1-acbc-44ab-bb86-5f1db497c369",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAK8AAABhCAYAAACgcPGxAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAC8ZJREFUeJzt3X9M1PUfB/DnB4Tj9hXu+HF+8UR+CI7YNZo7a7RMy3KJsdYf6mJXAZE5yvwamoUbgWlbSkUaG1S2w7Gr9GtupbS8pacSipXFpmBgxzEVAYNxXAYcHPf6/sH8fDkPBE7uq+9vr8fmxn3eP+71/txzn73vc59NiYgIjAko4E4XwJivOLxMWBxeJiwOLxMWh5cJi8PLhMXhZcLi8DJhcXiZsDi8TFgcXiYsDi8TFoeXCYvDy4TF4WXC4vAyYXF4mbA4vExYHF4mLA4vExaHlwmLw8uExeFlwuLwMmFxeJmwOLxMWBxeJiwOLxMWh5cJi8PLhMXhZcLi8DJhcXiZsDi8TFgcXiYsDi8TljDh/f333yfdt7e3F3/88Ycfq2F3A7+FV5IkSJKEpqYmr7YPPvgAkiShuLh4UnO9/vrr2L59+6TfOykpCQ0NDQAAk8kEnU436bF/RzqdDiaTacy21tZWSJKE1tbW/21Rk+DXK29UVBQqKyu9jhuNRoSFhU16nqleRbu6uuS/DQaDHGQ2toaGBhgMhjtdxpT5NbwGgwFVVVVwu93ysR9//BGDg4NYsGCBfCw7OxvZ2dkeYyVJwvHjx7Ft2zaYTCaYTCbcd999AIBTp05h6dKl0Gq1CAkJwcKFC1FXVwcASE5OBgCkp6dj586dqKysRHx8vDxvTU0NFi9eDLVajYSEBBQWFsLpdAIAiouLsXLlSjz77LNQq9WIiYlBQUGBPPbkyZNYuHAh1Go1kpKSsGHDBrhcrjHXLkkSSkpKEBsbi9DQUGRkZODq1aty+/fff48HHngAarXa68qXnZ2NVatWISUlBRqNBlar1WPuG1fDjRs3Ijw8HK+88goA4Msvv0RqaipUKhX0ej3MZvOkao+Pj5cvMn/++SeysrKgUqkwZ84cVFVVjfm53HDz+f3ll1/w6KOPIjw8HPPnz0dpaSn89l/9kZ8AILPZTBqNho4cOSIfX7t2Le3YsYOWLFlCRUVFRESUlZVFWVlZXuMtFotXe19fH0VERFBZWRkNDw/T9evXafXq1bRo0aIxxxqNRoqLiyMiot9++40UCgV9+OGH5HQ66eLFi5Samkrr168nIqKioiKSJIn27t1LLpeLqqurSZIkOn36NBERxcbGUmVlJRER2Ww20mq1dODAgXHXr9PpqKWlhex2Oz311FNyjfX19aRUKumrr74il8tFtbW1FBUVRd9995283pkzZ9K5c+eop6fHa26bzUYAaM2aNeR0Oqmnp4eqq6tJpVLRiRMnyOVy0aFDh2jmzJl0/vz5CWuPi4sjo9FIRETPPfccPfjgg9TZ2UldXV20bNkyAkA2m83r3N58ftva2kilUlFZWRkNDg5SQ0MDJSUlUUVFxZjn6Hb5NbwWi4U2bNhAmZmZRDQSvMjISGpvb/c5vC6Xi5qbm8ntdlNfXx+dP3+e8vLyKCEhYcyxo09uYWEh3X///R7vU11dTUqlkoaHh6moqIiSk5M92rVaLe3du5eIiJKTk2nFihV06NAhcjgcNDw8fMv1Hzx4UH7d1NQkhyAvL4+eeeYZj/4FBQWUkZEhr/fxxx8fd+4b4f3hhx/kY+np6fTmm2969MvMzKR169ZNWPuN8A4MDJBCoSCz2Sy31dfXTzq8O3bsoLS0NI8aPv74Y7r33nvHXcvtmOGf6/l/5eTkIC0tDQ6HA19//TUeeughREdH+zxfYGAgLBYL0tPTcf36deh0OgQFBXlsTcbT2dmJefPmeRxLSEhAf38/rl27BgBetY2e++jRoyguLsbLL7+M9vZ2LF++HOXl5YiJiRnz/ebPny//HRsbCwBob29Ha2srjh07BrVaLbcPDw8jMTFRfq3Vaidcz+g+ra2tOH78OMrLy+VjLpcLjz322KRr7+7uhtPplGsF4FHTRFpbW3H27FmPdbndbgQGBk56jqnw+62y1NRU3HPPPdi/fz+MRiNyc3O9+gQGBmJwcFB+PfoL183OnDmDV199Ffv27UNHRweOHj2KZcuWTaqW+Ph4r/2j1WqFQqFARETELccODAygsbER5eXluHTpEhoaGtDb24vXXntt3DFtbW3y3zabDcBIiGNiYpCdnQ273S7/a25uxrfffiv3lyRpwvWM7hMTE4O33nrLY87Gxkbs2bNn0rVHRUUhJCQELS0t8rErV6549AkICBj3s4qJicHSpUs9arDZbPj1118nXIsv/if3eXNyclBaWoqmpiasWLHCqz0lJQU1NTVoa2tDf38/tm7d6vHBhISEoLe3F8DIPdyAgAAolUoAQF1dHXbt2uVxQhUKhdx/tMzMTDQ2Nsr9rVYrtmzZAoPBgODg4FuuQZIkZGZm4v3334fL5UJ0dDSCgoIQFRU17pitW7eio6MDdrsdmzZtwvLlyzFnzhzk5ubi888/h9lshtvtxsWLF7F48WK89957tz6Rt/DSSy9h9+7d+OmnnwAAP//8M/R6Pb744otJ1x4cHIznn38ehYWFuHLlCnp7e7F582aPPikpKThw4ABcLhesVis+++wzuc1gMOD06dMwmUxwuVxob29HRkYG8vPzfV7XLfllM0Kee6Ouri4KDg6mzZs3y+2j97wOh4NWr15NkZGRpNVqaefOnRQXFyePP3bsGGk0Gpo7dy653W7Kz8+nyMhIUqvVtGDBAiopKaHAwEDq6OggIqL169eTUqmkLVu2eOzJiIhqampo0aJFpFKpKDo6mvLz86mvr4+IRr6wLVmyxGMdo7/M1NbWUlpaGoWFhVF4eDgZDAay2+3jrn/dunWUmJhIKpWKDAYDdXd3y+2HDx8mvV5PYWFhNHv2bMrPzyen00lEY38HGO3GnvfGPvQGo9FIOp2OQkNDKS4ujt555x1yu90T1j56jf39/ZSXl0dqtZo0Gg1t377d471OnDhBer2eQkNDSa/XU0lJicf5PXXqFD388MMUHh5OGo2GcnJyqLe3d9y13A6JyF/3Mf7eJEmCxWLBI488cqdL+b8lzM/DjN2Mw8uE5fdbZX9XvBvzP77yMmFxeJmwpm3b8FfHJTjt4/+4MB6FOgr/iI71Om42m1FVVYWQkBCUlZWhvr4eERERHr9aTdVw12W4Hd1THhcQFonAqLlex0+ePImKigo4nU68/fbbPj96SY5OoN8+tUFKNaSwf47bbLFYsG/fPlRUVPhU09DQ0LgPHd3KjBkzEBQU5HGsvr5efgx29+7dUKlUPtXk9V7TMclfHZdweFUK3IMDUx4bEByCjH9f8AowEcFoNKK0tBTNzc24cOEC4uPjfQ7vcNdlXPuXHhhyTn1wkAKzdp31CrDD4UBVVRXOnTsHs9nsU3jJ0QkyGoDhwYk7jxYYDOSYxgxwS0sLzp49i4GBqX8ewEhwrVarT/t2SZKQmJjoEeCPPvoIn376Kc6cOYP9+/djzZo1PtV1s2nZNjjtXT4FFwDcgwNjXrGfeOIJHDlyBN988w0OHjyIPXv24N1334XRaPTtfRzdvgUXAIacY16xMzIyMDg4iLKyMt+fh+23Tz24wMiYca7W8+bNw6ZNm3yrByPPRPj6hZOIvK7YbrcbCoUCs2fPRmdnp8913eyuvdtQW1uLJ598EtHR0airq8OLL76I+Pj4u+qmf09PD9544w0UFRVh1qxZd7qcu5ZSqcTQ0BDa29tv66Gsm9214b18+TJeeOEF9Pf3Y9u2bUhKSrrTJXnZuHEjrl69ioKCAjz99NNYuXLlnS7prpSbm4u1a9fC6XTik08+mbZ5p+XnYX/seaebP/a80+F29rzSOHve2zXde15/mbZnG6b7boM/TPfdhunij7sNt2s67zb4Cz+Yw4TFP1IwYXF4mbA4vExYHF4mLA4vExaHlwmLw8uExeFlwuLwMmFxeJmwOLxMWBxeJiwOLxMWh5cJi8PLhMXhZcLi8DJhcXiZsDi8TFgcXiYsDi8TFoeXCYvDy4TF4WXC4vAyYXF4mbA4vExYHF4mLA4vExaHlwmLw8uExeFlwuLwMmFxeJmwOLxMWBxeJqz/AP+koPo3t91AAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 200x100 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.patches as mpatches\n",
    "\n",
    "\n",
    "# Create dummy patches for the legend\n",
    "patch_3plus = mpatches.Patch(color='#A63602', label='3+')\n",
    "patch_2 = mpatches.Patch(color='#E6550E', label='2')\n",
    "patch_1 = mpatches.Patch(color='#FD8D3D', label='1')\n",
    "patch_0 = mpatches.Patch(color='#D6D6D6', label='0')\n",
    "\n",
    "# Create the plot (empty, only for the legend)\n",
    "fig = plt.figure(figsize=(2, 1))\n",
    "plt.axis('off')  # Hide axes\n",
    "\n",
    "# Add the legend\n",
    "plt.legend(handles=[patch_3plus, patch_2, patch_1, patch_0],\n",
    "            title='Mutations per residue',\n",
    "            loc='center',  ncol = 4, frameon = False)\n",
    "\n",
    "fig.savefig('plots/Fig5a_legend.pdf', dpi=300, bbox_inches='tight')\n",
    "\n",
    "# Display the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0532b308-5c9e-4bd0-a1d5-ba449c52ad88",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "notebooks_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
